---
title: "COD_20240830_HOST_8_revision_figure"
author: "Minsik Kim"
date: "2024-08-30"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
        df_print: paged
editor_options: 
  chunk_output_type: inline
---


## Loading packages

```{r setup}
#===============================================================================
#BTC.LineZero.Header.1.1.0
#===============================================================================
#R Markdown environment setup and reporting utility.
#===============================================================================
#RLB.Dependencies:
#   knitr, magrittr, pacman, rio, rmarkdown, rmdformats, tibble, yaml
#===============================================================================
#Input for document parameters, libraries, file paths, and options.
#=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=

knitr::opts_chunk$set(message=FALSE, warning = FALSE)



path_working <- 
        ifelse(sessionInfo()[1]$R.version$platform == "x86_64-pc-linux-gnu",
               "/mnt/4T_samsung/Dropbox",
               ifelse(sessionInfo()[1]$R.version$platform == "aarch64-apple-darwin20",
                      "/Volumes/macdrive/Dropbox/", 
                      "/Dropbox (Personal)"))

path_library <- 
        ifelse(sessionInfo()[1]$R.version$platform == "x86_64-pc-linux-gnu",
               "/home/bagel/R/x86_64-pc-linux-gnu-library/4.1/",
               "/Library/Frameworks/R.framework/Resources/library/")


str_libraries <- c("readxl", "phyloseq", "tidyverse", "pacman", "yaml", "kableExtra",
                   "ggplot2",
                   "vegan", "microbiome", "ggpubr", "viridis", "decontam", "gridExtra",
                   "ggpubr", "lme4", "lmerTest", "writexl", "harrietr", "Maaslin2",
                   "ggtext", "ggpmisc", "gridExtra", "gamm4", "reshape2", "AMR")
        
YAML_header <-
'---
title: "COD_20240830_HOST_8_revision_figure"
author: "Minsik Kim"
date: "2024.08.30"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
---'
seed <- "20240912"

#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Loads libraries, file paths, and other document options.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
FUN.LineZero.Boot <- function() {
    .libPaths(path_library)

    require(pacman)
    pacman::p_load(c("knitr", "rmarkdown", "rmdformats", "yaml"))

    knitr::opts_knit$set(root.dir = path_working)

    str_libraries |> unique() |> sort() -> str_libraries
    pacman::p_load(char = str_libraries)

    set.seed(seed)
}
FUN.LineZero.Boot()
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Outputs R environment report.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
FUN.LineZero.Report <- function() {
    cat("Line Zero Environment:\n\n")
    paste("R:", pacman::p_version(), "\n") |> cat()
    cat("Libraries:\n")
    for (str_libraries in str_libraries) {
        paste(
            "    ", str_libraries, ": ", pacman::p_version(package = str_libraries),
            "\n", sep = ""
        ) |> cat()
    }
    paste("\nOperating System:", pacman::p_detectOS(), "\n") |> cat()
    paste("    Library Path:", path_library, "\n") |> cat()
    paste("    Working Path:", path_working, "\n") |> cat()
    paste("Seed:", seed, "\n\n") |> cat()
    cat("YAML Header:\n")
    cat(YAML_header)
}
FUN.LineZero.Report()


```

#Data cleaning description#

1. Loading data 
2. Merging all the output
3. Tidifying metadata
4. Save RDS file


***1. Loading data***


```{r warning=F, `Loading data`}
#set working directory
setwd("Project_SICAS2_microbiome/2_Protocols/host_depletion/")

#Loading meta data
        #Loading the data
        sample_data1 <- read_excel("SOP_20220324_MGK_Host_extraction.xlsx", sheet = 2)
        #Cryopreserved data
        sample_data2 <- read_excel("SOP_20220404_MGK_Host_qPCR_cryopreserve.xlsx", sheet = 2)
        #BAL and sputum data
        sample_data3 <- read_excel("SOP_20220407_MGK_sputum_BAL_host_depletion.xlsx", sheet = 5)
        #Additional BAl and NS data
        sample_data4 <- read_excel("SOP_20220606_MGK_BAL_NB_host_depletion.xlsx", sheet = 2)
        #control_data
        sample_data5 <- read_excel("SOP_20230112_MGK_host_depletion_controls.xlsx", sheet = 2) %>% subset(., .$sample_type != "nasal_swab")
        
#Laoding the qPCR data
        #Loading data  - host
        qPCR_data_host <- function(data){
                data %>% data.frame() %>% mutate(Quantity.Mean = case_when(
                        is.na(Quantity.Mean) ~ Quantity,
                        TRUE ~ Quantity.Mean)) %>%
                rename(extraction_id = "Sample", DNA_host_well = Quantity.Mean) %>% dplyr::select(c("extraction_id", "DNA_host_well"))
        }
        host_simple1 <- read_excel("SOP_20220324_MGK_Host_extraction.xlsx", sheet = 6) %>% qPCR_data_host()
        host_simple2 <- read_excel("SOP_20220404_MGK_Host_qPCR_cryopreserve.xlsx", sheet = 5) %>% qPCR_data_host()
        host_simple3 <- read_excel("SOP_20220407_MGK_sputum_BAL_host_depletion.xlsx", sheet = 6) %>% qPCR_data_host()
        host_simple4 <- read_excel("SOP_20220606_MGK_BAL_NB_host_depletion.xlsx", sheet = 5) %>% qPCR_data_host()
        host_simple5 <- read_excel("SOP_20230112_MGK_host_depletion_controls.xlsx", sheet = 9) %>% qPCR_data_host()

#Loading data  - Bac
        qPCR_data_bac <- function(data){
                data %>% data.frame() %>% mutate(Quantity.Mean = case_when(
                        is.na(Quantity.Mean) ~ Quantity,
                        TRUE ~ Quantity.Mean)) %>%
                rename(extraction_id = "Sample", DNA_bac_well = Quantity.Mean) %>% dplyr::select(c("extraction_id", "DNA_bac_well"))
        }
        bac_simple1 <- read_excel("SOP_20220324_MGK_Host_extraction.xlsx", sheet = 7) %>% qPCR_data_bac()
        bac_simple2 <- read_excel("SOP_20220404_MGK_Host_qPCR_cryopreserve.xlsx", sheet = 6) %>% qPCR_data_bac()
        bac_simple3 <- read_excel("SOP_20220407_MGK_sputum_BAL_host_depletion.xlsx", sheet = 7) %>% qPCR_data_bac()
        bac_simple4 <- read_excel("SOP_20220606_MGK_BAL_NB_host_depletion.xlsx", sheet = 6) %>% qPCR_data_bac()
        bac_simple5 <- read_excel("SOP_20230112_MGK_host_depletion_controls.xlsx", sheet = 8) %>% qPCR_data_bac()

```


***Merging all the output***


```{r, warning=F, `Merging all the outputs`}
        #Mergineg sampledata, host and bacterial qPCR data
        sample_results <- list()
        for (i in 1:5) {
                  sample_data <- get(paste0("sample_data", i))
                  host_simple <- get(paste0("host_simple", i))
                  bac_simple <- get(paste0("bac_simple", i))
                  
                  sample_results[[i]] <- merge(sample_data, host_simple, by = "extraction_id") %>%
                                          merge(., bac_simple, by = "extraction_id") %>%
                                          subset(., !duplicated(extraction_id, fromLast = T)) %>%
                          mutate(collection_date = as.character(collection_date), extraction_date = as.character(extraction_date))
        }

        #Changing type of data
        sample_results[[3]]$aliquot <- as.character(sample_results[[3]]$aliquot)
        
        sample_result <- dplyr::bind_rows(sample_results[[1]], sample_results[[2]], sample_results[[3]], sample_results[[4]], sample_results[[5]])

```

***Tidyfying data***

1. Re-calculattion of DNA concentration (adjusting dilution factors, standard concentration, etc.)
2. creating new column (treatment)


```{r warning=F, `Caculation of DNA, assigning treatment groups`}

#Specification at Zymo was 20, 2, 0.2, ... per well. 2 uL of standards were used.
#qPCR standards were 10, 0.1, 0.01 ng / uL etc. 
#Prior calculatoin was based on zymo's specification. (20 ng / uL, which is wrong). This need to be modified

sample_result<- sample_result %>% 
        mutate(DNA_bac_ng_uL = DNA_host_well/2, 
               DNA_host_ng_uL = DNA_bac_well/2,
               DNA_bac_gene_copies_uL = DNA_bac_ng_uL * 198349.73, 
               DNA_bac_16S_copies_uL = DNA_bac_gene_copies_uL * 7, 
               DNA_host_nondil = DNA_host_ng_uL * 10, DNA_bac_nondil = DNA_bac_ng_uL * 10, #qPCR was conducted for 10-fold diluted samples.
               host_proportion = (DNA_host_nondil/(DNA_host_nondil + DNA_bac_nondil)))

#Method and sample_type labels modification
sample_result <- sample_result %>% mutate(
        treatment = case_when(
                sample_result$control == 1 ~ "control",
                sample_result$lypma == 1 ~ "lyPMA",
                sample_result$benzonase == 1 ~ "benzonase",
                sample_result$qiaamp == 1 ~ "qiaamp",
                sample_result$host_zero == 1 ~ "host_zero",
                sample_result$molysis == 1 ~ "molysis",
                TRUE ~ "control"),
        sample_type = case_when(
                sample_type == "host_depletion_negative_control" ~ "neg_depletion",
                sample_type == "host_depletion_positive_control" ~ "pos_depletion",
                TRUE ~ sample_type)
)


```



# C19. Cryoprotected vs non-cryoprotected

C19.  Limited applicability of results to fresh samples. The study focused on frozen samples without cryoprotectants, which may limit the applicability of the findings to fresh samples or those preserved with cryoprotectants.…..


list of samples

Due to limited number of swab samples, there was no identical nasal-swab samples that had croprotectant & no-cryoprotectant in different aliquot at the same time.

```{r}
sample_result %>% subset(.,.$sample_type == "nasal_swab") %>% 
        group_by(original_sample, cryo_preserve) %>%
        summarise(n = n(),
                  sequenced = ifelse(original_sample %in% (phyloseq_unfiltered$phyloseq_count %>% sample_data %>% .$original_sample), 1, 0)) %>%
        reactable::reactable(sortable = T)


```

Statistica test table 


```{r}


cr_model <- sample_result %>% 
        lmer(data = ., log(DNA_bac_nondil) ~ sample_type * treatment * cryo_preserve + (1|original_sample)) 

cr_model %>% anova


cr_model_anova <- cr_model %>% anova %>% data.frame(check.names = F) %>% round(3) %>% rownames_to_column(" ") %>%
        mutate(` ` = case_when(` ` == "sample_type" ~ "Sample type",
                               ` ` == "treatment" ~ "Treatment",
                               ` ` == "cryo_preserve" ~ "Glycerol",
                               ` ` == "sample_type:treatment" ~ "Sample type * Treatment",
                               ` ` == "sample_type:cryo_preserve" ~ "Sample type * Glycerol", 
                               ` ` == "treatment:cryo_preserve" ~ "Treatment * Glycerol", 
                               ` ` == "sample_type:treatment:cryo_preserve" ~ "Sample type * Treatment * Glycerol", 
                               .default = "")) %>% 
        column_to_rownames(" ") %>% 
        mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                               abs(`Pr(>F)`) < 0.01 ~ "**",
                               abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) 



tables15 <- cr_model_anova %>% 
    kbl(format = "html", escape = 0) %>%
        #add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")


save_kable(tables15, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tables14.html", self_contained = T)



```

figure




```{r}


figr3_bacteria <- ggplot(
        sample_result %>% subset(., .$sample_type %in% c("BAL", "Nasal swabs", "Sputum")), #%>% 
        #filter(original_sample %in% (phyloseq_unfiltered$phyloseq_count %>% sample_data %>% .$original_sample)),
        aes(x = sample_type, y = log(DNA_bac_nondil))) +
        geom_jitter(aes(col = treatment, x = treatment),
                    lwd = 0.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment, x = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), 
                           name = "Treatment", 
                           breaks = c("control", "lyPMA", "benzonase", "host_zero", "molysis", "qiaamp"),
                           labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Treatment")+
        guides(fill = guide_legend(nrow = 1),
               color = guide_legend(nrow = 1)) +
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              axis.text.x = element_blank()) +
        facet_grid(sample_type~cryo_preserve, scales = "free_y") +
        ylab("log<sub>10</sub>(qPCR bacterial DNA)<br>(ng/μL)")+
        labs(tag = "A")



figr3_host <- ggplot(
        sample_result %>% subset(., .$sample_type %in% c("BAL", "Nasal swabs", "Sputum")),
        #filter(original_sample %in% (phyloseq_unfiltered$phyloseq_count %>% sample_data %>% .$original_sample)),
                aes(x = sample_type, y = DNA_host_nondil)) +
        geom_jitter(aes(col = treatment, x = treatment),
                    lwd = 0.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment, x = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), 
                           name = "Treatment", 
                           breaks = c("control", "lyPMA", "benzonase", "host_zero", "molysis", "qiaamp"),
                           labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Treatment")+
        guides(fill = guide_legend(nrow = 1),
               color = guide_legend(nrow = 1)) +
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              axis.text.x = element_blank()) +
        facet_grid(sample_type~cryo_preserve, scales = "free_y") +
        ylab("log<sub>10</sub>(qPCR human DNA)<br>(ng/μL)") +
        labs(tag = "B")





figr3_host_prop <- ggplot(
        sample_result %>% subset(., .$sample_type %in% c("BAL", "Nasal swabs", "Sputum")),
        #filter(original_sample %in% (phyloseq_unfiltered$phyloseq_count %>% sample_data %>% .$original_sample)),
                aes(x = sample_type, y = DNA_host_nondil/(DNA_bac_nondil + DNA_host_nondil)*100)) +
        geom_jitter(aes(col = treatment, x = treatment),
                    lwd = 0.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment, x = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), 
                           name = "Treatment", 
                           breaks = c("control", "lyPMA", "benzonase", "host_zero", "molysis", "qiaamp"),
                           labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Treatment")+
        guides(fill = guide_legend(nrow = 1),
               color = guide_legend(nrow = 1)) +
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              axis.text.x = element_blank(), legend.position = "top") +
        facet_grid(sample_type~cryo_preserve, scales = "free_y") +
        ylab("Host DNA %")  +
        labs(tag = "B")


ggarrange(figr3_bacteria, figr3_host_prop, common.legend = T, ncol = 2)


png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureR3_cryo_qPCR.png",   # The directory you want to save the file in
    width = 190, # The width of the plot in inches
    height = 170, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue
ggarrange(figr3_bacteria, figr3_host_prop, common.legend = T, ncol = 2)

#ggarrange(figr3_bacteria, figr3_host, figr3_host_prop, common.legend = T, ncol = 3)

# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()

```



### Loading phyloseq data

```{r warning=F, message=FALSE, "Data loading"}
# Loading files -----------------------------------------------------------
#loading tidy phyloseq object

phyloseq_unfiltered <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20230521_MGK_host_tidy.rds")

#v_phyloseq <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20240212_MGK_host_marker_magu_Jessica.rds")
v_phyloseq <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20240212_MGK_host_marker_magu.rds")

sample_data <- sample_data(phyloseq_unfiltered$phyloseq_count)

```


### Loading library data from Juwan

```{r}

picogreen_1 <- readxl::read_excel("Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/4_Communications biology/data/DAR_20240920_JC_Pooling_Copy of P. Lai PQ Requests 9.19.2024.xlsx", 
                   sheet = 1)

picogreen_2 <- readxl::read_excel("Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/4_Communications biology/data/DAR_20240920_JC_Pooling_Copy of P. Lai PQ Requests 9.19.2024.xlsx", 
                   sheet = 2) %>%
        rename("picogreen_tem_ng_ul" = "Raw nucleic acids concentration (ng/ul)",
               "picogreen_lib_ng_ul" = "libraries  concentration (ng/ul)",
               "pooling_vol_ul" = "Pooling Volume For Target (ul)",
               "pooling_total_DNA_ng" = "Pooling Total ng/sample")

names(picogreen_1) <- names(picogreen_2)


picogreen <- rbind(picogreen_1, picogreen_2)


picogreen_host <- picogreen[lapply(
        sample_data$baylor_other_id, 
        function(id)
                grep(id, picogreen$Sample_Name)) %>%
        as.numeric, ]

sample_data$picogreen_tem_ng_ul <- picogreen_host$picogreen_tem_ng_ul %>%
        ifelse(. < 0, 0, .)
sample_data$picogreen_lib_ng_ul <- picogreen_host$picogreen_lib_ng_ul %>%
        ifelse(. < 0, 0, .)
sample_data$pooling_vol_ul <- picogreen_host$pooling_vol_ul
sample_data$pooling_total_DNA_ng <- picogreen_host$pooling_total_DNA_ng
        

```


# Table C12

```{r}

table1_mock_and_neg <- sample_data %>% data.frame() %>% 
        #dplyr::filter(sample_type %in% c("Sputum", "Nasal", "BAL")) %>% 
        # mutate(S.obs = case_when(S.obs==0 ~ NA,
        #                          .default = S.obs),
        #        F.obs = case_when(F.obs==0 ~ NA,
        #                          .default = F.obs),
        #        V.obs = case_when(V.obs==0 ~ NA,
        #                          .default = V.obs)) %>% 
        filter(!is.na(picogreen_tem_ng_ul)) %>%
        mutate(sample_type = case_when(sample_type == "Neg." & treatment == "Untreated" ~ "Neg. (extraction)",
                                       sample_type == "Neg." & treatment == "lyPMA" ~ "Neg. (lyPMA)",
                                       sample_type == "Neg." & treatment == "Benzonase" ~ "Neg. (Benzonase)",
                                       sample_type == "Neg." & treatment == "HostZERO" ~ "Neg. (HostZERO)",
                                       sample_type == "Neg." & treatment == "MolYsis" ~ "Neg. (MolYsis)",
                                       sample_type == "Neg." & treatment == "QIAamp" ~ "Neg. (QIAamp)",
                                       .default = sample_type),
               sample_type = factor(sample_type, 
                                    levels = c("Neg. (extraction)",
                                               "Neg. (lyPMA)",
                                               "Neg. (Benzonase)",
                                               "Neg. (HostZERO)",
                                               "Neg. (MolYsis)",
                                               "Neg. (QIAamp)",
                                               "Mock",
                                               "BAL",
                                               "Nasal",
                                               "Sputum"))) %>%
        group_by (sample_type) %>%
        summarise(`N` = n(),
            #      `Total DNA <br>ng/µL` = paste(format(round(median(picogreen_ng_ul),2), nsmall = 2, big.mark = ","), "<br>(", format(round(quantile(picogreen_ng_ul, 0.25),2), nsmall = 2, big.mark = ","), ", ", format(round(quantile(picogreen_ng_ul, 0.75),2), nsmall = 2, big.mark = ","), ")", sep = ""),
               `Human DNA <br>pg/µL` = 
                    paste(format(round(median(DNA_host_ng_uL*1000),1), nsmall = 1, big.mark = ",")," (", format(round(quantile(DNA_host_ng_uL*1000, 0.25),1), nsmall = 1, big.mark = ","), ", ", format(round(quantile(DNA_host_ng_uL*1000, 0.75),1), nsmall = 1, big.mark = ","), ")", sep = ""),
               `Bacterial DNA <br>pg/µL` = paste(format(round(median(DNA_bac_ng_uL*1000),1), nsmall = 1, big.mark = ","), " (", format(round(quantile(DNA_bac_ng_uL*1000, 0.25),1), nsmall = 1, big.mark = ","), ", ", format(round(quantile(DNA_bac_ng_uL*1000, 0.75),1), nsmall = 1, big.mark = ","), ")", sep = ""),
              `Template DNA <br>conc. (ng/µL)` = paste(format(round(median(picogreen_tem_ng_ul),1), nsmall = 1, big.mark = ","), " (", format(round(quantile(picogreen_tem_ng_ul, 0.25),1), nsmall = 1, big.mark = ","), ", ", format(round(quantile(picogreen_tem_ng_ul, 0.75),1), nsmall = 1, big.mark = ","), ")", sep = ""),
              `Library<br>conc. (ng/µL)` = paste(format(round(median(picogreen_lib_ng_ul),1), nsmall = 1, big.mark = ","), " (", format(round(quantile(picogreen_lib_ng_ul, 0.25),1), nsmall = 1, big.mark = ","), ", ", format(round(quantile(picogreen_lib_ng_ul, 0.75),1), nsmall = 1, big.mark = ","), ")", sep = ""),
              `Volume used for<br>pooling (µL)` = paste(format(round(median(pooling_vol_ul),1), nsmall = 1, big.mark = ","), " (", format(round(quantile(pooling_vol_ul, 0.25),1), nsmall = 1, big.mark = ","), ", ", format(round(quantile(pooling_vol_ul, 0.75),1), nsmall = 1, big.mark = ","), ")", sep = ""),
            
              
              `QC'd reads<br>reads x 10<sup>6</sup>` = paste(format(round(median(Reads_after_trim/1000000),1), nsmall = 1, big.mark = ","), " (", format(round(quantile(Reads_after_trim/1000000, 0.25),1), nsmall = 1, big.mark = ","), ", ", format(round(quantile(Reads_after_trim/1000000, 0.75),1), nsmall = 1, big.mark = ","), ")", sep = ""),
              `Host reads<br>%` = paste(format(round(median(sequencing_host_prop*100),1),
                                                nsmall = 1, big.mark = ","),
                                         " (",
                                         format(round(quantile(sequencing_host_prop * 100,
                                                               0.25),
                                                      1),
                                                nsmall = 1,
                                                big.mark = ","), 
                                         ", ", 
                                         format(round(quantile(sequencing_host_prop * 100, 0.75),1), 
                                                nsmall = 1, 
                                                big.mark = ","), 
                                         ")", 
                                         sep = ""),
             `Final reads<br>reads x 10<sup>6</sup>` = paste(format(round(median(Final_reads/1000000,
                                                                                 na.rm = T),
                                                                          1),
                                                                    nsmall = 1, big.mark = ","),
                                                             " (", 
                                                             format(round(quantile(Final_reads/1000000,
                                                                                            0.25,
                                                                                            na.rm = T),
                                                                                   1),
                                                                             nsmall = 1,
                                                                             big.mark = ","),
                                                             ", ",
                                                             format(round(quantile(Final_reads/1000000, 
                                                                                   0.75, 
                                                                                   na.rm = T),
                                                                          1), 
                                                                    nsmall = 1, 
                                                                    big.mark = ","), 
                                                             ")",
                                                             sep = "")
              
            ) %>% 
        data.frame(check.names = F) %>% 
        arrange(sample_type) %>%
        rename(`Sample` = sample_type) %>%
        mutate_all(linebreak) %>% kbl(format = "html", escape = F) %>% kable_styling(full_width = 0, html_font = "sans")

table1_mock_and_neg

save_kable(table1_mock_and_neg, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/table1_mock_neg.html", self_contained = T)


```

# Viability test 


```{r}
# Host Depletion Viability Experiments


# Making dymo aliquot labels ----------------------------------------------
filename_dymo <- "Summer_Student_Projects/People/@thi/SOP/TAT_Host_Depletion/LOG_20240411_PSL_Dymo_Label.xlsx"
dymo <- read_excel(filename_dymo , sheet = "HDV")
dymo_long <- dymo %>%
  pivot_longer(
    cols = c(Number7:Number19),
    values_to = "aliquot"
  ) %>% 
  mutate(aliquot = paste(Number6, aliquot, sep = " ")) %>%
  select(-c(Number6, name)) 
#write_csv(dymo_long, "Summer_Student_Projects/People/@thi/SOP/TAT_Host_Depletion/LOG_20240411_PSL_Dymo_Label_HDV_long.csv")


# 20240822 Freezing + DNA/RNA Shield --------------------------------------

rawdata.path.hdv <- "Summer_Student_Projects/People/@thi/SOP/TAT_Host_Depletion"
filename1 <- file.path(rawdata.path.hdv, "SOP_20240822_TAT_HDV_060-099_Viability_experiment_DNA-RNA_shield.xlsx")
file1 <- read_excel(filename1, sheet = "DAR_CFU_count") 

filename2 <- file.path(rawdata.path.hdv, "SOP_20240822_TAT_HDV_100-159_Viability_experiment_DNA-RNA_shield.xlsx")
file2 <- read_excel(filename2, sheet = "DAR_CFU_count") 

filename3 <- file.path(rawdata.path.hdv, "SOP_20240828_TAT_HDV_benzonase.xlsx")
file3 <- read_excel(filename3, sheet = "DAR_CFU_Count") 

filename4 <- file.path(rawdata.path.hdv, "SOP_20240829_TAT_HDV_HostZero.xlsx")
file4 <- read_excel(filename4, sheet = "DAR_CFU_Count") 

filename5 <- file.path(rawdata.path.hdv, "SOP_20240903_TAT_HDV_QIAamp.xlsx")
file5 <- read_excel(filename5, sheet = "DAR_CFU_count") 

filename6 <- file.path(rawdata.path.hdv, "SOP_20240904_TAT_HDV_lyPMA.xlsx")
file6 <- read_excel(filename6, sheet = "DAR_CFU_count")

filename7 <- file.path(rawdata.path.hdv, "SOP_20240905_TAT_HDV_fresh_frozen.xlsx") # repeat studies on S. aureus
file7 <- read_excel(filename7, sheet = "DAR_CFU_count") %>%
  select(expt_id:cfu_per_ml2)

filename8 <- file.path(rawdata.path.hdv, "SOP_20240910_TAT_HDV_S_aureus_fresh_frozen_benzonase_QIAamp.xlsx")
file8 <- read_excel(filename8, sheet = "DAR_CFU_count") %>%
  select(expt_id:cfu_per_ml2)

filename9 <- file.path(rawdata.path.hdv, "SOP_20240911_TAT_HDV_S_aureus_HostZero_lyPMA.xlsx")
file9 <- read_excel(filename9, sheet = "DAR_CFU_count") %>%
  select(expt_id:cfu_per_ml2)

filename10 <- file.path(rawdata.path.hdv, "SOP_20240912_TAT_HDV_S_aureus_MolYsis.xlsx")
file10 <- read_excel(filename10, sheet = "DAR_CFU_count") %>%
  select(expt_id:cfu_per_ml2)

filename11 <- file.path(rawdata.path.hdv, "SOP_20240917_TAT_HDV_P_aeruginosa_Achromobacter_fresh_Benzonase_QIAamp.xlsx")
file11 <- read_excel(filename11, sheet = "DAR_CFU_count") %>%
  select(expt_id:cfu_per_ml2)

filename12 <- file.path(rawdata.path.hdv, "SOP_20240918_TAT_HDV_P_aeruginosa_Achromobacter_frozen_hostzero_lyPMA.xlsx")
file12 <- read_excel(filename12, sheet = "DAR_CFU_count") %>%
  select(expt_id:cfu_per_ml2)


hdv_all <- bind_rows(file1, file2, file3, file4, file5, file6, file7, file8, file9, file10, file11, file12) %>%
  filter(!is.na(num_colonies1) & !is.na(num_colonies2)) %>%
  mutate(cfu_per_ml = ifelse(num_colonies2 > num_colonies1,
                             (num_colonies2*1000/1)*dilution_factor2,# replicator volume was 1 uL not 3 uL. all counts were devided by 3 but now should be devided by 1
                             (num_colonies1*1000/1)*dilution_factor1)) %>%
  mutate(cfu_per_ml1 = (num_colonies1*1000/1)*dilution_factor1,
         cfu_per_ml2 = (num_colonies2*1000/1)*dilution_factor2) %>%
  # rowwise() %>%
  # mutate(cfu_per_ml = max(c_across(c(cfu_per_ml1, cfu_per_ml2)), na.rm=TRUE)) %>% 
  # mutate(cfu_per_ml = median(c_across(c(cfu_per_ml1, cfu_per_ml2)), na.rm=TRUE)) %>%
  filter(treatment != "frozen_shield" & !is.na(treatment)) %>%
  mutate(treatment2 = case_when(
    treatment == "fresh" ~ "Unfrozen",
    treatment == "frozen" ~ "Frozen",
    treatment == "fresh_shield" ~ "Unfrozen DNA/RNA Shield",
    treatment == "frozen_glycerol" ~ "Frozen Glycerol",
    treatment %in% c("lyPMA", "lyPMA_frozen") ~ "Frozen lyPMA",
    treatment == "lyPMA_glycerol" ~ "Frozen Glycerol lyPMA",
    treatment %in% c("benzonase", "Benzonase", "benzonase_frozen") ~ "Frozen Benzonase",
    treatment %in% c("Benzonase_glycerol", "benzonase_glycerol") ~ "Frozen Glycerol Benzonase",
    treatment %in% c("hostzero", "hostzero_frozen") ~ "Frozen HostZERO",
    treatment == "hostzero_glycerol" ~ "Frozen glycerol HostZERO",
    treatment == "molysis_frozen" ~ "Frozen MolYsis",
    treatment == "molysis_glycerol" ~ "Frozen Glycerol MolYsis",
    treatment %in% c("QIAamp", "QIAamp_frozen") ~ "Frozen QIAamp",
    treatment == "QIAamp_glycerol" ~ "Frozen Glycerol QIAamp",
    TRUE ~ treatment
  )) %>%
  mutate(treatment2 = factor(treatment2, 
                             ordered = TRUE, 
                             levels = c("Unfrozen", "Unfrozen DNA/RNA Shield", "Frozen", "Frozen Glycerol", 
                                        "Frozen lyPMA", "Frozen Glycerol lyPMA",
                                        "Frozen Benzonase", "Frozen Glycerol Benzonase",
                                        "Frozen HostZERO", "Frozen glycerol HostZERO",
                                        "Frozen MolYsis", "Frozen Glycerol MolYsis",
                                        "Frozen QIAamp", "Frozen Glycerol QIAamp")
                            )
        ) %>%
  mutate(
    cfu_per_ml = ifelse(grepl("glycerol", treatment2), cfu_per_ml*(225/135), cfu_per_ml), # glycerol stock had 135 uL liquid culture + 90 uL 50% glycerol
    cfu_per_ml1 = ifelse(grepl("glycerol", treatment2), cfu_per_ml1*(225/135), cfu_per_ml1),
    cfu_per_ml2 = ifelse(grepl("glycerol", treatment2), cfu_per_ml2*(225/135), cfu_per_ml2),
    cfu_per_ml_forlog = cfu_per_ml + 1,
    cfu_per_ml_log10 = log10(cfu_per_ml_forlog)
  ) %>%
  mutate(species2 = case_when(
    species %in% c("Stenotrophomonas maltophila", "Stenotrophomonas maltophilia") ~ "Stenotrophomonas maltophila",
    species %in% c("TSB", "TSB broth") ~ "Tryptic Soy Broth",
    species %in% c("Achromobacter sp") ~ "Achromobacter spp",
    TRUE ~ species)
    ) %>%
  mutate(species2 = factor(species2,
                           ordered = TRUE, 
                           levels = c("Tryptic Soy Broth", "Staphylococcus aureus", "Pseudomonas aeruginosa", 
                                      "Achromobacter spp", "Escherichia coli", "Klebsiella pneumoniae", "Stenotrophomonas maltophila"))) %>%
  filter(!is.na(cfu_per_ml))

## checks
hdv_all %>% filter(is.na(treatment2)) %>% select(expt_id, species, cfb_strain, treatment, treatment2)
hdv_all %>% pull(treatment2) %>% table(useNA = "always")
hdv_all %>% pull(species2) %>% table(useNA = "always")
#write_csv(hdv_all, "Summer_Student_Projects/People/@thi/SOP/TAT_Host_Depletion/DAT_20240909_PSL_hdv_all.csv")

# Customize specific words in x-axis labels
# Initialize treatment_colored with the original values
hdv_all$treatment_colored <- hdv_all$treatment2

hdv_all$treatment_colored <- gsub("glycerol", "Glycerol", hdv_all$treatment2)
hdv_all$treatment_colored %>% table
# Apply gsub iteratively for different words, using hex codes if needed

#scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3")

#hdv_all$treatment_colored <- gsub("Unfrozen", "<span style='color:#;'>Unfrozen</span>", hdv_all$treatment_colored)
hdv_all$treatment_colored <- gsub("DNA/RNA Shield", "<span style='color:#b15928;'>DNA/RNA Shield</span>", hdv_all$treatment_colored)
hdv_all$treatment_colored <- gsub("Frozen", "<span style='color:#6a3d9a;'>Frozen</span>", hdv_all$treatment_colored)
hdv_all$treatment_colored <- gsub("Glycerol", "<span style='color:#cab2d6;'>Glycerol</span>", hdv_all$treatment_colored)
hdv_all$treatment_colored <- gsub("lyPMA", "<span style='color:#fb9a99;'>lyPMA</span>", hdv_all$treatment_colored)
hdv_all$treatment_colored <- gsub("Benzonase", "<span style='color:#33a02c;'>Benzonase</span>", hdv_all$treatment_colored)
hdv_all$treatment_colored <- gsub("HostZERO", "<span style='color:#b2df8a;'>HostZERO</span>", hdv_all$treatment_colored)
hdv_all$treatment_colored <- gsub("QIAamp", "<span style='color:#a6cee3;'>QIAamp</span>", hdv_all$treatment_colored)

unique(hdv_all$treatment_colored)


hdv_all$treatment_colored <- factor(hdv_all$treatment_colored, levels = c(unique(hdv_all$treatment_colored)[1],
                                               unique(hdv_all$treatment_colored)[3],
                                               unique(hdv_all$treatment_colored)[2],
                                               unique(hdv_all$treatment_colored)[8],
                                               unique(hdv_all$treatment_colored)[7],
                                               unique(hdv_all$treatment_colored)[12],
                                               unique(hdv_all$treatment_colored)[4],
                                               unique(hdv_all$treatment_colored)[10],
                                               unique(hdv_all$treatment_colored)[5],
                                               unique(hdv_all$treatment_colored)[11],
                                               unique(hdv_all$treatment_colored)[6],
                                               unique(hdv_all$treatment_colored)[9]))

figure_S16 <- hdv_all %>%
        filter(
                species2 %in% c("Staphylococcus aureus", "Pseudomonas aeruginosa", "Achromobacter spp") &
                        !(treatment2 %in% c("Frozen MolYsis", "Frozen Glycerol MolYsis"))
                ) %>%
        ggdotplot(
                x = "treatment_colored", 
                y = "cfu_per_ml_forlog",
                color = "treatment2",
                binwidth = 0.12,
                x.text.angle = 90,
                xlab = "", 
                ylab = "CFU/mL",
                add = "mean_ci",
                palette = c("black", "#b15928", "#6a3d9a", "#cab2d6", "#fc9272", "#fee0d2", "#006d2c", "#238b45", "#41ab5d", "#a1d99b", "#6baed6", "#c6dbef")) %>%
        ggpar(ylim = c(1, 10000000000), yscale = "log10", format.scale = TRUE) +
        facet_wrap(
                ~ species2, 
                ncol = 1, 
                labeller = as_labeller(c(
                        "Staphylococcus aureus" = "*Staphylococcus aureus*",
                        "Pseudomonas aeruginosa" = "*Pseudomonas aeruginosa*",
                        "Achromobacter spp" = "*Achromobacter* spp."))) +
        theme_classic(base_family = 'sans', base_size = 12) +
        rremove("legend") +
        theme(
                axis.text.x = element_markdown(angle = 45, vjust = 1, hjust = 1),
                strip.text = element_markdown()
                )
figure_S16


png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS16.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 210, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figure_S16
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()


```
### Figure v2

```{r}
# hdv_all %>%
#         mutate(Frozen = ifelse(hdv_all$Frozen == 1, "Frozen", "Unfrozen") %>%
#                        factor(., levels = c("Unfrozen", "Frozen")),
#                Glycerol = ifelse(hdv_all$Frozen == 1, "Glycerol", "No glycerol") %>%
#                        factor(., levels = c("No glycerol", "Glycerol")))# %>%
#         filter(
#                 species2 %in% c("Staphylococcus aureus", "Pseudomonas aeruginosa", "Achromobacter spp") &
#                         !(treatment2 %in% c("Frozen MolYsis", "Frozen Glycerol MolYsis"))
#                 ) %>%
#         ggdotplot(
#                 x = "treatment_colored", 
#                 y = "cfu_per_ml_forlog",
#                 color = "treatment2",
#                 binwidth = 0.12,
#                 x.text.angle = 90,
#                 xlab = "", 
#                 ylab = "CFU/mL",
#                 add = "mean_ci",
#                 palette = c("black", "#b15928", "#6a3d9a", "#cab2d6", "#fc9272", "#fee0d2", "#006d2c", "#238b45", "#41ab5d", "#a1d99b", "#6baed6", "#c6dbef")) %>%
#         ggpar(ylim = c(1, 10000000000), yscale = "log10", format.scale = TRUE) +
#         facet_wrap(Frozen
#                 ~ species2, 
#                 ncol = 1, 
#                 labeller = as_labeller(c(
#                         "Staphylococcus aureus" = "*Staphylococcus aureus*",
#                         "Pseudomonas aeruginosa" = "*Pseudomonas aeruginosa*",
#                         "Achromobacter spp" = "*Achromobacter* spp."))) +
#         theme_classic(base_family = 'sans', base_size = 12) +
#         rremove("legend") +
#         theme(
#                 axis.text.x = element_markdown(angle = 45, vjust = 1, hjust = 1),
#                 strip.text = element_markdown()
#                 )
```



### Data trimming

```{r}
hdv_all <- hdv_all %>% 
        mutate(Frozen = ifelse(grepl("Frozen", .$treatment2), 1, 0),
               treatment = case_when(grepl("lyPMA", .$treatment2) ~ "lyPMA",
                                     grepl("MolYsis", .$treatment2) ~ "MolYsis",
                                     grepl("Benzonase", .$treatment2) ~ "Benzonase",
                                     grepl("QIAamp", .$treatment2) ~ "QIAamp",
                                     grepl("HostZERO", .$treatment2) ~ "HostZERO",
                                     .default = "Untreated"),
               treatment = factor(treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")),
               `DNA/RNA shield` = ifelse(grepl("DNA", .$treatment2), 1, 0),
               Glycerol = ifelse(grepl("Glycerol|glycerol", .$treatment2), 1, 0))


```

### All species together

Statistical tests for C12

Model: 

LMER

cfu_per_ml_log10 ~ treatment + Frozen + `DNA/RNA shield` + Glycerol + species2 + (1|cfb_strain)

```{r}
hdv_all$dilution_factor1 != hdv_all$dilution_factor2
hdv_all %>%
        filter(
                species2 %in% c("Staphylococcus aureus", "Pseudomonas aeruginosa", "Achromobacter spp") &
                        !(treatment2 %in% c("Frozen MolYsis", "Frozen Glycerol MolYsis"))
                ) %>%
        mutate(colonies_large = case_when(num_colonies1 > num_colonies1 ~ num_colonies1,
                                          .default = num_colonies2)) %>%
        group_by(species, treatment2) %>%
        summarise(min = min(colonies_large),
                  max = max(colonies_large))

multi_model <- hdv_all %>% 
                filter(
                species2 %in% c("Staphylococcus aureus", "Pseudomonas aeruginosa", "Achromobacter spp") &
                        !(treatment2 %in% c("Frozen MolYsis", "Frozen Glycerol MolYsis"))
                ) %>%
        #mutate()
        lmerTest::lmer(data = .,
                       cfu_per_ml_log10 ~ treatment + Frozen + `DNA/RNA shield` + Glycerol + species2 + (1|cfb_strain)) 
        
multi_model %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              multi_model %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
            "Frozen",
            "Glycerol",
            "`DNA/RNA shield`",
            "lyPMA",
            "Benzonase",
            "HostZERO",
            #"MolYsis",
            "QIAamp"),] %>%
        kbl(format = "html", escape = 0) %>%
        #add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")

```

### interaction terms

Model: 

cfu_per_ml_log10 ~ treatment`*`species2 + Frozen`*`species2 + DNA/RNA shield`*`species2 + Glycerol`*`species2 + (1|strain)

--> change into gram-positive and gram-negative

```{r}
interaction_model <- hdv_all %>% 
                filter(
                species2 %in% c("Staphylococcus aureus", "Pseudomonas aeruginosa", "Achromobacter spp") &
                        !(treatment2 %in% c("Frozen MolYsis", "Frozen Glycerol MolYsis"))
                ) %>%
        lmerTest::lmer(data = .,
                       cfu_per_ml_log10 ~ treatment*species2 + Frozen*species2 + `DNA/RNA shield`*species2 + Glycerol*species2 + (1|cfb_strain)) 
        
#interaction_model %>% summary
interaction_model %>% anova 
#         
# 
# interaction_model <- hdv_all %>% 
#                 filter(
#                 species2 %in% c("Staphylococcus aureus", "Pseudomonas aeruginosa", "Achromobacter spp") &
#                         !(treatment2 %in% c("Frozen MolYsis", "Frozen Glycerol MolYsis"))
#                 ) %>%
#         lmerTest::lmer(data = .,
#                        cfu_per_ml_log10 ~ Glycerol * treatment * species2 + (1|species2)) 
#         
# interaction_model %>% anova 
# 
# interaction_model %>% summary
#         


```

### Stratified by species (log scale)

LMER : --> removed DNA RNA shield from the model.

```{r}

multi_model_sa <- hdv_all %>% 
                filter(
                species2 %in% c("Staphylococcus aureus") &
                        !(treatment2 %in% c("Frozen MolYsis", "Frozen Glycerol MolYsis")),
                `DNA/RNA shield` == 0
                ) %>%
        lmerTest::lmer(data = .,
                       cfu_per_ml_log10 ~ treatment + Frozen + Glycerol + (1|cfb_strain)) 
        

multi_model_pa <- hdv_all %>% 
                filter(
                species2 %in% c("Pseudomonas aeruginosa") &
                        !(treatment2 %in% c("Frozen MolYsis", "Frozen Glycerol MolYsis")),
                `DNA/RNA shield` == 0
                ) %>%
        lmerTest::lmer(data = .,
                       cfu_per_ml_log10 ~ treatment + Frozen + Glycerol + (1|cfb_strain)) 
        

multi_model_a <- hdv_all %>% 
                filter(
                species2 %in% c("Achromobacter spp") &
                        !(treatment2 %in% c("Frozen MolYsis", "Frozen Glycerol MolYsis")),
                `DNA/RNA shield` == 0
                ) %>%
        lmerTest::lmer(data = .,
                       cfu_per_ml_log10 ~ treatment + Frozen + Glycerol + (1|cfb_strain)) 
        

multi_sa <- multi_model_sa %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              multi_model_sa %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
            "Frozen",
            "Glycerol",
            #"`DNA/RNA shield`",
            "lyPMA",
            "Benzonase",
            "HostZERO",
            #"MolYsis",
            "QIAamp"),]


multi_pa <- multi_model_pa %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              multi_model_pa %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
            "Frozen",
            "Glycerol",
            #"`DNA/RNA shield`",
            "lyPMA",
            "Benzonase",
            "HostZERO",
            #"MolYsis",
            "QIAamp"),]


multi_a <- multi_model_a %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              multi_model_a %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
            "Frozen",
            "Glycerol",
            #"`DNA/RNA shield`",
            "lyPMA",
            "Benzonase",
            "HostZERO",
            #"MolYsis",
            "QIAamp"),]


lmer_viability_result <- cbind(multi_sa, multi_pa, multi_a) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "<i>Staphylococcus aureus</i>" = 3, "<i>Pseudomonas aeruginosa</i>" = 3,"<i>Achromobacter</i> spp."= 3), escape = F) %>% 
        kable_styling(full_width = 0, html_font = "sans")

lmer_viability_result


save_kable(lmer_viability_result, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS12-v2.html", self_contained = T)

lmer_viability_result

```


### Molysis on Staphylococcus


Model: 

LMER

cfu_per_ml_log10 ~ treatment + Frozen + `DNA/RNA shield` + Glycerol + species2 + (1|cfb_strain)

```{r}
hdv_all %>% 
        filter(grepl("Mol", treatment2))


multi_model <- hdv_all %>% 
                filter(
                species2 %in% c("Staphylococcus aureus", "Pseudomonas aeruginosa", "Achromobacter spp")) %>%
        mutate(species3 = factor(species2, levels = c("Staphylococcus aureus", "Pseudomonas aeruginosa", "Achromobacter spp"),
                                 ordered = F)) %>%
        lmerTest::lmer(data = .,
                       cfu_per_ml_log10 ~ treatment + Frozen + `DNA/RNA shield` + Glycerol + species3 + (1|cfb_strain)) 
        

tableS13 <- multi_model %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              multi_model %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% 
        mutate(x = gsub("treatment|sample_type|species3", "", x)) %>%
        mutate(x = gsub("Achromobacter spp", "<i>Achromobacter</i> spp.", x)) %>% 
        mutate(x = gsub("Pseudomonas aeruginosa", "<i>Pseudomonas aeruginosa</i>", x)) %>%
        mutate(x = gsub("`DNA/RNA shield`", "DNA/RNA shield", x)) %>% 
        mutate(x = gsub(":", " * ", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
            "<i>Achromobacter</i> spp.",
            "<i>Pseudomonas aeruginosa</i>",
            "Frozen",
            "Glycerol",
            "DNA/RNA shield",
            "lyPMA",
            "Benzonase",
            "HostZERO",
            "MolYsis",
            "QIAamp"),] %>%
        kbl(format = "html", escape = 0) %>%
        #add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")



save_kable(tableS13, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS13.html", self_contained = T)

```

### Alpha diversity indices

```{r}           

alpha_diversity <- function(data) {
        otu_table <- otu_table(data) #%>% .[, colSums(.) !=0]
        S.obs <- rowSums(t(otu_table) != 0)
        sample_data <- sample_data(data)
        data_evenness <- vegan::diversity(t(otu_table)) / log(vegan::specnumber(t(otu_table))) # calculate evenness index using vegan package
        data_shannon <- vegan::diversity(t(otu_table), index = "shannon") # calculate Shannon index using vegan package
        data_hill <- exp(data_shannon)                           # calculate Hills index
        data_dominance <- microbiome::dominance(otu_table, index = "all", rank = 1, aggregate = TRUE) # dominance (Berger-Parker index), etc.
        data_invsimpson <- vegan::diversity(t(otu_table), index = "invsimpson")                          # calculate Shannon index using vegan package
        alpha_diversity <- cbind(S.obs, data_shannon, data_hill, data_invsimpson, data_evenness,data_dominance) # combine all indices in one data table
        sample_data <- merge(data.frame(sample_data), alpha_diversity, by = 0, all = T) %>% column_to_rownames(var = "Row.names")
}
sample_data(v_phyloseq$viral_reads) <- sample_data(alpha_diversity(v_phyloseq$viral_reads))
#sample_data(v_phyloseq$viral_rpkm) <- sample_data(alpha_diversity(v_phyloseq$viral_rpkm))

sample_data(phyloseq_unfiltered$phyloseq_count)$V.obs <- 
        sample_data(v_phyloseq$viral_reads) %>% 
        data.frame %>% 
        dplyr::select("S.obs") %>% 
#        mutate(S.obs = case_when(is.na(S.obs) ~ 0,
#                                 .default = S.obs)) %>%
        .[sample_names(phyloseq_unfiltered$phyloseq_count),]

sample_data(phyloseq_unfiltered$phyloseq_rel)$V.obs <- 
        sample_data(v_phyloseq$viral_reads) %>% 
        data.frame %>% 
        dplyr::select("S.obs") %>% 
#        mutate(S.obs = case_when(is.na(S.obs) ~ 0,
#                                 .default = S.obs)) %>%
        .[sample_names(phyloseq_unfiltered$phyloseq_count),]

sample_data(phyloseq_unfiltered$phyloseq_path_rpk)$V.obs <- 
        sample_data(v_phyloseq$viral_reads) %>% 
        data.frame %>% 
        dplyr::select("S.obs") %>% 
#        mutate(S.obs = case_when(is.na(S.obs) ~ 0,
#                                 .default = S.obs)) %>%
        .[sample_names(phyloseq_unfiltered$phyloseq_count),]

phyloseq <- phyloseq_unfiltered



alpha_diversity <- function(data) {
        otu_table <- otu_table(data) # %>% .[, colSums(.) !=0]
        S.obs <- rowSums(t(otu_table) != 0)
        sample_data <- sample_data(data)
        data_evenness <- vegan::diversity(t(otu_table)) / log(vegan::specnumber(t(otu_table))) # calculate evenness index using vegan package
        data_shannon <- vegan::diversity(t(otu_table), index = "shannon") # calculate Shannon index using vegan package
        data_hill <- exp(data_shannon)                           # calculate Hills index
        data_dominance <- microbiome::dominance(otu_table, index = "all", rank = 1, aggregate = TRUE) # dominance (Berger-Parker index), etc.
        data_invsimpson <- vegan::diversity(t(otu_table), index = "invsimpson")                          # calculate Shannon index using vegan package
        alpha_diversity <- cbind(S.obs, data_shannon, data_hill, data_invsimpson, data_evenness,data_dominance) # combine all indices in one data table
        sample_data <- merge(data.frame(sample_data), alpha_diversity, by = 0, all = T) %>% column_to_rownames(var = "Row.names")
}



sample_data(phyloseq_unfiltered$phyloseq_rel) <- sample_data(alpha_diversity(phyloseq_unfiltered$phyloseq_rel))
sample_data(phyloseq_unfiltered$phyloseq_count) <- sample_data(alpha_diversity(phyloseq_unfiltered$phyloseq_count)) 
sample_data(phyloseq_unfiltered$phyloseq_path_rpk) <- sample_data(alpha_diversity(phyloseq_unfiltered$phyloseq_path_rpk))  
```


# C10. Rarefaction curve



## Rarefaction curve calculation {.tabset}


```{r}

# phyloseq_nz_microbe <- phyloseq_unfiltered$phyloseq_count %>%
#         subset_samples(S.obs != 0 & sample_type %in% c(#"Mock",
#                                                        "BAL", "Nasal", "Sputum"))
# 
# # phyloseq_nz_function <- phyloseq_unfiltered$phyloseq_path_rpk %>%
# #         subset_samples(S.obs != 0 & sample_type %in% c(#"Mock",
# #                                                        "BAL", "Nasal", "Sputum"))
# 
# phyloseq_nz_v <- v_phyloseq$viral_reads %>%
#         subset_samples(S.obs != 0 & sample_type %in% c(#"Mock",
#                                                        "BAL", "Nasal", "Sputum"))


```



## Calculating rarecurve with step = 10000

```{r}
# set.seed(seed)
# 
# 
# rare_microbe_10000 <- phyloseq.extended::ggrare(phyloseq_nz_microbe,
#                           step = 10000, color = "treatment", label = "treatment", se = T)
# 
# # rare_function <- phyloseq.extended::ggrare(phyloseq_nz_function,
# #                           step = 100000, color = "treatment", label = "subject_id", se = T)
# 
# rare_v_10000 <- phyloseq.extended::ggrare(phyloseq_nz_v,
#                           step = 10000, color = "treatment", label = "subject_id", se = T)

```

## Calculating rarecurve with step = 1000

```{r}
# set.seed(seed)
# 
# 
# rare_microbe_1000 <- phyloseq.extended::ggrare(phyloseq_nz_microbe,
#                           step = 1000, color = "treatment", label = "subject_id", se = T)
# 
# # rare_function <- phyloseq.extended::ggrare(phyloseq_nz_function,
# #                           step = 100000, color = "treatment", label = "subject_id", se = T)
# 
# rare_v_1000 <- phyloseq.extended::ggrare(phyloseq_nz_v,
#                           step = 1000, color = "treatment", label = "subject_id", se = T)

```

## Calculating rarecurve with step = 500 

```{r}
# set.seed(seed)
# 
# rare_microbe_500 <- phyloseq.extended::ggrare(phyloseq_nz_microbe,
#                           step = 500, color = "treatment", label = "subject_id", se = T)
# 
# # rare_function <- phyloseq.extended::ggrare(phyloseq_nz_function,
# #                           step = 100000, color = "treatment", label = "subject_id", se = T)
# 
# rare_v_500 <- phyloseq.extended::ggrare(phyloseq_nz_v,
#                           step = 500, color = "treatment", label = "subject_id", se = T)

```



### Calculating rarecurve with step = 200

```{r}
# set.seed(seed)
# 
# rare_microbe_200 <- phyloseq.extended::ggrare(phyloseq_nz_microbe,
#                           step = 200, color = "treatment", label = "subject_id", se = T)
# 
# # rare_function <- phyloseq.extended::ggrare(phyloseq_nz_function,
# #                           step = 100000, color = "treatment", label = "subject_id", se = T)
# 
# rare_v_200 <- phyloseq.extended::ggrare(phyloseq_nz_v,
#                           step = 200, color = "treatment", label = "subject_id", se = T)

rare_microbe <- readRDS("Git/Host_depletion_macmini/data/DAT_20240916_rarecurve_precal.rds")
```




```{r}



rare_figure_microbe <- 
{rare_microbe$rare_microbe_1000 +
        facet_wrap(~sample_type) +
        scale_color_manual(
                values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), 
                name = "Treatment", 
                labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_fill_manual(
                values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), 
                name = "Treatment", 
                labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using        
        #guides(fill = guide_legend()) +   
        scale_x_continuous(trans='log10') +
        theme_classic (base_size = 11, base_family = "sans")}%>%
        gginnards::delete_layers(., "GeomText") +
        theme(axis.title.x = element_blank(),
              axis.title.y = element_blank()) +
        #ylab("Species richness") +
        #xlab("Reads") +
        guides(fill = guide_legend(nrow = 1),
               color = guide_legend(nrow = 1))




rare_figure_virus <- 
{rare_microbe$rare_virus_1000 +
        facet_wrap(~sample_type) +
        scale_color_manual(
                values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), 
                name = "Treatment", 
                labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_fill_manual(
                values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), 
                name = "Treatment", 
                labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using        
        #guides(fill = guide_legend()) +   
        scale_x_continuous(trans='log10') +
        theme_classic (base_size = 11, base_family = "sans")}%>%
        gginnards::delete_layers(., "GeomText") +
        theme(axis.title.x = element_blank(),
              axis.title.y = element_blank()) +
        #ylab("Species richness") +
        #xlab("Reads") +
        guides(fill = guide_legend(nrow = 1),
               color = guide_legend(nrow = 1))
 
annotate_figure(ggarrange(rare_figure_microbe +
        labs(tag = "A"),
        rare_figure_virus +
        labs(tag = "B"),
        common.legend = T, ncol = 1),
                left = text_grob("Species richness",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
        bottom = text_grob("Microbial reads", rot = 0, family = "sans", size = 11)
                )


png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS17.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 160, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

annotate_figure(ggarrange(rare_figure_microbe +
        labs(tag = "A"),
        rare_figure_virus +
        labs(tag = "B"),
        common.legend = T, ncol = 1),
                left = text_grob("Species richness",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
        bottom = text_grob("Microbial reads", rot = 0, family = "sans", size = 11)
                )

# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()



```

rare_microbe$rare_virus_1000

```{r}
plot_rarefaction_curves <- function(otu_table, metadata, group_vars, step = 1, summarize = FALSE) {
  # Create grouping variable
  metadata$Group <- interaction(metadata[, group_vars], drop = TRUE)
  
  # Run rarecurve
  rare <- rarecurve(otu_table, step = step, sample = NULL, label = FALSE)
  
  # Extract data
  rare_df <- rarecurve_to_df(rare)
  
  # Merge with metadata
  rare_df <- left_join(rare_df, metadata, by = "SampleID")
  
  if (summarize) {
    # Summarize data
    summary_df <- rare_df %>%
      group_by(Group, Sequences) %>%
      summarize(
        MeanSpecies = mean(Species),
        SDSpecies = sd(Species),
        .groups = 'drop'
      )
    
    # Plot summarized data
    p <- ggplot(summary_df, aes(x = Sequences, y = MeanSpecies, color = Group)) +
      geom_line(size = 1) +
      geom_ribbon(aes(ymin = MeanSpecies - SDSpecies, ymax = MeanSpecies + SDSpecies, fill = Group), alpha = 0.2) +
      labs(
        x = "Number of Sequences",
        y = "Mean Species Richness",
        title = "Mean Rarefaction Curves with Standard Deviation"
      ) +
      theme_minimal() +
      theme(legend.title = element_blank())
  } else {
    # Plot individual curves
    p <- ggplot(rare_df, aes(x = Sequences, y = Species, color = Group, group = SampleID)) +
      geom_line(alpha = 0.7) +
      labs(
        x = "Number of Sequences",
        y = "Species Richness",
        title = "Rarefaction Curves Stratified by Group"
      ) +
      theme_minimal() +
      theme(legend.title = element_blank())
  }
  
  return(p)
}

# Usage
p <- plot_rarefaction_curves(
  otu_table = otu_table,
  metadata = metadata,
  group_vars = c("SampleType", "Treatment"),
  step = 10,
  summarize = TRUE
)
print(p)

```





# Bibliography

```{r warning=FALSE}
#===============================================================================
#BTC.LineZero.Footer.1.1.0
#===============================================================================
#R markdown citation generator.
#===============================================================================
#RLB.Dependencies:
#   magrittr, pacman, stringr
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#BTC.Dependencies:
#   LineZero.Header
#===============================================================================
#Generates citations for each explicitly loaded library.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
str_libraries <- c("r", str_libraries)
for (str_libraries in str_libraries) {
    str_libraries |>
        pacman::p_citation() |>
        print(bibtex = FALSE) |>sav
        capture.output() %>%
        .[-1:-3] %>% .[. != ""] |>
        stringr::str_squish() |>
        stringr::str_replace("_", "") |>
        cat()
    cat("\n")
}
#===============================================================================
```
