---
title: "COD_20230831_HOST_3_analysis_figure_table"
author: "Minsik Kim"
date: "2023-09-25"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
        df_print: paged
editor_options: 
  chunk_output_type: inline
  markdown: 
    wrap: 72
---

# Loading packages

```{r warning=FALSE, message=FALSE, setup}
#===============================================================================
#BTC.LineZero.Header.1.1.0
#===============================================================================
#R Markdown environment setup and reporting utility.
#===============================================================================
#RLB.Dependencies:
#   knitr, magrittr, pacman, rio, rmarkdown, rmdformats, tibble, yaml
#===============================================================================
#Input for document parameters, libraries, file paths, and options.
#=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=
knitr::opts_chunk$set(message=FALSE, warning = FALSE)



path_working <- 
        ifelse(sessionInfo()[1]$R.version$platform == "x86_64-pc-linux-gnu",
               "/home/bagel/minsik/",
               ifelse(sessionInfo()[1]$R.version$platform == "aarch64-apple-darwin20",
                      "/Volumes/macdrive/Dropbox/", 
                      "/Dropbox (Personal)"))

path_library <- 
        ifelse(sessionInfo()[1]$R.version$platform == "x86_64-pc-linux-gnu",
               "/home/bagel/R/x86_64-pc-linux-gnu-library/4.1/",
               "/Library/Frameworks/R.framework/Resources/library/")

str_libraries <- c("readxl", "phyloseq", "tidyverse", "pacman", "yaml", "ggplot2", "vegan", "microbiome", "ggpubr", "viridis", "decontam", "gridExtra", "ggpubr", "lme4", "lmerTest", "writexl", "harrietr", "Maaslin2", "ggtext", "ggpmisc", "gamm4", "reshape2", "kableExtra", "knitr", "ggtree", "car", "mediation", "lemon", "qvalue")
        
YAML_header <-
'---
title: "Host-DNA depletion analysis"
author: "Minsik Kim"
date: "2032.09.25"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
---'
seed <- "20230925"

#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Loads libraries, file paths, and other document options.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
FUN.LineZero.Boot <- function() {
    .libPaths(path_library)

    require(pacman)
    pacman::p_load(c("knitr", "rmarkdown", "rmdformats", "yaml"))

    knitr::opts_knit$set(root.dir = path_working)

    str_libraries |> unique() |> sort() -> str_libraries
    pacman::p_load(char = str_libraries)

    set.seed(seed)
}
FUN.LineZero.Boot()
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Outputs R environment report.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
FUN.LineZero.Report <- function() {
    cat("Line Zero Environment:\n\n")
    paste("R:", pacman::p_version(), "\n") |> cat()
    cat("Libraries:\n")
    for (str_libraries in str_libraries) {
        paste(
            "    ", str_libraries, ": ", pacman::p_version(package = str_libraries),
            "\n", sep = ""
        ) |> cat()
    }
    paste("\nOperating System:", pacman::p_detectOS(), "\n") |> cat()
    paste("    Library Path:", path_library, "\n") |> cat()
    paste("    Working Path:", path_working, "\n") |> cat()
    paste("Seed:", seed, "\n\n") |> cat()
    cat("YAML Header:\n")
    cat(YAML_header)
}
FUN.LineZero.Report()


```



# 1. Loading data

1.1. phyloseq obejct

1.2. qPCR data (controls)

## LIST OF PRIMARY QUESTIONS AND CORRESPONDING ANALYSES {.tabset}

STUDY AIMS

Aim 1. What is the efficiency of host depletion for each method?
        
        •	% host DNA measured by mNGS
        o	Sequencing failure rates
        •	% host DNA measured by qPCR 

Aim 2. Did host depletion change microbial community composition?

        •	Alpha diversity (microbial species richness, microbial predicted functional richness)
        •	Beta diversity (Morisita-Horn distance compared to control (not host-depleted) sample)
        •	Differential abundance

Aim 3: Is there effect modification by sample type?

Aim 4. Does host depletion increase the risk of contamination?



# Aim 1: What is the efficiency of host depletion for each method?

## 1a. Did treatment change % host DNA?

## Figure

        A: Raw reads (do you really need to log10 transform y-axis?)
        B: Host mapped reads
        C: Final reads (QC’d, non-human)
        D: % Host DNA 

## Statistical model

        ### linear mixed effects model to account for repeated measures (multiple aliquots per individual)
        ### Outcome = % host DNA (mNGS reads mapping to human genome/QC’d reads)
        ### Predictors
                Host depletion method (categorical, comparison group = control not host-depleted)
                Sample type
        ### testing for interaction term to justify stratified analysis
        ###  % host DNA ~ method + sample_type + method*sample_type + (1|subjid), report interaction p-value.
        ### Stratified analysis
        ###  %host ~ method + (1|subjid), report beta [95% CI], p-value for each sample type
        
## 1b. Did host depletion work successfully for sequencing?
        
## Statistical Model 
        
        ### Logistics mixed effects model
                ### Outcome = library prep and sequencing failure rate
                ### (==1 if failed library prep, ==0 if not)
                ### Report in text sequencing failure (n) stratified by sample type and treatment method.
        ## testing for interaction term to justify stratified analysis
                ### fail ~ method + sample_type + method*sample_type + (1|subjid), report interaction p-value.
        ### stratified analysis: 
                ### fail ~ method + (1|subjid), report OR [95% CI], p-value for each sample type

## Final reads

        ### Investigate distribution of final reads and apply proper transformation.

### Figure 

        A: Raw reads
        B: Host mapped reads
        C: Final reads
        D: Host DNA ratio
        
### model (to justify stratified analysis)
        
        ### final reads ~ method + sample_type + method*sample_type + (1|subjid), just report interaction p-value.

### model (stratified analysis)

        # final reads ~ method + (1|subjid), report effect size [95% CI], p-value for each sample type.
        
# Aim 2. Did host depletion change microbial community composition?

## Figure 

        # Alpha diversity.  Species richness by treatment, facetted by sample type

        # Beta diversity. Morisita-Horn distance of each treated compared to untreated sample  
        # forest plot - x-axis as distance (size of bias) and y axis as treatment. Facet data by sample type

## Alpha diversity

        # Linear mixed effects model
                ## Outcome 
                ### species richness
                ## inverse simpson
        # stratified analyses by sample type
                ## report the significant method*sample_type interaction term to justify stratified analysis)
                ## species_richness ~ method + sample_type + method * sample_type + (1|subjid)
                ## InvSimp~ sample type + treatment + sample type * treatment + (1|subject_id) 

## Beta diversity 
        
        ### Outcome = Morisita Horn 
        # PERMANOVA
                ## Overall: MH ~ sample type + treatment + sample type * treatment, strata = subject_id ) 
                ## Stratified: MH ~ method + strata = subject_id
        # Calculate Morisita-Horn distance between control and host depleted sample and use that as an outcome in linear model
                ## Change in MH ~ method + sample_type + method*sample_type + log10(reads))
## Differential abundance of species 

        # Outcome = relative abundance of species

## Figure

        # Volcano plot with Mock, BAL, Nasal and Sputum 
                ## Note: Mock community placed in Zymo DNA/RNA Protect prior to freezing. Zymo DNA/RNA protect is a mild detergent thus increases susceptibility of microbial cells to lysis and is not recommended to put these samples through host depletion 

## Statistical model
        
        # MaAsLin2
                ## Stratified by sample type
                        ### Species ~ + lyPMA + Benzonase + HostZERO + MolYsis + QIAamp + (1|subjid). Comparison group is untreated sample
                ## Figure: Balloon plots for BAL, Nasal and Sputum (Figure). Add q-val, mean relative abundance
        
## Proportion gram negative 
        
        # Species collapsed into single category: gram negative bacteria, gram positive bacteria, fungi
        # Statistical model
                ## % gram negative ~ sample type + treatment + sample type * treatment + (1|subject_id) ) 
                ## Stratified analysis
                        ### % gram negative ~ treatment + (1|subject_id) ) 

## Does change differ by sample type for results of predicted function as well? Repeat all analyses above analyses but use predicted microbial function (KEGG, CPM) instead

# Aim 3: Is there effect modification by sample type?
        
## All above analyses with interaction term and stratified analyses

## If effect modification present, which treatment is the best for each treatment? 

## Make a summary result by treatment, stratified by sample type

        ### Issues from sequencing (library failure, no change in host depletion, etc.)
        ### Changes in alpha diversity (Mean species richness change by each subject)
        ### Changes in beta diversity (statistical test results with significant factor)
        
# Secondary analyses
## 1.	Is qPCR an alternative to mNGS for estimating % Host DNA? 
### Figure
        
        ### A: Correlation Plot. x-axis with host DNA proportion measured with shotgun metagenomic sequencing vs y-axis as that by qPCR.
### Statistics
        ### Correlation coefficient
        ### Bland-Altman statistics

## 2. Mediation analysis. Host depletion treatment increases species and predicted microbial functional richness due higher effective sequencing depth 

        # Mediation R package
                ## outcome = species richness
                ## exposure = each host depletion treatment compared to untreated control
                ## mediator = final reads
                ## mediator-outcome confounders = sample type
                ## exposure-mediator confounders = NA
                ## outcome model = Mixed effects linear regression 
                ## mediator model = Mixed effects linear regression

# Aim 4. Does host depletion increase the risk of contamination?

1. Were there any contaminants in the sequencing result? If species richness were increased after treatment, is it due to increased coverage with higher final reads?

        # Run decontam (Davis 2018. PMID: 30558668)
                ## List potential contaminants with their prevalences in samples and negative controls 
                ## Sensitivity analysis where potential contaminant species identified removed
                        ### species richness  ~ treatment + (1|subjid)). 
        # Decontaminate data by estimating microbial population using mock community data.
                ## Run Tinyvamp  (arXiv: 2204.12733)
                ## Make adjusted relative abundance table by calculating taxon-specific detection efficiencies, relative to a reference taxa (Enterococcus).
                ## Known community for efficiency estimation: negative (no taxa) + positive (mock community
        

# Data inputs

*Meta data*

-   qPCR - bacteria

-   qPCR - human

-   qPCR host %

-   Raw reads

-   final reads

-   sequencing host %

-   library prep failure status

-   Raw reads

-   subject_id

-   treatment

-   sample_type

-   subject_id

*Sequencing result*

-   samples

-   controls

###  {-}

# Aanalysis preparation


## Analysis prep {.tabset}

### Loading data

```{r warning=F, message=FALSE, "Data loading"}
# Loading files -----------------------------------------------------------
#loading tidy phyloseq object
phyloseq_unfiltered <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20230521_MGK_host_tidy.rds")

v_phyloseq <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20240212_MGK_host_marker_magu_Jessica.rds")

sample_data <- sample_data(phyloseq_unfiltered$phyloseq_count)

```


```{r}
#Metagenote description was made as below
sample_data %>% 
        data.frame(check.names = F) %>% 
        dplyr::select(c("baylor_id", "baylor_other_id", "sample_type", "treated")) %>%
        mutate(nephele_description =
          case_when(
                  sample_type == "Mock" &
                    treated == 0 ~ "Control (untreated) positive mock community",
                  sample_type == "Mock" &
                    treated == 1 ~ "host DNA depleted positive mock community",
                  sample_type == "Neg." &
                    treated == 0 ~ "Control (untreated) negative reagent only controls",
                  sample_type == "Neg." &
                    treated == 1 ~ "host DNA depleted negative reagent only controls",
                  sample_type == "BAL" &
                    treated == 0 ~ "Control (untreated) BAL",
                  sample_type == "BAL" &
                    treated == 1 ~ "host DNA depleted BAL",
                  sample_type == "Nasal" &
                    treated == 0 ~ "Control (untreated) nasal swabs",
                  sample_type == "Nasal" &
                    treated == 1 ~ "host DNA depleted nasal swabs",
                  sample_type == "Sputum" &
                    treated == 0 ~ "Control (untreated) sputum",
                  sample_type == "Sputum" &
                    treated == 1 ~ "host DNA depleted sputum"
                  )
          ) %>%
        group_by(nephele_description) %>%
        mutate(rownum = row_number(nephele_description)) %>%
        mutate(nephele_description_numbered = 
                        paste(nephele_description,
                              rownum,
                              sep = "_")
                ) %>%
        write.csv("Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git/data/LOG_20230913_MGK_HOST_nephele_samplegroup.csv")

```


### Tinyvamp results

```{r warning=F, message=FALSE, "Data loading 2"}
# Loading files -----------------------------------------------------------
#loading tidy phyloseq object


tinyvamp_untreated <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/HOST_tinyvamp_decontaminated_Amy_Willis/20230807_amy/untreated_p_hats_all_v3.RDS") %>%
        t()

tinyvamp_lypma <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/HOST_tinyvamp_decontaminated_Amy_Willis/20230807_amy/lypma_p_hats_all_v3.RDS") %>%
        t()

tinyvamp_benzonase <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/HOST_tinyvamp_decontaminated_Amy_Willis/20230807_amy/benzonase_p_hats_all_v3.RDS") %>%
        t()

tinyvamp_host_zero <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/HOST_tinyvamp_decontaminated_Amy_Willis/20230807_amy/hostzero_p_hats_all_v3.RDS") %>%
        t()

tinyvamp_molysis <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/HOST_tinyvamp_decontaminated_Amy_Willis/20230807_amy/molysis_p_hats_all_v3.RDS") %>% 
        t()

tinyvamp_qiaamp <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/HOST_tinyvamp_decontaminated_Amy_Willis/20230807_amy/qiaamp_p_hats_all_v3.RDS") %>% 
        t()

        
#tinyvamp_benzonase <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/HOST_tinyvamp_decontaminated_Amy_Willis/benzonase_p_hats.RDS")

sample_data_tv <- phyloseq_unfiltered$phyloseq_rel %>%
        subset_samples(sample_type %in% c("BAL", "Nasal", "Sputum")) %>%
        sample_data %>% data.frame() %>%
        mutate(treatment = case_when(treatment == "MolYsis" ~ "Molysis",
                                     treatment == "HostZERO" ~ "Host zero",
                                     .default = treatment)) %>%
        mutate("names"= paste(.$original_sample, .$treatment, sep = "_")) %>%
        remove_rownames() %>% column_to_rownames("names") %>% sample_data()

phyloseq_tv <- merge_phyloseq(
        merge_phyloseq(sample_data_tv,
                       otu_table(tinyvamp_untreated,
                                 taxa_are_rows = T)),
        merge_phyloseq(sample_data_tv,
                       otu_table(tinyvamp_lypma,
                                 taxa_are_rows = T)),
        merge_phyloseq(sample_data_tv,
                       otu_table(tinyvamp_benzonase,
                                 taxa_are_rows = T)),
        merge_phyloseq(sample_data_tv,
                       otu_table(tinyvamp_host_zero,
                                 taxa_are_rows = T)),
        merge_phyloseq(sample_data_tv,
                       otu_table(tinyvamp_molysis,
                                 taxa_are_rows = T)),
        merge_phyloseq(sample_data_tv,
                       otu_table(tinyvamp_qiaamp,
                                 taxa_are_rows = T)),
        tax_table(phyloseq_unfiltered$phyloseq_rel)
) 

sample_data(phyloseq_tv)$treatment <-
        phyloseq_tv %>%
        sample_data() %>%
        data.frame(check.names = F) %>% 
        mutate(treatment =
                       case_when(
                               treatment == 
                                       "Molysis"~ 
                                       "MolYsis",
                               treatment == 
                                       "Host zero" ~
                                       "HostZERO", 
                               .default = treatment)) %>%
        .$treatment %>%
        factor(., levels = c("Untreated", 
                             "lyPMA",
                             "Benzonase",
                             "HostZERO",
                             "MolYsis",
                             "QIAamp"))
```


### Alpha diversity indices

```{r}           

alpha_diversity <- function(data) {
        otu_table <- otu_table(data) #%>% .[, colSums(.) !=0]
        S.obs <- rowSums(t(otu_table) != 0)
        sample_data <- sample_data(data)
        data_evenness <- vegan::diversity(t(otu_table)) / log(vegan::specnumber(t(otu_table))) # calculate evenness index using vegan package
        data_shannon <- vegan::diversity(t(otu_table), index = "shannon") # calculate Shannon index using vegan package
        data_hill <- exp(data_shannon)                           # calculate Hills index
        data_dominance <- microbiome::dominance(otu_table, index = "all", rank = 1, aggregate = TRUE) # dominance (Berger-Parker index), etc.
        data_invsimpson <- vegan::diversity(t(otu_table), index = "invsimpson")                          # calculate Shannon index using vegan package
        alpha_diversity <- cbind(S.obs, data_shannon, data_hill, data_invsimpson, data_evenness,data_dominance) # combine all indices in one data table
        sample_data <- merge(data.frame(sample_data), alpha_diversity, by = 0, all = T) %>% column_to_rownames(var = "Row.names")
}
sample_data(v_phyloseq$viral_rel) <- sample_data(alpha_diversity(v_phyloseq$viral_rel))

sample_data(phyloseq_unfiltered$phyloseq_count)$V.obs <- 
        sample_data(v_phyloseq$viral_rel) %>% 
        data.frame %>% 
        dplyr::select("S.obs") %>% 
#        mutate(S.obs = case_when(is.na(S.obs) ~ 0,
#                                 .default = S.obs)) %>%
        .[sample_names(phyloseq_unfiltered$phyloseq_count),]

sample_data(phyloseq_unfiltered$phyloseq_rel)$V.obs <- 
        sample_data(v_phyloseq$viral_rel) %>% 
        data.frame %>% 
        dplyr::select("S.obs") %>% 
#        mutate(S.obs = case_when(is.na(S.obs) ~ 0,
#                                 .default = S.obs)) %>%
        .[sample_names(phyloseq_unfiltered$phyloseq_count),]

sample_data(phyloseq_unfiltered$phyloseq_path_rpk)$V.obs <- 
        sample_data(v_phyloseq$viral_rel) %>% 
        data.frame %>% 
        dplyr::select("S.obs") %>% 
#        mutate(S.obs = case_when(is.na(S.obs) ~ 0,
#                                 .default = S.obs)) %>%
        .[sample_names(phyloseq_unfiltered$phyloseq_count),]

phyloseq <- phyloseq_unfiltered



alpha_diversity <- function(data) {
        otu_table <- otu_table(data) # %>% .[, colSums(.) !=0]
        S.obs <- rowSums(t(otu_table) != 0)
        sample_data <- sample_data(data)
        data_evenness <- vegan::diversity(t(otu_table)) / log(vegan::specnumber(t(otu_table))) # calculate evenness index using vegan package
        data_shannon <- vegan::diversity(t(otu_table), index = "shannon") # calculate Shannon index using vegan package
        data_hill <- exp(data_shannon)                           # calculate Hills index
        data_dominance <- microbiome::dominance(otu_table, index = "all", rank = 1, aggregate = TRUE) # dominance (Berger-Parker index), etc.
        data_invsimpson <- vegan::diversity(t(otu_table), index = "invsimpson")                          # calculate Shannon index using vegan package
        alpha_diversity <- cbind(S.obs, data_shannon, data_hill, data_invsimpson, data_evenness,data_dominance) # combine all indices in one data table
        sample_data <- merge(data.frame(sample_data), alpha_diversity, by = 0, all = T) %>% column_to_rownames(var = "Row.names")
}



sample_data(phyloseq_unfiltered$phyloseq_rel) <- sample_data(alpha_diversity(phyloseq_unfiltered$phyloseq_rel))
sample_data(phyloseq_unfiltered$phyloseq_count) <- sample_data(alpha_diversity(phyloseq_unfiltered$phyloseq_count)) 
sample_data(phyloseq_unfiltered$phyloseq_path_rpk) <- sample_data(alpha_diversity(phyloseq_unfiltered$phyloseq_path_rpk))  



        

```

## {-}

# **3.1. Screening of treatment effect**

# *i.	Did treatment change host % in qPCR results? *

## qPCR and seqeuncing {.tabset}                

### Fig. S1. qPCR figure

**Figure S1**. Host depletion effects measured by qPCR. (A) total DNA (16S bacterial DNA + human), (B) host DNA, (C) bacterial DNA, and (D) proportion of host DNA.

```{r}
#1A: Change in total DNA (qPCR)

fS1a <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(DNA_host_nondil + DNA_bac_nondil))) +
        geom_jitter(aes(color = treatment, x = treatment), lwd = 0.2,
                    alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment, x = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Treatment")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              axis.text.x = element_blank()) +
        facet_wrap(~sample_type, scale = "free_x") +
        ylab("log<sub>10</sub>(qPCR total DNA)<br>(ng/μL)") +
        labs(tag = "A") +
        guides(fill = guide_legend(nrow = 1))



#1B: Change in human DNA (qPCR)
fS1b <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(DNA_host_nondil))) +
        geom_jitter(aes(col = treatment, x = treatment), 
                    lwd = 0.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment, x = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Treatment")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              axis.text.x = element_blank()) +
        facet_wrap(~sample_type, scale = "free_x") +
        ylab("log<sub>10</sub>(qPCR host DNA)<br>(ng/μL)") +
        labs(tag = "B")

#1C: Change in 16S DNA (qPCR)
fS1c <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(DNA_bac_nondil))) +
        geom_jitter(aes(col = treatment, x = treatment),
                    lwd = 0.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment, x = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Treatment")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              axis.text.x = element_blank()) +
        facet_wrap(~sample_type, scale = "free_x") +
        ylab("log<sub>10</sub>(qPCR bacterial DNA)<br>(ng/μL)") +
        labs(tag = "C")
        
#1D. Change in % host (qPCR)
fS1d <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = host_proportion)) +
        geom_jitter(aes(col = treatment, x = treatment),
                    lwd = 0.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment, x = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Treatment")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              axis.text.x = element_blank()) +
        facet_wrap(~sample_type, scale = "free_x") +
        ylab("Host DNA ratio") +
        labs(tag = "D")


#output for markdown
figureS1 <- ggarrange(fS1a, fS1b, fS1c, fS1d, common.legend = T , align = "hv")

figureS1 



png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS1.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 170, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figureS1
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()


```

### Fig. S2. numbers

Peggy's comment on 20240912:

`I think you need to distinguish between the host depletion negative (stratified by each method ie lyPMA, benzonase, etc) and sequencing negative controls.`

`revise Table C12 in your R2R so that besides the row for negative control, also stratify by the type of negative control`

`create a stacked barplot where you facet by type of negative control`

Table updated- negative controls stratified by sample type

```{r warning=F}

table1_mock_and_neg <- sample_data(phyloseq$phyloseq_count) %>% data.frame() %>% 
        #dplyr::filter(sample_type %in% c("Sputum", "Nasal", "BAL")) %>% 
        # mutate(S.obs = case_when(S.obs==0 ~ NA,
        #                          .default = S.obs),
        #        F.obs = case_when(F.obs==0 ~ NA,
        #                          .default = F.obs),
        #        V.obs = case_when(V.obs==0 ~ NA,
        #                          .default = V.obs)) %>% 
        mutate(sample_type = case_when(sample_type == "Neg." & treatment == "Untreated" ~ "Neg. (extraction)",
                                       sample_type == "Neg." & treatment == "lyPMA" ~ "Neg. (lyPMA)",
                                       sample_type == "Neg." & treatment == "Benzonase" ~ "Neg. (Benzonase)",
                                       sample_type == "Neg." & treatment == "HostZERO" ~ "Neg. (HostZERO)",
                                       sample_type == "Neg." & treatment == "MolYsis" ~ "Neg. (MolYsis)",
                                       sample_type == "Neg." & treatment == "QIAamp" ~ "Neg. (QIAamp)",
                                       .default = sample_type),
               sample_type = factor(sample_type, 
                                    levels = c("Neg. (extraction)",
                                               "Neg. (lyPMA)",
                                               "Neg. (Benzonase)",
                                               "Neg. (HostZERO)",
                                               "Neg. (MolYsis)",
                                               "Neg. (QIAamp)",
                                               "Mock",
                                               "BAL",
                                               "Nasal",
                                               "Sputum"))) %>%
        group_by (sample_type) %>%
        summarise(`N` = n(),
            #      `Total DNA <br>ng/µL` = paste(format(round(median(picogreen_ng_ul),2), nsmall = 2, big.mark = ","), "<br>(", format(round(quantile(picogreen_ng_ul, 0.25),2), nsmall = 2, big.mark = ","), ", ", format(round(quantile(picogreen_ng_ul, 0.75),2), nsmall = 2, big.mark = ","), ")", sep = ""),
               `Human DNA <br>pg/µL` = paste(format(round(median(DNA_host_ng_uL*1000),1), nsmall = 1, big.mark = ","), " (", format(round(quantile(DNA_host_ng_uL*1000, 0.25),1), nsmall = 1, big.mark = ","), ", ", format(round(quantile(DNA_host_ng_uL*1000, 0.75),1), nsmall = 1, big.mark = ","), ")", sep = ""),
               `Bacterial DNA <br>pg/µL` = paste(format(round(median(DNA_bac_ng_uL*1000),1), nsmall = 1, big.mark = ","), " (", format(round(quantile(DNA_bac_ng_uL*1000, 0.25),1), nsmall = 1, big.mark = ","), ", ", format(round(quantile(DNA_bac_ng_uL*1000, 0.75),1), nsmall = 1, big.mark = ","), ")", sep = ""),
              `QC'd reads<br>reads x 10<sup>6</sup>` = paste(format(round(median(Reads_after_trim/1000000),1), nsmall = 1, big.mark = ","), " (", format(round(quantile(Reads_after_trim/1000000, 0.25),1), nsmall = 1, big.mark = ","), ", ", format(round(quantile(Reads_after_trim/1000000, 0.75),1), nsmall = 1, big.mark = ","), ")", sep = ""),
              `Host reads<br>%` = paste(format(round(median(sequencing_host_prop*100),1),
                                                nsmall = 1, big.mark = ","),
                                         " (",
                                         format(round(quantile(sequencing_host_prop * 100,
                                                               0.25),
                                                      1),
                                                nsmall = 1,
                                                big.mark = ","), 
                                         ", ", 
                                         format(round(quantile(sequencing_host_prop * 100, 0.75),1), 
                                                nsmall = 1, 
                                                big.mark = ","), 
                                         ")", 
                                         sep = ""),
             `Final reads<br>reads x 10<sup>6</sup>` = paste(format(round(median(Final_reads/1000000,
                                                                                 na.rm = T),
                                                                          1),
                                                                    nsmall = 1, big.mark = ","),
                                                             " (", 
                                                             format(round(quantile(Final_reads/1000000,
                                                                                            0.25,
                                                                                            na.rm = T),
                                                                                   1),
                                                                             nsmall = 1,
                                                                             big.mark = ","),
                                                             ", ",
                                                             format(round(quantile(Final_reads/1000000, 
                                                                                   0.75, 
                                                                                   na.rm = T),
                                                                          1), 
                                                                    nsmall = 1, 
                                                                    big.mark = ","), 
                                                             ")",
                                                             sep = "")
              
            ) %>% 
        data.frame(check.names = F) %>% 
        arrange(sample_type) %>%
        rename(`Sample` = sample_type) %>%
        mutate_all(linebreak) %>% kbl(format = "html", escape = F) %>% kable_styling(full_width = 0, html_font = "sans")

table1_mock_and_neg

save_kable(table1_mock_and_neg, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/table1_mock_neg.html", self_contained = T)



```

### Fig. S2. sequencing results

**Fig. S2.** (A) final reads after removing low quality reads and host-mapped reads and (B) sum of MetaPhlAn mapped reads by sample type.        

```{r warning=F, QC1.5.0}

sample_data %>% data.frame %>% subset(., .$sample_type == "Neg.")

#how were the samples failed in library prep?
sample_data %>%
        data.frame() %>%
        mutate(sample_type = factor(.$sample_type, 
                                    labels = c("Negative control", "Mock", "BAL", "Nasal", "Sputum")))
        
figS2_raw_reads <- sample_data %>% data.frame %>% mutate(total_read = phyloseq_unfiltered$phyloseq_count %>% otu_table %>% colSums()) %>%
        mutate(sample_type = factor(.$sample_type, 
                                    labels = c("Negative control", "Mock", "BAL", "Nasal", "Sputum"))) %>%
        ggplot(aes(x = reorder(baylor_other_id, -Raw_reads),
                               y = Raw_reads/1000000,
                               col = sample_type)) +
                geom_point() +
                theme_classic(base_family = "sans") +
                theme(axis.title.y = element_markdown(), axis.text.x = element_blank()) +
                ylab("Raw reads X 10<sup>6</sup>") +
                xlab("Samples") +
        guides(col=guide_legend(title="Sample type")) +
        scale_color_brewer(type = "qual", palette = 6) +
        labs(tag = "A") #+
        #ylim(c(0, 8.5))

figS2_raw_reads_log <- sample_data %>% data.frame %>% mutate(total_read = phyloseq_unfiltered$phyloseq_count %>% otu_table %>% colSums()) %>%
        mutate(sample_type = factor(.$sample_type, 
                                    labels = c("Negative control", "Mock", "BAL", "Nasal", "Sputum"))) %>%
        ggplot(aes(x = reorder(baylor_other_id, -Raw_reads),
                               y = log10(Raw_reads+1),
                               col = sample_type)) +
                geom_point() +
                theme_classic(base_family = "sans") +
                theme(axis.title.y = element_markdown(), axis.text.x = element_blank()) +
                ylab("log<sub>10</sub>(raw reads)") +
                xlab("Samples") +
        guides(col=guide_legend(title="Sample type")) +
        scale_color_brewer(type = "qual", palette = 6) +
        labs(tag = "A") #+
        #ylim(c(0, 8.5))

figS2_final_reads <- sample_data %>% data.frame %>% mutate(total_read = phyloseq_unfiltered$phyloseq_count %>% otu_table %>% colSums()) %>%
        mutate(sample_type = factor(.$sample_type, 
                                    labels = c("Negative control", "Mock", "BAL", "Nasal", "Sputum"))) %>%
        ggplot(aes(x = reorder(baylor_other_id, -Final_reads),
                               y = Final_reads/1000000,
                               col = sample_type)) +
                geom_point() +
                theme_classic(base_family = "sans") +
                theme(axis.title.y = element_markdown(), axis.text.x = element_blank()) +
                ylab("Final reads X 10<sup>6</sup>") +
                xlab("Samples") +
        guides(col=guide_legend(title="Sample type")) +
        scale_color_brewer(type = "qual", palette = 6) +
        labs(tag = "B") #+
        #ylim(c(0, 8.5))


figS2_final_reads_log <- sample_data %>% data.frame %>% mutate(total_read = phyloseq_unfiltered$phyloseq_count %>% otu_table %>% colSums()) %>%
        mutate(sample_type = factor(.$sample_type, 
                                    labels = c("Negative control", "Mock", "BAL", "Nasal", "Sputum"))) %>%
        ggplot(aes(x = reorder(baylor_other_id, -Final_reads),
                               y = log10(Final_reads + 1),
                               col = sample_type)) +
                geom_point() +
                theme_classic(base_family = "sans") +
                theme(axis.title.y = element_markdown(), axis.text.x = element_blank()) +
                ylab("log<sub>10</sub>(final reads)") +
                xlab("Samples") +
        guides(col=guide_legend(title="Sample type")) +
        scale_color_brewer(type = "qual", palette = 6) +
        labs(tag = "B")


figS2_total_reads <- sample_data %>% data.frame %>% mutate(total_read = phyloseq_unfiltered$phyloseq_count %>% otu_table %>% colSums()) %>%
        mutate(sample_type = factor(.$sample_type, 
                                    labels = c("Negative control", "Mock", "BAL", "Nasal", "Sputum"))) %>%
        ggplot(aes(x = reorder(baylor_other_id, -total_read),
                               y = total_read/1000000,
                               col = sample_type)) +
                geom_point() +
                theme_classic(base_family = "sans") +
                theme(axis.title.y = element_markdown(), axis.text.x = element_blank()) +
                ylab("Microbial reads X 10<sup>6</sup>") +
                xlab("Samples") +
        guides(col=guide_legend(title="Sample type")) +
        scale_color_brewer(type = "qual", palette = 6) +
        labs(tag = "C")


figS2_total_reads_log <- sample_data %>% data.frame %>% mutate(total_read = phyloseq_unfiltered$phyloseq_count %>% otu_table %>% colSums()) %>%
        mutate(sample_type = factor(.$sample_type, 
                                    labels = c("Negative control", "Mock", "BAL", "Nasal", "Sputum"))) %>%
        ggplot(aes(x = reorder(baylor_other_id, -total_read),
                               y = log10(total_read + 1),
                               col = sample_type)) +
                geom_point() +
                theme_classic(base_family = "sans") +
                theme(axis.title.y = element_markdown(), axis.text.x = element_blank()) +
                ylab("log<sub>10</sub>(Microbial reads)") +
                xlab("Samples") +
        guides(col=guide_legend(title="Sample type")) +
        scale_color_brewer(type = "qual", palette = 6) +
        labs(tag = "C") 


figS2 <- ggarrange(figS2_raw_reads, figS2_final_reads, figS2_total_reads, ncol = 1, nrow = 3, common.legend = T)


figS2_log <- ggarrange(figS2_raw_reads_log, figS2_final_reads_log, figS2_total_reads_log, ncol = 1, nrow = 3, common.legend = T)

figS2

figS2_log

png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS2_revised.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 210, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figS2
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()


png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS2_revised_log.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 210, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figS2_log
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()



```

### {-}

# *ii.	How were changes in sequencing results?*        

## Changes in sequencing output {.tabset}

**Table 1 is generated later (after QCing to remove sequencing failed samples, etc.)**

## Library failure {.tabset}


### Failure crosstab

```{r}
sample_data <- sample_data(phyloseq_unfiltered$phyloseq_count) %>% 
		  data.frame() %>%
		  mutate(
		      sample_sums = sample_sums(phyloseq_unfiltered$phyloseq_count),
		      lib_failed = case_match(lib_failed, TRUE ~ 1, FALSE ~0),
		      seq_failed = ifelse(sample_sums == 0, 1, 0),
		      lib_seq_failed = ifelse(lib_failed ==1 | seq_failed == 1, 1, 0),
		      prop_host = round((Host_mapped/Reads_after_trim), 1)
		      )
		xtabs(lib_failed ~ sample_type + treatment, data = sample_data)
		xtabs(seq_failed ~ sample_type + treatment, data = sample_data)
		xtabs(lib_seq_failed ~ sample_type + treatment, data = sample_data)
```


### Library failure - OR (all samples)

**Stratified OR cannot be calculated, as some samples showed 0 library failure.**

Effect size, standard error (SE) and t-value at a statistical test on library prep failure rate using generalized linear mixed effect model. glm ( sequencing fail \~ sample_type + treatment + sample_type \*
treatment + (1|subject_id) )

```{r warning=F, ''}

gm1 <- glmer(lib_failed ~ sample_type + treatment + sample_type * treatment + (1|subject_id),
      data = sample_data %>% 
              subset(sample_type %in% c("BAL", "Nasal", "Sputum")) %>% 
              data.frame(check.names = F),
      family = binomial)

gm1 %>% summary %>% .$coefficients

```

None of associations were significant at binomial generalized mixed effect model. 

With glmer(lib_failed ~ sample_type + (1|subject_id)),


```{r}
glmer(lib_failed ~ sample_type + (1|subject_id),
      data = sample_data %>% 
              subset(sample_type %in% c("BAL", "Nasal", "Sputum")) %>% 
              data.frame(check.names = F),
      family = binomial) %>%
        summary
```

It seems like there were sample_type specific effects (since some of the sample * treatment did not shoy any failure).

### Library failure (stratified)

glm ( sequencing fail \~ treatment + subject_id )

*--> Cannot run for Sputum (no failed sample). *

For BAL 

GLMER result

```{r warning=F, ''}

glmer_libfail_bal <- glmer(lib_failed ~ treatment + (1|subject_id), data = sample_data %>% data.frame %>% dplyr::filter(sample_type %in% c("BAL")), family = "binomial")
glmer_libfail_bal %>% summary

```
No treatment significantly affected for library failure for BAL.

For nasal, GLMER result does not converge. As all the variables are categorical, we cannot rescale.

```{r, include=FALSE, eval=FALSE}

glmer_libfail_ns <- glmer(lib_failed ~ treatment + (1|subject_id),
                             data = sample_data %>%
                                     data.frame %>%
                                     dplyr::filter(sample_type %in% c("Nasal")),
                             family = "binomial") 
 

glmer_libfail_ns %>% summary


```

### {-}

## Host ratio {.tabset}

### Host ratio (all samples)

**None-stratified. p-value of interaction term was 4.587e-16**

```{r warning=F}


lmer_host_ratio_all <- lmer(sequencing_host_prop * 100 ~ sample_type * treatment + (1|subject_id), 
                    data = sample_data %>%
                            data.frame %>%
                            mutate(total_read = phyloseq_unfiltered$phyloseq_count %>%
                            otu_table %>%
                            colSums()) %>%
                            subset(., total_read != 0))
        
lmer_host_ratio_all
```

ANOVA result

```{r}
lmer_host_ratio_all %>% anova() 
```

### Table S1. Host ratio (stratified) 


**Table S1. ** Effect size, standard error (SE) and p-value at a statistical test on host DNA ratio using linear mixed effect model with treatment as fixed effect and subject as random effect using r package lmer::lmer( Host DNA % ~ treatment + (1|subject_id) ). Stratified analyses were conducted for each sample type as an interaction term of sample type and treatment was significant at an ANOVA test (p-value < 0.001) using a model, LMER (Host DNA % ~ sample type + treatment + sample type `*` treatment  + (1|subject_id) ). The baseline of categorical variable is untreated, and statistical significances were noted with `*`: p-value < 0.05 and `***`: p-value < 0.001.

**HostZERO and Molysius was effect to to all QIAamp was effective for Nasal swab and Sputum**

Stratified (BAL)

```{r warning=F}
hr_lmer_bal <- lmer(sequencing_host_prop * 100 ~ treatment + (1|subject_id), 
                    data = sample_data %>%
                            data.frame %>%
                            mutate(total_read = phyloseq_unfiltered$phyloseq_count %>%
                            otu_table %>%
                            colSums()) %>%
                            subset(., .$sample_type %in% c("BAL")))

hr_lmer_bal %>%
        summary()
```

Stratified (nasal)

```{r}

hr_lmer_ns <- lmer(sequencing_host_prop * 100 ~ treatment + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Nasal")))

hr_lmer_ns %>% summary
```

Stratified (sputum)

```{r}
hr_lmer_spt <- lmer(sequencing_host_prop * 100 ~ treatment + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Sputum")))

hr_lmer_spt %>% summary
```

Merged table

```{r warning=F}

hr_lmer_bal_kbl <- hr_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              hr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]

hr_lmer_ns_kbl <- hr_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              hr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        



hr_lmer_spt_kbl <- hr_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              hr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        


tables1 <- cbind(hr_lmer_bal_kbl, hr_lmer_ns_kbl, hr_lmer_spt_kbl) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")
tables1
save_kable(tables1, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS1.html", self_contained = T)


```

### {-}


## Final reads {.tabset}

### Final reads (all samples)

**None-stratified. p-value of interaction term was 3.955e-09**

Final reads output

```{r warning=F}

lmer_final_reads <- lmer(Final_reads * 100 ~ sample_type * treatment + (1|subject_id), 
                    data = sample_data %>%
                            data.frame %>%
                            mutate(total_read = phyloseq_unfiltered$phyloseq_count %>%
                            otu_table %>%
                            colSums()) %>%
                            subset(., total_read != 0))

lmer_final_reads %>% summary
```

ANOVA on lmer result

```{r warning=F}

lmer_final_reads %>% anova() 
        
```


### Table S2 Final reads (stratified)

lmer( Host DNA ratio ~ treatment + (1\|subject_id) )

**Table S2.** Changes on final reads stratified by sample type tested with linear mixed effect models using r package lmer::lmer(log10(Final reads) ~ Treatment + (1|Subject id)). Stratified analyses were conducted for each sample type as an interaction term of sample type and treatment was significant at an ANOVA test (p-value < 0.001) using a model, LMER (log10(Final reads) ~ sample type + treatment + sample type * treatment  + (1|subject_id) ). Effect size with adjusted 95% confidence intervals and p-value were listed. The unit of final read is reads x 106. Statistical significances were noted with `*`: p-value < 0.05, `**`: p-value < 0.01 and `***`: p-value < 0.001.


**Sputum's final read increased after every treatment. Nasal swab showed improved reads with lyPMA, HostZERO and QIAamp. BAL also showed increased reads with most of treatment.**

Raw results, BAL, nasal and Sputum, respectively

```{r warning=F}

fr_lmer_bal <- lmer(log10(Final_reads/1000000) ~ treatment + (1|subject_id),
                    data = sample_data %>%
                                data.frame %>%
                                mutate(total_read = phyloseq_unfiltered$phyloseq_count %>%
                                otu_table %>%
                                colSums()) %>%
             subset(., .$sample_type %in% c("BAL")))


fr_lmer_bal %>% 
        summary()


fr_lmer_ns <- lmer(log10(Final_reads/1000000) ~ treatment + (1|subject_id),
                    data = sample_data %>%
                                data.frame %>%
                                mutate(total_read = phyloseq_unfiltered$phyloseq_count %>%
                                otu_table %>%
                                colSums()) %>%
             subset(., .$sample_type %in% c("Nasal")))


fr_lmer_ns %>% 
        summary()


fr_lmer_spt <- lmer(log10(Final_reads/1000000) ~ treatment + (1|subject_id),
                    data = sample_data %>%
                                data.frame %>%
                                mutate(total_read = phyloseq_unfiltered$phyloseq_count %>%
                                otu_table %>%
                                colSums()) %>%
             subset(., .$sample_type %in% c("Sputum"))) 


fr_lmer_spt %>% 
        summary()


```

Tidified output

```{r warning=F}

fr_lmer_bal_kbl <- fr_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              fr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        
fr_lmer_ns_kbl <- fr_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              fr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


fr_lmer_spt_kbl <- fr_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              fr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


tables2 <- cbind(fr_lmer_bal_kbl, fr_lmer_ns_kbl, fr_lmer_spt_kbl) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")
tables2

save_kable(tables2, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS2.html", self_contained = T)

```
### {-}


## Figure of sequencing result {.tabset}

### Fig. S3. Figure of sequencing result

*ii.	How were changes in sequencing results?*

 **Fig. S3.**  Host depletion effects measured by shotgun metagenomic sequencing. (A) Raw DNA reads, (B) host mapped reads by bowtie2, (C) final reads of microbes, and (D) proportion of host mapped among total mapped reads.

```{r}

fs3a <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(Raw_reads))) +
        geom_jitter(aes(col = treatment, x = treatment),
                    lwd = 0.2, alpha = 0.3, stroke = 0)+
        stat_summary(aes(color = treatment, x = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Treatment")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              axis.text.x = element_blank()) +
        ylab("log<sub>10</sub>(raw reads)") +
        labs(tag = "A") +
        facet_wrap(~sample_type, scale = "free_x") +
        guides(fill = guide_legend(nrow = 1))

#	- Host_mapped


fs3b <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(Host_mapped))) +
        geom_jitter(aes(col = treatment, x = treatment),
                    lwd = 0.2, alpha = 0.3, stroke = 0)+
        stat_summary(aes(color = treatment, x = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Treatment")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              axis.text.x = element_blank()) +
        ylab("log<sub>10</sub>(host reads)") +
        labs(tag = "B") +
        facet_wrap(~sample_type, scale = "free_x") +
        guides(fill = guide_legend(nrow = 1))







#	- % Host (we have used Host_mapped/Raw_reads in prior papers)

#	- Final_reads

fs3c <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(Final_reads))) +
        geom_jitter(aes(col = treatment, x = treatment),
                    lwd = 0.2, alpha = 0.3, stroke = 0)+
        stat_summary(aes(color = treatment, x = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Treatment")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              axis.text.x = element_blank()) +
        ylab("log<sub>10</sub>(final reads)") +
        labs(tag = "C") +
        facet_wrap(~sample_type, scale = "free_x") +
        guides(fill = guide_legend(nrow = 1))





#	- % Host (we have used Host_mapped/Raw_reads in prior papers)
fs3d <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = sequencing_host_prop)) +
        geom_jitter(aes(col = treatment, x = treatment),
                    lwd = 0.2, alpha = 0.3, stroke = 0)+
        #geom_linerange(aes(xmin=lower.ci, xmax=upper.ci)) +
        stat_summary(aes(color = treatment, x = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Treatment")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              axis.text.x = element_blank()) +
        
        ylab("Host ratio by sequencing") +
        labs(tag = "D") +
        facet_wrap(~sample_type, scale = "free_x") +
        guides(fill = guide_legend(nrow = 1))



figS3 <- ggarrange(fs3a, fs3b, fs3c, fs3d, common.legend = T, align = "hv")

figS3


png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS3.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 170, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figS3
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()



```

### {-}


# *iii.	Was host DNA proportion of sequencing similar to that of qPCR? (secondary analysis)*

## Fig. S4. Host ratio qPCR vs sequencing
        
Peggy's comment: metagenomics is gold standard for % host, but most people don't have the money to do deep sequencing. So secondary analysis is to calculate correlation between %host by qPCR vs %host by metagenomics (this can be a supplementary figure but would at least mention correlation in text)

**Fig. S4. ** (A) Correlation plot and (B) Bland-Altman plot between host DNA proportion measured with qPCR and shotgun metagenomic sequencing.

```{r, message=FALSE}

figS4a <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = host_proportion, y = sequencing_host_prop, col = sample_type)) +
        geom_point() +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#33a02c", "#1f78b4")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        #scale_x_discrete(name ="Sample type")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              legend.position = "top") +
        ylab("% Host DNA (mNGS)") +
        xlab("% Host DNA (qPCR)") +
        labs(col = "Sample type") +
        annotate(family = "sans",
                 geom='richtext',
                 x=0.5, y=1,
                 label = paste("R<sup>2</sup> = ",
                               lm(sequencing_host_prop ~ host_proportion,
                                  data = sample_data %>%
                                          data.frame %>% 
                                          subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")))
                                  %>% summary %>% .$r.squared %>% round(., 2) %>% format(nsmall = 2), sep = "")) +
        geom_smooth(method=lm , color="red", se=T, level = 0.95) +
        guides(fill = guide_legend(nrow = 1)) +
        labs(tag = "A")


bland_altman_data <- sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")) %>% data.frame %>%
        mutate(avg_two = (host_proportion + sequencing_host_prop)/2,
               diff_two = sequencing_host_prop - host_proportion)
        
figS4b <- ggplot(bland_altman_data, aes(x = avg_two, y = diff_two, col = sample_type)) +
        geom_point() +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#33a02c", "#1f78b4")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        #scale_x_discrete(name ="Sample type")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15)) +
        ylab("Difference between qPCR and mNGS") +
        xlab("% Host DNA (mean)") +
        geom_hline(yintercept = mean(bland_altman_data$diff_two), colour = "black", size = 0.5) +
        geom_hline(yintercept = mean(bland_altman_data$diff_two) - (1.96 * sd(bland_altman_data$diff_two)), colour = "black", size = 0.5, linetype = "dashed") +
        geom_hline(yintercept = mean(bland_altman_data$diff_two) + (1.96 * sd(bland_altman_data$diff_two)), colour = "black", size = 0.5, linetype = "dashed") +
        labs(tag = "B")



figS4 <- ggarrange(figS4a, figS4b, common.legend = T, ncol = 1, align = "hv")

figS4

png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS4.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height =180, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figS4
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()



```

# **3.2. Quality control of sequencing data**

# *iv.	Were there any contaminants in the sequencing result? (Do these host depletion methods introduce contamination)*

## decontam {.tabset}

### Table S10. decontam - stratified by sample_type

**Table S10.** Summary table of potential contaminants and their prevalence across all samples (N = 113) and within negative controls (total N = 31). The contaminants were identified using decontam37 combined method (Fisher’s exact test result of prevalence and frequency method results), using 16S qPCR bacterial DNA concentration as total bacterial load at prevalence threshold of 0.1. The analyses were stratified by sample types.

```{r warning=FALSE, message=FALSE}
#Stratified by sample type
neg_number_decontam <- 
        phyloseq_unfiltered$phyloseq_count %>% 
        subset_samples(sample_type == "Neg.") %>%
        subset_samples(S.obs != 0) %>%
        sample_names() %>%
        length

sample_number_decontam <- 
        phyloseq_unfiltered$phyloseq_count %>% 
        subset_samples(sample_type != "Mock") %>%
        subset_samples(sample_type != "Neg.") %>%
        subset_samples(S.obs != 0) %>%
        sample_names() %>%
        length

prev_neg <- ((phyloseq_unfiltered$phyloseq_count %>% 
          subset_samples(sample_type == "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (negative controls)" = ".")

prev_all <- ((phyloseq_unfiltered$phyloseq_count %>%
                      subset_samples(sample_type != "Mock") %>%
                      subset_samples(sample_type != "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (all)" = ".")


sample_data(phyloseq_unfiltered$phyloseq_rel)$is.neg <- grepl("Neg", sample_data(phyloseq_unfiltered$phyloseq_rel)$sample_type)

phyloseq_decontam_bal <- phyloseq_unfiltered$phyloseq_rel %>% 
        subset_samples(S.obs != 0) %>%
        subset_samples(sample_type == "Neg." | sample_type == "BAL")
phyloseq_decontam_ns <- phyloseq_unfiltered$phyloseq_rel %>% 
        subset_samples(S.obs != 0) %>%
        subset_samples(sample_type == "Neg." | sample_type == "Nasal")
phyloseq_decontam_spt <- phyloseq_unfiltered$phyloseq_rel %>% 
        subset_samples(S.obs != 0) %>%
        subset_samples(sample_type == "Neg." | sample_type == "Sputum")


contaminant_combined_bal <- 
data.frame("BAL", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_bal, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) %>% row.names
)

contaminant_combined_ns <- 
data.frame("Nasal swab", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_ns, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) %>% row.names
)


contaminant_combined_spt <- 
data.frame("Sputum", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_spt, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) %>% row.names
)

contaminants <- rbind(contaminant_combined_bal, contaminant_combined_ns, contaminant_combined_spt)

names(contaminants) <- c("Sample type", "Taxa")


neg_number_decontam
sample_number_decontam

merged_contaminants <- merge(contaminants, prev_all %>% 
                                     rownames_to_column("Taxa"), by = "Taxa") %>%
        merge(., prev_neg %>%
                      rownames_to_column("Taxa"), by = "Taxa") %>%
       dplyr::select(c("Sample type",
                       "Taxa",
                       "Prevalence (all)",
                       "Prevalence (negative controls)")) %>%
        mutate(`Prevalence (all)` = paste0(`Prevalence (all)`,
                                           " (",
                                           round(`Prevalence (all)`/
                                                         sample_number_decontam * 100, 2),
                                           "%)"),
                `Prevalence (negative controls)` = paste0(`Prevalence (negative controls)`,
                                           " (",
                                           round(`Prevalence (negative controls)`/
                                                         neg_number_decontam * 100, 2),
                                           "%)")) %>%
        rename(`Prevalence (all) N = 93` = `Prevalence (all)`,
               `Prevalence (negative controls) N = 31` = `Prevalence (negative controls)`) %>%
        .[order(.$"Sample type", .$"Taxa"),] %>%
        remove_rownames()


species_italic2 <- function(data){
  data <- gsub("_", " ", data)
  data <- gsub("[]]|[[]", "", data)
  data <- gsub(" sp", " sp.", data)
  data <- gsub(" sp.", "</em> sp.", data)
  data <- gsub(" group", "", data)
  data <- ifelse(grepl("[*]", data), paste("<em>", data, sep = ""), paste("<em>", data, "</em>", sep = ""))
  data
}

```

### Decontam result

Stratified-BAL


```{r}

isContaminant(phyloseq_decontam_bal, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) 

```

Stratified-Nasal

```{r}

isContaminant(phyloseq_decontam_ns, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) 

```

Stratified-Sputum

```{r}

isContaminant(phyloseq_decontam_spt, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) 

```


### Decontam -stratified by treatment

```{r}
#Stratified by sample type

prev_neg <- ((phyloseq_unfiltered$phyloseq_count %>% 
          subset_samples(sample_type == "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (negative controls)" = ".")

prev_all <- ((phyloseq_unfiltered$phyloseq_count %>%
                      subset_samples(sample_type != "Mock") %>%
                      subset_samples(sample_type != "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (all)" = ".")


sample_data(phyloseq_unfiltered$phyloseq_rel)$is.neg <- grepl("Neg", sample_data(phyloseq_unfiltered$phyloseq_rel)$sample_type)

phyloseq_decontam_lypma <- phyloseq_unfiltered$phyloseq_rel %>% 
        subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "lyPMA")

phyloseq_decontam_ben <- phyloseq_unfiltered$phyloseq_rel %>% 
        subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "Benzonase")

phyloseq_decontam_hostzero <- phyloseq_unfiltered$phyloseq_rel %>% 
        subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "HostZero")

phyloseq_decontam_molysis <- phyloseq_unfiltered$phyloseq_rel %>% 
        subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "MolYsis")

phyloseq_decontam_qiaamp <- phyloseq_unfiltered$phyloseq_rel %>% 
        subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "QIAamp")


contaminant_combined_lypma <- 
data.frame("lyPMA", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_lypma, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) %>% row.names
)

contaminant_combined_ben <- 
data.frame("Benzonase", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_ben, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) %>% row.names
)


contaminant_combined_hostzero <- 
data.frame("HostZero", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_hostzero, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) %>% row.names
)


contaminant_combined_molysis <- 
data.frame("MolYsis", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_molysis, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) %>% row.names
)


contaminant_combined_QIAamp <- 
data.frame("QIAamp", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_qiaamp, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) %>% row.names
)

contaminants_strat_treat <- rbind(contaminant_combined_lypma,
                      contaminant_combined_ben, 
                      contaminant_combined_hostzero,
                      contaminant_combined_molysis,
                      contaminant_combined_QIAamp
                      )

names(contaminants_strat_treat) <- c("Treatment", "Taxa")



```

12 taxa identified as contaminants. 7 Taxa identified as contaminants (sample type stratified).

### Decontam summary

Summary table of potential contaminants with all sample types and stratified sample types in all methods (prevalence, frequence, and combined)

```{r}

tableS10 <- merged_contaminants %>% 
        mutate(Taxa = species_italic2(Taxa)
               ) %>%
        kbl(format = "html", escape = F) %>%
        kable_styling(full_width = 0, html_font = "sans") 
tableS10


save_kable(tableS10, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS10.html", self_contained = T)
```

### {-}


## Filtering taxa {.tabset}

### Prevalence filtering 

**Prevalence & abundance filtering was conducted**

v.	M&M - Prevalence filtering

        1.	Prevalence filtration at 5%, except its abundance is over 0.75 quantile.
                a.	Information in the main text

```{r warning=FALSE, "Red flag"}

phyloseq_unfiltered$phyloseq_rel <- transform_sample_counts(phyloseq_unfiltered$phyloseq_rel,
                                                            function(x){x/sum(x)})
taxa_qc <- data.frame("species" =
                              otu_table(
                                      subset_samples(
                                              phyloseq_unfiltered$phyloseq_rel,S.obs != 0 &
                                                      sample_type %in% c("Mock", "BAL", "Nasal", "Sputum"))) %>%
                              t() %>% colnames(),
                      "prevalence" =
                              ifelse(subset_samples(phyloseq_unfiltered$phyloseq_rel, S.obs != 0 & 
                                                            sample_type %in% c("Mock", "BAL", "Nasal", "Sputum")) %>%
                                             otu_table() > 0, 1, 0)%>% 
                              t() %>%
                              colSums(), #Prevalence of taxa
                      "mean_rel_abd" = 
                              subset_samples(phyloseq_unfiltered$phyloseq_rel,
                                             S.obs != 0 & sample_type %in% c("Mock", "BAL", "Nasal", "Sputum")) %>%
                              otu_table() %>%
                              t() %>%
                              colMeans(na.rm = T) #mean relativ abundacne 
)


function_qc <- data.frame("function" =
                                  otu_table(
                                          subset_samples(
                                                  phyloseq_unfiltered$phyloseq_path_rpk,
                                                           S.obs != 0 & 
                                                          sample_type %in% 
                                                          c("Mock", "BAL", "Nasal", "Sputum")
                                                  )
                                          ) %>% 
                                  t() %>% 
                                  colnames(),
                          "prevalence" = 
                                  ifelse(subset_samples(phyloseq_unfiltered$phyloseq_path_rpk,
                                                        S.obs != 0 & 
                                                                sample_type %in% 
                                                                c("Mock", "BAL", "Nasal", "Sputum")
                                                        ) %>% 
                                                 otu_table() > 0, 
                                         1, 
                                         0
                                         ) %>% 
                                  t() %>% 
                                  colSums(), #Prevalence of taxa
                          "mean_rpk" = 
                                  subset_samples(phyloseq_unfiltered$phyloseq_path_rpk, 
                                                 S.obs != 0 & 
                                                         sample_type %in% 
                                                         c("Mock", "BAL", "Nasal", "Sputum")
                                                 ) %>% 
                                  otu_table() %>% 
                                  t() %>% 
                                  colMeans(na.rm = T), #mean relativ abundacne 
                          unidentified = 
                                  ifelse((subset_samples(phyloseq_unfiltered$phyloseq_path_rpk,
                                              S.obs != 0 & sample_type %in% c("Mock", "BAL", "Nasal", "Sputum")) %>% 
                                       otu_table() > 0) %>%
                                      row.names() %in% c("UNMAPPED", "UNINTEGRATED")
                              , 1, 0)
)

red_flag_taxa <- data.frame(species = taxa_qc$species,
                            prevalence = taxa_qc$prevalence,
                            mean_rel_abd = taxa_qc$mean_rel_abd,
                            red_flag_prev_abd = 
                                    ifelse(taxa_qc$prevalence < 
                                                   otu_table(
                                                           subset_samples(
                                                                   phyloseq_unfiltered$phyloseq_rel,
                                                                   S.obs != 0 & 
                                                                           sample_type %in% 
                                                                           c("Mock", "BAL", "Nasal", "Sputum"))) %>%
                                                               t %>% rownames() %>%
                                                   length * 0.05 &
                                                   #Removing taxa with zero prevalence - taxa from nasal swabs
                                                   taxa_qc$mean_rel_abd <
                                                   taxa_qc %>%
                                                   subset(., .$prevalence != 0) %>%
                                                   .$mean_rel_abd %>%
                                                   quantile(., 0.75), 1,0),
                           red_flag_prev =
                                    ifelse(taxa_qc$prevalence < 
                                                   otu_table(
                                                           subset_samples(
                                                                   phyloseq_unfiltered$phyloseq_rel,
                                                                   S.obs != 0 & 
                                                                           sample_type %in% 
                                                                           c("Mock", "BAL", "Nasal", "Sputum"))) %>%
                                                               t %>% rownames() %>%
                                                   length * 0.05,
                                           1,
                                           0)) %>%
        mutate(red_flag_decontam = species %in% (contaminants$Taxa %>% unique()))

subset(red_flag_taxa, red_flag_taxa$red_flag_prev == 1 & red_flag_taxa$red_flag_prev_abd == 0)

#Unampped function were removed

red_flag_function <- 
        data.frame(function. = function_qc$function.,
                   prevalence = function_qc$prevalence,
                   mean_rel_abd = function_qc$mean_rpk,
                   red_flag_prev_abd = 
                           ifelse(function_qc$prevalence < 
                                          otu_table(
                                                  subset_samples(
                                                          phyloseq_unfiltered$phyloseq_path_rpk,
                                                          S.obs != 0 & 
                                                                  sample_type %in% 
                                                                  c("Mock", "BAL", "Nasal", "Sputum"))) %>%
                                          t %>% 
                                          rownames() %>%
                                          length * 0.05 &
                                          #Removing taxa with zero prevalence - taxa from nasal swabs
                                          function_qc$mean_rpk <
                                          function_qc %>%
                                          subset(., .$prevalence != 0) %>%
                                          .$mean_rpk %>%
                                          quantile(., 0.75), 1,0),
                   red_flag_prev =
                           ifelse(function_qc$prevalence <
                                          otu_table(
                                                  subset_samples(
                                                          phyloseq_unfiltered$phyloseq_path_rpk,
                                                          S.obs != 0 & 
                                                                  sample_type %in% 
                                                                  c("Mock", "BAL", "Nasal", "Sputum"))) %>%
                                          t %>% 
                                          rownames() %>%
                                          length * 0.05,
                                          1,
                                          0)) %>%
        mutate(red_flag_prev_abd = case_when(function. %in% c("UNMAPPED", "UNINTEGRATED") ~ 1,
                                     .default = red_flag_prev_abd))

subset(red_flag_function, red_flag_function$red_flag_prev == 1 & red_flag_function$red_flag_prev_abd == 0)



#decontaminated phyloseq 
phyloseq_decontam <- phyloseq
phyloseq_decontam$phyloseq_count <- prune_taxa(subset(red_flag_taxa,
                                           red_flag_taxa$red_flag_prev_abd != 1 &
                                                   !red_flag_taxa$red_flag_decontam)$species,
                                    phyloseq$phyloseq_count)

phyloseq_decontam$phyloseq_rel <- prune_taxa(subset(red_flag_taxa,
                                           red_flag_taxa$red_flag_prev_abd != 1 &
                                                   !red_flag_taxa$red_flag_decontam)$species,
                                    phyloseq$phyloseq_rel) %>%
        transform_sample_counts(., function(x){x/sum(x)})
        
#phyloseq for analysis
phyloseq$phyloseq_count <- prune_taxa(subset(red_flag_taxa,
                                           red_flag_taxa$red_flag_prev_abd != 1)$species,
                                    phyloseq$phyloseq_count)
phyloseq$phyloseq_rel <- prune_taxa(subset(red_flag_taxa,
                                           red_flag_taxa$red_flag_prev_abd != 1)$species,
                                    phyloseq$phyloseq_rel) %>%
        transform_sample_counts(., function(x){x/sum(x)})


phyloseq$phyloseq_path_rpk <- prune_taxa(subset(red_flag_function, red_flag_function$red_flag_prev_abd != 1)$function., phyloseq$phyloseq_path_rpk)

#phyloseq$tree_phyloseq_count <- prune_taxa(subset(red_flag_taxa,
                                           #red_flag_taxa$red_flag_prev_abd != 1 & !red_flag_taxa$red_flag_decontam_prev)$species,
                                    #phyloseq$tree_phyloseq_count)

#phyloseq$tree_phyloseq_rel <- prune_taxa(subset(red_flag_taxa,
                                           #red_flag_taxa$red_flag_prev_abd != 1 & !red_flag_taxa$red_flag_decontam_prev)$species,
                                    #phyloseq$tree_phyloseq_rel)

```

### Filtration result

Taxa were filtered wehn `red_flag_prev_abd == 1`.

```{r}
red_flag_taxa
```

### {-}

## Extras {.tabset}

### Alpha diversity calculation

```{r}

#Calculation of alpha diversity indices for filtered samples

alpha_diversity <- function(data) {
        otu_table <- otu_table(data) #%>% .[, colSums(.) !=0]
        S.obs <- rowSums(t(otu_table) != 0)
        sample_data <- sample_data(data)
        data_evenness <- vegan::diversity(t(otu_table)) / log(vegan::specnumber(t(otu_table))) # calculate evenness index using vegan package
        data_shannon <- vegan::diversity(t(otu_table), index = "shannon") # calculate Shannon index using vegan package
        data_hill <- exp(data_shannon)                           # calculate Hills index
        data_dominance <- microbiome::dominance(otu_table, index = "all", rank = 1, aggregate = TRUE) # dominance (Berger-Parker index), etc.
        data_invsimpson <- vegan::diversity(t(otu_table), index = "invsimpson")                          # calculate Shannon index using vegan package
        alpha_diversity <- cbind(S.obs, data_shannon, data_hill, data_invsimpson, data_evenness,data_dominance) # combine all indices in one data table
        sample_data <- merge(data.frame(sample_data), alpha_diversity, by = 0, all = T) %>% column_to_rownames(var = "Row.names")
}
#sample_data(phyloseq$phyloseq_count) <- sample_data(alpha_diversity(phyloseq$phyloseq_count))
sample_data(phyloseq$phyloseq_rel) <- sample_data(alpha_diversity(phyloseq$phyloseq_count))
sample_data(phyloseq$phyloseq_count) <-sample_data(alpha_diversity(phyloseq$phyloseq_count)) 
sample_data(phyloseq$phyloseq_path_rpk) <- sample_data(alpha_diversity(phyloseq$phyloseq_path_rpk))  

sample_data(phyloseq_tv) <- sample_data(alpha_diversity(phyloseq_tv))  

sample_data(phyloseq_decontam$phyloseq_rel) <- sample_data(alpha_diversity(phyloseq_decontam$phyloseq_count))  

sample_data(phyloseq_decontam$phyloseq_count) <- sample_data(alpha_diversity(phyloseq_decontam$phyloseq_count))

sample_data <- sample_data(phyloseq$phyloseq_count)

```

### Alpha diversity calculation data

```{r}
sample_data(phyloseq$phyloseq_rel) %>% 
        data.frame() %>%
        dplyr::select(c("S.obs", "data_shannon", "data_invsimpson")) 
```


### Table 1 (revised)

**Revised Table 1 was constructed after prevalence & abundacne filtering**

**Table 1.** Sequencing results stratified by sample type and host depletion treatment. Number of samples in each experimental group, human and bacterial DNA measured by qPCR, number of samples did not pass the library QC or no microbial mapped reads, QC/d read, % host mapped reads, final reads, microbial species, function, and viral species richness. Microbial richness was calculated after employing prevalence and abundance filtration. Samples failed sequencing were removed from the calculation. Values depicted as N (%) or median (interquartile range). 

```{r, warning=F, error=FALSE, results='asis', `QC2 Table - cnt.2`}

sample_data(phyloseq$phyloseq_count)$sequencing_fail <- 
        ifelse(phyloseq$phyloseq_count %>% 
                       sample_data %>%
                       .$S.obs == 0,
               1,
               0)

sample_data(phyloseq$phyloseq_count) <- 
        
        merge(phyloseq$phyloseq_count %>% 
                      sample_data %>%
                      data.frame(check.names = F),
              phyloseq$phyloseq_path_rpk %>% #extracting functional richness as dataframe
                      sample_data %>%
                      data.frame(check.names = F) %>% 
                     dplyr::select(c("S.obs")) %>%
                      rename(F.obs = "S.obs"),
              by = 0
        ) %>%
                column_to_rownames("Row.names")

table1 <- sample_data(phyloseq$phyloseq_count) %>% data.frame() %>% 
        dplyr::filter(sample_type %in% c("Sputum", "Nasal", "BAL")) %>% 
        mutate(S.obs = case_when(S.obs==0 ~ NA,
                                 .default = S.obs),
               F.obs = case_when(F.obs==0 ~ NA,
                                 .default = F.obs),
               V.obs = case_when(V.obs==0 ~ NA,
                                 .default = V.obs)) %>% 
        group_by (sample_type, treatment) %>%
        summarise(`N` = n(),
            #      `Total DNA <br>ng/µL` = paste(format(round(median(picogreen_ng_ul),2), nsmall = 2, big.mark = ","), "<br>(", format(round(quantile(picogreen_ng_ul, 0.25),2), nsmall = 2, big.mark = ","), ", ", format(round(quantile(picogreen_ng_ul, 0.75),2), nsmall = 2, big.mark = ","), ")", sep = ""),
               `Human DNA <br>pg/µL` = paste(format(round(median(DNA_host_ng_uL*1000),1), nsmall = 1, big.mark = ","), " (", format(round(quantile(DNA_host_ng_uL*1000, 0.25),1), nsmall = 1, big.mark = ","), ", ", format(round(quantile(DNA_host_ng_uL*1000, 0.75),1), nsmall = 1, big.mark = ","), ")", sep = ""),
               `Bacterial DNA <br>pg/µL` = paste(format(round(median(DNA_bac_ng_uL*1000),1), nsmall = 1, big.mark = ","), " (", format(round(quantile(DNA_bac_ng_uL*1000, 0.25),1), nsmall = 1, big.mark = ","), ", ", format(round(quantile(DNA_bac_ng_uL*1000, 0.75),1), nsmall = 1, big.mark = ","), ")", sep = ""),
              `Sequencing fail<br>N (%)` = paste(sum(lib_failed + sequencing_fail), " (", sum(lib_failed +sequencing_fail) / n() * 100, " %)", sep = ""),
              `QC'd reads<br>reads x 10<sup>6</sup>` = paste(format(round(median(Reads_after_trim/1000000),1), nsmall = 1, big.mark = ","), " (", format(round(quantile(Reads_after_trim/1000000, 0.25),1), nsmall = 1, big.mark = ","), ", ", format(round(quantile(Reads_after_trim/1000000, 0.75),1), nsmall = 1, big.mark = ","), ")", sep = ""),
              `Host reads<br>%` = paste(format(round(median(sequencing_host_prop*100),1),
                                                nsmall = 1, big.mark = ","),
                                         " (",
                                         format(round(quantile(sequencing_host_prop * 100,
                                                               0.25),
                                                      1),
                                                nsmall = 1,
                                                big.mark = ","), 
                                         ", ", 
                                         format(round(quantile(sequencing_host_prop * 100, 0.75),1), 
                                                nsmall = 1, 
                                                big.mark = ","), 
                                         ")", 
                                         sep = ""),
             `Final reads<br>reads x 10<sup>6</sup>` = paste(format(round(median(Final_reads/1000000,
                                                                                 na.rm = T),
                                                                          1),
                                                                    nsmall = 1, big.mark = ","),
                                                             " (", 
                                                             format(round(quantile(Final_reads/1000000,
                                                                                            0.25,
                                                                                            na.rm = T),
                                                                                   1),
                                                                             nsmall = 1,
                                                                             big.mark = ","),
                                                             ", ",
                                                             format(round(quantile(Final_reads/1000000, 
                                                                                   0.75, 
                                                                                   na.rm = T),
                                                                          1), 
                                                                    nsmall = 1, 
                                                                    big.mark = ","), 
                                                             ")",
                                                             sep = ""),
             `Microbial<br>species<br>richness` = paste(median(S.obs, na.rm = T) %>% round,
                                                             " (", 
                                                             quantile(S.obs, 0.25, na.rm = T) %>% round,
                                                             ", ",
                                                             quantile(S.obs, 0.75, na.rm = T) %>% round,
                                                             ")",
                                                             sep = ""),
            `Microbial function<br>richness` = paste(median(F.obs, na.rm = T) %>% round,
                                                             " (", 
                                                             quantile(F.obs, 0.25, na.rm = T) %>% round,
                                                             ", ",
                                                             quantile(F.obs, 0.75, na.rm = T) %>% round,
                                                             ")",
                                                             sep = ""),
                `Viral<br>species<br>richness` = paste(median(V.obs, na.rm = T) %>% round,
                                                             " (", 
                                                             quantile(V.obs, 0.25, na.rm = T) %>% round,
                                                             ", ",
                                                             quantile(V.obs, 0.75, na.rm = T) %>% round,
                                                             ")",
                                                             sep = "")
              
            ) %>% 
        data.frame(check.names = F) %>% 
        arrange(sample_type, treatment) %>%
        rename(`Sample` = sample_type, Treatment = treatment) %>%
        mutate_all(linebreak) %>% kbl(format = "html", escape = F) %>% kable_styling(full_width = 0, html_font = "sans")

table1 


save_kable(table1, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/table1.html", self_contained = T)


```



```{r}

mock_otu <- subset_samples(phyloseq$phyloseq_count, sample_type =="Mock") %>% otu_table() %>% data.frame()

mock_otu_nonzero <- mock_otu[rowSums(mock_otu[])>0,]


```

### {-}

## **+ α** *Reproducing Amy's decontamination*

This code takes a wile to run + 1800 lines of codes, therefore not included in this manuscript. (will be submitted separately).


```{r, message=FALSE, warning=FALSE}

```

## Calculations used for main text

## Figure 1 numbers {.tabset}

**These tables were generated to note average med(IQR) of qPCR results**

### Figure 1 numbers (Host DNA) - treated vs untreated

This is calculations used for writing values of Figure 1 in the main text (as figure does not show exact value)

```{r}

sample_data %>% 
        subset(., .$S.obs != 0) %>%
        group_by(sample_type, treated) %>%
        summarise(med = median(DNA_host_nondil + DNA_bac_nondil),
                  lo = quantile(DNA_host_ng_uL + DNA_host_ng_uL, 0.25),
                  high = quantile(DNA_host_nondil + DNA_bac_nondil, 0.75))

sample_data %>% subset(., .$S.obs != 0) %>%
        group_by(sample_type, treated) %>% summarise(med = median(DNA_host_ng_uL), lo = quantile(DNA_host_ng_uL, 0.25), high = quantile(DNA_host_ng_uL, 0.75))

```

### Figure 1 numbers (Host prooportion) - treated vs untreated


```{r}
sample_data %>% subset(., .$S.obs != 0) %>% group_by(sample_type, treated) %>% summarise(med = median(host_proportion), lo = quantile(host_proportion, 0.25), high = quantile(host_proportion, 0.75))
```


### Figure 1 numbers (bacterial DNA) - treated vs untreated

```{r}
sample_data %>% subset(., .$S.obs != 0) %>% group_by(sample_type, treated) %>% summarise(med = median(DNA_bac_ng_uL), lo = quantile(DNA_bac_ng_uL, 0.25), high = quantile(DNA_bac_ng_uL, 0.75))

```

### {-}




# **3.3. Effects of treatments on taxonomic composition**





# *Did taxonomic composition change? 

## Fig. 2 Overview of sequencing results

- PSL comments: I would move this to the supplement and replace with a figure that demonstrates the relative abundance of the 10 most abundant species (stratified by sample type) and 10 most abundant KEGG functions. The whole point is to highlight why you would want to do metagenomics rather than amplicon sequencing. Consider using the following qualitative color palette from colorbrewer: 


**Fig. 2. ** Relative abundances at species level by sample type after employing prevalence and abundance filtering and top 10 species in each sample type by mean abundance. (A) BAL, (B) nasal swabs, (C) sputum, and (D) mock community. Empty space indicates samples showed no microbial reads, i.e., sequencing failed samples.


```{r, warning=FALSE, message=FALSE}

my_plot_bar = function (physeq, x = "Sample", y = "Abundance", fill = NULL, title = NULL, 
                        facet_grid = NULL) {
    mdf = psmelt(physeq)
    p = ggplot(mdf, aes_string(x = x, y = y, fill = fill))
    p = p + geom_bar(stat = "identity")
    p = p + theme(axis.text.x = element_text(angle = -90, hjust = 0)) +
            scale_x_discrete(drop = F)
    if (!is.null(facet_grid)) {
        p <- p + facet_grid(facet_grid)
    }
    if (!is.null(title)) {
        p <- p + ggtitle(title)
    }
    return(p)
}

a <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Mock")) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Mock") %>% 
           subset_samples(., S.obs != 0)) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Mock") %>% 
           subset_samples(., S.obs != 0)
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        my_plot_bar(., fill="species20") + 
        xlab("Sample") +
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.text.x = element_text(color = "white")) +
        guides(fill=guide_legend(title="Top 10 species")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        labs(tag = "A")
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +
        
neg_bar_labels <- c(
                    `Untreated` = "Extraction",
                    `lyPMA` = "lyPMA",
                    `Benzonase` = "Benzonase",
                    `HostZERO` = "HostZERO",
                    `MolYsis` = "MolYsis",
                    `QIAamp` = "QIAamp"
                    )

plot_labeller <- function(variable,value){
  return(neg_bar_labels[value])
}

b <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Neg.") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
        ) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Neg.") %>%
                                      subset_samples(.,
                                                     baylor_other_id != "20220606_Neg") %>%
                                      transform_sample_counts(.,
                                                              function(x){ifelse(is.na(x),
                                                                                 0, 
                                                                                 x)})
                              ) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Neg.") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>% 
        my_plot_bar(., fill="species20") + 
        ylab("") +
        xlab("Sample") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.text.x = element_text(color = "white")) +
        guides(fill=guide_legend(title="Top 10 species")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        #facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        facet_wrap ( ~ treatment,
                    scales= "free_x", nrow=1, labeller = plot_labeller
                    ) +
        ggtitle("Negative controls")

c <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "BAL") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
        ) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "BAL") %>%
                                      transform_sample_counts(.,
                                                              function(x){ifelse(is.na(x),
                                                                                 0, 
                                                                                 x)})
                              ) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "BAL") %>%
           transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E"))
   phyloseq_temp
  } %>% 
        my_plot_bar(., x = "subject_id", fill="species20") + 
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 species")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        #facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        facet_wrap ( ~ treatment,
                    scales= "free_x", nrow=1) +
        ggtitle("A    BAL")


d <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Nasal") %>% 
           subset_samples(., S.obs != 0)) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Nasal") %>% 
           subset_samples(., S.obs != 0)) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Nasal") %>% 
           subset_samples(., S.obs != 0)
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E", "F", "G", "H", "I", "J")) %>%
           as.character()
   phyloseq_temp
  } %>% 
        my_plot_bar(., x = "subject_id", fill="species20") + 
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 species")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
#        scale_x_discrete(drop=) +
        facet_wrap(~ treatment,  nrow = 1, drop = T, scales = "free_x") +
        ggtitle("B    Nasal swabs")

e <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Sputum") %>% 
           subset_samples(., S.obs != 0)) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Sputum") %>% 
           subset_samples(., S.obs != 0)) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Sputum") %>% 
           subset_samples(., S.obs != 0)
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E"))
   phyloseq_temp
  } %>%
        my_plot_bar(., x = "subject_id", fill="species20")+
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 species")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap(~ treatment, scales= "free_x", nrow = 1) +
        ggtitle("C    Sputum")


fig2 <- ggarrange(c, c %>% lemon::g_legend() %>% as_ggplot,
                  d, d %>% lemon::g_legend() %>% as_ggplot,
                  e, e %>% lemon::g_legend() %>% as_ggplot,
                  #a, a %>% lemon::g_legend() %>% as_ggplot,
                  #b, b %>% lemon::g_legend() %>% as_ggplot,
                  ncol=2,nrow=3, widths = c(3, 1),
                  legend = "none",
                  align = "hv")

annotate_figure(fig2,
                left = text_grob("Relative abundance",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Subject",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11)
)


          

png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/Figure2.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 160, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

annotate_figure(fig2,
                left = text_grob("Relative abundance",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Subject",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11)
)

# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()

```

** 3 Candidat and 2 Staphylococcus --> can be used for a reason why we should be doing shotgun sequencing (species level)

Dolosgranulum - repored present in nose
Malassezia - fungi
[others] --> [Other]

### Fig. S5. Changes of Mock microbial community

This figure will be updated, after calculating species richenss of all samples

```{r}
barplot_mock_microbe <- a


barplot_mock_microbe


```

Why **Cryptococcus** is missing? Double-check with a plot of top 20 taxa

```{r}

tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Mock")) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Mock") %>% 
           subset_samples(., S.obs != 0)) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 20)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Mock") %>% 
           subset_samples(., S.obs != 0)
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        my_plot_bar(., fill="species20") + 
        xlab("Subject") +
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.text.x = element_text(color = "white")) +
        guides(fill=guide_legend(title="Top 20 species")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 9, name = "Set3"),
                                     RColorBrewer::brewer.pal(n = 12, name = "Paired")
                                     )) +
        facet_wrap (~ treatment, scales = "free_x", nrow = 1)
        #ggtitle("D    Mock")
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +
        

```
There were **Cryptococcus neoformans**, however, it seems like they were assigned as **Cryptococcus gattii**.

### Negative control plot for R2R

Peggy's  on 20240912: create a stacked barplot where you facet by type of negative control


```{r, warning=FALSE, message=FALSE}
barplot_neg_microbe <- b        


annotate_figure(barplot_neg_microbe,
                left = text_grob("Relative abundance",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11))


          

png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureC12.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 80, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue


annotate_figure(barplot_neg_microbe,
                left = text_grob("Relative abundance",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11))


# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()

```

### Predicted function barplot

**This figure is not included in the main text as it is having too taxonomic groups**

```{r, warning=FALSE, message=FALSE}


phyloseq$phyloseq_path_cpm <- transform_sample_counts(phyloseq$phyloseq_path_rpk, function(x){x/sum(x)*1000000})

a <- tax_table(subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "Mock") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
        ) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "Mock") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "pathway"])
   .[, 3] <- gsub("s__", " ", .[, 3])
   .[, 3] <- gsub("_", " ", .[, 3])
   .[, 3] <- gsub("[]]|[[]", "",  .[, 3])
   .[, 3] <- gsub(" sp", " sp.",  .[, 3])
   .[, 3] <- gsub(" sp.", "</i> sp.",  .[, 3])
   .[, 3] <- gsub(" group", "</i> group.",  .[, 3])
   .[, 3] <- ifelse(grepl("Other",.[, 3]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 3]),
                           paste("<i>",  .[, 3], sep = ""),
                           paste("<i>",  .[, 3], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "Mock") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        my_plot_bar(., fill="species20") + 
        ylab("Relative abundancne") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(), 
              axis.text.x = element_text(size = 0)) +
        guides(fill=guide_legend(title="Top 10 species")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap (~ treatment, scales= "free_x", nrow = 1) +
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +
        labs(tag="D") + 
        ggtitle("Mock")



b <- tax_table(subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "Neg.") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
        ) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "Neg.") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "pathway"])
   .[, 3] <- gsub("s__", " ", .[, 3])
   .[, 3] <- gsub("_", " ", .[, 3])
   .[, 3] <- gsub("[]]|[[]", "",  .[, 3])
   .[, 3] <- gsub(" sp", " sp.",  .[, 3])
   .[, 3] <- gsub(" sp.", "</i> sp.",  .[, 3])
   .[, 3] <- gsub(" group", "</i> group.",  .[, 3])
   .[, 3] <- ifelse(grepl("Other",.[, 3]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 3]),
                           paste("<i>",  .[, 3], sep = ""),
                           paste("<i>",  .[, 3], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "Neg.") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        my_plot_bar(., fill="species20") + 
        ylab("Relative abundancne") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(), 
              axis.text.x = element_text(size = 0)) +
        guides(fill=guide_legend(title="Top 10 species")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap (~ treatment, scales= "free_x", nrow = 1) +
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +
        labs(tag="E") + 
        ggtitle("Negative controls")



c <- tax_table(subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "BAL") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "BAL") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "pathway"])
   .[, 3] <- gsub("s__", " ", .[, 3])
   .[, 3] <- gsub("_", " ", .[, 3])
   .[, 3] <- gsub("[]]|[[]", "",  .[, 3])
   .[, 3] <- gsub(" sp", " sp.",  .[, 3])
   .[, 3] <- gsub(" sp.", "</i> sp.",  .[, 3])
   .[, 3] <- gsub(" group", "</i> group.",  .[, 3])
   .[, 3] <- ifelse(grepl("Other",.[, 3]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 3]),
                           paste("<i>",  .[, 3], sep = ""),
                           paste("<i>",  .[, 3], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "BAL") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        my_plot_bar(., fill="species20") + 
        ylab("Relative abundancne") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(), axis.text.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 species")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap (~ treatment, scales= "free_x", nrow = 1) +
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +
        labs(tag="A") + 
        ggtitle("BAL")



d <- tax_table(subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "Nasal") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
        ) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "Nasal") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "pathway"])
   .[, 3] <- gsub("s__", " ", .[, 3])
   .[, 3] <- gsub("_", " ", .[, 3])
   .[, 3] <- gsub("[]]|[[]", "",  .[, 3])
   .[, 3] <- gsub(" sp", " sp.",  .[, 3])
   .[, 3] <- gsub(" sp.", "</i> sp.",  .[, 3])
   .[, 3] <- gsub(" group", "</i> group.",  .[, 3])
   .[, 3] <- ifelse(grepl("Other",.[, 3]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 3]),
                           paste("<i>",  .[, 3], sep = ""),
                           paste("<i>",  .[, 3], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "Nasal") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        my_plot_bar(., fill="species20") + 
        ylab("Relative abundancne") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(), axis.text.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 species")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap (~ treatment, scales= "free_x", nrow = 1) +
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +
        labs(tag="B") + 
        ggtitle("Nasal swabs")



e <- tax_table(subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "Sputum") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "Sputum") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "pathway"])
   .[, 3] <- gsub("s__", " ", .[, 3])
   .[, 3] <- gsub("_", " ", .[, 3])
   .[, 3] <- gsub("[]]|[[]", "",  .[, 3])
   .[, 3] <- gsub(" sp", " sp.",  .[, 3])
   .[, 3] <- gsub(" sp.", "</i> sp.",  .[, 3])
   .[, 3] <- gsub(" group", "</i> group.",  .[, 3])
   .[, 3] <- ifelse(grepl("Other",.[, 3]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 3]),
                           paste("<i>",  .[, 3], sep = ""),
                           paste("<i>",  .[, 3], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_path_cpm,
                              sample_type == "Neg.") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        my_plot_bar(., fill="species20") + 
        ylab("Relative abundancne") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(), axis.text.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 species")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap (~ treatment, scales= "free_x", nrow = 1) +
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +
        labs(tag="C") + 
        ggtitle("Sputum")


a
b
c
d
e


```



# *v.	Were there any bias in Mock community? *


## Gram-stain analysis {.tabset}

### Fig. S6. Bar plot of gram-stain

**Fig. S6.** Bar plot annotated with gram-stain information of (A) BAL, (B) nasal swabs, (C) sputum, and (D) mock communities after each treatment. 
  

                
```{r, warning=FALSE}

barplot_mock_gramproportion <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Mock")) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Mock") %>% 
           subset_samples(., S.obs != 0)) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Mock") %>% 
           subset_samples(., S.obs != 0)
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        my_plot_bar(., fill="gram_stain") + 
        ylab("") +
        xlab("Subject") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(), 
              legend.position = "right",
              axis.title.y = element_blank(),
              axis.text.x = element_blank()) +
        guides(fill=guide_legend(title="Gram-stain")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        labs(tag = "B")
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +


b <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Neg.") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
        ) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Neg.") %>%
                                      subset_samples(.,
                                                     baylor_other_id != "20220606_Neg") %>%
                                      transform_sample_counts(.,
                                                              function(x){ifelse(is.na(x),
                                                                                 0, 
                                                                                 x)})
                              ) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Neg.") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>% 
        my_plot_bar(., fill="gram_stain") + 
        ylab("") +
        xlab("Subject") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(),
              axis.text.x = element_blank()) +
        guides(fill=guide_legend(title="Gram-stain")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        #facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        facet_wrap ( ~ treatment,
                    scales= "free_x", nrow=1) +
        ggtitle("E  Negative controls")

c <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "BAL") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
        ) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "BAL") %>%
                                      transform_sample_counts(.,
                                                              function(x){ifelse(is.na(x),
                                                                                 0, 
                                                                                 x)})
                              ) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "BAL") %>%
           transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E"))
   phyloseq_temp
  } %>% 
        my_plot_bar(., x = "treatment", fill="gram_stain") + 
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(),
              axis.title.y = element_blank(),
              axis.text.x = element_text(angle = 45, hjust=1, size = 7,
                                         color = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3")),

              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Gram-stain")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        #facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        facet_wrap ( ~ subject_id,
                    scales= "free_x", nrow=1) +
        ggtitle("A  BAL")


d <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Nasal") %>% 
           subset_samples(., S.obs != 0)) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Nasal") %>% 
           subset_samples(., S.obs != 0)) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Nasal") %>% 
           subset_samples(., S.obs != 0)
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E", "F", "G", "H", "I", "J")) %>% 
           as.character()
   sample_data(phyloseq_temp)$treatment_order <- sample_data(phyloseq_temp)$treatment %>% 
           as.numeric() %>%
           as.character()
   phyloseq_temp
  } %>% 
        my_plot_bar(., x = "treatment_order", fill="gram_stain") + 
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.position = "none",
              axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 45, hjust=1, size = 7,
                                         color = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3")),
              axis.title.y = element_blank()) +
        guides(fill=guide_legend(title="Gram-stain")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        scale_x_discrete(breaks = c("1", "2", "3", "4", "5", "6"),
                         labels = c("Untreated", "lyPMA", "Benzonase", "HostZero", "MolYsis", "QIAamp")) +
        ggtitle("B  Nasal swabs")

d <- ggplotGrob(d + facet_grid(~ subject_id, drop = T, scales = "free", space = "free"))



d$grobs[[22]]$children[[2]]$grobs[[2]]$children[[1]]$label
d$grobs[[23]]$children[[2]]$grobs[[2]]$children[[1]]$label
d$grobs[[22]]$children[[2]]$grobs[[2]]$children[[1]]$gp$col <- c("#e31a1c", "#fb9a99", "#a6cee3")
d$grobs[[23]]$children[[2]]$grobs[[2]]$children[[1]]$gp$col <- c("#e31a1c", "#fb9a99", "#a6cee3")

d$grobs[[24]]$children[[2]]$grobs[[2]]$children[[1]]$label
d$grobs[[25]]$children[[2]]$grobs[[2]]$children[[1]]$label
d$grobs[[26]]$children[[2]]$grobs[[2]]$children[[1]]$label
d$grobs[[27]]$children[[2]]$grobs[[2]]$children[[1]]$label
d$grobs[[24]]$children[[2]]$grobs[[2]]$children[[1]]$gp$col <- c("#e31a1c", "#33a02c", "#b2df8a", "#1f78b4")
d$grobs[[25]]$children[[2]]$grobs[[2]]$children[[1]]$gp$col <- c("#e31a1c", "#33a02c", "#b2df8a", "#1f78b4")
d$grobs[[26]]$children[[2]]$grobs[[2]]$children[[1]]$gp$col <- c("#e31a1c", "#33a02c", "#b2df8a", "#1f78b4")
d$grobs[[27]]$children[[2]]$grobs[[2]]$children[[1]]$gp$col <- c("#e31a1c", "#33a02c", "#b2df8a", "#1f78b4")



d$grobs[[28]]$children[[2]]$grobs[[2]]$children[[1]]$label
d$grobs[[28]]$children[[2]]$grobs[[2]]$children[[1]]$gp$col <- c("#e31a1c","#fb9a99", "#33a02c", "#b2df8a")

d$grobs[[29]]$children[[2]]$grobs[[2]]$children[[1]]$label
d$grobs[[29]]$children[[2]]$grobs[[2]]$children[[1]]$gp$col <- c("#e31a1c", "#1f78b4", "#a6cee3")

d$grobs[[30]]$children[[2]]$grobs[[2]]$children[[1]]$label
d$grobs[[31]]$children[[2]]$grobs[[2]]$children[[1]]$label
d$grobs[[30]]$children[[2]]$grobs[[2]]$children[[1]]$gp$col <- c("#e31a1c", "#fb9a99", "#a6cee3")
d$grobs[[31]]$children[[2]]$grobs[[2]]$children[[1]]$gp$col <- c("#e31a1c", "#fb9a99", "#a6cee3")




e <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Sputum") %>% 
           subset_samples(., S.obs != 0)) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Sputum") %>% 
           subset_samples(., S.obs != 0)) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Sputum") %>% 
           subset_samples(., S.obs != 0)
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E"))
   phyloseq_temp
  } %>%
        my_plot_bar(., x = "treatment", fill="gram_stain")+
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(),
              axis.title.y = element_blank(),
              axis.text.x = element_text(angle = 45, hjust=1, size = 7, 
                                         color = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3")),

              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Gram-stain")) +
        #scale_x_discrete(drop=F) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap ( ~ subject_id,
                    scales= "free_x", nrow=1) +
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +
        ggtitle("C  Sputum")



figS6 <- ggarrange(c, d %>% as_ggplot, e, ncol = 1, common.legend = T, legend = "top")



annotate_figure(figS6,
                left = text_grob("Relative abundance",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Treatment",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11)
)



png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS6_updated.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 200, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue


annotate_figure(figS6,
                left = text_grob("Relative abundance",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Treatment",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11)
)


# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()





```                


### Gram-stain stats (all samples)
                

Effect size, standard error (SE) and p-value at a statistical test on gram-negative proportion using linear mixed effect model. lmer( Gram-negative proportion vs sample_type + treatment + sample_type \* treatment +
(1\|subject_id) )

**Interaction term was significant (p-value = 0.018771)**

Raw results

```{r warning=F}
lmer_gram_stain_proportion <- lmer(gram_neg_prop ~ sample_type + treatment + sample_type * treatment + (1|subject_id),
     data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")))

lmer_gram_stain_proportion %>% summary
```

ANOVA on lmer result

```{r warning=F}
lmer_gram_stain_proportion %>% anova 
```

### Table S5. Gram-stain stats -stratified 

**Stratified analsyis**

Table S5. Effect size, standard error (SE) and p-value at a statistical test on gram-negative 
proportion using linear mixed effect model. LMER(Gram-negative proportion vs sample type + treatment + sample type * treatment + (1|subject id)). Stratified analyses were conducted for each sample type as an interaction term of sample type and treatment was significant at an ANOVA test (p-value < 0.001) using a model, LMER (Gram-negative proportion ~ sample type + treatment + sample type * treatment  + (1|subject_id) ). The baseline of categorical variables is untreated BAL, and statistical significances were noted with `*: p-value < 0.05 and ***: p-value < 0.001.`

Raw results - Mock

```{r}
##Mock
gram_neg_prop_mock <- 
        lm(gram_neg_prop ~ treatment,
     data = sample_data(phyloseq$phyloseq_rel) %>% data.frame %>% subset(., .$sample_type %in% c("Mock"))) 

gram_neg_prop_mock %>%
        summary
```

BAL

```{r}
##BAL
gram_neg_prop_bal <- 
lmer(gram_neg_prop ~ treatment + (1|subject_id),
     data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("BAL"))) 


gram_neg_prop_bal %>%
        summary
```

Nasal

```{r}

##Nasal
gram_neg_prop_ns <- 
lmer(gram_neg_prop ~ treatment + (1|subject_id),
     data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Nasal"))) 


gram_neg_prop_ns %>%
        summary
```

Sputum

```{r}
##Sputum
gram_neg_prop_spt <- 
lmer(gram_neg_prop ~ treatment + (1|subject_id),
     data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Sputum"))) 

gram_neg_prop_spt %>%
        summary
```


Tidy summarized table


```{r}

gram_neg_prop_mock_kbl <- gram_neg_prop_mock %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              gram_neg_prop_mock %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]

gram_neg_prop_ns_kbl  <- gram_neg_prop_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              gram_neg_prop_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


gram_neg_prop_spt_kbl <- gram_neg_prop_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              gram_neg_prop_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]



gram_neg_prop_bal_kbl <- gram_neg_prop_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              gram_neg_prop_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


tableS5 <- cbind(gram_neg_prop_mock_kbl, gram_neg_prop_bal_kbl, gram_neg_prop_ns_kbl, gram_neg_prop_spt_kbl) %>%
        kbl(format = "html", escape = 0) %>% kable_styling(full_width = 0, html_font = "sans") %>% 
        add_header_above(c(" " = 1, "Mock" = 3, "BAL" = 3, "Nasal swab" = 3, "Sputum" = 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans") 

tableS5

save_kable(tableS5, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS5.html", self_contained = T)


```

### {-}



# *vi. Did diversity matrices change? 



## Taxa change {.tabset}

### Fig. 3 Alpha and beta diversity


**Fig. 3.** Alpha and beta diversity by sample type and treatment method after removing potential contaminants and rare taxa. (A) Species richness with statistical test results (linear mixed effect model stratified by sample type), (B) Morisita-Horn dissimilarity within subject between treatment, representing squares for median value and bars for 95% confidence intervals.  

- PSL comments (20230630): 
		- Figure 3A: it is hard to see some of the boxplot colors due to the narrow interquartile range. Can you use stat_summary and geom = "pointrange" like what Maghini DG et al did for their Figure 3b or create dotplot + pointrange geom like their Figure 3c? I found some example code goodgling though you might want to show median + iqr rather than mean + sd
			stat_summary(fun = mean,
               geom = "pointrange",
               fun.max = function(x) mean(x) + sd(x),
               fun.min = function(x) mean(x) - sd(x))
    - Figure 3A: we also need to talk because the mock community is only supposed to have a species richness = 10. I know that some of this may be due to taxonomic misclassification but that immediately will raise issues with the reviewer so we need to anticipate how to address this in advance
    - Figure 3A: add relevant p-values using horizontal bars and stars for significance like Maghini DG et al did for Figure 3B
    - Figure 3A: why not just do what Maghini DG et al did for Figure 3e? But have one panel per sample type


```{r, warning=FALSE, message=FALSE}

f3a <- ggplot(subset(sample_data(phyloseq$phyloseq_count) %>% 
                             data.frame, sample_data(phyloseq$phyloseq_count)$sample_type %in% c(#"Mock",
                                     "Sputum", "Nasal", "BAL")), aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment),
                    position = position_jitter(0.2), size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        #stat_summary(fun = mean,
        #       geom = "pointrange",
        #       fun.max = function(x) mean(x) + sd(x),
        #       fun.min = function(x) mean(x) - sd(x))+
        ylab("Species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        labs(tag = "A") +
        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 

dat_text <- data.frame(
  label = c(
          #"", "***", "***", "**", "***", #label for Mock
          "", "", "", "*", "", #label for BAL
          "", "", "***", "*", "**", 
          "**", "***", "***", "***", "***"),
  sample_type = c(
          #"Mock", "Mock", "Mock", "Mock", "Mock", 
          "BAL", "BAL", "BAL", "BAL", "BAL", 
          "Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          "Sputum", "Sputum", "Sputum", "Sputum", "Sputum"),
  treatment     = c(
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
  S.obs = c(
          #50, 30, 35, 33, 31,
          30, 35, 50, 52, 50,
          30, 30, 30, 35, 30,
          100, 120, 147, 140, 125)
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f3a <- f3a + geom_text(
  data    = dat_text,
  mapping = aes(x = treatment, y = S.obs, label = label)
)



f5S_mock <- ggplot(subset(sample_data(phyloseq$phyloseq_count) %>% 
                             data.frame,
                          sample_data(phyloseq$phyloseq_count)$sample_type %in%
                                  c("Mock")),
                   aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment), position = position_jitter(0.2),
                    size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        ylab("Species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        labs(tag = "C") +
        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 

dat_text <- data.frame(
  label = c(
          "", "***", "***", "**", "***"#, #label for Mock
          #"", "", "", "*", "", #label for BAL
          #"", "", "***", "*", "**", 
          #"**", "***", "***", "***", "***"
          ),
  sample_type = c(
          "Mock", "Mock", "Mock", "Mock", "Mock"#, 
          #"BAL", "BAL", "BAL", "BAL", "BAL", 
          #"Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          #"Sputum", "Sputum", "Sputum", "Sputum", "Sputum"
          ),
  treatment = c(
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"#, 
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"
          ),
  S.obs = c(
          50, 30, 35, 33, 31
          #30, 35, 50, 52, 50,
          #30, 30, 30, 35, 30,
          #100, 120, 147, 140, 125
          )
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("Mock"#, 
                                                                #"BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f5S_mock <- f5S_mock + geom_text(
  data    = dat_text,
  mapping = aes(x = treatment, y = S.obs, label = label)
)



#Making subset of non-zero samples without neg

phyloseq_rel_nz <- phyloseq$phyloseq_rel %>%
        subset_samples(S.obs != 0 & sample_type %in% c("Mock", 
                                                       "BAL", "Nasal", "Sputum"))

#distances of betadiversity - boxplots
horn_dist_long <- distance(phyloseq_rel_nz, method="horn") %>% as.matrix() %>% melt_dist() #making long data of distance matrices

#Adding sample type and treatment name. 
#this can be also done by merging metadata into the `horn_dist_long`
names <- data.frame(str_split_fixed(horn_dist_long$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(horn_dist_long$iso2, "_", 3))
horn_dist_long$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
horn_dist_long$method_1 <- ifelse(grepl("lyPMA", horn_dist_long$iso1),"lypma", 
                                         ifelse(grepl("ben", horn_dist_long$iso1),"benzonase", 
                                                ifelse(grepl("host", horn_dist_long$iso1),"host_zero", 
                                                       ifelse(grepl("qia", horn_dist_long$iso1),"qiaamp", 
                                                              ifelse(grepl("moly", horn_dist_long$iso1),"molysis", 
                                                                     "control")))))


#Adding data for iso 2 also should be done
horn_dist_long$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")
horn_dist_long$method_2 <-ifelse(grepl("lyPMA", horn_dist_long$iso2),"lypma", 
                                        ifelse(grepl("ben", horn_dist_long$iso2),"benzonase", 
                                               ifelse(grepl("host", horn_dist_long$iso2),"host_zero", 
                                                      ifelse(grepl("qia", horn_dist_long$iso2),"qiaamp", 
                                                             ifelse(grepl("moly", horn_dist_long$iso2),"molysis", 
                                                                    "control")))))


#subsetting distances of my interest
horn_dist_long$sample_id_1 <- ifelse(grepl("pos", horn_dist_long$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_1, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_1))
horn_dist_long$sample_id_2 <- ifelse(grepl("pos", horn_dist_long$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_2, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_2))


horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long, horn_dist_long$sample_id_1 == horn_dist_long$sample_id_2) # data within samples

horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long_within_sampleid_from_control,
                                                           horn_dist_long_within_sampleid_from_control$method_1 != horn_dist_long_within_sampleid_from_control$method_2) # remove irrelevant association

horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long_within_sampleid_from_control, (horn_dist_long_within_sampleid_from_control$method_1 == "control") + (horn_dist_long_within_sampleid_from_control$method_2 == "control") != 0)


horn_dist_long_within_sampleid_from_control$treatment <- horn_dist_long_within_sampleid_from_control$method_1

horn_dist_long_within_sampleid_from_control$treatment <- ifelse(horn_dist_long_within_sampleid_from_control$treatment == "control", horn_dist_long_within_sampleid_from_control$method_2, horn_dist_long_within_sampleid_from_control$treatment) 


#Setting key method
horn_dist_long_within_sampleid_from_control$sample_type <- ifelse(grepl("NS", horn_dist_long_within_sampleid_from_control$iso1), "Nasal",
                                                                  ifelse(grepl("CFB", horn_dist_long_within_sampleid_from_control$iso1), "Sputum",
                                                                         ifelse(grepl("BAL", horn_dist_long_within_sampleid_from_control$iso1), "BAL",
                                                                                ifelse(grepl("pos|POS", horn_dist_long_within_sampleid_from_control$iso1, ignore.case = T), "Mock",
                                                                                       ifelse(grepl("neg|N_EXT", horn_dist_long_within_sampleid_from_control$iso1), "Neg.",NA)))))

#Making a column for baseline (controls, from where?)
horn_dist_long_within_sampleid_from_control <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(dist_from = case_when(method_1 == "control" ~ iso1,
                                     method_2 == "control" ~ iso2))

dummy <- data.frame(iso1 = horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           iso2 = horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           dist = 0,
           treatment = "Untreated",
           method_1 = "control",
           method_2 = "control"
           )
names <- data.frame(str_split_fixed(dummy$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(dummy$iso2, "_", 3))
dummy$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
#Adding data for iso 2 also should be done
dummy$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")


#subsetting distances of my interest
dummy$sample_id_1 <- ifelse(grepl("pos", dummy$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_1, ignore.case = T),"Neg.",
                                        dummy$sample_id_1))
dummy$sample_id_2 <- ifelse(grepl("pos", dummy$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_2, ignore.case = T),"Neg.",
                                        dummy$sample_id_2))
dummy$sample_type <- ifelse(grepl("NS", dummy$iso1), "Nasal",
                            ifelse(grepl("CFB", dummy$iso1), "Sputum",
                                   ifelse(grepl("BAL", dummy$iso1), "BAL",
                                          ifelse(grepl("pos|POS", dummy$iso1, ignore.case = T), "Mock",
                                                 ifelse(grepl("neg|N_EXT", dummy$iso1), "Neg.",NA)))))
dummy <- subset(dummy, !is.na(dummy$sample_type))
horn_dist_long_within_sampleid_from_control <- bind_rows(horn_dist_long_within_sampleid_from_control, dummy)

#Here, sample id is the same as subject id.
horn_dist_long_within_sampleid_from_control$subject_id <- horn_dist_long_within_sampleid_from_control$sample_id_1

horn_dist_long_within_sampleid_from_control$treatment <-
        factor(horn_dist_long_within_sampleid_from_control$treatment,
               levels = c("Untreated", "lypma", "benzonase", "host_zero", "molysis", "qiaamp"))

#Making figure of beta diversity distances

## This only includes samples (BAL, Nasal and Sputum)
f3b2 <- rbind(cbind("Sputum",
            lmer(dist ~ treatment + (1|subject_id),
                data = horn_dist_long_within_sampleid_from_control %>%
                        subset(sample_type == "Sputum")) %>% 
                    confint()
            ),
      cbind("BAL",
            lmer(dist ~ treatment + (1|subject_id),
                 data = horn_dist_long_within_sampleid_from_control %>%
                         subset(sample_type == "BAL")) %>% 
                    confint()
            ),
      cbind("Nasal",
            lmer(dist ~ treatment + (1|subject_id),
                 data = horn_dist_long_within_sampleid_from_control %>%
                         subset(sample_type == "Nasal")) %>% 
                    confint()
     )
) %>% {
        row_names <- rownames(.)
        data_frame <- data.frame(.)
        data_frame$treatment <- row_names
        data_frame %>% 
                remove_rownames() %>%
                rename(sample_type = "V1",
               "2.5%" = "X2.5..",
               "97.5%" = "X97.5..") %>% 
                mutate(`2.5%` = as.numeric(`2.5%`),
                       `97.5%` = as.numeric(`97.5%`),
                       mean = (`97.5%`+`2.5%`)/2,
                       treatment = case_when(treatment == "(Intercept)" ~ "Untreated",
                                             treatment == "treatmentlypma" ~ "lyPMA",
                                             treatment == "treatmentbenzonase" ~ "Benzonase",
                                             treatment == "treatmenthost_zero" ~ "HostZERO",
                                             treatment == "treatmentmolysis" ~ "MolYsis",
                                             treatment == "treatmentqiaamp" ~ "QIAamp"
                                             ),
                       treatment = factor(treatment,
                                          levels = c("Untreated", "lyPMA", "Benzonase",
                                      "HostZERO", "MolYsis", "QIAamp")),
                       sample_type = factor(sample_type,
                                          levels = c("BAL", "Nasal", "Sputum"))
                       ) %>%
                subset(treatment %in% c(#"Untreated", 
                                        "lyPMA", "Benzonase",
                                      "HostZERO", "MolYsis", "QIAamp"))
} %>%
        ggplot(aes(x = mean, y = treatment, col = treatment)) +
        geom_point(aes(x=mean), shape=15, size=3) +
        geom_linerange(aes(xmin=`2.5%`, xmax=`97.5%`)) +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c(#"#e31a1c",
                                      "#fb9a99","#33a02c",
                                      "#b2df8a","#1f78b4","#a6cee3"),
                           name = "Treatment",
                           breaks = c(#"Untreated", 
                                      "lyPMA", "Benzonase",
                                      "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Morisita-Horn dissimilarity from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              legend.position = "none") +
        labs(tag = "B") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "sans") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "sans") +
        scale_x_continuous(breaks = c(-0.25, 0, 0.25, 0.5, 0.75),
                           labels = c(-0.25, "0 (low bias)", 0.25, 0.5, "0.75 (high bias)"))


dat_text <- data.frame(
  label = c(
          #"", "***", "***", "**", "***", #label for Mock
          "*", "", "", "", "", #label for BAL
          "*", "*", "", "**", "", 
          "**", "***", "***", "***", "***"),
  sample_type = c(
          #"Mock", "Mock", "Mock", "Mock", "Mock", 
          "BAL", "BAL", "BAL", "BAL", "BAL", 
          "Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          "Sputum", "Sputum", "Sputum", "Sputum", "Sputum"),
  treatment     = c(
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
  mean = c(
          #50, 30, 35, 33, 31,
          0.6, 0, 0, 0, 0,
          0.32, 0.3, 0, 0.37, 0,
          0.58, 0.75, 0.85, 0.83, 0.85)
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f3b2 <- f3b2 + geom_text(
  data    = dat_text,
  mapping = aes(y = treatment, x = mean, label = label),
  col = "black"
)


f3b2_box <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(across(sample_type, factor, levels=c(#"Mock", 
                                                    "BAL", "Nasal","Sputum"
                                                    ))) %>%
        subset(., .$sample_type != "Neg.") %>% 
        subset(., .$treatment != "Untreated") %>%
  mutate(treatment = factor(treatment, levels = c("lypma", "benzonase", "host_zero", "molysis", "qiaamp"),
                            labels = c("lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))) %>%
        ggplot(aes(x = dist, y = treatment, col = treatment)) +
        geom_boxplot(aes(x=dist)) +
        #geom_linerange(aes(xmin=`2.5%`, xmax=`97.5%`)) +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c(#"#e31a1c",
                                      "#fb9a99","#33a02c",
                                      "#b2df8a","#1f78b4","#a6cee3"),
                           name = "Treatment",
                           breaks = c(#"Untreated", 
                                      "lyPMA", "Benzonase",
                                      "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Morisita-Horn dissimilarity from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              legend.position = "none") +
        labs(tag = "B") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "sans") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "sans") +
        scale_x_continuous(breaks = c(-0.25, 0, 0.25, 0.5, 0.75),
                           labels = c(-0.25, "0 (low bias)", 0.25, 0.5, "0.75 (high bias)"))


dat_text <- data.frame(
  label = c(
          #"", "***", "***", "**", "***", #label for Mock
          "*", "", "", "", "", #label for BAL
          "*", "*", "", "**", "", 
          "**", "***", "***", "***", "***"),
  sample_type = c(
          #"Mock", "Mock", "Mock", "Mock", "Mock", 
          "BAL", "BAL", "BAL", "BAL", "BAL", 
          "Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          "Sputum", "Sputum", "Sputum", "Sputum", "Sputum"),
  treatment     = c(
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
  dist = c(
          #50, 30, 35, 33, 31,
          0.9, 0, 0, 0, 0,
          0.48, 0.4, 0, 0.55, 0,
          0.8, 0.82,1, 1.02, 0.98)
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f3b2_box <- f3b2_box + geom_text(
  data    = dat_text,
  mapping = aes(y = treatment, x = dist, label = label),
  col = "black"
)



# Mock community's beta-diversity plot.
## This only includes positive control samples (Mock)

f5S_mock_mh <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(across(sample_type, factor, levels=c("Mock"#, 
                                                    #"BAL", "Nasal","Sputum"
                                                    ))) %>%
        subset(., .$sample_type != "Neg.") %>% 
        group_by(sample_type, treatment) %>%
        summarise(mean = mean(dist, na.rm = TRUE),
            sd = sd(dist, na.rm = TRUE),
            n = n()) %>%
        subset(., .$treatment != "Untreated") %>%
  mutate(se = sd / sqrt(n),
         lower.ci = mean - qt(1 - (0.05 / 2), n - 1) * se,
         upper.ci = mean + qt(1 - (0.05 / 2), n - 1) * se,
         treatment = factor(treatment, levels = c(#"Untreated",
                                                  "lypma", "benzonase", "host_zero", "molysis", "qiaamp"),
                            labels = c(#"Untreated",
                                       "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))) %>%
        #,
         #text = paste(sprintf("%.2f", round(mean, digits = 2)), " [", sprintf("%.2f", round(lower.ci, digits = 2)), ", ", sprintf("%.2f", round(upper.ci, digits = 2)), "]", sep = "")) %>%
        ggplot(aes(x = mean, y = treatment, col = treatment)) +
        geom_point(aes(x=mean), shape=15, size=3) +
        geom_linerange(aes(xmin=lower.ci, xmax=upper.ci)) +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c(#"#e31a1c", 
                                      "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", 
                           labels = c(#"Untreated",
                                      "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Morisita-Horn dissimilarity from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              legend.position = "none") +
        labs(tag = "D") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "sans") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "sans") +
        scale_x_continuous(breaks = c(-0.5, 0, 0.5, 1, 1.5), labels = c(-0.5, "0 (low bias)", 0.5, 1, "1.5 (high bias)"))



f5S_mock_mh_box <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(across(sample_type, factor, levels=c("Mock"#, 
                                                    #"BAL", "Nasal","Sputum"
                                                    ))) %>%
        subset(., .$sample_type != "Neg.") %>% 
        subset(., .$treatment != "Untreated") %>%
  mutate(treatment = factor(treatment, levels = c(#"Untreated",
                                                  "lypma", "benzonase", "host_zero", "molysis", "qiaamp"),
                            labels = c(#"Untreated",
                                       "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))) %>%
        #,
         #text = paste(sprintf("%.2f", round(mean, digits = 2)), " [", sprintf("%.2f", round(lower.ci, digits = 2)), ", ", sprintf("%.2f", round(upper.ci, digits = 2)), "]", sep = "")) %>%
        ggplot(aes(x = dist, y = treatment, col = treatment)) +
        geom_boxplot() +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c(#"#e31a1c", 
                                      "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", 
                           labels = c(#"Untreated",
                                      "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Morisita-Horn dissimilarity from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              legend.position = "none") +
        labs(tag = "D") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "sans") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "sans") +
        scale_x_continuous(breaks = c(-0.5, 0, 0.5, 1, 1.5), labels = c(-0.5, "0 (low bias)", 0.5, 1, "1.5 (high bias)"))


fig3 <- ggarrange(f3a, f3b2_box, ncol = 1, common.legend = T, align = "hv")

fig3

png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/Figure3.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 220, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

fig3
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()
```


### {-}


# *vii.	& viii. Do these host depletion methods introduce bias in the sequenced community? *

+ Does bias differ by sample type?


## Alpha diversity changes {.tabset}

### Calculation for log centered Final reads

**Distribution of centralized final reads**

```{r}
sample_data <- sample_data(phyloseq$phyloseq_count)
sample_data$log_centered_final_reads <- log(sample_data$Final_reads + 1) - median(log((subset(sample_data, sample_data$sample_type %in% c("BAL") & sample_data$treatment %in% c("Untreated")) %>% .$Final_reads) + 1))

sample_data$bal_log_centered_final_reads <- log(sample_data$Final_reads + 1) - median(log((subset(sample_data, sample_data$sample_type %in% c("BAL") & sample_data$treatment %in% c("Untreated"))%>% .$Final_reads) + 1))

sample_data$ns_log_centered_final_reads <- log(sample_data$Final_reads + 1) - median(log((subset(sample_data, sample_data$sample_type %in% c("Nasal") & sample_data$treatment %in% c("Untreated"))%>% .$Final_reads) + 1))

sample_data$spt_log_centered_final_reads <- log(sample_data$Final_reads + 1) - median(log((subset(sample_data, sample_data$sample_type %in% c("Sputum") & sample_data$treatment %in% c("Untreated"))%>% .$Final_reads) + 1))


subset(sample_data, sample_data$sample_type %in% c("BAL", "Nasal", "Sputum")) %>% .$log_centered_final_reads %>% hist (main = "Histogram of centered log10 final reads of BAL, Nasal, Sputum")

subset(sample_data, sample_data$sample_type %in% c("BAL")) %>% .$log_centered_final_reads %>% hist (main = "Histogram of centered log10 final reads of BAL")

subset(sample_data, sample_data$sample_type %in% c("Nasal")) %>% .$log_centered_final_reads %>% hist (main = "Histogram of centered log10 final reads of Nasal")

subset(sample_data, sample_data$sample_type %in% c("Sputum")) %>% .$log_centered_final_reads %>% hist (main = "Histogram of centered log10 final reads of Sputum")

```

### Species richness (all sample)

**This table is too redundant. Removed from the manuscript.**. 

Effect size, standard error (SE) and p-value of a statistical test for species richness with an interaction term using linear mixed effect model (Species richness ~ sample_type * treatment + log10 (Final_reads) + (1|subject_id) ).

**Interaction term was highly significant (p = 2.2e-16)**

Raw result

```{r}
lmer_sr <- lmer(S.obs ~ treatment * sample_type + log10(Final_reads) + (1|subject_id),
                    data = sample_data %>% 
                            data.frame %>% 
                            subset(., .$S.obs != 0))

lmer_sr %>% summary
        
```

ANOVA result on LMER result for species richness

```{r}
lmer_sr %>% anova()
        
```

### Table S3. Species richness (stratified) without depth

**Table S3.** Effect size, standard error (SE) and p-value of a statistical test for species richness using linear mixed effect model stratified by sample type (Species richness ~ treatment + (1|subject_id) ). Stratified analyses were conducted for each sample type as an interaction term of sample type and treatment was significant at an ANOVA test (p-value < 0.001) using a model, lmer(species richness ~ sample type + treatment + sample type * treatment  + (1|subject_id)). Statistical significances were noted with `*: p-value < 0.05, **: p-value < 0.01, and ***: p-value < 0.001.`

LMER raw result - Mock

```{r}

sr_lmer_mock <- lm(S.obs ~ treatment,
                    data = sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Mock") & S.obs != 0))

sr_lmer_mock %>% summary() 

```

LMER raw result - BAL

```{r}

sr_lmer_bal <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("BAL") & S.obs != 0))
sr_lmer_bal %>% summary()
        
```

LMER raw result - Nasal

```{r}

sr_lmer_ns <- lmer(S.obs ~ treatment + (1|subject_id),
                   data = sample_data %>% 
                           data.frame %>% 
                           subset(., .$sample_type %in% c("Nasal")))
sr_lmer_ns %>% summary()
```

LMER raw result - Sputum

```{r}
sr_lmer_spt <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Sputum")))
sr_lmer_spt %>% summary()
        
sr_lmer_spt %>% confint()

```

Tidy summarized table

```{r}

sr_lmer_mock_kbl <-  sr_lmer_mock %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_mock %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

sr_lmer_bal_kbl <-  sr_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        
sr_lmer_ns_kbl <-  sr_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


sr_lmer_spt_kbl <-  sr_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

tables3 <- cbind(#sr_lmer_mock_kbl,
                 sr_lmer_bal_kbl,
                 sr_lmer_ns_kbl,
                 sr_lmer_spt_kbl) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1,
                           #"Mock" = 3,
                           "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")

tables3

save_kable(tables3, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS3.html", self_contained = T)

```



**Table S3 viral.** Effect size, standard error (SE) and p-value of a statistical test for viral species richness using linear mixed effect model stratified by sample type (Species richness ~ treatment + (1|subject_id) ). Stratified analyses were conducted for each sample type as an interaction term of sample type and treatment was significant at an ANOVA test (p-value < 0.001) using a model, lmer(species richness ~ sample type + treatment + sample type * treatment  + (1|subject_id)). Statistical significances were noted with `*: p-value < 0.05, **: p-value < 0.01, and ***: p-value < 0.001.`


LMER raw result - BAL

```{r}

sr_lmer_bal_v <- lmerTest::lmer(V.obs ~ treatment + (1|subject_id),
                    data = sample_data %>% 
                            data.frame %>% 
                           subset(., .$sample_type %in% c("BAL")))
sr_lmer_bal_v %>% summary()
        
```

LMER raw result - Nasal

```{r}

sr_lmer_ns_v <- lmerTest::lmer(V.obs ~ treatment + (1|subject_id),
                   data = sample_data %>% 
                           data.frame %>% 
                           subset(., .$sample_type %in% c("Nasal")))
sr_lmer_ns_v %>% summary()
```

LMER raw result - Sputum

```{r}
sr_lmer_spt_v <- lmerTest::lmer(V.obs ~ treatment + (1|subject_id),
                    data = sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Sputum")))
sr_lmer_spt_v %>% summary()
        


```

Tidy summarized table

```{r}



sr_lmer_bal_kbl_v <-  sr_lmer_bal_v %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        
sr_lmer_ns_kbl_v <-  sr_lmer_ns_v %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


sr_lmer_spt_kbl_v <-  sr_lmer_spt_v %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        
 
tables3 <- 
        
        rbind(
                cbind(Outcome = "Microbial species richness",
                      sr_lmer_bal_kbl %>% rownames_to_column("Treatment"), 
                      sr_lmer_ns_kbl, 
                      sr_lmer_spt_kbl) %>% remove_rownames() %>% as.matrix(),
                cbind(Outcome = "Viral species richness",
                      sr_lmer_bal_kbl_v %>% rownames_to_column("Treatment"), 
                      sr_lmer_ns_kbl_v, 
                      sr_lmer_spt_kbl_v) %>% remove_rownames() %>% as.matrix()
                ) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 2,
                           #"Mock" = 3,
                           "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")

tables3

save_kable(tables3, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS3_updated.html", self_contained = T)

```

### Species richness - stratified - with depth

**This table is too redundant. Removed from the manuscript.**. 

Sequencing depth adjusted effect size, standard error (SE) and p-value of a statistical test for species richness using linear mixed effect model stratified by sample type (Species richness ~ treatment + log10 (Final_reads) + (1|subject_id) ).

```{r}

sr_lmer_bal_w_depth <- lmer(S.obs ~ treatment + bal_log_centered_final_reads + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("BAL") & S.obs != 0)) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = 1), 
                                " (", 
                                round(Estimate - 1.96 * abs(t_value), 1) %>% format(nsmall = 1),
                                ", ",
                                round(Estimate + 1.96 * abs(t_value), 1) %>% format(nsmall = 1),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " "))
        
sr_lmer_ns_w_depth <- lmer(S.obs ~ treatment + ns_log_centered_final_reads + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Nasal"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("ns_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = ), 
                                " (", 
                                round(Estimate - 1.96 * abs(t_value), 1) %>% format(nsmall = ),
                                ", ",
                                round(Estimate + 1.96 * abs(t_value), 1) %>% format(nsmall = ),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " "))


sr_lmer_spt_w_depth <- lmer(S.obs ~ treatment + spt_log_centered_final_reads + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Sputum"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("spt_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
                rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = 1), 
                                " (", 
                                round(Estimate - 1.96 * abs(t_value), 1) %>% format(nsmall = 1),
                                ", ",
                                round(Estimate + 1.96 * abs(t_value), 1) %>% format(nsmall = 1),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " "))
        

cbind(sr_lmer_bal_w_depth, sr_lmer_ns_w_depth, sr_lmer_spt_w_depth) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")



#save_kable(tableS7, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS7.html", self_contained = T)




```

### {-}


## Beta diversity distances - Morisita Horn dissimilarity {.tabset}

### PERMANOVA - all samples 

**This table is too redundant. Removed from the manuscript.**. 

Degree of freedom, effect size (residual, R2) and p-value of permutational ANOVA for Morisita-Horn dissimilarity with an interaction term and strata term (MH-index of species composition ~ sample type `*` treatment + subject + log10(final reads), strata = subject id). Statistical significances were noted with `***:` p-value < 0.001.



```{r}
set.seed(seed)
horn_perm_all <- vegan::adonis2(by = "terms",
                                distance(phyloseq_rel_nz, method="horn") ~ sample_type * treatment + subject_id,
                                  data = phyloseq_rel_nz %>% sample_data %>% data.frame(check.names = F),
                                  permutations = 10000)
set.seed(seed)
horn_perm_ns <- vegan::adonis2(by = "terms", distance(subset_samples(phyloseq_rel_nz, sample_type == "Nasal"), method="horn") ~ lypma + benzonase + host_zero + molysis + qiaamp,
                               data = subset_samples(phyloseq_rel_nz, sample_type == "Nasal") %>%
                                       sample_data %>% data.frame(check.names = F),
                               strata = subset_samples(phyloseq_rel_nz, sample_type == "Nasal") %>% 
                                       sample_data %>% data.frame(check.names = F) %>% .$subject_id, permutations = 10000)
set.seed(seed)
horn_perm_bal  <- vegan::adonis2(by = "terms", distance(subset_samples(phyloseq_rel_nz, sample_type == "BAL"), method="horn") ~  lypma + benzonase + host_zero + molysis + qiaamp,
                                 data = subset_samples(phyloseq_rel_nz, sample_type == "BAL") %>% sample_data %>% data.frame(check.names = F),
                                 strata = subset_samples(phyloseq_rel_nz, sample_type == "BAL") %>%
                                         sample_data %>% data.frame(check.names = F) %>% .$subject_id,
                                  permutations = 10000)
set.seed(seed)
horn_perm_spt <- vegan::adonis2(by = "terms", distance(subset_samples(phyloseq_rel_nz, sample_type == "Sputum"), method="horn") ~ lypma + benzonase + host_zero + molysis + qiaamp,
                                data = subset_samples(phyloseq_rel_nz, sample_type == "Sputum") %>% sample_data %>% data.frame(check.names = F),
                                strata = subset_samples(phyloseq_rel_nz, sample_type == "Sputum")
                                %>% sample_data %>% data.frame(check.names = F) %>% .$subject_id,
                                  permutations = 10000) 

```

PERMANOVA raw result

```{r}
horn_perm_all 
```

Tidy permanova result

```{r}

tableS5_A <- horn_perm_all %>% data.frame(check.names = F) %>% rownames_to_column("row.names") %>% 
        mutate(row.names = case_when(row.names == "sample_type" ~ 'Sample type',
                                     row.names == "treatment" ~ 'Treatment',
                                     row.names == "subject_id" ~ 'Subject',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "sample_type:treatment" ~ 'Sample type * Treatment',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
        mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        mutate(`<i>p</i>-value` = format(`<i>p</i>-value`, nsmall = 3)) %>%
        dplyr::select(c("Degree of freedom", "R<sup>2</sup>", "<i>p</i>-value", " ")) %>% 
        kbl(format = "html", escape = 0) %>%
        kable_styling(full_width = 0, html_font = "sans")

tableS5_A



```

### Table S6. PERMANOVA - Stratified

Raw result - BAL

```{r}
horn_perm_bal
```


Raw result - Nasal

```{r}
horn_perm_ns
```


Raw result - BAL

```{r}
horn_perm_spt
```


**This table was additionally added to the manuscript as to determine bias after treatment with sputum**. 

**Table S6.** Degree of freedom, effect size (residual, R^2) and p-value of permutational ANOVA for Morisita-Horn distiances for species richness (MH-distance ~ lyPMA + Benzoase + HostZERO + MolYsis + QIAamp, strata = subject_id). Sample types were stratified as interaction term of (MH-distance ~ sample type + treatment + sample type * treatment, strata = subject_id) was significant (p < 0.001).

Tidy permanova result (stratified)

```{r}

a <- horn_perm_bal %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'HostZERO',
                                     row.names == "molysis" ~ 'MolYsis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        dplyr::select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 

b <- horn_perm_ns %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'HostZERO',
                                     row.names == "molysis" ~ 'MolYsis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        dplyr::select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 

c <- horn_perm_spt %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'HostZERO',
                                     row.names == "molysis" ~ 'MolYsis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        dplyr::select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 


tableS6 <- cbind(a, b, c) %>% 
        kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")


save_kable(tableS6, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS6.html", self_contained = T)


tableS6


```

### Table S7. LM on M-H distance 

**Table S7.** Effect size, standard error (SE) and p-value of a statistical test for Morisita-Horn dissimilarity from untreated to each treated within subject, stratified by sample type. Stratified analyses were conducted for each sample type as an interaction term of sample type and treatment was significant (p-value < 0.001) using a model, ANOVA(species richness ~ sample type + treatment + sample type * treatment). The tested was conducted with a linear mixed effect model LM(distance within subject by treatment ~ treatment + (1|subject)). The baseline of sample type is untreated (0 distance, untreated from untreated), and statistical significance were noted with `*: p-value < 0.05, **: p-value < 0.01 and ***: `p-value < 0.001.

```{r}

horn_dist_long_within_sampleid_from_control$treatment <- factor(horn_dist_long_within_sampleid_from_control$treatment,
                                                                     levels = c("Untreated",
                                                                                "lypma",
                                                                                "benzonase",
                                                                                "host_zero",
                                                                                "molysis",
                                                                                "qiaamp"))
horn_dist_long_within_sampleid_from_control$sample_type <- factor(horn_dist_long_within_sampleid_from_control$sample_type,
                                                                     levels = c("BAL",
                                                                                "Nasal",
                                                                                "Sputum"))
```

Raw lmer result on M-H distances

```{r}

lmer(dist ~ treatment * sample_type + (1|subject_id),
     data = horn_dist_long_within_sampleid_from_control %>%
             subset(sample_type != "Mock")) %>%
        summary() 


```

Justifying stratified analysis - ANOVA(LM(dist ~ sample type * treatment))

```{r}
lmer(dist ~ treatment * sample_type + (1|subject_id),
     data = horn_dist_long_within_sampleid_from_control %>%
             subset(sample_type != "Mock")) %>%
        anova() 

```

Stratified analysis - BAL

```{r}
mh_lmer_bal <- lmer(dist ~ treatment + (1|subject_id),
     data = horn_dist_long_within_sampleid_from_control %>%
             subset(sample_type == "BAL")) 

mh_lmer_bal %>% 
        summary
```


Stratified analysis - Nasal

```{r}
mh_lmer_ns <- lmer(dist ~ treatment + (1|subject_id),
     data = horn_dist_long_within_sampleid_from_control %>%
             subset(sample_type == "Nasal")) 

mh_lmer_ns %>% 
        summary

```



Stratified analysis - Sputum

```{r}
mh_lmer_spt <- lmer(dist ~ treatment + (1|subject_id),
     data = horn_dist_long_within_sampleid_from_control %>%
             subset(sample_type == "Sputum")) 

mh_lmer_spt %>% 
        summary

```

Viral beta diversity (LMER)

```{r}


#Making subset of non-zero samples without neg

phyloseq_rel_nz_v <- v_phyloseq$viral_rel %>% 
        subset_samples(S.obs != 0 & sample_type %in% c("Mock", 
                                                       "BAL", "Nasal", "Sputum"))

#distances of betadiversity - boxplots
horn_dist_long <- distance(phyloseq_rel_nz_v, method="horn") %>% as.matrix() %>% melt_dist() #making long data of distance matrices

#Adding sample type and treatment name. 
#this can be also done by merging metadata into the `horn_dist_long`
names <- data.frame(str_split_fixed(horn_dist_long$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(horn_dist_long$iso2, "_", 3))
horn_dist_long$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
horn_dist_long$method_1 <- ifelse(grepl("lyPMA", horn_dist_long$iso1),"lypma", 
                                         ifelse(grepl("ben", horn_dist_long$iso1),"benzonase", 
                                                ifelse(grepl("host", horn_dist_long$iso1),"host_zero", 
                                                       ifelse(grepl("qia", horn_dist_long$iso1),"qiaamp", 
                                                              ifelse(grepl("moly", horn_dist_long$iso1),"molysis", 
                                                                     "control")))))


#Adding data for iso 2 also should be done
horn_dist_long$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")
horn_dist_long$method_2 <-ifelse(grepl("lyPMA", horn_dist_long$iso2),"lypma", 
                                        ifelse(grepl("ben", horn_dist_long$iso2),"benzonase", 
                                               ifelse(grepl("host", horn_dist_long$iso2),"host_zero", 
                                                      ifelse(grepl("qia", horn_dist_long$iso2),"qiaamp", 
                                                             ifelse(grepl("moly", horn_dist_long$iso2),"molysis", 
                                                                    "control")))))


#subsetting distances of my interest
horn_dist_long$sample_id_1 <- ifelse(grepl("pos", horn_dist_long$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_1, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_1))
horn_dist_long$sample_id_2 <- ifelse(grepl("pos", horn_dist_long$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_2, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_2))


horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long, horn_dist_long$sample_id_1 == horn_dist_long$sample_id_2) # data within samples

horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long_within_sampleid_from_control,
                                                           horn_dist_long_within_sampleid_from_control$method_1 != horn_dist_long_within_sampleid_from_control$method_2) # remove irrelevant association

horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long_within_sampleid_from_control, (horn_dist_long_within_sampleid_from_control$method_1 == "control") + (horn_dist_long_within_sampleid_from_control$method_2 == "control") != 0)


horn_dist_long_within_sampleid_from_control$treatment <- horn_dist_long_within_sampleid_from_control$method_1

horn_dist_long_within_sampleid_from_control$treatment <- ifelse(horn_dist_long_within_sampleid_from_control$treatment == "control", horn_dist_long_within_sampleid_from_control$method_2, horn_dist_long_within_sampleid_from_control$treatment) 


#Setting key method
horn_dist_long_within_sampleid_from_control$sample_type <- ifelse(grepl("NS", horn_dist_long_within_sampleid_from_control$iso1), "Nasal",
                                                                  ifelse(grepl("CFB", horn_dist_long_within_sampleid_from_control$iso1), "Sputum",
                                                                         ifelse(grepl("BAL", horn_dist_long_within_sampleid_from_control$iso1), "BAL",
                                                                                ifelse(grepl("pos|POS", horn_dist_long_within_sampleid_from_control$iso1, ignore.case = T), "Mock",
                                                                                       ifelse(grepl("neg|N_EXT", horn_dist_long_within_sampleid_from_control$iso1), "Neg.",NA)))))

#Making a column for baseline (controls, from where?)
horn_dist_long_within_sampleid_from_control <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(dist_from = case_when(method_1 == "control" ~ iso1,
                                     method_2 == "control" ~ iso2))

dummy <- data.frame(iso1 = horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           iso2 = horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           dist = 0,
           treatment = "Untreated",
           method_1 = "control",
           method_2 = "control"
           )
names <- data.frame(str_split_fixed(dummy$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(dummy$iso2, "_", 3))
dummy$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
#Adding data for iso 2 also should be done
dummy$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")


#subsetting distances of my interest
dummy$sample_id_1 <- ifelse(grepl("pos", dummy$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_1, ignore.case = T),"Neg.",
                                        dummy$sample_id_1))
dummy$sample_id_2 <- ifelse(grepl("pos", dummy$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_2, ignore.case = T),"Neg.",
                                        dummy$sample_id_2))
dummy$sample_type <- ifelse(grepl("NS", dummy$iso1), "Nasal",
                            ifelse(grepl("CFB", dummy$iso1), "Sputum",
                                   ifelse(grepl("BAL", dummy$iso1), "BAL",
                                          ifelse(grepl("pos|POS", dummy$iso1, ignore.case = T), "Mock",
                                                 ifelse(grepl("neg|N_EXT", dummy$iso1), "Neg.",NA)))))
dummy <- subset(dummy, !is.na(dummy$sample_type))
horn_dist_long_within_sampleid_from_control <- bind_rows(horn_dist_long_within_sampleid_from_control, dummy)

#Here, sample id is the same as subject id.
horn_dist_long_within_sampleid_from_control$subject_id <- horn_dist_long_within_sampleid_from_control$sample_id_1

horn_dist_long_within_sampleid_from_control$treatment <-
        factor(horn_dist_long_within_sampleid_from_control$treatment,
               levels = c("Untreated", "lypma", "benzonase", "host_zero", "molysis", "qiaamp"))


#mh_lmer_bal_v <- lmer(dist ~ treatment + (1|subject_id),
#     data = horn_dist_long_within_sampleid_from_control %>%
#             subset(sample_type == "BAL")) 

mh_lmer_ns_v <- lmer(dist ~ treatment + (1|subject_id),
     data = horn_dist_long_within_sampleid_from_control %>%
             subset(sample_type == "Nasal")) 

mh_lmer_spt_v <- lmer(dist ~ treatment + (1|subject_id),
     data = horn_dist_long_within_sampleid_from_control %>%
             subset(sample_type == "Sputum")) 


```

Tidy, summarized-stratified analysis

```{r}
mh_lmer_bal_kbl <-  mh_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              mh_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column("treatment") %>%
        mutate(treatment = case_when(treatment == "(Intercept)" ~ "Untreated",
                                             treatment == "treatmentlypma" ~ "lyPMA",
                                             treatment == "treatmentbenzonase" ~ "Benzonase",
                                             treatment == "treatmenthost_zero" ~ "HostZERO",
                                             treatment == "treatmentmolysis" ~ "MolYsis",
                                             treatment == "treatmentqiaamp" ~ "QIAamp"
                                             )) %>%
        column_to_rownames("treatment") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("Untreated",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


mh_lmer_ns_kbl <-  mh_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              mh_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
                column_to_rownames("Row.names") %>%
        rownames_to_column("treatment") %>%
        mutate(treatment = case_when(treatment == "(Intercept)" ~ "Untreated",
                                             treatment == "treatmentlypma" ~ "lyPMA",
                                             treatment == "treatmentbenzonase" ~ "Benzonase",
                                             treatment == "treatmenthost_zero" ~ "HostZERO",
                                             treatment == "treatmentmolysis" ~ "MolYsis",
                                             treatment == "treatmentqiaamp" ~ "QIAamp"
                                             )) %>%
        column_to_rownames("treatment")  %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("Untreated",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


mh_lmer_spt_kbl <-  mh_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              mh_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
                column_to_rownames("Row.names") %>%
        rownames_to_column("treatment") %>%
        mutate(treatment = case_when(treatment == "(Intercept)" ~ "Untreated",
                                             treatment == "treatmentlypma" ~ "lyPMA",
                                             treatment == "treatmentbenzonase" ~ "Benzonase",
                                             treatment == "treatmenthost_zero" ~ "HostZERO",
                                             treatment == "treatmentmolysis" ~ "MolYsis",
                                             treatment == "treatmentqiaamp" ~ "QIAamp"
                                             )) %>%
        column_to_rownames("treatment") %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("Untreated",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]



mh_lmer_bal_kbl_v <-  mh_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              mh_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column("treatment") %>%
        mutate(treatment = case_when(treatment == "(Intercept)" ~ "Untreated",
                                             treatment == "treatmentlypma" ~ "lyPMA",
                                             treatment == "treatmentbenzonase" ~ "Benzonase",
                                             treatment == "treatmenthost_zero" ~ "HostZERO",
                                             treatment == "treatmentmolysis" ~ "MolYsis",
                                             treatment == "treatmentqiaamp" ~ "QIAamp"
                                             )) %>%
        column_to_rownames("treatment") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = "-",
               "<i>p</i>-value" = "-",
               " " = "") %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("Untreated",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


mh_lmer_ns_kbl_v <-  mh_lmer_ns_v %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              mh_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
                column_to_rownames("Row.names") %>%
        rownames_to_column("treatment") %>%
        mutate(treatment = case_when(treatment == "(Intercept)" ~ "Untreated",
                                             treatment == "treatmentlypma" ~ "lyPMA",
                                             treatment == "treatmentbenzonase" ~ "Benzonase",
                                             treatment == "treatmenthost_zero" ~ "HostZERO",
                                             treatment == "treatmentmolysis" ~ "MolYsis",
                                             treatment == "treatmentqiaamp" ~ "QIAamp"
                                             )) %>%
        column_to_rownames("treatment")  %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("Untreated",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


mh_lmer_spt_kbl_v <-  mh_lmer_spt_v %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              mh_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
                column_to_rownames("Row.names") %>%
        rownames_to_column("treatment") %>%
        mutate(treatment = case_when(treatment == "(Intercept)" ~ "Untreated",
                                             treatment == "treatmentlypma" ~ "lyPMA",
                                             treatment == "treatmentbenzonase" ~ "Benzonase",
                                             treatment == "treatmenthost_zero" ~ "HostZERO",
                                             treatment == "treatmentmolysis" ~ "MolYsis",
                                             treatment == "treatmentqiaamp" ~ "QIAamp"
                                             )) %>%
        column_to_rownames("treatment") %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("Untreated",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]



tableS7 <- 
        rbind(
                cbind(Outcome = "Microbial beta diversity",
                      mh_lmer_bal_kbl %>% rownames_to_column("Treatment"), 
                      mh_lmer_ns_kbl, 
                      mh_lmer_spt_kbl) %>% remove_rownames() %>% as.matrix(),
                cbind(Outcome = "Viral beta diversity",
                      mh_lmer_bal_kbl_v %>% rownames_to_column("Treatment"), 
                      mh_lmer_ns_kbl_v, 
                      mh_lmer_spt_kbl_v) %>% remove_rownames() %>% as.matrix()
                ) %>%
        
        kbl(format = "html", escape = 0) %>% 
        add_header_above(c(" " = 2, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        #add_rownames(c("Microbial species" = 6, "Viral species"= 6)) %>% 
        kable_styling(full_width = 0, html_font = "sans")


save_kable(tableS7, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS7.html", self_contained = T)

tableS7


```

### Fig. S7 PCA figure

**Fig. S7.** Principal coordinate analysis plot based on Morisita-Horn dissimilarity of taxonomic sequencing results stratified by sample type.

```{r}
figS7 <- ordinate(subset_samples(phyloseq_rel_nz, sample_type != "Neg." & sample_type != "Mock"), method = "PCoA", distance = "horn") %>%
        plot_ordination(phyloseq_rel_nz, ., col = "treatment") +
        #scale_color_viridis(discrete = 6, name = "Treatment", labels = c("Mock theoretical", "Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"),
                           name = "Treatment",
                           breaks = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
                           labels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        #scale_shape(name = "Sample type", labels = c("Mock theoretical", "Mock")) +
        geom_point(size = 3) +
        theme_classic (base_size = 12, base_family = "sans") +
        facet_wrap(~sample_type, scales = "free") +
        theme(plot.tag = element_text(size = 15), legend.position = "top")# +
        #stat_ellipse(type = "norm") +
        #stat_ellipse(type = "t")

figS7

png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS7.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 90, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue
figS7
dev.off()


```

### {-}

# Fig. S6. Mock community summary



Meanwhile, mock community controls are showing higher species richness than its original design.

*Plot of taxa that they shouldn't be found at Mock samples*


```{r}
#Manipulating phyloseq - only top 10 
phyloseq_control_rel <- subset_samples(phyloseq_rel_nz, sample_type == "Mock" | sample_type == "Neg.")
phyloseq_control_rel_contam <- subset_taxa(phyloseq_control_rel,
                                           !(taxa_names(phyloseq_control_rel) %in%
                                                       head(
                                                               taxa_sums(
                                                                       subset_samples(
                                                                               phyloseq_control_rel,
                                                                               sample_type == "Mock" &
                                                                                       S.obs != 0)) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10))
)

phyloseq_control_rel_contam <- subset_taxa(phyloseq_control_rel_contam, taxa_sums(phyloseq_control_rel_contam) != 0)
phyloseq_control_rel_contam <- subset_samples(phyloseq_control_rel_contam, sample_type != "Neg." & S.obs != 0)


tax_table(phyloseq_control_rel_contam) %>%
        cbind(species20 = "[Others]") %>%
        {top20species <- head(taxa_sums(phyloseq_control_rel_contam) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- .[, 9] %>% gsub("s__", "", .) %>% gsub("_", " ", .) %>% paste("<i>", ., "</i>", sep = "")
   phyloseq_temp <- phyloseq_control_rel_contam
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        plot_bar(., fill="species20") + 
        ylab("Relative abundancne") +
        theme_classic(base_size = 11, base_family = "serif") +
        ggtitle("Contaminants in Zymo mock (denominator is total microbes in samples)") +
        theme(legend.text = element_markdown()) +
        guides(fill=guide_legend(title="Top 10 species")) +
        facet_wrap (~ factor(treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")),
                    scales= "free_x", nrow=1)

#Manipulating phyloseq - only top 10 


tax_table(phyloseq_control_rel_contam) %>%
        cbind(species20 = "[Others]") %>%
        {top20species <- head(taxa_sums(phyloseq_control_rel_contam) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 20)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- .[, 9] %>% gsub("s__", "", .) %>% gsub("_", " ", .) %>% paste("<i>", ., "</i>", sep = "")
   phyloseq_temp <- phyloseq_control_rel_contam
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp %>%
           transform_sample_counts(., function(x){x/sum(x)})
  } %>%
        plot_bar(., fill="species20") + 
        ylab("Relative abundancne") +
        theme_classic(base_size = 11, base_family = "serif") +
        ggtitle("Contaminants in Zymo mock (denominator is total comtaminant)") +
        theme(legend.text = element_markdown()) +
        guides(fill=guide_legend(title="Top 20 species")) +
        facet_wrap (~ factor(treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")),
                    scales= "free_x", nrow=1)

```
The proportion of unexpected taxa was maintained throughout all the samples besides 1 lyPMA terated Mock sample. It lookws like that 1 sample had contaminant of **Cupriavidus** sp. and **Cutibacterium**. For the others, as their composition looks similar, it seems like they are taxa that assigned with wrong species name. Because,

1. If they were contaminants after host depletion, all of them should appear in treated samples as well but only some abundant taxa were remained.

2. If they were lab-contaminants before host depletion, their proportion should be different across samples but they are not.

3. Most of taxa were in the same taxonomic clades with mock communities at genus level (Cryptococcus, Listeria, Saccharomyces, Staphylococcus).

4. At least, the species richness decreased in the treated group. Therefore, it cannot be said that they were contaminants from lab wares, host depletion, extraction, sequencing, etc. If they were, it is possible that they should have been contained in the Mock community from the beginning. 

# Mock community filtering


```{r}
phyloseq_control_rel_10zymo <- subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Mock")
subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Mock") %>% taxa_names %>% .[c(grep("lacto", .))]

zymo_speceis <- c("s__Pseudomonas_aeruginosa",
                  "s__Escherichia_coli", 
                  "s__Salmonella_enterica",
                  "s__Lactobacillus_fermentum",
                  "s__Enterococcus_faecalis",
                  "s__Staphylococcus_aureus",
                  "s__Listeria_monocytogenes",
                  "s__Bacillus_subtilis",
                  "s__Saccharomyces_cerevisiae",
                  "s__Cryptococcus_neoformans")



phyloseq_control_rel_10zymo <- subset_taxa(phyloseq_control_rel_10zymo,
                                           Species %in% zymo_speceis)

sample_data(phyloseq_control_rel_10zymo)$S.obs <-
        rowSums(t(otu_table(phyloseq_control_rel_10zymo)) != 0)


f5S_mock_filtered <- ggplot(subset(sample_data(phyloseq_control_rel_10zymo) %>% 
                             data.frame,
                          sample_data(phyloseq_control_rel_10zymo)$sample_type %in%
                                  c("Mock")),
                   aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment), position = position_jitter(0.2),
                    size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        ylab("Species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        labs(tag = "C") +
        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        ylim(c(0, 10)) +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1))


phyloseq_control_rel_10zymo <- transform_sample_counts(phyloseq_control_rel_10zymo,
                                                       function(x){x/sum(x)})

barplot_mock_microbe_zymo_filtered <- phyloseq_control_rel_10zymo %>%
        tax_table() %>%
        {
   .[, 7] <- gsub("s__", " ", .[, 7])
   .[, 7] <- gsub("_", " ", .[, 7])
   .[, 7] <- gsub("[]]|[[]", "",  .[, 7])
   .[, 7] <- gsub(" sp", " sp.",  .[, 7])
   .[, 7] <- gsub(" sp.", "</i> sp.",  .[, 7])
   .[, 7] <- gsub(" group", "</i> group.",  .[, 7])
   .[, 7] <- ifelse(grepl("Other",.[, 7]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 7]),
                           paste("<i>",  .[, 7], sep = ""),
                           paste("<i>",  .[, 7], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq_control_rel_10zymo,
                              sample_type == "Mock") %>% 
           subset_samples(., S.obs != 0)
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        my_plot_bar(., fill="Species") + 
        xlab("Sample") +
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.text.x = element_text(color = "white")) +
        guides(fill=guide_legend(title="Top 10 species")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 9, name = "Paired"))) +
        facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        labs(tag = "A")
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +
        


```

# LMER test on species richness of mock - zymo species only

```{r}

phyloseq_control_rel_10zymo


sr_lmer_mock_zymo10 <- lm(S.obs ~ treatment,
                    data = sample_data(phyloseq_control_rel_10zymo) %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Mock") & S.obs != 0))

sr_lmer_mock_zymo10 %>% summary() 

```

# Species richness of mock - zymo species only - figure

```{r}

dat_text <- data.frame(
  label = c(
          "", "***", "***", "", ""#, #label for Mock
          #"", "", "", "*", "", #label for BAL
          #"", "", "***", "*", "**", 
          #"**", "***", "***", "***", "***"
          ),
  sample_type = c(
          "Mock", "Mock", "Mock", "Mock", "Mock"#, 
          #"BAL", "BAL", "BAL", "BAL", "BAL", 
          #"Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          #"Sputum", "Sputum", "Sputum", "Sputum", "Sputum"
          ),
  treatment = c(
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"#, 
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"
          ),
  S.obs = c(
          9, 8.5, 7.5, 9, 9
          #30, 35, 50, 52, 50,
          #30, 30, 30, 35, 30,
          #100, 120, 147, 140, 125
          )
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("Mock"#, 
                                                                #"BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f5S_mock_filtered <- f5S_mock_filtered + geom_text(
  data    = dat_text,
  mapping = aes(x = treatment, y = S.obs, label = label)
)
f5S_mock_filtered
```


# Mock community figure

Alpha and beta diversity indices were calculated for mock community samples and merged to FigS5.


```{r}


ggarrange(barplot_mock_microbe_zymo_filtered,
          barplot_mock_gramproportion,
          f5S_mock_filtered,
          f5S_mock_mh_box,
          align = "hv",
          heights = c(2,2,3,2),
          widths = c(1, 1, 0.8,1),
          ncol=1, nrow=4)


png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS5.png",   # The directory you want to save the file in
    width = 190, # The width of the plot in inches
    height = 200, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

ggarrange(barplot_mock_microbe_zymo_filtered,
          barplot_mock_gramproportion,
          f5S_mock_filtered,
          f5S_mock_mh_box,
          align = "hv",
          heights = c(2,2,3,2),
          widths = c(1, 1, 0.8,1),
          ncol=1, nrow=4)



dev.off()


```



# *iv. Mediation analysis*

## mediation analysis {.tabset}

### Table S4. Mediation analysis (treatment-stratified)

outcome = S.obs
exposure = treatment (stratified)
mediator = Final_reads
mediator-outcome confounders = sample_type
exposure-mediator confounders = NA
outcome model = Mixed effects linear regression 
mediator model = Mixed effects linear regression

**Mediation analysis was conducted stratified treatment.**

**Table S4.** Mediation analysis results in estimated effect sizes and p-values of indirect effect, direct effect, and proportion of mediation. The analysis employed treatment as exposure, log10(final reads) as mediator, species richness as outcome, and sample type as mediator-exposure confounder. Analysis was stratified by each treatment and treated as binary variables. 

```{r}

## lypma
# only mediator-outcome confounders
detach_package <- function(pkg, character.only = FALSE)
{
  if(!character.only)
  {
    pkg <- deparse(substitute(pkg))
  }
  search_item <- paste("package", pkg, sep = ":")
  while(search_item %in% search())
  {
    detach(search_item, unload = TRUE, character.only = TRUE)
  }
}

detach_package(lmerTest)

## all treatment groups

sample_data_respiratory <- subset(phyloseq_rel_nz %>% sample_data %>% data.frame, sample_data$sample_type == "Sputum" | sample_data$sample_type == "BAL" | sample_data$sample_type == "Nasal")


med.fit <- lmer(log10(Final_reads) ~ lypma +  sample_type + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$lypma == 1))

out.fit <- lmer(S.obs ~ lypma * log10(Final_reads) + sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$lypma == 1))
set.seed(seed)
med.out.lypma <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "lypma",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out.lypma)

final.out.lypma <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))

## benzonase


med.fit <- lmer(log10(Final_reads) ~ benzonase + sample_type + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$benzonase == 1))

out.fit <- lmer(S.obs ~ benzonase * log10(Final_reads) +  sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$benzonase == 1))

set.seed(seed)
med.out.benzonase <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "benzonase",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out.benzonase)

final.out.benzonase <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))

## host zero


med.fit <- lmer(log10(Final_reads) ~ host_zero +  sample_type + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$host_zero == 1))

out.fit <- lmer(S.obs ~ host_zero * log10(Final_reads) +  sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$host_zero == 1))
set.seed(seed)
med.out.hostzero <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "host_zero",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out.hostzero)

final.out.host_zero <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))


## molysis


med.fit <- lmer(log10(Final_reads) ~ molysis +  sample_type + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$molysis == 1))

out.fit <- lmer(S.obs ~ molysis * log10(Final_reads) +  sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$molysis == 1))
set.seed(seed)
med.out.molysis <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "molysis",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out.molysis)

final.out.molysis <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))

## qiaamp


med.fit <- lmer(log10(Final_reads) ~ qiaamp +  sample_type + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$qiaamp == 1))

out.fit <- lmer(S.obs ~ qiaamp * log10(Final_reads) + sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$qiaamp == 1))

set.seed(seed)
med.out_qiaamp <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "qiaamp",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out_qiaamp)

final.out.qiaamp <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))

```

Raw mediation outputs - lyPMA

```{r}
med.out.lypma %>% summary
```


Raw mediation outputs - Benzonase

```{r}
med.out.benzonase %>% summary
```




Raw mediation outputs - host zero

```{r}
med.out.hostzero %>% summary
```




Raw mediation outputs - MolYsis

```{r}
med.out.molysis %>% summary
```


Raw mediation outputs - qiaamp

```{r}
med.out_qiaamp %>% summary
```


Tidy table of mediation analysis result

```{r}
tableS4_a <- rbind(final.out.lypma,
      final.out.benzonase,
      final.out.host_zero, 
      final.out.molysis,
      final.out.qiaamp) %>%
mutate(Treatment = c("lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"), .before = "Indirect est.") 

tableS4_a

```

### Mediation analysis for function

outcome = S.obs (with phyloseq$)
exposure = treatment (stratified)
mediator = Final_reads
mediator-outcome confounders = sample_type
exposure-mediator confounders = NA
outcome model = Mixed effects linear regression 
mediator model = Mixed effects linear regression

**Mediation analysis was conducted stratified treatment.**

**Table S4.** Mediation analysis results in estimated effect sizes and p-values of indirect effect, direct effect, and proportion of mediation. The analysis employed treatment as exposure, log10(final reads) as mediator, species richness as outcome, and sample type as mediator-exposure confounder. Analysis was stratified by each treatment and treated as binary variables. 

```{r}

detach_package(lmerTest)

## all treatment groups

sample_data_respiratory_path <- subset(phyloseq$phyloseq_path_rpk %>% sample_data %>% data.frame, sample_data$sample_type == "Sputum" | sample_data$sample_type == "BAL" | sample_data$sample_type == "Nasal")


med.fit <- lmer(log10(Final_reads) ~ lypma +  sample_type + (1|subject_id),
                data = subset(sample_data_respiratory_path, sample_data_respiratory_path$control == 1 | sample_data_respiratory_path$lypma == 1) %>% subset(., !is.na(.$S.obs)))

out.fit <- lmer(S.obs ~ lypma * log10(Final_reads) + sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory_path, sample_data_respiratory_path$control == 1 | sample_data_respiratory_path$lypma == 1) %>% subset(., !is.na(.$S.obs)))
set.seed(seed)

med.out.lypma <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "lypma",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out.lypma)

final.out.lypma <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))

## benzonase


med.fit <- lmer(log10(Final_reads) ~ benzonase + sample_type + (1|subject_id),
                data = subset(sample_data_respiratory_path, sample_data_respiratory_path$control == 1 | sample_data_respiratory_path$benzonase == 1) %>% subset(., !is.na(.$S.obs)))

out.fit <- lmer(S.obs ~ benzonase * log10(Final_reads) +  sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory_path, sample_data_respiratory_path$control == 1 | sample_data_respiratory_path$benzonase == 1) %>% subset(., !is.na(.$S.obs)))

set.seed(seed)
med.out.benzonase <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "benzonase",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out.benzonase)

final.out.benzonase <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))

## host zero


med.fit <- lmer(log10(Final_reads) ~ host_zero +  sample_type + (1|subject_id),
                data = subset(sample_data_respiratory_path, sample_data_respiratory_path$control == 1 | sample_data_respiratory_path$host_zero == 1) %>% subset(., !is.na(.$S.obs)))

out.fit <- lmer(S.obs ~ host_zero * log10(Final_reads) +  sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory_path, sample_data_respiratory_path$control == 1 | sample_data_respiratory_path$host_zero == 1) %>% subset(., !is.na(.$S.obs)))
set.seed(seed)
med.out.hostzero <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "host_zero",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out.hostzero)

final.out.host_zero <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))


## molysis


med.fit <- lmer(log10(Final_reads) ~ molysis +  sample_type + (1|subject_id),
                data = subset(sample_data_respiratory_path, sample_data_respiratory_path$control == 1 | sample_data_respiratory_path$molysis == 1) %>% subset(., !is.na(.$S.obs)))

out.fit <- lmer(S.obs ~ molysis * log10(Final_reads) +  sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory_path, sample_data_respiratory_path$control == 1 | sample_data_respiratory_path$molysis == 1) %>% subset(., !is.na(.$S.obs)))
set.seed(seed)
med.out.molysis <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "molysis",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out.molysis)

final.out.molysis <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))

## qiaamp


med.fit <- lmer(log10(Final_reads) ~ qiaamp +  sample_type + (1|subject_id),
                data = subset(sample_data_respiratory_path, sample_data_respiratory_path$control == 1 | sample_data_respiratory_path$qiaamp == 1) %>% subset(., !is.na(.$S.obs)))

out.fit <- lmer(S.obs ~ qiaamp * log10(Final_reads) + sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory_path, sample_data_respiratory_path$control == 1 | sample_data_respiratory_path$qiaamp == 1) %>% subset(., !is.na(.$S.obs)))

set.seed(seed)
med.out_qiaamp <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "qiaamp",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out_qiaamp)

final.out.qiaamp <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))

```

Raw mediation outputs - lyPMA

```{r}
med.out.lypma %>% summary
```


Raw mediation outputs - Benzonase

```{r}
med.out.benzonase %>% summary
```


Raw mediation outputs - host zero

```{r}
med.out.hostzero %>% summary
```


Raw mediation outputs - MolYsis

```{r}
med.out.molysis %>% summary
```


Raw mediation outputs - qiaamp

```{r}
med.out_qiaamp %>% summary
```


Tidy table of mediation analysis result (for functional richness)

```{r}
tableS4_b <- rbind(final.out.lypma,
      final.out.benzonase,
      final.out.host_zero, 
      final.out.molysis,
      final.out.qiaamp) %>%
mutate(Treatment = c("lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"), .before = "Indirect est.") 

tableS4_b

```


### Table S4. Mediation analysis for viruses

outcome = V.obs
exposure = treatment (stratified)
mediator = Final_reads
mediator-outcome confounders = sample_type
exposure-mediator confounders = NA
outcome model = Mixed effects linear regression 
mediator model = Mixed effects linear regression

**Mediation analysis was conducted stratified treatment.**

**Table S4.** Mediation analysis results in estimated effect sizes and p-values of indirect effect, direct effect, and proportion of mediation. The analysis employed treatment as exposure, log10(final reads) as mediator, species richness as outcome, and sample type as mediator-exposure confounder. Analysis was stratified by each treatment and treated as binary variables. 

```{r}

## lypma
# only mediator-outcome confounders


detach_package(lmerTest) #lmerTest induces error

## all treatment groups

sample_data_respiratory <- subset(phyloseq_unfiltered$phyloseq_count %>% 
        subset_samples(#S.obs != 0 & 
                               sample_type %in% c("Mock", 
                                                       "BAL", "Nasal", "Sputum")) %>% sample_data %>% data.frame, 
                                  sample_data$sample_type == "Sputum" | sample_data$sample_type == "BAL" | sample_data$sample_type == "Nasal") %>%
        subset(., !is.na(.$V.obs))


med.fit <- lmer(log10(Final_reads) ~ lypma +  sample_type + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$lypma == 1))

out.fit <- lmer(V.obs ~ lypma * log10(Final_reads) + sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$lypma == 1))
set.seed(seed)
med.out.lypma_v <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "lypma",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out.lypma_v)

final.out.lypma_v <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))

## benzonase


med.fit <- lmer(log10(Final_reads) ~ benzonase + sample_type + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$benzonase == 1))

out.fit <- lmer(V.obs ~ benzonase * log10(Final_reads) +  sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$benzonase == 1))

set.seed(seed)
med.out.benzonase_v <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "benzonase",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out.benzonase_v)

final.out.benzonase_v <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))

## host zero


med.fit <- lmer(log10(Final_reads) ~ host_zero +  sample_type + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$host_zero == 1))

out.fit <- lmer(V.obs ~ host_zero * log10(Final_reads) +  sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$host_zero == 1))
set.seed(seed)
med.out.hostzero_v <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "host_zero",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out.hostzero_v)

final.out.host_zero_v <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))


## molysis


med.fit <- lmer(log10(Final_reads) ~ molysis +  sample_type + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$molysis == 1))

out.fit <- lmer(V.obs ~ molysis * log10(Final_reads) +  sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$molysis == 1))
set.seed(seed)
med.out.molysis_v <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "molysis",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out.molysis_v)

final.out.molysis_v <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))

## qiaamp


med.fit <- lmer(log10(Final_reads) ~ qiaamp +  sample_type + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$qiaamp == 1))

out.fit <- lmer(V.obs ~ qiaamp * log10(Final_reads) + sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$qiaamp == 1))

set.seed(seed)
med.out_qiaamp_v <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "qiaamp",
                     mediator = "log10(Final_reads)",
                     sims = 10000)

out.sum <- summary(med.out_qiaamp_v)

final.out.qiaamp_v <- data.frame(`Indirect est.`=out.sum$d.avg, 
                          `Indirect p-value`=out.sum$d.avg.p,
                          `Direct est.`=out.sum$z.avg, 
                          `Direct p-value`=out.sum$z.avg.p, 
                          `Total est.`=out.sum$tau.coef, 
                          `Total p-value`=out.sum$tau.p,
                          `Proportion mediation est.`=out.sum$n.avg, 
                          `Proportion mediation p-value`=out.sum$n.avg.p,
                          check.names = F) %>% 
    mutate_all(~round(., 3))

```

Raw mediation outputs - lyPMA

```{r}
med.out.lypma_v %>% summary
```


Raw mediation outputs - Benzonase

```{r}
med.out.benzonase_v %>% summary
```




Raw mediation outputs - host zero

```{r}
med.out.hostzero_v %>% summary
```




Raw mediation outputs - MolYsis

```{r}
med.out.molysis_v %>% summary
```


Raw mediation outputs - qiaamp

```{r}
med.out_qiaamp_v %>% summary
```


Tidy table of mediation analysis result

```{r}
tableS4_c <- rbind(final.out.lypma_v,
      final.out.benzonase_v,
      final.out.host_zero_v, 
      final.out.molysis_v,
      final.out.qiaamp_v) %>%
mutate(Treatment = c("lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"), .before = "Indirect est.") 

tableS4_c

```


### Mediation analsyis - combined table (Species and functional richness)

```{r}


tables4 <- rbind (data.frame(Outcome = "Microbial species richness",
           tableS4_a),
       data.frame(Outcome = "Functional richness",
           tableS4_b)
       ) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "sans")

save_kable(tables4, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS4.html", self_contained = T)



tables4_updated <- rbind (data.frame(Outcome = "Microbial species richness",
           tableS4_a),
       data.frame(Outcome = "Functional richness",
           tableS4_b),
       data.frame(Outcome = "Viral species richness",
           tableS4_c)
       ) %>%
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "sans")

save_kable(tables4_updated, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS4_updated.html", self_contained = T)





```



# *x.	How were changes of individual taxa?*

*x.	How were changes of individual taxa? Differential abundance analysis *
       
        1.	Volcano plot with Mock, BAL, Nasal and Sputum (Figure S5)
                a.	Maaslin (feature ~ lyPMA + Benzonase + HostZERO + MolYsis + QIAamp, RE = subject_id)
                        i.	make sure that you are doing this analysis accounting for the compositional nature of this data
                b.	Or Maaslin ( feature ~ sample type + treatment + sample type * treatment, RE = subject_id) 
        2.	Balloon plots BAL, Nasal and Sputum. (Figure 4)
                        a.	Add q-val, mean relative abundance, gram strain, phylogenetic information
        3.	List of differentiallly abundant taxa by treatment method (Table S8)
                        a.	Subset of low q-val (<0.1) & high fold-change (|est.| > 1)
                        

## DA taxa {.tabset}

### Factors affecting DA taxa (main text)

```{r, warning=FALSE, message=FALSE}
setwd("Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git")

filt_maaslin_all <- read.csv("data/da_lmer_filt_maaslin_all.csv")
#filt_maaslin_interaction <- read.csv("data/da_lmer_filt_maaslin_interaction.csv")
filt_fit_data_bal <- read.csv("data/da_lmer_filt_fit_data_bal.csv")
filt_fit_data_spt <- read.csv("data/da_lmer_filt_fit_data_spt.csv")
filt_fit_data_ns <- read.csv("data/da_lmer_filt_fit_data_ns.csv")
#filt_fit_data_pos <- read.csv("data/da_lmer_filt_fit_data_pos.csv")

```

### MaAsLin output (raw data)

```{r}
filt_maaslin_all
```

### MaAsLin output (stratified)

BAL stratified MaAsLin result

```{r}
filt_fit_data_bal
```

Nasal stratified MaAsLin result

```{r}
filt_fit_data_ns
```

Sputum stratified MaAsLin result

```{r}
filt_fit_data_spt
```

### Number of taxa on some conditions

These calculations were made to write the details in the main text

```{r}
cat("Factors affecting DA taxa (q<0.1)")
filt_maaslin_all %>% subset(., .$qval < 0.1 ) %>% subset(., abs(.$Estimate) > 1) %>% .$metadata %>% table
filt_maaslin_all %>% subset(., .$qval < 0.1 ) %>% .$metadata %>% table

cat("Number of positive & significant changes")
filt_maaslin_all %>% subset(., .$qval < 0.1 ) %>% subset(., .$Estimate > 0) %>% .$feature %>% unique

cat("Number of negative & significant changes")
filt_maaslin_all %>% subset(., .$qval < 0.1 ) %>% subset(., .$Estimate < 0) %>% .$feature %>% unique

cat("BAL highly changed taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_bal %>% subset(., .$qval < 0.1 ) %>% subset(., abs(.$Estimate) >1 ) %>% .$metadata %>% table

cat("Sputum highly changed taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_spt %>% subset(., .$qval < 0.1 ) %>% subset(., abs(.$Estimate) >1 ) %>% .$metadata %>% table

cat("NS highly changed taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_ns %>% subset(., .$qval < 0.1 ) %>% subset(., abs(.$Estimate) >1 ) %>% .$metadata %>% table



cat("NS taxa (q<0.1)")
filt_fit_data_bal %>% subset(., .$qval < 0.1 ) %>% .$feature %>% unique

cat("NS taxa (q<0.1)")
filt_fit_data_ns %>% subset(., .$qval < 0.1 ) %>% .$feature %>% unique

cat("Sputum taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_spt %>% subset(., .$qval < 0.1 ) %>% .$feature %>% unique


cat("Sputum taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_spt %>% subset(., .$qval < 0.1 ) %>% .$metadata %>% table




```

### Fig. S8. Volcano plot


**Fig. S8.** Volcano plot of differential abundance of microbes by each treatment with a model MaAsLin (relative abundance of each taxa ~ sample type + lyPMA + Benzonase + HostZERO + MolYsis + QIAamp, random effect = subject id).

```{r warning=FALSE, message=FALSE, "MaAslin Function continued2"}

#Making significance table for figure
        # Define a function to make species names italicized
# Make a significance table for each figure (top 20 taxa)
species_italic <- function(data) {
  names <- gsub("_", " ", rownames(data))
  names <- gsub("[]]|[[]", "", names)
  names <- gsub(" sp", " sp.", names)
  names <- gsub(" sp.", "* sp.", names)
  names <- gsub(" group", "", names)
  names <- ifelse(grepl("[*]", names), paste("*", names, sep = ""), paste("*", names, "*", sep = ""))
  rownames(data) <- names
  data
}
species_revise <- function(data) {
  data$feature <- gsub("Saccharomyces_cerevisiae_x_Saccharomyces_kudriavzevii", "S._cerevisiae x S._kudriavzevii", data$feature)
  data$feature <- gsub("Pseudomonas_aeruginosa_group", "Pseudomonas_aeruginosa", data$feature)
  data$feature <- gsub("Pseudomonas_fluorescens_group", "Pseudomonas_fluorescens", data$feature)
  data
}



make_sig_table <- function(data) {
  sig_data <- spread(data[order(data$qval), c("feature", "metadata", "qval")], metadata, qval)
  sig_data <- species_revise(sig_data)
  sig_data$min <- apply(sig_data %>% dplyr::select(c("lypma", "benzonase", "molysis", "host_zero", "qiaamp")), 1, FUN = min)
  sig_data <- sig_data[order(sig_data$min),] %>% dplyr::select("feature", "lypma", "benzonase", "host_zero", "molysis", "qiaamp") %>% .[1:20,]
  sig_data[["feature"]] <- ifelse(sig_data[["feature"]] == "X.Collinsella._massiliensis", "[Collinsella]_massiliensis", sig_data[["feature"]])
  sig_data_italic <- sig_data %>% rownames_to_column(var = "-") %>%
          column_to_rownames(var = "feature") %>% species_italic %>% dplyr::select(-c("-")) %>%
          rename(lyPMA = lypma,  Benzonase = benzonase, `HostZERO` = host_zero, MolYsis = molysis, QIAamp = qiaamp)
  sig_data_sig <- ifelse(sig_data_italic < 0.1, "*", NA) %>% data.frame(check.names = F)
  return(list(data = sig_data, data_italic = sig_data_italic, data_sig = sig_data_sig))
}



#filt_fit_data_pos <- make_sig_table(filt_fit_data_pos)
filt_fit_data_bal <- make_sig_table(filt_fit_data_bal)
filt_fit_data_ns <- make_sig_table(filt_fit_data_ns)
filt_fit_data_spt <- make_sig_table(filt_fit_data_spt)

#filt_pos_sig <- subset_taxa(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Mock"),
#                                       taxa_names(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Mock")) %in% filt_fit_data_pos$data$feature)

#filt_fit_data_pos$rel <- cbind(filt_pos_sig %>% otu_table %>% t, filt_pos_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>% 
#        .[row.names(filt_fit_data_pos$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")

filt_spt_sig <- subset_taxa(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Sputum"), taxa_names(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Sputum")) %in% filt_fit_data_spt$data$feature)

filt_fit_data_spt$rel <- cbind(filt_spt_sig %>% otu_table %>% t, filt_spt_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>% 
        .[row.names(filt_fit_data_spt$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")

filt_ns_sig <- subset_taxa(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Nasal"),
                                       taxa_names(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Nasal")) %in% filt_fit_data_ns$data$feature)

filt_fit_data_ns$rel <- cbind(filt_ns_sig %>% otu_table %>% t, filt_ns_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>% 
        .[row.names(filt_fit_data_ns$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")

filt_bal_sig <- subset_taxa(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "BAL"),
                                       taxa_names(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "BAL")) %in% filt_fit_data_bal$data$feature)

filt_fit_data_bal$rel <- cbind(filt_bal_sig %>% otu_table %>% t, filt_bal_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>%
        .[row.names(filt_fit_data_bal$data_italic),] %>%
        mutate_all(~na_if(., 0)) %>% rownames_to_column("feature") %>% subset(., !grepl("NA", .$feature))


```


```{r}

#Volcano plot

figS7 <- ggplot(filt_maaslin_all, aes(y = -log10(qval), x = Estimate, col = metadata)) +
        theme_classic(base_family = "sans") +
        #labs(tag = "A") +
        geom_point(size = 2, alpha = 0.3, stroke = 0) +
        xlab("Change estimate of CLR-normalized read counts") +
        ylab("-log<sub>10</sub>(*q*-value)") +
        #ylim(c(-1, 35)) +
        geom_hline(yintercept = 1, col = "gray") +
        geom_vline(xintercept = 0, col = "gray") +
        annotate(family = "sans",
                 geom='richtext',
                 x=0, y=20,
                 label = "<i>q</i>-value = 0.1, fold-change = 0") +
        theme(legend.position = "top", axis.title.y = ggtext::element_markdown(), legend.text = element_markdown()) +
        scale_color_manual(values = c("#a65628",
                                      "grey",
                                      "#fb9a99",
                                      "#33a02c",
                                      "#b2df8a",
                                      "#1f78b4",
                                      "#a6cee3"),
                           breaks = c("log10.Final_reads",
                                      "sample_type",
                                      "lypma",
                                      "benzonase",
                                      "host_zero",
                                      "molysis",
                                      "qiaamp"), 
                           labels = c("log<sub>10</sub>(Final reads)",
                                      "Sample type",
                                      "lyPMA",
                                      "Benzonase",
                                      "HostZERO",
                                      "MolYsis",
                                      "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        guides(col = guide_legend(title = "Factors", title.position = "top", nrow = 1))

figS7
png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS8.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 90, # The height of the plot in inches/
    units = "mm",
    res = 600
) #fixing multiple page issue
figS7
dev.off()

```


### Fig. 4. Balloon plot

**Fig. 4.** Mean relative abundance of top 20 significant taxa by q-value identified by differential abundance analysis using MaAsLin. Analyses were stratified by sample type. (A) bronchoalveolor lavage, (B) nasal swabs, and (C) sputum. Statistical significances were noted at the level of q-value < 0.1



```{r}

#ffff33 qia

f5b <- merge(filt_fit_data_bal$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_bal$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(filt_fit_data_bal$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(is.na(qval) ~ "> 0.1",
                               qval < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%

#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        theme_classic(base_family = "sans") +
        #colors for qvalues
        #xlab("Experimental group") +
        #ylab("Species") +
        labs(tag = "A")  +
        ggtitle("BAL") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              #axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 8),
              axis.title.y = element_blank(),
              axis.title.x = element_blank(),
              plot.margin = unit(c(0,0.2,0,1), 'lines'))  +
        scale_fill_manual(values = c("grey", "red"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1)
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "MolYsis",
                                  "qiaamp" = "QIAamp")
                         )

f5c <- merge(filt_fit_data_ns$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_ns$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(filt_fit_data_ns$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "sans") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        ylab("Species") +
        labs(tag = "B")  +
        ggtitle("Nasal swab") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              #axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 8),
              axis.title.x = element_blank(),
              plot.margin = unit(c(0,0.2,0,1), 'lines'))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top",
                                   override.aes = list(size=5)),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "MolYsis",
                                  "qiaamp" = "QIAamp")
                         )

f5d <- merge(filt_fit_data_spt$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_spt$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        merge(filt_fit_data_spt$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "sans") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        #ylab("Species") +
        labs(tag = "C")  +
        ggtitle("Sputum") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              #axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 8),
              #axis.title.x = element_blank(),
              axis.title.x = element_text(margin = margin(t = 0)),
              axis.title.y = element_blank(),
              plot.margin = unit(c(0,0.2,0,1), 'lines'))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "MolYsis",
                                  "qiaamp" = "QIAamp")
                         )

fig4 <- ggarrange(f5c %>% lemon::g_legend() %>% as_ggplot,
                  f5b,
                  f5c,
                  f5d,
                  #b, b %>% lemon::g_legend() %>% as_ggplot,
                  ncol=1, heights = c(1.5, 4, 4, 4),
                  legend = "none",
                  align = "hv")

fig4

png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/Figure4.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 220, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

fig4

# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()

```


### Table S8. List of DA taxa

**Table S8.** List of differentially abundant taxa identified by MaAsLin (Beghini et al., 2021). Significant associations (q-value < 0.1) that is showing high change ( | log2(fold-change) | > 0.5) were listed.

```{r, warning=FALSE, message=FALSE}

filt_maaslin_all %>% subset(., .$qval < 0.1 & abs(.$Estimate) > 4) 

tableS8 <- filt_maaslin_all %>% subset(., .$qval < 0.1 & abs(.$Estimate) > 4) %>% dplyr::select(c("feature", "metadata", "Estimate", "qval")) %>%
        mutate(feature = paste("<i>", gsub("_", " ", feature), sep = "") %>%
                       gsub(" sp", "</i> sp.", .) %>% gsub(" group", "", .)) %>%
        mutate(feature = case_when(!grepl("</i>", feature) ~ paste(feature, "</i>", sep = ""),
                                   .default = feature))%>% 
        rename(Taxa = "feature",
                     `Fixed effect` = "metadata",
                     "Change estimate of CLR transformed count" = "Estimate",
                     `<i>q<i/>-value` = "qval") %>% 
        mutate(`Fixed effect` = case_when(
                         `Fixed effect` == "sample_type" ~ "Sample type",
                         `Fixed effect` == "lypma" ~ "lyPMA",
                         `Fixed effect` == "host_zero" ~ "HostZERO",
                         `Fixed effect` == "benzonase" ~ "Benzonase",
                         `Fixed effect` == "molysis" ~ "MolYsis",
                         `Fixed effect` == "qiaamp" ~ "QIAamp",
                         .default = `Fixed effect`),
               "Change estimate of CLR transformed count" = round(`Change estimate of CLR transformed count`, 3),
               `<i>q<i/>-value` = round(`<i>q<i/>-value`, 3)) %>%
        remove_rownames() %>% 
        kbl(format = "html", escape = FALSE) %>%
        kable_styling(full_width = 0, html_font = "sans")

tableS8

save_kable(tableS8, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS8.html", self_contained = T)

```



### {-}

# **3.4. Effect of treatments on functional analysis results**

# *xi.	Did depletion methods change diversity, by sample type?*


## Function diveristy {.tabset}

### Fig. S9. Figure of alpha and beta 

**Fig. S9.** Alpha and beta diversity by sample type and treatment method of predicted functions. (A) Species richness with statistical test results (linear mixed effect model stratified by sample type), (B) Morisita-Horn dissimilarity within subject between treatment, representing squares for median value and bars for 95% confidence intervals.  

```{r warning=FALSE, "Functional richness"}

sample_data <- sample_data(phyloseq$phyloseq_path_rpk) %>% data.frame(check.names = F) %>% subset(., !is.nan(.$simpson))
phyloseq_rel_nz_f <- subset_samples(phyloseq$phyloseq_path_rpk, S.obs != 0 & sample_type %in% c("BAL", "Nasal", "Sputum", "Mock"))

sample_data(phyloseq_rel_nz_f)$log10.Final_reads <- log10(sample_data(phyloseq_rel_nz_f)$Final_reads)
sample_data(phyloseq_rel_nz_f)$sampletype_treatment <- paste(sample_data(phyloseq_rel_nz_f)$sample_type, sample_data(phyloseq_rel_nz_f)$treatment, sep = ":")

```

```{r}


f4a <- ggplot(subset(sample_data(phyloseq$phyloseq_path_rpk) %>% 
                             data.frame,
                     sample_data(phyloseq$phyloseq_path_rpk)$sample_type %in% 
                             c("Sputum","Nasal", "BAL"#, "Mock"
                               )), aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment), position = position_jitter(0.2), size = 1.2,
                    alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        ylab("Functional richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        labs(tag = "A") +
        theme(plot.tag = element_text(size = 15),  axis.text.x = element_blank(), axis.ticks.x = element_blank()) +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 

dat_text <- data.frame(
  label = c(
          #"**", "***", "***", "", "***", #label for Mock
          "", "*", "**", "***", "*", #label for BAL
          "", "", "***", "**", "***", 
          "*", "**", "***", "***", "***"),
  sample_type = c(
          #"Mock", "Mock", "Mock", "Mock", "Mock", 
          "BAL", "BAL", "BAL", "BAL", "BAL", 
          "Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          "Sputum", "Sputum", "Sputum", "Sputum", "Sputum"),
  treatment     = c(
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
  S.obs = c(
          #420, 400, 330, 410, 400,
          250, 230, 300, 320, 300,
          210, 220, 230, 240, 250,
          330, 350, 370, 350, 360)
)

dat_text$sample_type <- factor(dat_text$sample_type, levels = c(#"Mock", 
                                                                "BAL", "Nasal", "Sputum"))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))



f4a <- f4a + geom_text(
  data    = dat_text,
  mapping = aes(x = treatment, y = S.obs, label = label)
)


#f3a <- f3a + geom_text(
#  data    = dat_text,
#  mapping = aes(x = treatment, y = S.obs, label = label)
#)


#distances of betadiversity - boxplots
horn_dist_long <- distance(phyloseq_rel_nz_f, method="horn") %>% as.matrix() %>% melt_dist() #making long data of distance matrices

#Adding sample type and treatment name. 
#this can be also done by merging metadata into the `horn_dist_long`
names <- data.frame(str_split_fixed(horn_dist_long$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(horn_dist_long$iso2, "_", 3))
horn_dist_long$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
horn_dist_long$method_1 <- ifelse(grepl("lyPMA", horn_dist_long$iso1),"lypma", 
                                         ifelse(grepl("ben", horn_dist_long$iso1),"benzonase", 
                                                ifelse(grepl("host", horn_dist_long$iso1),"host_zero", 
                                                       ifelse(grepl("qia", horn_dist_long$iso1),"qiaamp", 
                                                              ifelse(grepl("moly", horn_dist_long$iso1),"molysis", 
                                                                     "control")))))


#Adding data for iso 2 also should be done
horn_dist_long$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")
horn_dist_long$method_2 <-ifelse(grepl("lyPMA", horn_dist_long$iso2),"lypma", 
                                        ifelse(grepl("ben", horn_dist_long$iso2),"benzonase", 
                                               ifelse(grepl("host", horn_dist_long$iso2),"host_zero", 
                                                      ifelse(grepl("qia", horn_dist_long$iso2),"qiaamp", 
                                                             ifelse(grepl("moly", horn_dist_long$iso2),"molysis", 
                                                                    "control")))))


#subsetting distances of my interest
horn_dist_long$sample_id_1 <- ifelse(grepl("pos", horn_dist_long$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_1, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_1))
horn_dist_long$sample_id_2 <- ifelse(grepl("pos", horn_dist_long$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_2, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_2))


path_horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long, horn_dist_long$sample_id_1 == horn_dist_long$sample_id_2) # data within samples

path_horn_dist_long_within_sampleid_from_control <- subset(path_horn_dist_long_within_sampleid_from_control,
                                                           path_horn_dist_long_within_sampleid_from_control$method_1 != path_horn_dist_long_within_sampleid_from_control$method_2) # remove irrelevant association

path_horn_dist_long_within_sampleid_from_control <- subset(path_horn_dist_long_within_sampleid_from_control, (path_horn_dist_long_within_sampleid_from_control$method_1 == "control") + (path_horn_dist_long_within_sampleid_from_control$method_2 == "control") != 0)


path_horn_dist_long_within_sampleid_from_control$treatment <- path_horn_dist_long_within_sampleid_from_control$method_1

path_horn_dist_long_within_sampleid_from_control$treatment <- ifelse(path_horn_dist_long_within_sampleid_from_control$treatment == "control", path_horn_dist_long_within_sampleid_from_control$method_2, path_horn_dist_long_within_sampleid_from_control$treatment) 


#Setting key method
path_horn_dist_long_within_sampleid_from_control$sample_type <- ifelse(grepl("NS", path_horn_dist_long_within_sampleid_from_control$iso1), "Nasal",
                                                                  ifelse(grepl("CFB", path_horn_dist_long_within_sampleid_from_control$iso1), "Sputum",
                                                                         ifelse(grepl("BAL", path_horn_dist_long_within_sampleid_from_control$iso1), "BAL",
                                                                                ifelse(grepl("pos|POS", path_horn_dist_long_within_sampleid_from_control$iso1, ignore.case = T), "Mock",
                                                                                       ifelse(grepl("neg|N_EXT", path_horn_dist_long_within_sampleid_from_control$iso1), "Neg.",NA)))))

#Making a column for baseline (controls, from where?)
path_horn_dist_long_within_sampleid_from_control <- path_horn_dist_long_within_sampleid_from_control %>% 
        mutate(dist_from = case_when(method_1 == "control" ~ iso1,
                                     method_2 == "control" ~ iso2))

dummy <- data.frame(iso1 = path_horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           iso2 = path_horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           dist = 0,
           treatment = "Untreated",
           method_1 = "control",
           method_2 = "control"
           )
names <- data.frame(str_split_fixed(dummy$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(dummy$iso2, "_", 3))
dummy$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
#Adding data for iso 2 also should be done
dummy$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")


#subsetting distances of my interest
dummy$sample_id_1 <- ifelse(grepl("pos", dummy$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_1, ignore.case = T),"Neg.",
                                        dummy$sample_id_1))
dummy$sample_id_2 <- ifelse(grepl("pos", dummy$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_2, ignore.case = T),"Neg.",
                                        dummy$sample_id_2))
dummy$sample_type <- ifelse(grepl("NS", dummy$iso1), "Nasal",
                            ifelse(grepl("CFB", dummy$iso1), "Sputum",
                                   ifelse(grepl("BAL", dummy$iso1), "BAL",
                                          ifelse(grepl("pos|POS", dummy$iso1, ignore.case = T), "Mock",
                                                 ifelse(grepl("neg|N_EXT", dummy$iso1), "Neg.",NA)))))
dummy <- subset(dummy, !is.na(dummy$sample_type))
path_horn_dist_long_within_sampleid_from_control <- bind_rows(path_horn_dist_long_within_sampleid_from_control, dummy)


path_horn_dist_long_within_sampleid_from_control$subject_id <- path_horn_dist_long_within_sampleid_from_control$sample_id_1

path_horn_dist_long_within_sampleid_from_control$treatment <-
        factor(path_horn_dist_long_within_sampleid_from_control$treatment,
               levels = c("Untreated", "lypma", "benzonase", "host_zero", "molysis", "qiaamp"))


f4b2 <- path_horn_dist_long_within_sampleid_from_control %>% 
        mutate(across(sample_type, factor, levels=c(#"Mock", 
                                                    "BAL", "Nasal","Sputum"
                                                    ))) %>%
        subset(., .$sample_type != "Neg.") %>% 
        subset(., .$treatment != "Untreated") %>%
  mutate(treatment = factor(treatment, levels = c("lypma", "benzonase", "host_zero", "molysis", "qiaamp"),
                            labels = c("lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))) %>%
        ggplot(aes(x = dist, y = treatment, col = treatment)) +
        geom_boxplot() + 
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c(#"#e31a1c",
                                      "#fb9a99","#33a02c",
                                      "#b2df8a","#1f78b4","#a6cee3"),
                           name = "Treatment",
                           breaks = c(#"Untreated", 
                                      "lyPMA", "Benzonase",
                                      "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Morisita-Horn dissimilarity from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              legend.position = "none") +
        labs(tag = "B") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "sans") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "sans") +
        scale_x_continuous(breaks = c(-0.25, 0, 0.25, 0.5, 0.75),
                           labels = c(-0.25, "0 (low bias)", 0.25, 0.5, "0.75 (high bias)"))




figS8 <- ggarrange(f4a, f4b2, ncol = 1, common.legend = T, align = "hv") +
        guides(fill = guide_legend(nrow = 1))


figS8

png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS10.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 170, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue
figS8

# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()


```


### Functional richness - LMER all samples

**Effect of some treatment was neutralized by interaction term. Therefore, the association was sample_type specific.**

Effect size, standard error (SE) and p-value of a statistical test for functional richness with an interaction term using linear mixed effect model (Species richness ~ sample type * treatment + (1|subject_id) ).


raw result

```{r}
library(lmerTest)
sample_data <- sample_data(phyloseq$phyloseq_path_rpk)
sample_data$log_centered_final_reads <- log(sample_data$Final_reads + 1) - median(log((subset(sample_data, sample_data$sample_type %in% c("BAL") & sample_data$treatment %in% c("Untreated")) %>% .$Final_reads) + 1))

lmer(S.obs ~ sample_type * treatment + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum"))) 

```

Tidy table

```{r}


lmer(S.obs ~ sample_type * treatment + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("sample_type|treatment", "", x))  %>% mutate(x = gsub(":", " * ", x)) %>% 
        column_to_rownames(var = "x") %>%  
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = 1), 
                                " (", 
                                round(Estimate - 1.96 * abs(t_value), 1) %>% format(nsmall = 1),
                                ", ",
                                round(Estimate + 1.96 * abs(t_value), 1) %>% format(nsmall = 1),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>% 
        kbl(format = "html", escape = 0) %>% kable_styling(full_width = 0, html_font = "sans")



```

### Table S9. Functional richness - all & stratified

**Table S9.** Effect size, standard error (SE) and p-value of a statistical test for functional richness with an interaction term using linear mixed effect model (functional richness ~ treatment + (1|subject_id)). Stratified analyses were conducted for each sample type as an interaction term of sample type and treatment was significant (p-value < 0.001) at ANOVA of LMER(functional richness ~ sample type + treatment + sample type * treatment + (1|subject_id)). The baseline of categorical variables is untreated group. Statistical significances were noted with **: p-value < 0.01 and ***: p-value < 0.001.

Association was not adjusted with sequencing depth.

Raw result - Mock

```{r}
pfr_lmer_mock <- lm(S.obs ~ treatment, data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Mock"))) 
pfr_lmer_mock %>% summary
```

Raw result - BAL

```{r}



dummy1 <- ((phyloseq$phyloseq_path_rpk %>% otu_table %>% data.frame(check.names = F)) != 0) %>% colSums()


pfr_lmer_bal <- lm(S.obs ~ treatment, data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("BAL"))) 

pfr_lmer_bal %>% summary
```

Raw result - BAL

```{r}
pfr_lmer_ns <- lm(S.obs ~ treatment, data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Nasal"))) 
pfr_lmer_ns %>% summary
```



Raw result - BAL

```{r}
pfr_lmer_spt <- lm(S.obs ~ treatment, data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Sputum"))) 
pfr_lmer_spt %>% summary
```


Tidy table

```{r}

pfr_lmer_mock_kbl <- pfr_lmer_mock %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              pfr_lmer_mock %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

pfr_lmer_bal_kbl <- pfr_lmer_bal %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              pfr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        
pfr_lmer_ns_kbl <- pfr_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              pfr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        
pfr_lmer_spt_kbl <-  pfr_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              pfr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


tables9 <- cbind(pfr_lmer_bal_kbl, pfr_lmer_ns_kbl, pfr_lmer_spt_kbl) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")

tables9



save_kable(tables9, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS9.html", self_contained = T)


```

### Function beta - all samples 

Rwa result, with interaction term

```{r}

phyloseq_rel_nz_f <- transform_sample_counts(phyloseq$phyloseq_path_rpk, function(x) {x/sum(x)}) %>%
        subset_samples(S.obs != 0 & sample_type %in% c("Mock", "BAL", "Nasal", "Sputum"))


set.seed(seed)
horn_perm_inter_fun <- vegan::adonis2(by = "terms", distance(phyloseq_rel_nz_f, method="horn") ~ sample_type * treatment + subject_id,
                                  data = phyloseq_rel_nz_f %>% sample_data %>% data.frame(check.names = F), 
                                  strata = phyloseq_rel_nz_f %>% sample_data %>% data.frame(check.names = F) %>% .$subject_id,
                                  permutations = 10000)
set.seed(seed)
horn_perm_ns_fun <- vegan::adonis2(by = "terms", distance(subset_samples(phyloseq_rel_nz_f, sample_type == "Nasal"), method="horn") ~ lypma + benzonase + host_zero + molysis + qiaamp,
                               data = subset_samples(phyloseq_rel_nz_f, sample_type == "Nasal") %>%
                                       sample_data %>% data.frame(check.names = F),
                               strata = subset_samples(phyloseq_rel_nz_f, sample_type == "Nasal") %>% 
                                       sample_data %>% data.frame(check.names = F) %>% .$subject_id, permutations = 10000)
set.seed(seed)
horn_perm_bal_fun  <- vegan::adonis2(by = "terms", distance(subset_samples(phyloseq_rel_nz_f, sample_type == "BAL"), method="horn") ~  lypma + benzonase + host_zero + molysis + qiaamp,
                                 data = subset_samples(phyloseq_rel_nz_f, sample_type == "BAL") %>% sample_data %>% data.frame(check.names = F),
                                 strata = subset_samples(phyloseq_rel_nz_f, sample_type == "BAL") %>%
                                         sample_data %>% data.frame(check.names = F) %>% .$subject_id,
                                  permutations = 10000)
set.seed(seed)
horn_perm_spt_fun <- vegan::adonis2(by = "terms", distance(subset_samples(phyloseq_rel_nz_f, sample_type == "Sputum"), method="horn") ~ lypma + benzonase + host_zero + molysis + qiaamp,
                                data = subset_samples(phyloseq_rel_nz_f, sample_type == "Sputum") %>% sample_data %>% data.frame(check.names = F),
                                strata = subset_samples(phyloseq_rel_nz_f, sample_type == "Sputum")
                                %>% sample_data %>% data.frame(check.names = F) %>% .$subject_id,
                                  permutations = 10000)


horn_perm_inter_fun
```

Tidy table of PERMANOVA result

```{r}
horn_perm_inter_fun %>% data.frame(check.names = F) %>% rownames_to_column("row.names") %>% 
        mutate(row.names = case_when(row.names == "sample_type" ~ 'Sample type',
                                     row.names == "treatment" ~ 'Treatment',
                                     row.names == "subject_id" ~ 'Subject',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "sample_type:treatment" ~ 'Sample type * Treatment',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
        mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3),
               `Pr(>F)` = format(`Pr(>F)`, nsmall = 3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        dplyr::select(c("Degree of freedom", "R<sup>2</sup>", "<i>p</i>-value", " ")) %>% 
        kbl(format = "html", escape = 0) %>%
        kable_styling(full_width = 0, html_font = "sans")


```

### Function beta - stratified

**Not included in the main text**

**Table**. Degree of freedom, effect size (residual, R^2) and p-value of permutational ANOVA for functional Horn-Morisita distances with an interaction term and strata term (BC-distance of functions ~ sample type * treatment + log10(final reads), strata = subject id).

Raw result - BAL

```{r}
horn_perm_bal_fun
```


Raw result - Nasal

```{r}
horn_perm_ns_fun
```

Raw result - Sputum

```{r}
horn_perm_spt_fun
```
Tidy table

```{r}

a <- horn_perm_bal_fun %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'HostZERO',
                                     row.names == "molysis" ~ 'MolYsis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        dplyr::select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 

b <- horn_perm_ns_fun %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'HostZERO',
                                     row.names == "molysis" ~ 'MolYsis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        dplyr::select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 

c <- horn_perm_spt_fun %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'HostZERO',
                                     row.names == "molysis" ~ 'MolYsis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        dplyr::select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 


tables9_A <- cbind(a, b, c) %>% 
        kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")


tables9_A

```


### {-}



# *xii.	What type of fuction were affected by the treatment?*

## Function DA analysis {.tabset}

```{r warning=FALSE, message=FALSE,  "MaAslin Function"}

#DA analysis - MaAslin
#Running MaAslin for all sample without decontam
#for taxa differentially abundant by host depletion method, look to see which ones overlap with potential contaminant taxa

# Maaslin - # # y ~ log(final reads) + sample_type + treatment  -----------

#all samples
```


```{r}

f_maaslin_all <- read.csv("Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git/data/da_lmer_filt_f_maaslin_all.csv")
f_fit_data_bal <- read.csv("Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git/data/da_lmer_filt_f_fit_data_bal.csv")
f_fit_data_spt <- read.csv("Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git/data/da_lmer_filt_f_fit_data_spt.csv")
f_fit_data_ns <- read.csv("Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git/data/da_lmer_filt_f_fit_data_ns.csv")
#f_fit_data_pos <- read.csv("Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git/data/da_lmer_filt_f_fit_data_pos.csv")


```


**Again, most of DA functions were sample type specific**

```{r warning=FALSE, message=FALSE, "MaAslin Function continued"}

#Making significance table for figure
        # Define a function to make species names italicized
# Make a significance table for each figure (top 20 taxa)
make_sig_table <- function(data) {
  sig_data <- spread(data[order(data$qval), c("feature", "metadata", "qval")], metadata, qval)
  sig_data$feature <- gsub("[.]", "-", sig_data$feature)
  sig_data$min <- apply(sig_data %>% dplyr::select(c("lypma", "benzonase", "molysis", "host_zero", "qiaamp")), 1, FUN = min)
  sig_data <- sig_data[order(sig_data$min),] %>% dplyr::select("feature", "lypma", "benzonase", "host_zero", "molysis", "qiaamp") %>% .[1:20,]
  sig_data[["feature"]] <- ifelse(sig_data[["feature"]] == "X.Collinsella._massiliensis", "[Collinsella]_massiliensis", sig_data[["feature"]])
  sig_data_italic <- sig_data %>% rownames_to_column(var = "-") %>%
          column_to_rownames(var = "feature") %>% dplyr::select(-c("-")) %>%
          rename(lyPMA = lypma,  Benzonase = benzonase, `HostZERO` = host_zero, MolYsis = molysis, QIAamp = qiaamp)
  sig_data_sig <- ifelse(sig_data_italic < 0.1, "*", NA) %>% data.frame(check.names = F)
  return(list(data = sig_data, data_italic = sig_data_italic, data_sig = sig_data_sig))
}

#f_fit_data_pos <- make_sig_table(f_fit_data_pos)
f_fit_data_bal <- make_sig_table(f_fit_data_bal)
f_fit_data_ns <- make_sig_table(f_fit_data_ns)
f_fit_data_spt <- make_sig_table(f_fit_data_spt)

#f_pos_sig <- subset_taxa(subset_samples(phyloseq_rel_nz, sample_type == "Mock"),
#                                       taxa_names(subset_samples(phyloseq_rel_nz, sample_type == "Mock")) %in% f_fit_data_pos$data$feature)
#f_fit_data_pos$rel <- cbind(f_pos_sig %>% otu_table %>% t, f_pos_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% data.frame(check.names = F) %>% 
#        .[row.names(f_fit_data_pos$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")


f_spt_sig <- subset_taxa(subset_samples(phyloseq_rel_nz_f, sample_type == "Sputum"),
                                       taxa_names(subset_samples(phyloseq_rel_nz_f, sample_type == "Sputum")) %in% f_fit_data_spt$data$feature)


f_fit_data_spt$rel <- cbind(f_spt_sig %>% otu_table %>% t, f_spt_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% data.frame(check.names = F) %>% 
        .[row.names(f_fit_data_spt$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")

f_ns_sig <- subset_taxa(subset_samples(phyloseq_rel_nz_f, sample_type == "Nasal"),
                                       taxa_names(subset_samples(phyloseq_rel_nz_f, sample_type == "Nasal")) %in% f_fit_data_ns$data$feature)

f_fit_data_ns$rel <- cbind(f_ns_sig %>% otu_table %>% t, f_ns_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% data.frame(check.names = F) %>% 
        .[row.names(f_fit_data_ns$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")
f_fit_data_ns$rel$feature <- row.names(f_fit_data_ns$data_sig)

f_bal_sig <- subset_taxa(subset_samples(phyloseq_rel_nz_f, sample_type == "BAL"),
                                       taxa_names(subset_samples(phyloseq_rel_nz_f, sample_type == "BAL")) %in% f_fit_data_bal$data$feature)

f_fit_data_bal$rel <- cbind(f_bal_sig %>% otu_table %>% t, f_bal_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% data.frame(check.names = F) %>%
        .[row.names(f_fit_data_bal$data_italic),] %>%
        mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")



```

### MaAslin renaming

```{r}

rename_function_gruop <- function(list_maaslin_result){
        taxa_df <- tax_table(phyloseq$phyloseq_path_cpm) %>% 
                data.frame %>% remove_rownames() %>% rename(feature = "pathway")
        tax_table(phyloseq$phyloseq_path_cpm)
        list_maaslin_result$data <-
                list_maaslin_result$data %>% 
                merge(., taxa_df, by = "feature") %>% 
                dplyr::select(-c("feature")) %>%
                rename(feature = "group") %>%
                dplyr::select(c("feature", "lypma", "benzonase", "host_zero", "molysis", "qiaamp"))
        list_maaslin_result$data_italic <- 
                list_maaslin_result$data_italic %>% 
                rownames_to_column("feature") %>%
                merge(., taxa_df, by = "feature") %>% 
                dplyr::select(-c("feature")) %>%
                column_to_rownames("group")
        list_maaslin_result$data_sig <-
                list_maaslin_result$data_sig  %>% 
                        rownames_to_column("feature") %>%
                merge(., taxa_df, by = "feature") %>% 
                dplyr::select(-c("feature")) %>%
                column_to_rownames("group")
        list_maaslin_result$rel <-
                list_maaslin_result$rel %>% 
                merge(., taxa_df, by = "feature") %>% 
                dplyr::select(-c("feature")) %>%
                rename(feature = "group") %>%
                dplyr::select(c("feature", "Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))
        list_maaslin_result
        
}


#f_fit_data_pos <- rename_function_gruop(f_fit_data_pos)
f_fit_data_bal <- rename_function_gruop(f_fit_data_bal)
f_fit_data_ns <- rename_function_gruop(f_fit_data_ns)
f_fit_data_spt <- rename_function_gruop(f_fit_data_spt)

```

### Functional MaAsLin raw results

```{r}
f_maaslin_all
```


### Fig. S10 MaAslin function - volcano plot 

Fig. S10. Volcano plot of differential abundance of function by each treatment with a model MaAsLin (copies per million of each function ~ sample type + lyPMA + Benzonase + HostZERO + MolYsis + QIAamp, random effect = subject id).

```{r}

#Volcano plot

figS9 <- ggplot(f_maaslin_all, aes(y = -log10(qval), x = Estimate, col = metadata)) +
        theme_classic(base_family = "sans") +
        #labs(tag = "A") +
        geom_point(size = 2, alpha = 0.3, stroke = 0) +
        xlab("Change estimate of CLR-normalized CPM") +
        ylab("-log<sub>10</sub>(*q*-value)") +
        geom_hline(yintercept = 1, col = "gray") +
        geom_vline(xintercept = 0, col = "gray") +
        annotate(family = "sans",
                 geom='richtext',
                 x=0, y=10,
                 label = "<i>q</i>-value = 0.1, fold-change = 0") +
        theme(legend.position = "top", axis.title.y = ggtext::element_markdown()) +
        scale_color_manual(values = c("#a65628",
                                      "grey",
                                      "#fb9a99",
                                      "#33a02c",
                                      "#b2df8a",
                                      "#1f78b4",
                                      "#a6cee3"),
                           breaks = c("log10.Final_reads",
                                      "sample_type",
                                      "lypma",
                                      "benzonase",
                                      "host_zero",
                                      "molysis",
                                      "qiaamp"), 
                           labels = c("log<sub>10</sub>(Final reads)",
                                      "Sample type",
                                      "lyPMA",
                                      "Benzonase",
                                      "HostZERO",
                                      "MolYsis",
                                      "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        guides(col = guide_legend(title = "Factors", title.position = "top", nrow = 2))


png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS11.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 90, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figS9
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()
figS9

```

### Fig. S11 Function baloon plot 

**Fig. S11.** Mean copies per million of top 20 significant function identified by differential abundance analysis using MaAsLin. Analyses were stratified by sample type. (A) Mock community, (B) bronchoalveolar lavage, (C) nasal swabs, and (D) sputum. Statistical significances were noted at the level of q-value < 0.1.
 
```{r}



#ffff33 qia

f5b <- merge(f_fit_data_bal$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      f_fit_data_bal$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(f_fit_data_bal$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1"),
               value = value * 1000000) %>%



#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        theme_classic(base_family = "sans") +
        #colors for qvalues
        #xlab("Experimental group") +
        #ylab("Species") +
        #labs(tag = "A")  +
        ggtitle("A  BAL") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 8),
              axis.title.y = element_blank(),
              axis.title.x = element_blank(),
              plot.margin = unit(c(0,0.2,0,1), 'lines'))  +
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top",
                                   nrow = 2,
                                   override.aes = list(size=3)),
               size = guide_legend(title = "Copies per million",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1)
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "MolYsis",
                                  "qiaamp" = "QIAamp")
                         )

f5c <- merge(f_fit_data_ns$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      f_fit_data_ns$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(f_fit_data_ns$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1"),
               value = value * 1000000) %>%


#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        theme_classic(base_family = "sans") +
        #colors for qvalues
        #xlab("Experimental group") +
        #ylab("Species") +
        #labs(tag = "B")  +
        ggtitle("B  Nasal swab") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 8),
              axis.title.y = element_blank(),
              axis.title.x = element_blank(),
              plot.margin = unit(c(0,0.2,0,1), 'lines'))  +
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top",
                                   override.aes = list(size=3)),
               size = guide_legend(title = "Copies per million",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1)
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "MolYsis",
                                  "qiaamp" = "QIAamp")
                         )

f5d <- merge(f_fit_data_spt$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      f_fit_data_spt$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        merge(f_fit_data_spt$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1"),
               value = value * 1000000) %>%


#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        theme_classic(base_family = "sans") +
        #colors for qvalues
        #xlab("Experimental group") +
        #ylab("Species") +
        #labs(tag = "C")  +
        ggtitle("C  Sputum") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 8),
              axis.title.y = element_blank(),
              axis.title.x = element_blank(),
              plot.margin = unit(c(0,0.2,0,1), 'lines'))  +
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top",
                                   override.aes = list(size=5)),
               size = guide_legend(title = "Copies per million",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1)
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "MolYsis",
                                  "qiaamp" = "QIAamp")
                         )

figS10 <- ggarrange(f5d %>% lemon::g_legend() %>% as_ggplot,
                  f5b,
                  f5c,
                  f5d,
                  ncol=1, heights = c(1.5, 4, 4, 4),
                  legend = "none",
                  align = "hv")


annotate_figure(figS10,
                left = text_grob("Predicted function",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Treatment",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11, hjust = -3)
                
)





png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS12.png",   # The directory you want to save the file in
    width = 240, # The width of the plot in inches
    height = 220, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue



annotate_figure(figS10,
                left = text_grob("Predicted function",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Treatment",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11, hjust = -3)
                
)


# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()



```

### {-}


# 3.5. Sensitivity analysis after decontamination

# *xiii. Is species richness similar after decontamination?*


## Sensitivity analysis {.tabset}


### Fig. S12. Rarefraction curve of species richenss

**Fig. S12.** Rarefaction curve for (A) species richness and (B) function richenss stratified by sample type, after removing possible contaminant-taxa identified by decontam and low prevalent taxa.

**As a sanity check, rarefaction curves were generated and seemed to be saturated**

```{r}


fig_rarefraction <- phyloseq$phyloseq_count %>% 
        sample_data %>% 
        data.frame %>%
        subset(., !is.na(.$treatment) & sample_type %in% c("BAL", "Nasal", "Sputum")) %>%
ggplot(., aes(x = log10(Final_reads/1000000), y = S.obs, col = treatment)) +
        geom_point() +
        theme_classic(base_family = "sans") +
        #xlab("Final reads x 10<sup>6</sup>") +
        ylab("Microbial species richness") +
        labs(tag = "A") +
        theme(axis.title.x = element_blank(), legend.position = "top") +
        guides(col = guide_legend(title = "Treatment", title.position = "top", nrow = 1)) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"),
                          name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) +
        facet_wrap(~sample_type, scales = "free", nrow = 1)

fig_rarefraction_function <- phyloseq$phyloseq_path_rpk %>% 
        sample_data %>% 
        data.frame %>%
        subset(., !is.na(.$treatment) & sample_type %in% c("BAL", "Nasal", "Sputum")) %>%
ggplot(., aes(x = log10(Final_reads/1000000), y = S.obs, col = treatment)) +
        geom_point() +
        theme_classic(base_family = "sans") +
        xlab("log<sub>10</sub>(Final reads x 10<sup>6</sup>)") +
        ylab("Functional richness") +
        labs(tag = "C") +
        theme(axis.title.x = element_markdown(), legend.position = "top") +
        guides(col = guide_legend(title = "Treatment", title.position = "top", nrow = 1)) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"),
                          name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) +
        facet_wrap(~sample_type, scales = "free", nrow = 1)



fig_rarefraction_viral <- phyloseq$phyloseq_count %>% 
        sample_data %>% 
        data.frame %>%
        subset(., !is.na(.$treatment) & sample_type %in% c("BAL", "Nasal", "Sputum")) %>%
ggplot(., aes(x = log10(Final_reads/1000000), y = V.obs, col = treatment)) +
        geom_point() +
        theme_classic(base_family = "sans") +
        #xlab("Final reads x 10<sup>6</sup>") +
        ylab("Viral richness") +
        labs(tag = "B") +
        theme(axis.title.x = element_blank(), legend.position = "top",) +
        guides(col = guide_legend(title = "Treatment", title.position = "top", nrow = 1)) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"),
                          name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) +
        facet_wrap(~sample_type, scales = "free", nrow = 1)



png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS13_log.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 180, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

ggarrange(fig_rarefraction, fig_rarefraction_viral, fig_rarefraction_function, common.legend = T, ncol = 1)

dev.off()



fig_rarefraction <- phyloseq$phyloseq_count %>% 
        sample_data %>% 
        data.frame %>%
        subset(., !is.na(.$treatment) & sample_type %in% c("BAL", "Nasal", "Sputum")) %>%
ggplot(., aes(x = Final_reads/1000000, y = S.obs, col = treatment)) +
        geom_point() +
        theme_classic(base_family = "sans") +
        #xlab("Final reads x 10<sup>6</sup>") +
        ylab("Microbial species richness") +
        labs(tag = "A") +
        theme(axis.title.x = element_blank(), legend.position = "top") +
        guides(col = guide_legend(title = "Treatment", title.position = "top", nrow = 1)) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"),
                          name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) +
        facet_wrap(~sample_type, scales = "free", nrow = 1)

fig_rarefraction_function <- phyloseq$phyloseq_path_rpk %>% 
        sample_data %>% 
        data.frame %>%
        subset(., !is.na(.$treatment) & sample_type %in% c("BAL", "Nasal", "Sputum")) %>%
ggplot(., aes(x = Final_reads/1000000, y = S.obs, col = treatment)) +
        geom_point() +
        theme_classic(base_family = "sans") +
        xlab("Final reads x 10<sup>6</sup>") +
        ylab("Functional richness") +
        labs(tag = "C") +
        theme(axis.title.x = element_markdown(), legend.position = "top") +
        guides(col = guide_legend(title = "Treatment", title.position = "top", nrow = 1)) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"),
                          name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) +
        facet_wrap(~sample_type, scales = "free", nrow = 1)



fig_rarefraction_viral <- phyloseq$phyloseq_count %>% 
        sample_data %>% 
        data.frame %>%
        subset(., !is.na(.$treatment) & sample_type %in% c("BAL", "Nasal", "Sputum")) %>%
ggplot(., aes(x = Final_reads/1000000, y = V.obs, col = treatment)) +
        geom_point() +
        theme_classic(base_family = "sans") +
        #xlab("Final reads x 10<sup>6</sup>") +
        ylab("Viral richness") +
        labs(tag = "B") +
        theme(axis.title.x = element_blank(), legend.position = "top",) +
        guides(col = guide_legend(title = "Treatment", title.position = "top", nrow = 1)) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"),
                          name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) +
        facet_wrap(~sample_type, scales = "free", nrow = 1)

ggarrange(fig_rarefraction, fig_rarefraction_viral, fig_rarefraction_function, common.legend = T, ncol = 1)

png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS13_updated.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 180, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

ggarrange(fig_rarefraction, fig_rarefraction_viral, fig_rarefraction_function, common.legend = T, ncol = 1)

dev.off()



```    

### Fig. S13. Species richness of decontaminated output

As some of the species richness got way higher, this data cannot be used for alpha diversity indices.

**Fig. S13.** Species richness of (A) raw data after prevalence and abundance filtering, (B) decontaminated species richness with decontam37, and (C) decontaminated data using tinyvamp.

```{r}


f10a <- ggplot(subset(sample_data(phyloseq$phyloseq_count) %>% 
                             data.frame, sample_data(phyloseq$phyloseq_count)$sample_type %in% c("Sputum", "Nasal", "BAL"#, "Mock"
                                                                                                 )), aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment), position = position_jitter(0.2),
                    size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        ylab("Species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        labs(tag = "A") +
        ggtitle("Prevalence & abundance filtered data") +
        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(),
              axis.title.x = element_blank(),
              axis.title.y = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 

f10b <- ggplot(subset(sample_data(phyloseq$phyloseq_count) %>% 
                             data.frame, sample_data(phyloseq$phyloseq_count)$sample_type %in% c("Sputum", "Nasal", "BAL"#, "Mock"
                                                                                                 )), aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment), position = position_jitter(0.2),
                    size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        ylab("Species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        labs(tag = "B") +
        ggtitle("Decontaminated data 1") +
        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(),
              axis.title.x = element_blank(),
              axis.title.y = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 


f10c <- ggplot(subset(sample_data(phyloseq_tv) %>% 
                             data.frame, sample_data(phyloseq_tv)$sample_type %in% c("Sputum", "Nasal", "BAL", "Mock")), aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment), position = position_jitter(0.2),
                    size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        ylab("Species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        labs(tag = "C") +
        ggtitle("Decontaminated data 2") +
        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(), 
              axis.title.x = element_blank(),
              axis.title.y = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 

figa1 <- ggarrange(f10a, f10b, f10c, common.legend = T, ncol = 1)



annotate_figure(figa1,
                left = text_grob("Species richness",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Treatment",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11)
)



png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS14.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 180, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

annotate_figure(figa1,
                left = text_grob("Species richness",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Treatment",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11)
)


dev.off()




```


### Table S11. Species richness change after decontamination

**Table S11.** Effect size (95% confidence interval) and p-value of decontaminated species richness using decontam37 (decontaminated data 1) and tinyvamp (decontaminated data 2). The change was tested using a model lmer(species richness~ treatment +  (1|subject id)). Statistical significances were noted with *: p-value < 0.05, **: p-value < 0.01, and ***: p-value < 0.001. 

Raw results - decontam/BAL

```{r}

sr_lmer_bal_decontam <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = sample_data(phyloseq_decontam$phyloseq_rel) %>% 
                             data.frame %>% 
                            subset(.,.$sample_type == "BAL"))
sr_lmer_bal_decontam %>% summary
```

Raw results - decontam/nasal

```{r}
sr_lmer_ns_decontam <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = sample_data(phyloseq_decontam$phyloseq_rel) %>% 
                             data.frame %>% 
                            subset(.,.$sample_type == "Nasal"))
sr_lmer_ns_decontam %>% summary

```


Raw results - decontam/spt

```{r}
sr_lmer_spt_decontam <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = sample_data(phyloseq_decontam$phyloseq_rel) %>% 
                             data.frame %>% 
                            subset(.,.$sample_type == "Nasal"))
sr_lmer_spt_decontam %>% summary

```

Raw results - Tinyvamp/BAL


```{r}
sr_lmer_bal_tv <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = sample_data(phyloseq_tv) %>% 
                             data.frame %>% 
                            subset(.,.$sample_type == "BAL"))
sr_lmer_bal_tv %>% summary()

```

Raw results - Tinyvamp/Nasal


```{r}
sr_lmer_ns_tv <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = sample_data(phyloseq_tv) %>% 
                             data.frame %>% 
                            subset(.,.$sample_type == "Nasal")) 
sr_lmer_ns_tv %>% summary()

```


Raw results - Tinyvamp/Sputum


```{r}
sr_lmer_spt_tv <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = sample_data(phyloseq_tv) %>% 
                             data.frame %>% 
                            subset(.,.$sample_type == "Sputum")) 
sr_lmer_spt_tv %>% summary()

```



Tidy table of LMER for decontaminated species richness


```{r}

sr_lmer_bal_decontam_kbl <- sr_lmer_bal_decontam %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_bal_decontam %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        
sr_lmer_ns_decontam_kbl  <- sr_lmer_ns_decontam %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_ns_decontam %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
                        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

sr_lmer_spt_decontam_kbl <- sr_lmer_spt_decontam %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_spt_decontam %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
                        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        


```

```{r}
sr_lmer_bal_tv_kbl <- sr_lmer_bal_tv %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_bal_tv %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
                        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        
sr_lmer_ns_tv_kbl  <- sr_lmer_ns_tv %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_ns_tv %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
                        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


sr_lmer_spt_tv_kbl <- sr_lmer_spt_tv %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_spt_tv %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        


tableS11 <- cbind(
        cbind(sr_lmer_bal_decontam_kbl, sr_lmer_ns_decontam_kbl, sr_lmer_spt_decontam_kbl),
        cbind(sr_lmer_bal_tv_kbl, sr_lmer_ns_tv_kbl, sr_lmer_spt_tv_kbl) %>% remove_rownames()) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" "= 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        add_header_above(c(" " = 1, "Decontaminated data 1" = 9, "Decontaminated data 2"= 9)) %>% 
        kable_styling(full_width = 0, html_font = "sans")


tableS11

save_kable(tableS11, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS11.html", self_contained = T)



```

### Beta diversity plot

```{r}
figd2a <- ordinate(subset_samples(phyloseq_rel_nz, sample_type != "Neg." & sample_type != "Mock"), method = "PCoA", distance = "horn") %>%
        plot_ordination(phyloseq_rel_nz, ., col = "treatment") +
        #scale_color_viridis(discrete = 6, name = "Treatment", labels = c("Mock theoretical", "Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"),
                           name = "Treatment",
                           breaks = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
                           labels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        #scale_shape(name = "Sample type", labels = c("Mock theoretical", "Mock")) +
        geom_point(size = 3) +
        theme_classic (base_size = 12, base_family = "sans") +
        facet_wrap(~sample_type, scales = "free") +
        labs(tag = "A") +
        ggtitle("Prevalence & abundance filtered data") +
        theme(plot.tag = element_text(size = 15), legend.position = "top")# +
        #stat_ellipse(type = "norm") +
        #stat_ellipse(type = "t")


figd2b <- ordinate(subset_samples(phyloseq_decontam$phyloseq_rel, sample_type != "Neg." & sample_type != "Mock" &
                                          S.obs != 0), method = "PCoA", distance = "horn") %>%
        plot_ordination(phyloseq_decontam$phyloseq_rel, ., col = "treatment") +
        #scale_color_viridis(discrete = 6, name = "Treatment", labels = c("Mock theoretical", "Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"),
                           name = "Treatment",
                           breaks = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
                           labels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        #scale_shape(name = "Sample type", labels = c("Mock theoretical", "Mock")) +
        geom_point(size = 3) +
        theme_classic (base_size = 12, base_family = "sans") +
        facet_wrap(~sample_type, scales = "free") +
        labs(tag = "B") +
        ggtitle("Decontaminated data 1") +
        theme(plot.tag = element_text(size = 15), legend.position = "top")# +
        #stat_ellipse(type = "norm") +
        #stat_ellipse(type = "t")


figd2c <- ordinate(subset_samples(phyloseq_tv, sample_type != "Neg." & sample_type != "Mock"), method = "PCoA", distance = "horn") %>%
        plot_ordination(phyloseq_tv, ., col = "treatment") +
        #scale_color_viridis(discrete = 6, name = "Treatment", labels = c("Mock theoretical", "Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"),
                           name = "Treatment",
                           breaks = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
                           labels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        #scale_shape(name = "Sample type", labels = c("Mock theoretical", "Mock")) +
        geom_point(size = 3) +
        theme_classic (base_size = 12, base_family = "sans") +
        facet_wrap(~sample_type, scales = "free") +
        labs(tag = "C") +
        ggtitle("Decontaminated data 2") +
        theme(plot.tag = element_text(size = 15), legend.position = "top")# +
        #stat_ellipse(type = "norm") +
        #stat_ellipse(type = "t")


figA3 <- ggarrange(figd2a, figd2b, figd2c, common.legend = T, nrow = 3)
figA3



```

### List of contaminants (Tinyvamp)


```{r}

#Stratified by sample type

prev_neg
prev_all

sample_data(phyloseq_unfiltered$phyloseq_rel)$is.neg <- grepl("Neg", sample_data(phyloseq_unfiltered$phyloseq_rel)$sample_type)

contaminants_tv <- data.frame(
        Taxa = subset(taxa_names(phyloseq_unfiltered$phyloseq_count),
       !(taxa_names(phyloseq_unfiltered$phyloseq_count) %in%
               taxa_names(phyloseq_tv))
)) 
       


merged_contaminants <- merge(contaminants_tv, prev_all %>% rownames_to_column("Taxa"), by = "Taxa") %>%
        merge(., prev_neg %>% rownames_to_column("Taxa"), by = "Taxa") %>%
        dplyr::select(c("Taxa", "Prevalence (all)", "Prevalence (negative controls)")) %>%
        .[order(-.$"Prevalence (all)", -.$"Prevalence (negative controls)"),] %>%
        remove_rownames() %>%
        subset(., .$"Prevalence (all)" != 0)


tableA3 <- merged_contaminants %>% 
        mutate(Taxa = species_italic2(Taxa)) %>%
        kbl(format = "html", escape = F) %>%
        kable_styling(full_width = 0, html_font = "sans") 

tableA3



```      
               


# 3.6. Summary

## Mean of species richness change by sample

```{r}
phyloseq$phyloseq_count %>% 
        sample_data %>%
        data.frame() %>%
        group_by(subject_id) %>%
        summarise(subject_id = subject_id,
                  treatment = treatment,
                  S.obs = S.obs,
                  S.obs_untreated = S.obs)

```



## Table 2. Sequencing summary

**Table 2**. Summary table of sequencing issues, significant effects linear mixed effect model (species richness ~ treatment + (1|subject)), changes in microbial beta diversity and significant effects of linear mixed effect model (functional richness ~ treatment + (1|subject)). Linear mixed effect models were stratified by sample type and employed data after prevalence and abundance filtering.

**MolYsis for BAL, QIAamp for nasal swab, and HostZERO for sputum. **


```{r}

summary_host_ratio <- rbind(
        hr_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              hr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("% Host" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("% Host")),
        
        hr_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              hr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("% Host" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("% Host")) ,
        
 hr_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              hr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("% Host" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%  
         column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("% Host")) 
)
        

summary_final_reads <- 
        rbind(
        fr_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              fr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("log<sub>10</sub>(Final reads)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("log<sub>10</sub>(Final reads)")),
        
        fr_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              fr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("log<sub>10</sub>(Final reads)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("log<sub>10</sub>(Final reads)")) ,
        
 fr_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              fr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("log<sub>10</sub>(Final reads)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%  
         column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("log<sub>10</sub>(Final reads)")) 
)
        

summary_species_richness <-
        rbind(
        sr_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              sr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Species richness" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Species richness")),
        
        sr_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              sr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Species richness" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Species richness")) ,
        
 sr_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              sr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Species richness" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%  
         column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Species richness")) 
) %>% 
        rename("Microbial species richness" = "Species richness")
        

summary_function_richness <-
        rbind(
        pfr_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              pfr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Functional richness" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Functional richness")),
        
        pfr_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              pfr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Functional richness" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Functional richness")) ,
        
 pfr_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              pfr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Functional richness" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%  
         column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Functional richness")) 
)
        

summary_viral_richness <-
        rbind(
        sr_lmer_bal_v %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              sr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Species richness" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Species richness")),
        
        sr_lmer_ns_v %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              sr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Species richness" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Species richness")) ,
        
 sr_lmer_spt_v %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              sr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Species richness" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%  
         column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Species richness")) 
) %>%
        rename("Viral species richness" = "Species richness")
        

summary_bias <-
        rbind(
        
 mh_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              mh_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Bias (Morisita-Horn)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = ""),
                "Row.names" = case_when(`Row.names` == "benzonase" ~ "Benzonase",
                                      `Row.names` == "host_zero" ~ "HostZERO",
                                      `Row.names` == "lypma" ~ "lyPMA",
                                      `Row.names` == "molysis" ~ "MolYsis",
                                      `Row.names` == "qiaamp" ~ "QIAamp",
                                      .default = `Row.names`)
                ) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Bias (Morisita-Horn)")),
        
        
 mh_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              mh_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Bias (Morisita-Horn)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = ""),
                "Row.names" = case_when(`Row.names` == "benzonase" ~ "Benzonase",
                                      `Row.names` == "host_zero" ~ "HostZERO",
                                      `Row.names` == "lypma" ~ "lyPMA",
                                      `Row.names` == "molysis" ~ "MolYsis",
                                      `Row.names` == "qiaamp" ~ "QIAamp",
                                      .default = `Row.names`)
                ) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Bias (Morisita-Horn)")),
        
 
 mh_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              mh_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Bias (Morisita-Horn)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = ""),
                "Row.names" = case_when(`Row.names` == "benzonase" ~ "Benzonase",
                                      `Row.names` == "host_zero" ~ "HostZERO",
                                      `Row.names` == "lypma" ~ "lyPMA",
                                      `Row.names` == "molysis" ~ "MolYsis",
                                      `Row.names` == "qiaamp" ~ "QIAamp",
                                      .default = `Row.names`)
                ) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Bias (Morisita-Horn)"))
 
)  %>%
        rename("Microbial bias (Morisita-Horn)" = "Bias (Morisita-Horn)")
        
summary_gram_negative <-
        rbind(
        gram_neg_prop_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              gram_neg_prop_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("% gram negative" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("% gram negative")),
        
        gram_neg_prop_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              gram_neg_prop_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("% gram negative" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("% gram negative")) ,
        
 gram_neg_prop_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              gram_neg_prop_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("% gram negative" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = "")) %>%  
         column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("% gram negative")) 
) 
        


summary_bias_viral <-
        rbind(     
                
mh_lmer_bal_kbl_v <-  
       mh_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              mh_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Bias (Morisita-Horn)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = ""),
                "Row.names" = case_when(`Row.names` == "benzonase" ~ "Benzonase",
                                      `Row.names` == "host_zero" ~ "HostZERO",
                                      `Row.names` == "lypma" ~ "lyPMA",
                                      `Row.names` == "molysis" ~ "MolYsis",
                                      `Row.names` == "qiaamp" ~ "QIAamp",
                                      .default = `Row.names`)
                ) %>%
                column_to_rownames("Row.names") %>%
          mutate("Bias (Morisita-Horn)" = "-") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Bias (Morisita-Horn)")),
       
        
 mh_lmer_ns_v %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              mh_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Bias (Morisita-Horn)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = ""),
                "Row.names" = case_when(`Row.names` == "benzonase" ~ "Benzonase",
                                      `Row.names` == "host_zero" ~ "HostZERO",
                                      `Row.names` == "lypma" ~ "lyPMA",
                                      `Row.names` == "molysis" ~ "MolYsis",
                                      `Row.names` == "qiaamp" ~ "QIAamp",
                                      .default = `Row.names`)
                ) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Bias (Morisita-Horn)")),
        
 
 mh_lmer_spt_v %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        merge(., 
              mh_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
         mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(`Row.names` = gsub("treatment|sample_type", "", `Row.names`)) %>%
         mutate(`Row.names` = gsub(":", " * ", `Row.names`)) %>%
         mutate("Bias (Morisita-Horn)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             " ",
                             conf, 
                             ` `,
                             sep = ""),
                "Row.names" = case_when(`Row.names` == "benzonase" ~ "Benzonase",
                                      `Row.names` == "host_zero" ~ "HostZERO",
                                      `Row.names` == "lypma" ~ "lyPMA",
                                      `Row.names` == "molysis" ~ "MolYsis",
                                      `Row.names` == "qiaamp" ~ "QIAamp",
                                      .default = `Row.names`)
                ) %>%
                column_to_rownames("Row.names") %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),] %>%
         dplyr::select(c("Bias (Morisita-Horn)"))
) %>% rename("Viral bias (Morisita-Horn)" = "Bias (Morisita-Horn)")
        

#summary_comment <- 
#        c("", "Low efficiency", "Low efficiency", 
#          "No incrased richness", "Optimal", "Low efficiency",
#          
#          "","No incrased richness", "Low efficiency",
#          "High rate of library prep failure", "High bias", "Optimal",
#          
#          "", "Low efficiency", "Low efficiency",
#          "Optimal<sup>1</sup>", "High bias<sup>2</sup>", "High bias")



table3 <- cbind(`Sample type` = c("", "BAL", "", "", "", "",
        "", "Nasal swabs", "", "", "", "",
        "", "Sputum", "", "", "", ""),
      `Treatment` = c("", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
        "", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
        "", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
      summary_host_ratio,
      summary_final_reads,
      summary_species_richness,
      summary_function_richness,
      summary_viral_richness,
      summary_bias,
      summary_bias_viral,
      summary_gram_negative
      #Comment = summary_comment
      ) %>% 
        subset(., .$Treatment != "") %>%
        remove_rownames() %>%
        kbl(format = "html", escape = 0) %>%
        add_footnote(c("Unable to run statistical tests"), notation = "alphabet") %>%
        kable_styling(full_width = 0, html_font = "sans")

save_kable(table3, file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/table2_updated.html", self_contained = T)

table3

```

Done.

# Bibliography

```{r warning=FALSE}
#===============================================================================
#BTC.LineZero.Footer.1.1.0
#===============================================================================
#R markdown citation generator.
#===============================================================================
#RLB.Dependencies:
#   magrittr, pacman, stringr
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#BTC.Dependencies:
#   LineZero.Header
#===============================================================================
#Generates citations for each explicitly loaded library.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
str_libraries <- c("r", str_libraries)
for (str_libraries in str_libraries) {
    str_libraries |>
        pacman::p_citation() |>
        print(bibtex = FALSE) |>sav
        capture.output() %>%
        .[-1:-3] %>% .[. != ""] |>
        stringr::str_squish() |>
        stringr::str_replace("_", "") |>
        cat()
    cat("\n")
}
#===============================================================================
```
