---
title: "COD_20240212_HOST_6_phage_analysis_figure_table"
author: "Minsik Kim"
date: "2024-02-12"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
        df_print: paged
editor_options: 
  chunk_output_type: inline
  markdown: 
    wrap: 72
---

# Loading packages

```{r warning=FALSE, message=FALSE, setup}
#===============================================================================
#BTC.LineZero.Header.1.1.0
#===============================================================================
#R Markdown environment setup and reporting utility.
#===============================================================================
#RLB.Dependencies:
#   knitr, magrittr, pacman, rio, rmarkdown, rmdformats, tibble, yaml
#===============================================================================
#Input for document parameters, libraries, file paths, and options.
#=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=
#knitr::opts_chunk$set(message=FALSE, warning = FALSE)


path_working <- 
        ifelse(sessionInfo()[1]$R.version$platform == "x86_64-pc-linux-gnu",
               "/home/bagel/minsik/",
               ifelse(sessionInfo()[1]$R.version$platform == "aarch64-apple-darwin20",
                      "/Volumes/macdrive/Dropbox/", 
                      "/Users/minsikkim/Dropbox (Personal)"))

path_library <- 
        ifelse(sessionInfo()[1]$R.version$platform == "x86_64-pc-linux-gnu",
               "/home/bagel/R/x86_64-pc-linux-gnu-library/4.1/",
               "/Library/Frameworks/R.framework/Resources/library/")

str_libraries <- c("readxl", "phyloseq", "tidyverse", "pacman", "yaml", "ggplot2", "vegan", "microbiome", "ggpubr", "viridis", "decontam", "gridExtra", "ggpubr", "lme4", "lmerTest", "writexl", "harrietr", "Maaslin2", "ggtext", "ggpmisc", "gamm4", "reshape2", "kableExtra", "knitr", "ggtree", "car", "mediation", "lemon", "qvalue")
        
YAML_header <-
'---
title: "Host-DNA depletion analysis of bacterio phage"
author: "Minsik Kim"
date: "2024.02.12"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
---'
seed <- "20230925"

#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Loads libraries, file paths, and other document options.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
FUN.LineZero.Boot <- function() {
    .libPaths(path_library)

    require(pacman)
    pacman::p_load(c("knitr", "rmarkdown", "rmdformats", "yaml"))

    knitr::opts_knit$set(root.dir = path_working)

    str_libraries |> unique() |> sort() -> str_libraries
    pacman::p_load(char = str_libraries)

    set.seed(seed)
}
FUN.LineZero.Boot()
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Outputs R environment report.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

FUN.LineZero.Report <- function() {
    cat("Line Zero Environment:\n\n")
    paste("R:", pacman::p_version(), "\n") |> cat()
    cat("Libraries:\n")
    for (str_libraries in str_libraries) {
        paste(
            "    ", str_libraries, ": ", pacman::p_version(package = str_libraries),
            "\n", sep = ""
        ) |> cat()
    }
    paste("\nOperating System:", pacman::p_detectOS(), "\n") |> cat()
    paste("    Library Path:", path_library, "\n") |> cat()
    paste("    Working Path:", path_working, "\n") |> cat()
    paste("Seed:", seed, "\n\n") |> cat()
    cat("YAML Header:\n")
    cat(YAML_header)
}
FUN.LineZero.Report()

```


# Contents

0. Exploratory data analysis

        •	Summary table of species richness
        •	Marker-MAGu microbial output
        •	Marker-MAGu phage output
        •	MetaPhlAnN3 output
        •	Mapping failure & viral mapped reads
        •	decontam

        
Question 1. How does the Marker-MAGu microbial community looks like?
        
        •	Microbial communities of samples at Domain level 
        •	Positive controls
        
Question 2. Was phage community changed the same as microbial community, by sample type and treatment?

        •	Alpha diversity (phage species richness)
        
Question 3. How was the changes of species richness, stratified by domain?

        •	Alpha diversity of microbial species richness, stratified by each domain
        
Conclusion


## Analysis prep {.tabset}

### Loading data

```{r warning=F, message=FALSE, "Data loading"}
# Loading files -----------------------------------------------------------
#loading tidy phyloseq object
phyloseq <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20240212_MGK_host_marker_magu.rds")

m_phyloseq <- phyloseq$microbe_rpkm
v_phyloseq <- phyloseq$viral_rpkm


```


### Alpha diversity indices

```{r}           
alpha_diversity <- function(data) {
        otu_table <- otu_table(data) %>% .[colSums(.) !=0]
        S.obs <- rowSums(t(otu_table) != 0)
        sample_data <- sample_data(data)
        data_evenness <- vegan::diversity(t(otu_table)) / log(vegan::specnumber(t(otu_table))) # calculate evenness index using vegan package
        data_shannon <- vegan::diversity(t(otu_table), index = "shannon") # calculate Shannon index using vegan package
        data_hill <- exp(data_shannon)                           # calculate Hills index
        data_dominance <- microbiome::dominance(otu_table, index = "all", rank = 1, aggregate = TRUE) # dominance (Berger-Parker index), etc.
        data_invsimpson <- vegan::diversity(t(otu_table), index = "invsimpson")                          # calculate Shannon index using vegan package
        alpha_diversity <- cbind(S.obs, data_shannon, data_hill, data_invsimpson, data_evenness,data_dominance) # combine all indices in one data table
        sample_data <- merge(data.frame(sample_data), alpha_diversity, by = 0, all = T) %>% column_to_rownames(var = "Row.names")
}

sample_data(m_phyloseq) <- sample_data(alpha_diversity(m_phyloseq)) 

sample_data(v_phyloseq) <- sample_data(alpha_diversity(v_phyloseq)) 



#sample data loading
m_sample_data <- sample_data(m_phyloseq) %>% data.frame()

sample_data(v_phyloseq)$Viral_reads <- sample_sums(v_phyloseq)
v_sample_data <- sample_data(v_phyloseq) %>% data.frame()


m_phyloseq_rel <- m_phyloseq %>%
        transform_sample_counts(., function(x){x/sum(x)})

v_phyloseq_rel <- v_phyloseq %>%
        transform_sample_counts(., function(x){x/sum(x)})

```

## {-}

# Summary of exploratory data analysis
## Summary table {.tabset}                

### Summary table of species richness

**Summary table of microbial community from Marker-MAGu**. 

```{r}

m_sample_data %>%
        group_by(sample_type, treatment) %>%
        summarise(Mean_SObs = mean(S.obs),
                  SD_SObs = sd(S.obs),
                  Max_SObs = max(S.obs),
                  Min_SObs = min(S.obs),
                  N = n())

```

**Summary table of phage community from Marker-MAGu**. 

```{r}
v_sample_data %>%
        group_by(sample_type, treatment) %>%
        summarise(Mean_SObs = mean(S.obs),
                  SD_SObs = sd(S.obs),
                  N = n())
        
```

1. Mock community had 8 microbial taxa in average. Did not show fungi reads.

2. The Marker-MAGu + metaphlan 4 outcome showed relatively lower viral speciese richness compared to microbial richness.

*3. The Marker-MAGu viral community seems less sensitive to contaminants. The increase in species richenss was lower than microbial community analysis result.*


### Marker-MAGu microbial output

*Marker-MAGu didn't detect Eukaryotes.*

```{r}
m_phyloseq %>% 
        tax_table %>%
        data.frame() %>%
        .$Kingdom %>% 
        table %>% 
        data.frame() %>% 
        rename(Domain = ".",
               Frequency = Freq)
```

### Marker-MAGu phage output

*Marker-MAGu didn't detect phage that infects Eukaryotes (There is no such phage).*

```{r}
v_phyloseq %>% 
        tax_table %>%
        data.frame() %>%
        .$H_Kingdom %>% 
        table(.,  useNA = "always") %>% 
        data.frame() %>% 
        rename(Domain = ".",
               Frequency = Freq)
```

### Metaphlan 3

```{r}

# Loading files -----------------------------------------------------------
#loading tidy phyloseq object
phyloseq_unfiltered <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20230521_MGK_host_tidy.rds")

rbind(

phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(sample_type == "BAL") %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>%
        tax_table %>%
        data.frame %>%
        .$Domain %>%
        table(.,  useNA = "always") %>% 
        data.frame() %>%
        mutate(`Sample type` = "BAL", .before = "."),

phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(sample_type == "Nasal") %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>%
        tax_table %>%
        data.frame %>%
        .$Domain %>%
        table(.,  useNA = "always") %>% 
        data.frame() %>%
        mutate(`Sample type` = "Nasal", .before = "."),


phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(sample_type == "Sputum") %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>%
        tax_table %>%
        data.frame %>%
        .$Domain %>%
        table(.,  useNA = "always") %>% 
        data.frame() %>%
        mutate(`Sample type` = "Sputum", .before = ".")
) %>% 
        rename(Domain = ".",
               Frequency = Freq)


```

*By sample type, the proportion of Eukaryota (mainly fungi) was different at Metaphlan3 output. Nasal swabs's community had 14 Eukaryotes out of 59 taxa.*


### {-}

# Exploratory QC

## Library failure {.tabset}


### Failure crosstab

*More samples showed 0 mapped reads, relative to metaphlan data. Samples with 0 taxa were not eliminated for further analyses*

```{r}
sample_data <- sample_data(v_phyloseq) %>% 
		  data.frame() %>%
		  mutate(
		      sample_sums = sample_sums(v_phyloseq),
		      lib_failed = case_match(lib_failed, TRUE ~ 1, FALSE ~0),
		      seq_failed = ifelse(sample_sums == 0, 1, 0),
		      lib_seq_failed = ifelse(lib_failed ==1 | seq_failed == 1, 1, 0),
		      prop_host = round((Host_mapped/Reads_after_trim), 1)
		      )
cat("Sequencing failed")		
xtabs(seq_failed ~ sample_type + treatment, data = sample_data)
cat("Sequencing and library failed")		
xtabs(lib_seq_failed ~ sample_type + treatment, data = sample_data)
```

### Figure of sequencing result (Viral reads)

Host depletion effects on total viral reads measured by shotgun metagenomic sequencing. 

```{r}

fs3a <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(Viral_reads))) +
        geom_jitter(aes(col = treatment, x = treatment),
                    lwd = 0.2, alpha = 0.3, stroke = 0)+
        stat_summary(aes(color = treatment, x = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Treatment")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              axis.text.x = element_blank()) +
        ylab("log<sub>10</sub>(Viral RPKM)") +
        labs(tag = "A") +
        facet_wrap(~sample_type, scale = "free_x") +
        guides(fill = guide_legend(nrow = 1))

#	- Host_mapped



fs3a


#png(file = "/Users/minsikkim/Dropbox (Personal)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS3.png",   # The directory you want to save the file in
#    width = 180, # The width of the plot in inches
#    height = 170, # The height of the plot in inches
#    units = "mm",
#    res = 600
#) #fixing multiple page issue

#figS3
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

#dev.off()



```

For BAL and Sputum, viral reads clearly increased after treatment. For nasal swabs, such pattern was not observed.

### {-}

## decontam {.tabset}

### Table of decontam - stratified by sample_type (prevalence method)

**Note: decontam was conducted only with prevalence method, as quantification on viruses was not conducted via qPCR.**

*At sample type stratified analysis, decontam identified 2 possible contaminants (that were identified in negative controls).*



```{r warning=FALSE, message=FALSE}
#Stratified by sample type
neg_number_decontam <- 
        v_phyloseq %>% 
        subset_samples(sample_type == "Neg.") %>%
        sample_names() %>%
        length

sample_number_decontam <- 
        v_phyloseq %>% 
        subset_samples(sample_type != "Mock") %>%
        subset_samples(sample_type != "Neg.") %>%
        #subset_samples(S.obs != 0) %>%
        sample_names() %>%
        length

prev_neg <- ((v_phyloseq %>% 
          subset_samples(sample_type == "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (negative controls)" = ".")

prev_all <- ((v_phyloseq %>%
                      subset_samples(sample_type != "Mock") %>%
                      subset_samples(sample_type != "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (all)" = ".")


sample_data(v_phyloseq)$is.neg <- grepl("Neg", sample_data(v_phyloseq)$sample_type)



phyloseq_decontam_bal <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(sample_type == "Neg." | sample_type == "BAL")
phyloseq_decontam_ns <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(sample_type == "Neg." | sample_type == "Nasal")
phyloseq_decontam_spt <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(sample_type == "Neg." | sample_type == "Sputum")



contaminant_combined_bal <- 
data.frame("BAL", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_bal, method="prev", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)


contaminant_combined_ns <- 
data.frame("Nasal", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_ns, method="prev", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)



contaminant_combined_spt <- 
data.frame("Sputum", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_spt, method="prev", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)

contaminants <- rbind(contaminant_combined_bal, contaminant_combined_ns, contaminant_combined_spt)

names(contaminants) <- c("Sample type", "Taxa")


merged_contaminants <- merge(contaminants, prev_all %>% 
                                     rownames_to_column("Taxa"), by = "Taxa") %>%
        merge(., prev_neg %>%
                      rownames_to_column("Taxa"), by = "Taxa") %>%
       dplyr::select(c("Sample type",
                       "Taxa",
                       "Prevalence (all)",
                       "Prevalence (negative controls)")) %>%
        subset(., .$"Prevalence (all)" != 0) %>%
        mutate(`Prevalence (all)` = paste0(`Prevalence (all)`,
                                           " (",
                                           round(`Prevalence (all)`/
                                                         sample_number_decontam * 100, 2),
                                           "%)"),
                `Prevalence (negative controls)` = paste0(`Prevalence (negative controls)`,
                                           " (",
                                           round(`Prevalence (negative controls)`/
                                                         neg_number_decontam * 100, 2),
                                           "%)")) %>%
        rename(`Prevalence (all) N = 94` = `Prevalence (all)`,
               `Prevalence (negative controls) N = 31` = `Prevalence (negative controls)`) %>%
        .[order(.$"Sample type", .$"Taxa"),] %>%
        remove_rownames()


merged_contaminants



```

### Decontam (stratified by treatment)

As a sanity check, decontam analysis was conducted after stratifying by treatment. Only 3 SGBs were nominated as potential candidates.

```{r, warning=FALSE}
#Stratified by sample type

prev_neg <- ((v_phyloseq %>% 
          subset_samples(sample_type == "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (negative controls)" = ".")

prev_all <- ((v_phyloseq %>%
                      subset_samples(sample_type != "Mock") %>%
                      subset_samples(sample_type != "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (all)" = ".")


sample_data(v_phyloseq)$is.neg <- grepl("Neg", sample_data(v_phyloseq)$sample_type)

phyloseq_decontam_lypma <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "lyPMA")

phyloseq_decontam_ben <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "Benzonase")

phyloseq_decontam_hostzero <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "HostZero")

phyloseq_decontam_molysis <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "MolYsis")

phyloseq_decontam_qiaamp <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "QIAamp")


#lypma showed zero contaminant
#isContaminant(phyloseq_decontam_lypma, method="prevalence", neg = "is.neg", threshold = 0.1) %>% subset(.,.$contaminant) 

contaminant_combined_ben <- 
data.frame("Benzonase", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_ben, method="prevalence", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)

#Host zero showed zero contaminant
#isContaminant(phyloseq_decontam_hostzero, method="prevalence", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant)

contaminant_combined_molysis <- 
data.frame("MolYsis", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_molysis, method="prevalence", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)

contaminant_combined_QIAamp <- 
data.frame("QIAamp", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_qiaamp, method="prevalence", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)

contaminants_strat_treat <- rbind(#contaminant_combined_lypma,
                      contaminant_combined_ben, 
                      #contaminant_combined_hostzero,
                      contaminant_combined_molysis,
                      contaminant_combined_QIAamp
                      )

names(contaminants_strat_treat) <- c("Treatment", "Taxa")

contaminants_strat_treat

```

### {-}


**Prevalence & abundance filtering was not conducted due to much lower species richness**

# Question 1. How does the Marker-MAGu microbial community looks like?


## Overview of sequencing results {.tabset}

### Fig. Viral community of samples

Relative abundances at SGB level by sample type after employing prevalence and abundance filtering and Top 10 phage in each sample type by mean abundance. (A) BAL, (B) nasal swabs, (C) sputum, and (D) mock community. Empty space indicates samples showed no microbial reads, i.e., sequencing failed samples.


```{r, warning=FALSE, message=FALSE}

my_plot_bar = function (physeq, x = "Sample", y = "Abundance", fill = NULL, title = NULL, 
                        facet_grid = NULL) {
    mdf = psmelt(physeq)
    p = ggplot(mdf, aes_string(x = x, y = y, fill = fill))
    p = p + geom_bar(stat = "identity")
    p = p + theme(axis.text.x = element_text(angle = -90, hjust = 0)) +
            scale_x_discrete(drop = F)
    if (!is.null(facet_grid)) {
        p <- p + facet_grid(facet_grid)
    }
    if (!is.null(title)) {
        p <- p + ggtitle(title)
    }
    return(p)
}

a <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "Mock")) %>%
        cbind(species20 = "Other") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq_rel,
                              sample_type == "Mock")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "Mock") 
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        my_plot_bar(., fill="species20") + 
        xlab("Sample") +
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.text.x = element_text(color = "white")) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        labs(tag = "A")
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +
        


b <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "Neg.")) %>%
        cbind(species20 = "Other") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq,
                              sample_type == "Neg.")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "Neg.") 
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>% 
        my_plot_bar(., fill="species20") + 
        ylab("") +
        xlab("Subject") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.text.x = element_text(color = "white")) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        #facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        facet_wrap ( ~ treatment,
                    scales= "free_x", nrow=1) +
        ggtitle("E    Negative control")

c <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "BAL")) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq,
                              sample_type == "BAL")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "BAL")
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E")) %>%
           as.character()
   phyloseq_temp
  } %>% 
        my_plot_bar(., x = "subject_id", fill="species20") + 
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        #facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        facet_wrap ( ~ treatment,
                    scales= "free_x", nrow=1) +
        ggtitle("A    BAL")


d <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "Nasal")) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq,
                              sample_type == "Nasal")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "Nasal")
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E", "F", "G", "H", "I", "J")) %>%
           as.character()
   phyloseq_temp
  } %>% 
        my_plot_bar(., x = "subject_id", fill="species20") + 
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
#        scale_x_discrete(drop=) +
        facet_wrap(~ treatment,  nrow = 1, drop = T, scales = "free_x") +
        ggtitle("B    Nasal swabs")

e <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "Sputum")) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq,
                              sample_type == "Sputum")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "Sputum")
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E"))
   phyloseq_temp
  } %>%
        my_plot_bar(., x = "subject_id", fill="species20")+
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap(~ treatment, scales= "free_x", nrow = 1) +
        ggtitle("C    Sputum")


fig2 <- ggarrange(c, c %>% lemon::g_legend() %>% as_ggplot,
                  d, d %>% lemon::g_legend() %>% as_ggplot,
                  e, e %>% lemon::g_legend() %>% as_ggplot,
                  #a, a %>% lemon::g_legend() %>% as_ggplot,
                  #b, b %>% lemon::g_legend() %>% as_ggplot,
                  ncol=2,nrow=3, widths = c(3, 1),
                  legend = "none",
                  align = "hv")

annotate_figure(fig2,
                left = text_grob("Viral relative abundance",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Subject",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11)
)



```


### Fig. Viral community of Zymo Mock

Lower species richness was observed, after treatment. 


```{r}

barplot_mock_viral <- a


barplot_mock_viral

```
### {-}



# Question 2. Does phage community changed the same as microbial community, by sample type and treatment?

## Taxa change {.tabset}

### Fig. Phage species richness


```{r, warning=FALSE, message=FALSE}

f3a <- ggplot(subset(sample_data(v_phyloseq) %>% 
                             data.frame, sample_data(v_phyloseq)$sample_type %in% c(#"Mock",
                                     "Sputum", "Nasal", "BAL")), aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment),
                    position = position_jitter(0.2), size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        #stat_summary(fun = mean,
        #       geom = "pointrange",
        #       fun.max = function(x) mean(x) + sd(x),
        #       fun.min = function(x) mean(x) - sd(x))+
        ylab("Phage species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        labs(tag = "A") +
        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 
f3a


```
### Fig. Microbial species richness

*Microbial species richness did not increase at nasal swabs.*

```{r, warning=FALSE, message=FALSE}

f3a <- ggplot(subset(sample_data(m_phyloseq) %>% 
                             data.frame, sample_data(m_phyloseq)$sample_type %in% c(#"Mock",
                                     "Sputum", "Nasal", "BAL")), aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment),
                    position = position_jitter(0.2), size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        #stat_summary(fun = mean,
        #       geom = "pointrange",
        #       fun.max = function(x) mean(x) + sd(x),
        #       fun.min = function(x) mean(x) - sd(x))+
        ylab("Microbial species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        labs(tag = "A") +
        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 
f3a

```

### Species richness (stratified, suppl) 

**This table is too redundant. Removed from the manuscript.**. 

Effect size, standard error (SE) and p-value of a statistical test for species richness using linear mixed effect model stratified by sample type (Species richness ~ treatment + (1|subject_id) ). Stratified analyses were conducted for each sample type as an interaction term of sample type and treatment was significant at an ANOVA test (p-value < 0.001) using a model, lmer(species richness ~ sample type + treatment + sample type * treatment  + (1|subject_id)). Statistical significances were noted with `*: p-value < 0.05, **: p-value < 0.01, and ***: p-value < 0.001.`

LMER raw result - Mock

```{r}

sr_lmer_mock <- lm(S.obs ~ treatment,
                    data = v_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Mock") & S.obs != 0))

sr_lmer_mock %>% summary() 

```

LMER raw result - BAL

```{r}

sr_lmer_bal <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = v_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("BAL")))
sr_lmer_bal %>% summary()
        
```

LMER raw result - Nasal

```{r}

sr_lmer_ns <- lmer(S.obs ~ treatment + (1|subject_id),
                   data = v_sample_data %>% 
                           data.frame %>% 
                           subset(., .$sample_type %in% c("Nasal")))
sr_lmer_ns %>% summary()
```

LMER raw result - Sputum

```{r}
sr_lmer_spt <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = v_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Sputum")))
sr_lmer_spt %>% summary()
        
sr_lmer_spt %>% confint()

```

### Phage species richness (stratified) 

Tidy summarized table of LME tests stratified by sample type.

*The results shows some treatments significantly increased the species richess of phage, but the effect become smaller with nasal swab samples.*

```{r}

sr_lmer_mock_kbl <-  sr_lmer_mock %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_mock %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

sr_lmer_bal_kbl <-  sr_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        
sr_lmer_ns_kbl <-  sr_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               abs(`Pr(>|t|)`) < 0.1 ~ ".",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


sr_lmer_spt_kbl <-  sr_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

tables3 <- cbind(#sr_lmer_mock_kbl,
                 sr_lmer_bal_kbl,
                 sr_lmer_ns_kbl,
                 sr_lmer_spt_kbl) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1,
                           #"Mock" = 3,
                           "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")

tables3

#save_kable(tables3, file = "/Users/minsikkim/Dropbox (Personal)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS3.html", self_contained = T)

```


### Species richness (stratified, suppl) 

**This table is too redundant. Removed from the manuscript.**. 

Effect size, standard error (SE) and p-value of a statistical test for species richness using linear mixed effect model stratified by sample type (Species richness ~ treatment + (1|subject_id) ). Stratified analyses were conducted for each sample type as an interaction term of sample type and treatment was significant at an ANOVA test (p-value < 0.001) using a model, lmer(species richness ~ sample type + treatment + sample type * treatment  + (1|subject_id)). Statistical significances were noted with `*: p-value < 0.05, **: p-value < 0.01, and ***: p-value < 0.001.`

LMER raw result - Mock

```{r}

sr_lmer_mock <- lm(S.obs ~ treatment,
                    data = m_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Mock") & S.obs != 0))

sr_lmer_mock %>% summary() 

```

LMER raw result - BAL

```{r}

sr_lmer_bal <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = m_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("BAL")))
sr_lmer_bal %>% summary()
        
```

LMER raw result - Nasal

```{r}

sr_lmer_ns <- lmer(S.obs ~ treatment + (1|subject_id),
                   data = m_sample_data %>% 
                           data.frame %>% 
                           subset(., .$sample_type %in% c("Nasal")))
sr_lmer_ns %>% summary()
```

LMER raw result - Sputum

```{r}
sr_lmer_spt <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = m_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Sputum")))
sr_lmer_spt %>% summary()
        
sr_lmer_spt %>% confint()

```

### Bacterial species richness (stratified) 

Tidy summarized table of LME tests stratified by sample type.

*The results shows some treatments significantly increased the species richess of phage, but the effect become smaller with nasal swab samples.*

```{r}

sr_lmer_mock_kbl <-  sr_lmer_mock %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_mock %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

sr_lmer_bal_kbl <-  sr_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        
sr_lmer_ns_kbl <-  sr_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               abs(`Pr(>|t|)`) < 0.1 ~ ".",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


sr_lmer_spt_kbl <-  sr_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

tables3 <- cbind(#sr_lmer_mock_kbl,
                 sr_lmer_bal_kbl,
                 sr_lmer_ns_kbl,
                 sr_lmer_spt_kbl) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1,
                           #"Mock" = 3,
                           "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")

tables3

#save_kable(tables3, file = "/Users/minsikkim/Dropbox (Personal)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS3.html", self_contained = T)

```

### {-}

# Question 3. How was the changes of species richness, stratified by domain?

## Species richenss (metaphlan) by domain {.tabset}

### Stratrifying phyloseq object

Increased species richness in Eukaryota was not observed

```{r}
metaphlan_euk <- phyloseq_unfiltered$phyloseq_count %>% 
        subset_taxa(., Domain == "k__Eukaryota")
sample_data(metaphlan_euk)$S.obs <- ((metaphlan_euk %>% otu_table) != 0 )%>% colSums()
sample_data(metaphlan_euk) %>% group_by(sample_type, treatment) %>% summarise(mean_Sobs = mean(S.obs))

```

*Subsetting phyloseq object having no Eukaryotes*

```{r}
metaphlan_bac_arc <- phyloseq_unfiltered$phyloseq_count %>% 
        subset_taxa(., Domain != "k__Eukaryota")

sample_data(metaphlan_bac_arc) <- alpha_diversity(metaphlan_bac_arc)

```


### Analysis on higher level taxonomy

*Were there differences in phylum richness after treatment?*

No.

```{r}
metaphlan_bac_arc_p <- metaphlan_bac_arc %>% tax_glom(., taxrank = "Phylum")
taxa_names(metaphlan_bac_arc_p) <- metaphlan_bac_arc_p %>% tax_table() %>% data.frame %>% .$Phylum
sample_data(metaphlan_bac_arc_p)$Phylum.obs <- ((metaphlan_bac_arc_p %>% otu_table) != 0)%>% colSums()
sample_data(metaphlan_bac_arc_p) %>% group_by(sample_type, treatment) %>% summarise(mean_phylum_richness = mean(Phylum.obs))
```

*Were there differences in class richness after treatment?*

It seems like a new class was observed at QIAamp treated nasal swab, by metaphan3.

```{r}
metaphlan_bac_arc_class <- metaphlan_bac_arc %>% tax_glom(., taxrank = "Class")
taxa_names(metaphlan_bac_arc_class) <- metaphlan_bac_arc_class %>% tax_table() %>% data.frame %>% .$Class
sample_data(metaphlan_bac_arc_class)$Class.obs <- ((metaphlan_bac_arc_class %>% otu_table) != 0)%>% colSums()
sample_data(metaphlan_bac_arc_class) %>% group_by(sample_type, treatment) %>% 
        summarise(mean_class_richness = mean(Class.obs),
                  sd_class_richness = sd(Class.obs))
```

*Were there differences in class richness after treatment?*

It seems like a new class was observed at QIAamp treated nasal swab, by metaphan3.

### Listing newly found taxa at nasal swab after QIAamp treatment

```{r}
NS_qiaamp_samples <- sample_data %>% 
        subset(., .$sample_type == "Nasal") %>% 
        subset(., .$treatment == "QIAamp") %>%
        .$subject_id


metaphlan3_ns6 <- phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(subject_id == "NS_6") %>% 
        prune_taxa(taxa_sums(.) != 0, .)
control_ns_species <- metaphlan3_ns6 %>%  
        subset_samples(treatment == "Untreated")  %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>% taxa_names()
new_taxa_ns6 <- prune_taxa(!(taxa_names(metaphlan3_ns6) %in% control_ns_species), metaphlan3_ns6)  %>% 
        prune_taxa(taxa_sums(.) != 0, .) 

metaphlan3_ns8 <- phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(subject_id == "NS_8") %>% 
        prune_taxa(taxa_sums(.) != 0, .)
control_ns_species <- metaphlan3_ns8 %>%  
        subset_samples(treatment == "Untreated")  %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>% taxa_names()
new_taxa_ns8 <- prune_taxa(!(taxa_names(metaphlan3_ns8) %in% control_ns_species), metaphlan3_ns8)  %>% 
        prune_taxa(taxa_sums(.) != 0, .) 

metaphlan3_ns17 <- phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(subject_id == "NS_17") %>% 
        prune_taxa(taxa_sums(.) != 0, .)
control_ns_species <- metaphlan3_ns17 %>%  
        subset_samples(treatment == "Untreated")  %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>% taxa_names()
new_taxa_ns17 <- prune_taxa(!(taxa_names(metaphlan3_ns17) %in% control_ns_species), metaphlan3_ns17)  %>% 
        prune_taxa(taxa_sums(.) != 0, .) 

metaphlan3_ns18 <- phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(subject_id == "NS_18") %>% 
        prune_taxa(taxa_sums(.) != 0, .)
control_ns_species <- metaphlan3_ns18 %>%  
        subset_samples(treatment == "Untreated")  %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>% taxa_names()
new_taxa_ns18 <- prune_taxa(!(taxa_names(metaphlan3_ns18) %in% control_ns_species), metaphlan3_ns18)  %>% 
        prune_taxa(taxa_sums(.) != 0, .) 

metaphlan3_ns37 <- phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(subject_id == "NS_37") %>% 
        prune_taxa(taxa_sums(.) != 0, .)
control_ns_species <- metaphlan3_ns37 %>%  
        subset_samples(treatment == "Untreated")  %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>% taxa_names()
new_taxa_ns37 <- prune_taxa(!(taxa_names(metaphlan3_ns37) %in% control_ns_species), metaphlan3_ns37)  %>% 
        prune_taxa(taxa_sums(.) != 0, .) 

newly_found_taxa <- merge_phyloseq(new_taxa_ns6, new_taxa_ns8, new_taxa_ns17, new_taxa_ns18, new_taxa_ns37) %>%
        tax_table %>%
        data.frame


marker_magu_db <- read.csv("/Users/minsikkim/Dropbox (Personal)/@minsik/project_ucas/Marker-magu_20240112/Marker-magu_DB_v1.1_metadata/mpa_vOct22_CHOCOPhlAnSGB_202212_SGB_metadata.tsv", sep = "\t") %>% .$lineage %>% str_split_fixed(., "[|]", 8) %>% 
        data.frame() %>% .[,7]


newly_found_taxa$Species[!(newly_found_taxa$Species %in% marker_magu_db)]

```


Corynebacterium_pseudogenitalium:In Metaphan4 dataset, not in Marker-MAGu DB (from mpa_v_Oct22_CHOCOPhlAnSGB)
Propionibacterium_humerusii:In Metaphan4 dataset, not in Marker-MAGu DB (from mpa_v_Oct22_CHOCOPhlAnSGB)
Propionibacterium_namnetense: Not in both Metaphan4 dataset and Marker-MAGu DB (from mpa_v_Oct22_CHOCOPhlAnSGB)
Olsenella_scatoligenes: In Metaphan4 dataset, not in Marker-MAGu DB (from mpa_v_Oct22_CHOCOPhlAnSGB)
Collinsella_massiliensis: In both Metaphan4 dataset and Marker-MAGu DB (from mpa_v_Oct22_CHOCOPhlAnSGB)

### {-}

# Summary

1. Viral species richness increases for

`BAL + MolYsis` and `Sputum + HostZERO or MolYsis or QIAamp.` 

For nasal swabs, `Nasal swabs + QIAamp` was the most effect combination but the p-value become less significant (Effect size 3.9, p-value: 0.073)

2. With Marker-MAGu, effect of treatments on increase of microbial species richness decreased.

The hypotheiss `Marker-MAGu does not have Eukaryotes in the databse therefore the change was underestimated`, and `Marker-MAGu does not have suffient amount of data base` was rejected.

It could be due to the options of Marker-MAGu: 

Marker-MAGu run with `--detection relaxed`: 33% of marker genes required for detection. 

With Marker-MAGu `--detection default`: 75% of marker genes required for detection.

# To do list


1. It would be helpful if we can run the another option (Marker-MAGu run with `--detection relaxed`) but it would take several more weeks

2. I suggest adding phage data (stratified LMR run) as the last main table to the manuscript. The figures (community data and boxplots of viral species richness) can be added as a supplementary material.


## EXTRA

### PERMANOVA stratified by samples

For it is unable to run statistical test on beta diversity as distances of phage community in BAL samples can not be calculted (too many sample with 0 reads)


```{r}
set.seed(seed)

horn_perm_ns <- vegan::adonis2(distance(subset_samples(v_phyloseq_rel, sample_type == "Nasal"), method="horn") ~ lypma + benzonase + host_zero + molysis + qiaamp,
                               data = subset_samples(v_phyloseq_rel, sample_type == "Nasal") %>%
                                       sample_data %>% data.frame(check.names = F),
                               strata = subset_samples(v_phyloseq_rel, sample_type == "Nasal") %>% 
                                       sample_data %>% data.frame(check.names = F) %>% .$subject_id, permutations = 10000)
#set.seed(seed)
#horn_perm_bal  <- vegan::adonis2(distance(subset_samples(v_phyloseq_rel, sample_type == "BAL"), method="horn") ~  lypma + benzonase + host_zero + molysis + qiaamp, data = subset_samples(v_phyloseq_rel, sample_type == "BAL") %>% sample_data %>% data.frame(check.names = F), strata = subset_samples(v_phyloseq_rel, sample_type == "BAL") %>% sample_data %>% data.frame(check.names = F) %>% .$subject_id, permutations = 10000)
set.seed(seed)
horn_perm_spt <- vegan::adonis2(distance(subset_samples(v_phyloseq_rel, sample_type == "Sputum"), method="horn") ~ lypma + benzonase + host_zero + molysis + qiaamp,
                                data = subset_samples(v_phyloseq_rel, sample_type == "Sputum") %>% sample_data %>% data.frame(check.names = F),
                                strata = subset_samples(v_phyloseq_rel, sample_type == "Sputum")
                                %>% sample_data %>% data.frame(check.names = F) %>% .$subject_id,
                                  permutations = 10000)

```



### {-}

Done.

# Bibliography

```{r warning=FALSE}
#===============================================================================
#BTC.LineZero.Footer.1.1.0
#===============================================================================
#R markdown citation generator.
#===============================================================================
#RLB.Dependencies:
#   magrittr, pacman, stringr
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#BTC.Dependencies:
#   LineZero.Header
#===============================================================================
#Generates citations for each explicitly loaded library.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
str_libraries <- c("r", str_libraries)
for (str_libraries in str_libraries) {
    str_libraries |>
        pacman::p_citation() |>
        print(bibtex = FALSE) |>
        capture.output() %>%
        .[-1:-3] %>% .[. != ""] |>
        stringr::str_squish() |>
        stringr::str_replace("_", "") |>
        cat()
    cat("\n")
}
#===============================================================================
```
