---
title: "COD_20230423_HOST_3_analysis_figure_table"
author: "Minsik Kim"
date: "2023-04-23"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
        df_print: paged
editor_options: 
  chunk_output_type: inline
  markdown: 
    wrap: 72
---

# Loading packages

```{r warning=FALSE, message=FALSE, setup}
#===============================================================================
#BTC.LineZero.Header.1.1.0
#===============================================================================
#R Markdown environment setup and reporting utility.
#===============================================================================
#RLB.Dependencies:
#   knitr, magrittr, pacman, rio, rmarkdown, rmdformats, tibble, yaml
#===============================================================================
#Input for document parameters, libraries, file paths, and options.
#=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=
knitr::opts_chunk$set(message=FALSE, warning = FALSE)

path_working <- "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git/"
path_library <- "/Library/Frameworks/R.framework/Resources/library"
str_libraries <- c(
    "readxl", "phyloseq", "tidyverse", "pacman", "yaml"
)

path_working <- "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git"
path_library <- "/Library/Frameworks/R.framework/Resources/library"
str_libraries <- c("readxl", "phyloseq", "tidyverse", "pacman", "yaml", "ggplot2", "vegan", "microbiome", "ggpubr", "viridis", "decontam", "gridExtra", "ggpubr", "lme4", "lmerTest", "writexl", "harrietr", "Maaslin2", "ggtext", "ggpmisc", "gridExtra", "gamm4", "reshape2", "kableExtra", "knitr", "ggtree", "car", "mediation")
        
YAML_header <-
'---
title: "Host-DNA depletion 2: analysis - filtered"
author: "Minsik Kim"
date: "2032.04.23"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
---'
seed <- "20230427"

#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Loads libraries, file paths, and other document options.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
FUN.LineZero.Boot <- function() {
    .libPaths(path_library)

    require(pacman)
    pacman::p_load(c("knitr", "rmarkdown", "rmdformats", "yaml"))

    knitr::opts_knit$set(root.dir = path_working)

    str_libraries |> unique() |> sort() -> str_libraries
    pacman::p_load(char = str_libraries)

    set.seed(seed)
}
FUN.LineZero.Boot()
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Outputs R environment report.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
FUN.LineZero.Report <- function() {
    cat("Line Zero Environment:\n\n")
    paste("R:", pacman::p_version(), "\n") |> cat()
    cat("Libraries:\n")
    for (str_libraries in str_libraries) {
        paste(
            "    ", str_libraries, ": ", pacman::p_version(package = str_libraries),
            "\n", sep = ""
        ) |> cat()
    }
    paste("\nOperating System:", pacman::p_detectOS(), "\n") |> cat()
    paste("    Library Path:", path_library, "\n") |> cat()
    paste("    Working Path:", path_working, "\n") |> cat()
    paste("Seed:", seed, "\n\n") |> cat()
    cat("YAML Header:\n")
    cat(YAML_header)
}
FUN.LineZero.Report()


```



# 1. Loading data

1.1. phyloseq obejct

1.2. qPCR data (controls)

## LIST OF PRIMARY QUESTIONS AND CORRESPONDING ANALYSES {.tabset}

### **3.1. Screening of treatment effect**

*i.	Did treatment change host % in qPCR results?*
        1.	Figure of 4 different matrices (Fig 2)
        
                a.	A: Raw reads (do you really need to log10 transform y-axis?)
                b.	B: Host mapped reads
                c.	C: Final reads
                d.	D: Host DNA ratio

*ii.	How were changes in sequencing results?*
        1.	Sequencing reads (Final reads or total reads) by sample type, descending order (Figure S1)
        2.	Large table of all the information (Table 1) - re-analyzed afterward
        
                a.	Table (summary table of metagenomics output stratified by sample type - see Table 2 in Birdy's DNA extraction method paper, please use only 1 significance digits for effect size, 2 significance digits for p-value to make the table easier to read).  Dataset: % host by metagenomics for nasal, sputum, BAL. People only care about the results of the actual respiratory samples, you would never host deplete mock community samples
                        i.	- total DNA (by Picogreen; this is important since many sequencing centers have cutoffs for minimum DNA concentration to decide whether or not to proceed with sequencing; this is why we were using a low input library prep)
                        ii.	- library prep failure - report both n and % (Table S1)
                                1.	report logistic mixed effects model (please convert to odds ratio)
                                2.	report in text sequencing failure (n) stratified by sample type and treatment method
                                3.	outcome = fail (==1 if failed library prep, ==0 if not)
                                        a.	model (to justify stratified analysis: fail ~ method + sample_type + method*sample_type + (1|subjid), just report interaction p-value
                                        b.	model (stratified analysis): fail ~ method + (1|subjid), report OR [95% CI], p-value for each sample type
                        iii.	qc'd reads
                        iv.	host ratio (Table S2)
                                1.	report linear mixed effects model
                                2.	model (to justify stratified analysis: %host ~ method + sample_type + method*sample_type + (1|subjid), just report interaction p-value
                                3.	model (stratified analysis): %host ~ method + (1|subjid), report beta [95% CI], p-value for each sample type
                        v.	final reads (Table S3)
        3.	Figure of 4 different matrices (Fig. S2)
                a.	A: Raw reads
                b.	B: Host mapped reads
                c.	C: Final reads
                d.	D: Host DNA ratio
                
*iii.	Was host DNA proportion of sequencing similar to that of qPCR? (secondary analysis)*

```
1.	metagenomics is gold standard for % host, but most people don't have the money to do deep sequencing. So secondary analysis is to calculate correlation between %host by qPCR vs %host by metagenomics (this can be a supplementary figure but would at least mention correlation in text) (Fig. S3)
```
        
        
### **3.2. Quality control of sequencing data**

*iv.	Were there any contaminants in the sequencing result? (Do these host depletion methods introduce contamination)*
        
        1.	Negative control – decontam, prevalence method
                a.	Table of decontam analysis results by sample type (Table S4) 
                b.	    - decontam on ALL samples prior to prevalence filtration. I would run this prior to prevalence filtration so people don't accuse us of cheating
                c.	show bugs that show up in negative controls stratified by depletion method after prevalence filtration as a supplemental figure (Figure S4)
                
*v.	Were there any bias in Mock community?*
        
        1.	Stacked bar plot of mock (Figure S5)
                a.	A  - Species level information
                b.	B - gram strain
        2.	Statistical test results on gram stain changes (Table S5)
                a.	Lmer(sqrt(gram-stain) ~ sample type + treatment + sample type * treatment + (1|subject_id) ) 
                
*vi.	M&M - Prevalence filtering*
        
        1.	Prevalence filtration at 5%, except its abundance is over 0.75 quantile.
                a.	Information in the main text
                
### **3.3. Effects of treatments on taxonomic composition**

*vii.	Did diversity matrices change? *

*vii. Do these host depletion methods introduce bias in the sequenced community? *

*viii.	Does bias differ by sample type? *
        
        1.	Figure of alpha and beta diversity changes (Figure 3)
                a.	Alpha - you currently have a box plot stratified by sample type, another approach is to use a forest plot using predictions from your model
                b.	Beta - can consider a forest plot for this one as well instead of a boxplot. For a forest plot you can annotate the x-axis to indicate which direction is better ie towards 0 you can annotate "less bias" and towards higher number you can annotate "more bias"
        2.	Alpha diversity-
                a.	 Run below models to justify stratified analyses (just report the significant method*sample_type interaction term) 
                        i.	species_richness ~ method + sample_type + method * sample_type + log10(reads) + (1|subjid)
                b.	Stratified Species richness - Does host depletion method introduce bias? (Table S6, 7)
                        i.	LMER(species_richness ~ method + log10(reads) + (1|subjid))
                                1.	By sample type, for bal, nasal, sputum
        3.	Beta diversity 
                a.	Justify stratified analysis (Table S8)
                        i.	PERM(horn dist. ~ sample type + treatment + sample type * treatment, strata = subject_id ) 
                                1.	beta_diversiety ~ method + sample_type + method*sample_type + log10(reads) + (1|subjid)
                        ii.	Or, PERM(horn dist. ~ sample type + treatment, strata = subject_id )
                b.	Stratified beta diversity ananlysis (Table 3, Figure S6)
                        i.	beta_diversity ~ method + log10(reads) + (1|subjid)
                c.	Calculate Horn-Morisita distance between control and host depleted sample type and use that as an outcome in mixed effecs model (Table S9)


*ix.	Mediation analysis *



*x.	How werer changes of individual taxa? *
        
        1.	Volcano plot with Mock, BAL, Nasal and Sputum (Figure S7)
                a.	Maaslin (feature ~ log10(Final reads) + lyPMA + Benzonase + Host zero + Molysis + QIAamp, RE = subject_id)
                        i.	make sure that you are doing this analysis accounting for the compositional nature of this data
                b.	Or Maaslin ( feature ~ log10(Final reads) + sample type + treatment + sample type * treatment, RE = subject_id) 
        2.	Balloon plots Mock, BAL, Nasal and Sputum. (Figure 4)
                        a.	Add q-val, mean relative abundance, gram strain, phylogenetic information
                        
                        
### **3.4. Effect of treatments on functional analysis results**

*xi.	Did depletion methods change diversity, by sample type?*
        
        1.	How alpha-diversity changed along treatment?
                a.	Figure of alpha (Species richness) and beta diversity (Fig. S8)
                b.	Stratified LMER for alpha diversity changes (Table S10)
                        i.	Lmer(S.obs~ sample type + treatment + sample type * treatment + (1|subject_id) ) 
                c.	Table of  beta diversity, Mock, BAL, Nasal, Sputum (Table S11)
                        i.	PERM( dist. ~ log10(Final reads) + sample type + lyPMA + Benzonase + Host zero + Molysis + QIAamp, strata = subject_id)
                                1.	Or, PERM(horn dist. ~ log10(Final reads) + sample type + treatment + sample_type * treatment, strata = subject_id)
                                
*xii.	What type of bugs were affected by the treatment?*
        
        1.	DA analysis result by sample type (Fig. S9)
                a.	A: Mock
                b.	B: BAL
                c.	C: Nasal
                d.	D: Sputum
                        i.	Including gram strain information, higher level of phylogenetic tree
        2.	Volcano plot with Mock, BAL, Nasal and Sputum (Figure S3)
                a.	Maaslin ( feature ~ log10(Final reads) + sample type +  lyPMA + Benzonase + Host zero + Molysis + QIAamp, RE = subject_id)
                b.	Or Maaslin ( feature ~ log10(Final reads) + sample type + treatment + sample type * treatment, RE = subject_id)


**3.5. Create a summary table like Table 5 in Birdy's DNA extraction methods paper to help drive home the takehome message. There are pros and cons to each method** (Table 4)


### Data inputs

*Meta data*

-   qPCR - bacteria

-   qPCR - human

-   qPCR host %

-   Raw reads

-   final reads

-   sequencing host %

-   library prep failure status

-   Raw reads

-   subject_id

-   treatment

-   sample_type

-   subject_id

*Sequencing result*

-   samples

-   controls

###  {-}





## Aanalysis preparation


## Analysis prep {.tabset}

### Loading data

```{r warning=F, message=FALSE, "Data loading"}
# Loading files -----------------------------------------------------------
#loading tidy phyloseq object
phyloseq <- read_rds("/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20230521_MGK_host_tidy.rds")

#sample data loading
sample_data <- sample_data(phyloseq$phyloseq_count)

```

### Alpha diversity indices

```{r}           



alpha_diversity <- function(data) {
        otu_table <- otu_table(data) %>% .[colSums(.) !=0]
        S.obs <- rowSums(t(otu_table) != 0)
        sample_data <- sample_data(data)
        data_evenness <- vegan::diversity(t(otu_table)) / log(vegan::specnumber(t(otu_table))) # calculate evenness index using vegan package
        data_shannon <- vegan::diversity(t(otu_table), index = "shannon") # calculate Shannon index using vegan package
        data_hill <- exp(data_shannon)                           # calculate Hills index
        data_dominance <- microbiome::dominance(otu_table, index = "all", rank = 1, aggregate = TRUE) # dominance (Berger-Parker index), etc.
        data_invsimpson <- vegan::diversity(t(otu_table), index = "invsimpson")                          # calculate Shannon index using vegan package
        alpha_diversity <- cbind(S.obs, data_shannon, data_hill, data_invsimpson, data_evenness,data_dominance) # combine all indices in one data table
        sample_data <- merge(data.frame(sample_data), alpha_diversity, by = 0, all = T) %>% column_to_rownames(var = "Row.names")
}
phyloseq_unfiltered <- phyloseq
sample_data(phyloseq_unfiltered$phyloseq_rel) <- sample_data(alpha_diversity(phyloseq$phyloseq_count))
sample_data(phyloseq_unfiltered$phyloseq_count) <- sample_data(alpha_diversity(phyloseq$phyloseq_count)) 
sample_data(phyloseq_unfiltered$phyloseq_path_rpk) <- sample_data(alpha_diversity(phyloseq$phyloseq_path_rpk))  
```

## {-}

# **3.1. Screening of treatment effect**

# *i.	Did treatment change host % in qPCR results? *



*i.	Did treatment change host % in qPCR results? *
        1.	Figure of 4 different matrices (Fig 2)
                a.	A: Raw reads (do you really need to log10 transform y-axis?)
                b.	B: Host mapped reads
                c.	C: Final reads
                d.	D: Host DNA ratio
                
                
## qPCR and seqeuncing (Figure 2) {.tabset}                

### qPCR figure

**Figure 2**. qPCR result of host depletion study. A. Total DNA B. Host DNA
C. Bacterial DNA D. Host DNA ratio
                

```{r}
#2A: Change in total DNA (qPCR)
f2a <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(DNA_host_nondil + DNA_bac_nondil))) +
        geom_boxplot(aes(fill = treatment), lwd = 0.2) +
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        ylab("log<sub>10</sub>(qPCR total DNA)<br>(ng/μL)") +
        xlab("Sample type") +
        theme_classic (base_size = 12, base_family = "serif") + 
        labs(tag = "A") +
        #scale_x_discrete(label = c( "Mock", "Neg.", "BAL", "Nasal", "Sputum")) +
        theme(plot.tag = element_text(size = 15), axis.title.y = element_markdown()) +              # Plot title size
        guides(fill = guide_legend(nrow = 1, title = "Treatment"))


#2B: Change in human DNA (qPCR)
f2b <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(DNA_host_nondil))) +
        geom_boxplot(aes(fill = treatment), lwd = 0.2) +
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3")) +
        ylab("log<sub>10</sub>(qPCR host DNA)<br>(ng/μL)") +
        xlab("Sample type") +
        theme_classic (base_size = 12, base_family = "serif")+ 
        labs(tag = "B") +
        #scale_x_discrete(label = c("BAL", "Nasal", "Sputum", "Mock", "Neg.")) +
        theme(plot.tag = element_text(size = 15), axis.title.y = element_markdown()) +              # Plot title size
        guides(fill = guide_legend(nrow = 1, title = "Treatment"))
#2C: Change in 16S DNA (qPCR)
f2c <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(DNA_bac_nondil))) +
        geom_boxplot(aes(fill = treatment), lwd = 0.2) +
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3")) +
        ylab("log<sub>10</sub>(qPCR bacterial DNA)<br>(ng/μL)") +
        xlab("Sample type") +
        theme_classic (base_size = 12, base_family = "serif")+ 
        labs(tag = "C") +
        #scale_x_discrete(label = c("BAL", "Nasal", "Sputum", "Mock", "Neg.")) +
        theme(plot.tag = element_text(size = 15), axis.title.y = element_markdown()) +              # Plot title size
        guides(fill = guide_legend(nrow = 1, title = "Treatment"))

#2D. Change in % host (qPCR)
f2d <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = host_proportion)) +
        geom_boxplot(aes(fill = treatment), lwd = 0.2) +
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3")) +
        ylab("Host DNA ratio") +
        xlab("Sample type") +
        theme_classic (base_size = 12, base_family = "serif") + 
        labs(tag = "D") +
        #scale_x_discrete(label = c("BAL", "Nasal", "Sputum", "Mock", "Neg.")) +
        theme(plot.tag = element_text(size = 15), axis.title.y = element_markdown()) +              # Plot title size
        guides(fill = guide_legend(nrow = 1, title = "Treatment"))

#output for markdown
figure2 <- ggarrange(f2a, f2b, f2c, f2d, common.legend = T , align = "hv")

figure2 



png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/Figure2.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 170, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figure2
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()


```


### Total reads by sample type (Figure S1)

*ii.	How were changes in sequencing results?*
        1.	Sequencing reads (Final reads or total reads) by sample type, descending order (Figure S1)
        
**Figure S1. **Sum of MetaPhlan mapped reads by sample type

```{r warning=F, QC1.5.0}
#how were the samples failed in library prep?
figS1 <- sample_data %>% data.frame %>% mutate(total_read = phyloseq$phyloseq_count %>% otu_table %>% colSums()) %>%
        ggplot(aes(x = reorder(baylor_other_id, -total_read),
                               y = log10(total_read + 1),
                               col = sample_type)) +
                geom_point() +
                theme_classic(base_family = "serif") +
                theme(axis.title.y = element_markdown(), axis.text.x = element_blank()) +
                ylab("log<sub>10</sub>(Sum of MetaPhlan mapped reads)") +
                xlab("Samples") +
        guides(col=guide_legend(title="Sample type")) +
        scale_color_brewer(type = "qual", palette = 6)




figS1


png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS1.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 90, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figS1
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()



```

### {-}





# *ii.	How were changes in sequencing results?*        




## Changes in sequencing output {.tabset}

### Table of sequencing result

*ii.	How were changes in sequencing results?*        
        
        2.	Large table of all the information (Table 1)
                a.	Table (summary table of metagenomics output stratified by sample type - see Table 2 in Birdy's DNA extraction method paper, please use only 1 significance digits for effect size, 2 significance digits for p-value to make the table easier to read).  Dataset: % host by metagenomics for nasal, sputum, BAL. People only care about the results of the actual respiratory samples, you would never host deplete mock community samples
                
                        i.	- total DNA (by Picogreen; this is important since many sequencing centers have cutoffs for minimum DNA concentration to decide whether or not to proceed with sequencing; this is why we were using a low input library prep)
                        ii.	- library prep failure - report both n and %
                                1.	report logistic mixed effects model (please convert to odds ratio)
                                2.	report in text sequencing failure (n) stratified by sample type and treatment method
                                3.	outcome = fail (==1 if failed library prep, ==0 if not)
                                        a.	model (to justify stratified analysis: fail ~ method + sample_type + method*sample_type + (1|subjid), just report interaction p-value
                                        b.	model (stratified analysis): fail ~ method + (1|subjid), report OR [95% CI], p-value for each sample type
                        iii.	qc'd reads
                        iv.	host ratio
                                1.	report linear mixed effects model
                                2.	model (to justify stratified analysis: %host ~ method + sample_type + method*sample_type + (1|subjid), just report interaction p-value
                                3.	model (stratified analysis): %host ~ method + (1|subjid), report beta [95% CI], p-value for each sample type
                        v.	final reads


**Table 1 is generated later (after QCing to remove sequencing failed samples, etc.)**

### Statistical tests for sequencing results

*ii.	Statistical tests on how were changes in sequencing results*
                        
                        ii.	- library prep failure - report both n and %
                                3.	outcome = fail (==1 if failed library prep, ==0 if not)
                                        a.	model (to justify stratified analysis: fail ~ method + sample_type + method*sample_type + (1|subjid), just report interaction p-value
                                        b.	model (stratified analysis): fail ~ method + (1|subjid), report OR [95% CI], p-value for each sample type
                        iv.	host ratio
                                1.	report linear mixed effects model
                                2.	model (to justify stratified analysis: %host ~ method + sample_type + method*sample_type + (1|subjid), just report interaction p-value
                                3.	model (stratified analysis): %host ~ method + (1|subjid), report beta [95% CI], p-value for each sample type
                                
                        
## Library failure (Table S1) {.tabset}

### Library failure (all samples)

**Table S1. ** Effect size, standard error (SE) and t-value at a statistical test on library prep failure rate using generalized linear mixed effect model. glm ( sequencing fail \~ sample_type + treatment + sample_type \*
treatment + (1|subject_id) )

```{r warning=F, ''}

tableS1 <- glmer(lib_failed ~ sample_type + treatment + sample_type * treatment + (1|subject_id), data = sample_data %>% data.frame %>% dplyr::filter(sample_type %in% c("Sputum", "Nasal", "BAL"))) %>%
        summary %>% .$coefficients %>% data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`t value`) > 2 ~ "*", .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("sample_type|treatment", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>% 
        column_to_rownames(var = "x") %>% 
        rename("<i>t</i>-value" = "t value",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 2) %>% format(nsmall = 2), 
                                " (", 
                                round(Estimate - 1.96 * SE, 2),
                                ", ",
                                round(Estimate + 1.96 * SE, 2),
                                ")", 
                                sep = ""),
               "<i>t</i>-value" = round(`<i>t</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>t</i>-value", " ")) %>% 
        kbl(format = "html", escape = 0) %>% kable_styling(full_width = 0, html_font = "serif")



        


tableS1

save_kable(tableS1, file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS1.html", self_contained = T)

```
                                    
### Library failure (stratified)

glm ( sequencing fail \~ treatment + subject_id )

*--> Cannot run for Sputum (no failed sample). *

```{r warning=F, ''}

glmer(lib_failed ~ treatment + (1|subject_id), data = sample_data %>% data.frame %>% dplyr::filter(sample_type %in% c("BAL"))) %>%
        summary %>% .$coefficients %>% data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`t value`) > 2 ~ "*", .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("sample_type|treatment", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>% 
        rename("<i>t</i>-value" = "t value",
               SE = "Std. Error",
               "Effect size" = "Estimate") %>%
        mutate(across(is.numeric, round, digits=2)) %>% 
        column_to_rownames(var = "x") %>%
        kbl(format = "html", escape = 0) %>% kable_styling(full_width = 0, html_font = "serif")


glmer(lib_failed ~ treatment + (1|subject_id), data = sample_data %>% data.frame %>% dplyr::filter(sample_type %in% c("Nasal"))) %>%
        summary %>% .$coefficients %>% data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`t value`) > 2 ~ "*", .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub(":", " * ", x)) %>% 
        mutate(across(is.numeric, round, digits=2)) %>% mutate(x = gsub("sample_type|treatment", "", x)) %>% 
        column_to_rownames(var = "x") %>% 
        kbl(format = "html") %>% kable_styling(full_width = 0, html_font = "serif")

### Library failure (Spt) - glm ( sequencing fail \~ treatment + subject_id ) - **Cannot run test; no failed sample **



```

### Library failure (OR)

Both OR and RR cannot be calculated (no cases in untreated group, infinite or 1)

```{r warning=F, ''}
sample_data %>%
        data.frame %>%
        group_by(sample_type, treated, lib_failed) %>%
        summarise(N = n())
```

### {-}

## Host ratio (Table S2) {.tabset}




### Host ratio (all samples)

**None-stratified. p-value of interaction term was 4.587e-16**

```{r warning=F}


lmer(sequencing_host_prop * 100 ~ sample_type * treatment + (1|subject_id), 
                    data = sample_data %>%
                            data.frame %>%
                            mutate(total_read = phyloseq$phyloseq_count %>%
                            otu_table %>%
                            colSums()) %>%
                            subset(., total_read != 0)) %>%
        anova() 
        
```

### Host ratio (stratified) (Table S2)


**Table S2. ** Effect size, standard error (SE) and p-value at a statistical test on host DNA ratio using linear mixed effect model. lmer( Host DNA ratio vs sample_type + treatment + sample_type \* treatment +
(1\|subject_id) )

**Host zero and Molysius was effect to to all QIAamp was effective for Nasal swab and Sputum**

```{r warning=F}


hr_lmer_bal <- lmer(sequencing_host_prop * 100 ~ treatment + (1|subject_id), 
                    data = sample_data %>%
                            data.frame %>%
                            mutate(total_read = phyloseq$phyloseq_count %>%
                            otu_table %>%
                            colSums()) %>%
                            subset(., .$sample_type %in% c("BAL") & total_read != 0)) %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = 1), 
                                " (", 
                                round(Estimate - 1.96 * SE, 1) %>% format(nsmall = 1),
                                ", ",
                                round(Estimate + 1.96 * SE, 1) %>% format(nsmall = 1),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " "))
        
hr_lmer_ns <- lmer(sequencing_host_prop * 100 ~ treatment + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Nasal"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("ns_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = ), 
                                " (", 
                                round(Estimate - 1.96 * SE, 1) %>% format(nsmall = ),
                                ", ",
                                round(Estimate + 1.96 * SE, 1) %>% format(nsmall = ),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " "))


hr_lmer_spt <- lmer(sequencing_host_prop * 100 ~ treatment + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Sputum"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("spt_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
                rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = 1), 
                                " (", 
                                round(Estimate - 1.96 * SE, 1) %>% format(nsmall = 1),
                                ", ",
                                round(Estimate + 1.96 * SE, 1) %>% format(nsmall = 1),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " "))


tables2 <- cbind(hr_lmer_bal, hr_lmer_ns, hr_lmer_spt) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "serif")
tables2
save_kable(tables2, file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS2.html", self_contained = T)


```

### {-}


## Final reads (Table S3) {.tabset}


### Final reads (all samples)

**None-stratified. p-value of interaction term was 3.955e-09**

```{r warning=F}


lmer(Final_reads * 100 ~ sample_type * treatment + (1|subject_id), 
                    data = sample_data %>%
                            data.frame %>%
                            mutate(total_read = phyloseq$phyloseq_count %>%
                            otu_table %>%
                            colSums()) %>%
                            subset(., total_read != 0)) %>%
        anova() 
        
```


### Final reads (stratified) (Table S3)

lmer( Host DNA ratio ~ treatment + (1\|subject_id) )

**Table S3.** Changes on final reads stratified by sample type tested with linear mixed effect models (log10(Final reads) ~ Treatment + (1|Subject id)). Effect size with adjusted 95% confidence intervals and p-value were listed. The unit of final read is reads x 10^6. Statistical significances were noted with `*`: p-value < 0.05, `**`: p-value < 0.01 and `***`: p-value < 0.001.

**Sputum's final read increased after every treatment. Nasal swab showed improved reads with lyPMA, Host zero and QIAamp. BAL also showed increased reads with most of treatment.**

```{r warning=F}

fr_lmer_bal <- lmer(log10(Final_reads/1000000) ~ treatment + (1|subject_id),
                    data = sample_data %>%
                                data.frame %>%
                                mutate(total_read = phyloseq$phyloseq_count %>%
                                otu_table %>%
                                colSums()) %>%
             subset(., .$total_read != 0 & .$sample_type %in% c("BAL"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = 1), 
                                " (", 
                                round(Estimate - 1.96 * SE, 1) %>% format(nsmall = 1),
                                ", ",
                                round(Estimate + 1.96 * SE, 1) %>% format(nsmall = 1),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " "))
        
fr_lmer_ns <- lmer(log10(Final_reads/1000000) ~ treatment + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Nasal"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("ns_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = ), 
                                " (", 
                                round(Estimate - 1.96 * SE, 1) %>% format(nsmall = ),
                                ", ",
                                round(Estimate + 1.96 * SE, 1) %>% format(nsmall = ),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " "))


fr_lmer_spt <- lmer(log10(Final_reads/1000000) ~ treatment + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Sputum"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("spt_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
                rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = 1), 
                                " (", 
                                round(Estimate - 1.96 * SE, 1) %>% format(nsmall = 1),
                                ", ",
                                round(Estimate + 1.96 * SE, 1) %>% format(nsmall = 1),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " "))


tables3 <- cbind(fr_lmer_bal, fr_lmer_ns, fr_lmer_spt) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "serif")
tables3

save_kable(tables3, file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS3.html", self_contained = T)

```
### {-}




## Figure of sequencing result (Fig. S2-3) {.tabset}



### Figure of sequencing result (Fig. S2)

*ii.	How were changes in sequencing results?*

        3.	Figure of 4 different matrices (Fig. S2)
        
                a.	A: Raw reads
                b.	B: Host mapped reads
                c.	C: Final reads
                d.	D: Host DNA ratio
                
                
**Figure S2.**  Sequencing result of host depletion study. A. Total DNA B.
Host DNA C. Bacterial DNA D. Host DNA ratio

```{r}

f3a <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(Raw_reads))) +
        geom_boxplot(aes(fill = treatment), lwd = 0.2) +
        theme_classic (base_size = 12, base_family = "serif") + 
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "Host zero", "Molysis", "QIAaamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Sample type") +
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15)) +
        ylab("log<sub>10</sub>(raw reads)") +
        labs(tag = "A") +
        guides(fill = guide_legend(nrow = 1))

#	- Host_mapped


f3b <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(Host_mapped))) +
        theme_classic (base_size = 12, base_family = "serif")+ 
        geom_boxplot(aes(fill = treatment), lwd = 0.2) +
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "Host zero", "Molysis", "QIAaamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Sample type")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15)) +
        ylab("log<sub>10</sub>(host reads)") +
        labs(tag = "B") +
        guides(fill = guide_legend(nrow = 1))


#	- % Host (we have used Host_mapped/Raw_reads in prior papers)

#	- Final_reads

f3c <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(Final_reads))) +
        geom_boxplot(aes(fill = treatment), lwd = 0.2) +
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "Host zero", "Molysis", "QIAaamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Sample type")+
        ylab("log<sub>10</sub>(final reads)") +
        theme_classic (base_size = 12, base_family = "serif") + 
        theme(axis.title.y = element_markdown(),
              plot.tag = element_markdown(size = 15)) +
        labs(tag = "C") +
        guides(fill = guide_legend(nrow = 1))


#	- % Host (we have used Host_mapped/Raw_reads in prior papers)
f3d <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = sequencing_host_prop)) +
        geom_boxplot(aes(fill = treatment), lwd = 0.2) +
        theme_classic (base_size = 12, base_family = "serif")+ 
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "Host zero", "Molysis", "QIAaamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Sample type")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15)) +
        ylab("Host ratio by sequencing") +
        labs(tag = "D") +
        guides(fill = guide_legend(nrow = 1))



figS2 <- ggarrange(f3a, f3b, f3c, f3d, common.legend = T, align = "hv")

figS2


png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS2.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 170, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figS2
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()



```

### {-}


# *iii.	Was host DNA proportion of sequencing similar to that of qPCR? (secondary analysis)*



## Host ratio qPCR vs sequencing (Figure 3)
        
        1.	metagenomics is gold standard for % host, but most people don't have the money to do deep sequencing. So secondary analysis is to calculate correlation between %host by qPCR vs %host by metagenomics (this can be a supplementary figure but would at least mention correlation in text)

**Figure S3.** Correlation between host DNA proportion measured with qPCR and shotgun metagenomic sequencing.

```{r, message=FALSE}

figS3 <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = host_proportion, y = sequencing_host_prop, col = sample_type)) +
        geom_point() +
        theme_classic (base_size = 12, base_family = "serif")+ 
        scale_color_manual(values = c("#e31a1c", "#33a02c", "#1f78b4")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        #scale_x_discrete(name ="Sample type")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              legend.position = c(0.9,0.2)) +
        ylab("Host ratio by sequencing") +
        xlab("Host ratio by qPCR") +
        labs(col = "Sample type") +
        annotate(family = "serif",
                 geom='richtext',
                 x=0.5, y=1,
                 label = paste("R<sup>2</sup> = ",
                               lm(sequencing_host_prop ~ host_proportion,
                                  data = sample_data %>%
                                          data.frame %>% 
                                          subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")))
                                  %>% summary %>% .$r.squared %>% round(., 2) %>% format(nsmall = 2), sep = "")) +
        geom_smooth(method=lm , color="red", se=T, level = 0.95) +
        guides(fill = guide_legend(nrow = 1))
 
figS3


png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS3.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 90, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figS3
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()



```


# **3.2. Quality control of sequencing data**

# *iv.	Were there any contaminants in the sequencing result? (Do these host depletion methods introduce contamination)*

        1.	Negative control (Fig. S4)
                a.	Table of decontam analysis results by sample type (Table S) 
                b.	    - decontam on ALL samples prior to prevalence filtration. I would run this prior to prevalence filtration so people don't accuse us of cheating
                c.	show bugs that show up in negative controls stratified by depletion method after prevalence filtration as a supplemental figure?

## Negative controls {.tabset}

### Bar plot of negative controls (Fig. S4)

**Fig. S4**. Barplot of negative controls by treatment methods at species level
                
```{r, warning=FALSE}

#Loading theoretical mock community
zymo_mock <- read_excel("/Users/minsikkim/Dropbox (Partners HealthCare)/@minsik/project_sicas2/data_raw/DAR_20210929_zymo_mock_data.xlsx") %>%
        data.frame(row.names = T) %>% rename(mock_theoretical = Mock) %>% mutate(mock_theoretical = mock_theoretical/100) %>%
        merge_phyloseq(otu_table(., taxa_are_rows = T), tax_table(phyloseq$phyloseq_count))

phyloseq_mock <- rbind(c("mock_theoretical", "Mock theoretical", "-")) %>% data.frame() %>%
        column_to_rownames(var = "X1") %>% rename(sample_type = X2, treatment = X3) %>% #making sample_data of mock community
        merge_phyloseq(sample_data(.), zymo_mock)

phyloseq_control_rel <- subset_samples(phyloseq$phyloseq_rel, sample_type == "Mock" | sample_type == "Neg.") #adding data of controls
sample_data(phyloseq_control_rel)$treatment <- sample_data(phyloseq_control_rel)$treatment %>% as.character()
sample_data(phyloseq_control_rel)$sample_type <- sample_data(phyloseq_control_rel)$sample_type %>% as.character()
phyloseq_control_rel <- merge_phyloseq(phyloseq_control_rel, phyloseq_mock) 


#Species richness of each control groups
sample_data(phyloseq_control_rel)$S.obs <- rowSums(t(otu_table(phyloseq_control_rel)) != 0)
sample_data(phyloseq_control_rel)$sample_type <- 
        factor(sample_data(phyloseq_control_rel)$sample_type, levels = c("Mock theoretical", "Mock", "Neg."))
sample_data(phyloseq_control_rel)$treatment <-
        factor(sample_data(phyloseq_control_rel)$treatment, levels = c("-", "Control", "lyPMA", "Benzonase", "Host zero", "Molysis", "QIAamp"))


#Manipulating phyloseq - only top 10 

my_plot_bar = function (physeq, x = "Sample", y = "Abundance", fill = NULL, title = NULL, 
                        facet_grid = NULL) {
    mdf = psmelt(physeq)
    p = ggplot(mdf, aes_string(x = x, y = y, fill = fill))
    p = p + geom_bar(stat = "identity")
    p = p + theme(axis.text.x = element_text(angle = -90, hjust = 0))
    if (!is.null(facet_grid)) {
        p <- p + facet_grid(facet_grid)
    }
    if (!is.null(title)) {
        p <- p + ggtitle(title)
    }
    return(p)
}


figS4 <- tax_table(phyloseq_control_rel) %>%
        cbind(species20 = "[Others]") %>%
        {top20species <- head(taxa_sums(phyloseq_control_rel) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Others",.[, 9]),
                    "Others",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .) %>%
                            gsub("s ", " ", .)
   )
   phyloseq_temp <- phyloseq_control_rel
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        subset_samples(., sample_type == "Neg.") %>%
        my_plot_bar(., fill="species20") + 
        ylab("Relative abundancne") +
        theme_classic(base_size = 11, base_family = "serif") +
        scale_fill_brewer(type = "qual", palette = 8) +
        theme(legend.text = element_markdown(), axis.text.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 species")) +
        facet_wrap (~ factor(treatment, levels = c("Untreated", "lyPMA", "Benzonase", "Host zero", "Molysis", "QIAamp")),
                    scales= "free_x", nrow=1)

figS4

png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS4.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 90, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figS4
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()



```

### {-}


## decontam {.tabset}

### decontam - stratified by sample_type 


**Stratified analysis** showed **no contaminants** in **NS** and **BAL**

Decontam analysis result stratified by sample type

```{r warning=FALSE, message=FALSE}
#Stratified by sample type
sample_data(phyloseq_unfiltered$phyloseq_rel)$is.neg <- grepl("Neg", sample_data(phyloseq_unfiltered$phyloseq_rel)$sample_type)

contaminant <- data.frame() 

#cat("decontam prevalence - all")
contaminant1 <- 
data.frame("prevalence", "all", fix.empty.names = F, 
phyloseq_unfiltered$phyloseq_rel %>% subset_samples(S.obs != 0 & sample_type != "Mock") %>%
        isContaminant(., method="prevalence", neg = "is.neg", threshold = 0.1) %>% subset(.,.$contaminant) %>% row.names()
)

#cat("decontam prevalence - BAL")
contaminant2 <- 
data.frame("prevalence", "BAL", fix.empty.names = F, 
subset_samples(phyloseq_unfiltered$phyloseq_rel %>% subset_samples(S.obs != 0), sample_type %in% c("BAL", "Neg.")) %>%
        isContaminant(., method="prevalence", neg = "is.neg", threshold = 0.1) %>% subset(.,.$contaminant) %>% row.names())

#cat("decontam prevalence - Nasal")
contaminant3 <- 
        data.frame("prevalence", "Nasal swab", fix.empty.names = F, 
subset_samples(phyloseq_unfiltered$phyloseq_rel %>% subset_samples(S.obs != 0), sample_type %in% c("Nasal", "Neg.")) %>%
        isContaminant(., method="prevalence", neg = "is.neg", threshold = 0.1) %>% subset(.,.$contaminant) %>% row.names())

#cat("prevalence", "decontam prevalence - Sputum")
contaminant4 <- 
data.frame("prevalence", "Sputum", fix.empty.names = F,
subset_samples(phyloseq_unfiltered$phyloseq_rel %>% subset_samples(S.obs != 0), sample_type %in% c("Sputum", "Neg.")) %>%
        isContaminant(., method="prevalence", neg = "is.neg", threshold = 0.1) %>% subset(.,.$contaminant) %>% row.names())

#cat("decontam frequency - All")
contaminant5 <- 
data.frame("frequency", "all", fix.empty.names = F,
phyloseq_unfiltered$phyloseq_rel %>% subset_samples(S.obs != 0) %>%
        isContaminant(method="frequency", conc="DNA_bac_well") %>% subset(.,.$contaminant) %>% row.names())

#cat("decontam frequency - BAL")
#contaminant6 <- 
#data.frame("frequency", "BAL", fix.empty.names = F,
#subset_samples(phyloseq_unfiltered$phyloseq_rel %>% subset_samples(S.obs != 0), sample_type %in% c("BAL")) %>%
#        isContaminant(method="frequency", conc="DNA_bac_well") %>% subset(.,.$contaminant) %>% row.names()
#)

#cat("decontam frequency - Nasal")
#contaminant7 <- 
#data.frame("frequency", "Nasal swab", fix.empty.names = F,
#subset_samples(phyloseq_unfiltered$phyloseq_rel %>% subset_samples(S.obs != 0), sample_type %in% c("Nasal")) %>%
#        isContaminant(method="frequency", conc="DNA_bac_well") %>% subset(.,.$contaminant) %>% row.names()
#)

#cat("decontam frequency - Sputum")
contaminant8 <- 
data.frame("frequency", "Sputum", fix.empty.names = F,
subset_samples(phyloseq_unfiltered$phyloseq_rel %>% subset_samples(S.obs != 0), sample_type %in% c("Sputum")) %>%
        isContaminant(method="frequency", conc="DNA_bac_well") %>% subset(.,.$contaminant) %>% row.names()
)
#cat("decontam combined - All")
contaminant9 <- 
data.frame("combined", "all", fix.empty.names = F,

phyloseq_unfiltered$phyloseq_rel %>% subset_samples(S.obs != 0) %>%
        isContaminant(method="combined", neg="is.neg", conc = "DNA_bac_well") %>% subset(.,.$contaminant) %>% row.names())

#cat("decontam combined - BAL")
contaminant10 <- 
data.frame("combined", "BAL", fix.empty.names = F,
subset_samples(phyloseq_unfiltered$phyloseq_rel %>% subset_samples(S.obs != 0), sample_type %in% c("BAL", "Neg.")) %>%
        isContaminant(method="combined", neg="is.neg", conc = "DNA_bac_well") %>% subset(.,.$contaminant) %>% row.names())

#cat("decontam combined - Nasal")
#contaminant11 <- 
#data.frame("combined", "Nasal swab", fix.empty.names = F,
#subset_samples(phyloseq_unfiltered$phyloseq_rel %>% subset_samples(S.obs != 0), sample_type %in% c("Nasal", "Neg.")) %>%
#        isContaminant(method="combined", neg="is.neg", conc = "DNA_bac_well") %>% subset(.,.$contaminant) %>% row.names() 
#)

#cat("decontam combined - Sputum")
contaminant12 <- 
data.frame("combined", "Sputum", fix.empty.names = F,
subset_samples(phyloseq_unfiltered$phyloseq_rel %>% subset_samples(S.obs != 0), sample_type %in% c("Sputum", "Neg.")) %>%
        isContaminant(method="combined", neg="is.neg", conc = "DNA_bac_well") %>% subset(.,.$contaminant) %>% row.names())

#Some contaminant table was not included as they were empty. (Contaminant 6, 7, 11)
contaminant <- rbind(contaminant1, contaminant2, contaminant3, contaminant4, contaminant5, contaminant8, contaminant9, contaminant10, contaminant12)
names(contaminant) <- c("method", "sample_type", "contaminant")
cat("List of contaminant (prevalence method) - from stratified analysis\n\n")
contaminant %>% subset(., .$method == "prevalence" & .$sample_type != "all") %>% .$contaminant %>% unique()

cat("Prevalence decontam table\n\n")
contaminant %>% subset(., .$method == "prevalence") %>% .$sample_type %>% table
cat("Frequency decontam table\n\n")
contaminant %>% subset(., .$method == "frequency") %>% .$sample_type %>% table
cat("Combined decontam table\n\n")
contaminant %>% subset(., .$method == "combined") %>% .$sample_type %>% table
```

### Decontam summary (Table S4)

**Table S4.** Summary table of potential contaminants with all sample types and stratified sample types in all methods (prevalence, frequence, and combined)

```{r}


tableS4 <- matrix(nrow=3,ncol=4) %>% data.frame() %>% rename(" " = X1, "BAL" = X2, "Nasal swab" = X3, "Sputum" = X4) %>%
        rownames_to_column("x") %>% mutate(x = c("Prevalence method", "Frequency method", "Combined"),
                                           " " = c(16, 3, 6),
                                           `BAL` = c(14, 0, 1),
                                           `Nasal swab` = c(12, 0, 0),
                                           `Sputum` = c(12, 3, 2)
                                           ) %>% column_to_rownames("x") %>%
        kbl(format = "html") %>%
        add_header_above(c(" " = 1, "Non-stratified" = 1, "Stratified" = 3)) %>% 
        kable_styling(full_width = 0, html_font = "serif") 
tableS4

save_kable(tableS4, file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS4.html", self_contained = T)
```

### {-}



# *v.	Were there any bias in Mock community? *


        1.	Stacked bar plot of mock (Figure S2)
                a.	A  - Species level information
                b.	B - gram strain

        2.	Statistical test results on gram stain changes (Table S5)
                a.	Lmer(sqrt(gram-stain) ~ sample type + treatment + sample type * treatment + (1|subject_id) ) 

## Controls {.tabset}

### Positive controls (Fig. S5)

                
**Fig. S5**. Barplot of positive controls by treatment methods at (A) species level and (B) gram-stain information

                
```{r, warning=FALSE}

#Manipulating phyloseq - only top 10 


a <- tax_table(subset_samples(phyloseq_control_rel, sample_type == "Mock")) %>% 
        cbind(species20 = "[Others]") %>%
        {top20species <- head(taxa_sums(phyloseq_control_rel) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Others",.[, 9]),
                    "Others",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .) %>%
                            gsub("s ", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq_control_rel,sample_type == "Mock" & S.obs != 0)
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        my_plot_bar(., fill="species20") + 
        ylab("Relative abundancne") +
        theme_classic(base_size = 11, base_family = "serif") +
        theme(legend.text = element_markdown(), axis.text.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 species")) +
        facet_wrap (~ factor(treatment, levels = c("Untreated", "lyPMA", "Benzonase", "Host zero", "Molysis", "QIAamp")),
                    scales= "free_x", nrow=1) +
        labs(tag = "A")

#there could be opportunistic pathogens...
                                


b <- phyloseq_control_rel %>% 
        subset_samples(., sample_type == "Mock") %>% 
        my_plot_bar(., fill="gram_stain") + 
        ylab("Relative abundancne") +
        theme_classic(base_size = 11, base_family = "serif") +
        theme(legend.text = element_markdown(), axis.text.x = element_blank()) +
        guides(fill=guide_legend(title="Gram-stain")) +
        facet_wrap (~ factor(treatment, levels = c("Untreated", "lyPMA", "Benzonase", "Host zero", "Molysis", "QIAamp")),
                    scales= "free_x", nrow=1) +
        labs(tag = "B")

figS5 <- ggarrange(a, b, ncol = 1)

figS5

png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS5.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 170, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figS5
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()



```                


### Gram-stain stats (all samples)
                

**Table S5. ** Effect size, standard error (SE) and p-value at a statistical test on gram-negative proportion using linear mixed effect model. lmer( Gram-negative proportion vs sample_type + treatment + sample_type \* treatment +
(1\|subject_id) )

**Interaction term was significant (p-value = 0.018771)**

```{r warning=F, 'gram strain %'}
lmer(sqrt(gram_neg_prop) ~ sample_type + treatment + sample_type * treatment + (1|subject_id),
     data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum"))) %>%
        anova 
```

### Gram-stain stats -stratified (Table S5)

**Stratified analsyis**

```{r}

tableS5_bal <- lmer(sqrt(gram_neg_prop) ~ treatment + (1|subject_id),
     data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("BAL"))) %>%
        summary %>% .$coefficients %>% data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("sample_type|treatment", "", x))  %>% mutate(x = gsub(":", " * ", x)) %>% 
        column_to_rownames(var = "x") %>%  
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 2) %>% format(nsmall = 2), 
                                " (", 
                                round(Estimate - 1.96 * SE, 2) %>% format(nsmall = 2),
                                ", ",
                                round(Estimate + 1.96 * SE, 2) %>% format(nsmall = 2),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) 

tableS5_ns <- lmer(sqrt(gram_neg_prop) ~ treatment + (1|subject_id),
     data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Nasal"))) %>%
        summary %>% .$coefficients %>% data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("sample_type|treatment", "", x))  %>% mutate(x = gsub(":", " * ", x)) %>% 
        column_to_rownames(var = "x") %>%  
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 2) %>% format(nsmall = 2), 
                                " (", 
                                round(Estimate - 1.96 * SE, 2) %>% format(nsmall = 2),
                                ", ",
                                round(Estimate + 1.96 * SE, 2) %>% format(nsmall = 2),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) 


tableS5_spt <- lmer(sqrt(gram_neg_prop) ~ treatment + (1|subject_id),
     data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Sputum"))) %>%
        summary %>% .$coefficients %>% data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("sample_type|treatment", "", x))  %>% mutate(x = gsub(":", " * ", x)) %>% 
        column_to_rownames(var = "x") %>%  
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 2) %>% format(nsmall = 2), 
                                " (", 
                                round(Estimate - 1.96 * SE, 2) %>% format(nsmall = 2),
                                ", ",
                                round(Estimate + 1.96 * SE, 2) %>% format(nsmall = 2),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) 




tableS5 <- cbind(tableS5_bal, tableS5_ns, tableS5_spt) %>%
        kbl(format = "html", escape = 0) %>% kable_styling(full_width = 0, html_font = "serif") %>% 
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3, "Sputum" = 3)) %>% 
        kable_styling(full_width = 0, html_font = "serif") 

tableS5

save_kable(tableS5, file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS5.html", self_contained = T)


```

### {-}


## Filtering taxa {.tabset}



### Prevalence filtering 

**Prevalence & abundance filtering was conducted**

v.	M&M - Prevalence filtering

        1.	Prevalence filtration at 5%, except its abundance is over 0.75 quantile.
                a.	Information in the main text

```{r warning=FALSE, "Red flag"}
taxa_qc <- data.frame("species" =  otu_table(subset_samples(phyloseq_unfiltered$phyloseq_count,
                                                            S.obs != 0 & sample_type %in% c("Mock", "BAL", "Nasal", "Sputum"))) %>%
                              t() %>% colnames(),
        "prevalence" = ifelse(subset_samples(phyloseq_unfiltered$phyloseq_count, S.obs != 0 & sample_type %in% c("Mock", "BAL", "Nasal", "Sputum")) %>% otu_table() > 0, 1, 0) %>% t() %>% colSums(), #Prevalence of taxa
        "mean_rel_abd" = subset_samples(phyloseq_unfiltered$phyloseq_count, S.obs != 0 & sample_type %in% c("Mock", "BAL", "Nasal", "Sputum")) %>%
                otu_table() %>%
                t() %>%
                colMeans(na.rm = T) #mean relativ abundacne 
)

function_qc <- data.frame("function" =  otu_table(subset_samples(phyloseq_unfiltered$phyloseq_path_rpk, S.obs != 0 & sample_type %in% c("Mock", "BAL", "Nasal", "Sputum"))) %>% t() %>% colnames(),
        "prevalence" = ifelse(subset_samples(phyloseq_unfiltered$phyloseq_path_rpk, S.obs != 0 & sample_type %in% c("Mock", "BAL", "Nasal", "Sputum")) %>% otu_table() > 0, 1, 0) %>% t() %>% colSums(), #Prevalence of taxa
        "mean_rpk" = subset_samples(phyloseq_unfiltered$phyloseq_path_rpk, S.obs != 0 & sample_type %in% c("Mock", "BAL", "Nasal", "Sputum")) %>% otu_table() %>% t() %>% colMeans(na.rm = T) #mean relativ abundacne 
)

red_flag_taxa <- data.frame(species = taxa_qc$species,
                            red_flag_prev_abd = ifelse(taxa_qc$prevalence < otu_table(phyloseq_unfiltered$phyloseq_rel) %>%
                                                               t %>% rownames() %>%
                                length * 0.05 & taxa_qc$mean_rel_abd < quantile(taxa_qc$mean_rel_abd, 0.75), 1,0)) %>%
        mutate(red_flag_decontam_prev = species %in% (contaminant %>%
                                    subset(., .$method == "prevalence" & .$sample_type != "all") %>%
                                    .$contaminant %>% unique()))


red_flag_function <- data.frame(function. = function_qc$function., red_flag_prev_abd = ifelse(function_qc$prevalence < otu_table(phyloseq$phyloseq_path_rpk) %>% t %>% rownames() %>% length * 0.05 & function_qc$mean_rpk < quantile(function_qc$mean_rpk, 0.75), 1, 0))

#phyloseq of table 1
phyloseq$phyloseq_count_prev <- prune_taxa(subset(red_flag_taxa,
                                           red_flag_taxa$red_flag_decontam_prev != 1)$species,
                                    phyloseq$phyloseq_count)
#phyloseq for analysis
phyloseq$phyloseq_rel <- prune_taxa(subset(red_flag_taxa,
                                           red_flag_taxa$red_flag_prev_abd != 1 & !red_flag_taxa$red_flag_decontam_prev)$species,
                                    phyloseq$phyloseq_rel)
phyloseq$phyloseq_count <- prune_taxa(subset(red_flag_taxa,
                                           red_flag_taxa$red_flag_prev_abd != 1 & !red_flag_taxa$red_flag_decontam_prev)$species,
                                    phyloseq$phyloseq_count)
phyloseq$phyloseq_path_rpk <- prune_taxa(subset(red_flag_taxa, red_flag_function$red_flag_prev_abd != 1)$function., phyloseq$phyloseq_path_rpk)

#phyloseq$tree_phyloseq_count <- prune_taxa(subset(red_flag_taxa,
                                           #red_flag_taxa$red_flag_prev_abd != 1 & !red_flag_taxa$red_flag_decontam_prev)$species,
                                    #phyloseq$tree_phyloseq_count)

#phyloseq$tree_phyloseq_rel <- prune_taxa(subset(red_flag_taxa,
                                           #red_flag_taxa$red_flag_prev_abd != 1 & !red_flag_taxa$red_flag_decontam_prev)$species,
                                    #phyloseq$tree_phyloseq_rel)
```

### {-}

## Extras {.tabset}


### Alpha diversity calculation

```{r}

#Calculation of alpha diversity indices for filtered samples

alpha_diversity <- function(data) {
        otu_table <- otu_table(data) %>% .[colSums(.) !=0]
        S.obs <- rowSums(t(otu_table) != 0)
        sample_data <- sample_data(data)
        data_evenness <- vegan::diversity(t(otu_table)) / log(vegan::specnumber(t(otu_table))) # calculate evenness index using vegan package
        data_shannon <- vegan::diversity(t(otu_table), index = "shannon") # calculate Shannon index using vegan package
        data_hill <- exp(data_shannon)                           # calculate Hills index
        data_dominance <- microbiome::dominance(otu_table, index = "all", rank = 1, aggregate = TRUE) # dominance (Berger-Parker index), etc.
        data_invsimpson <- vegan::diversity(t(otu_table), index = "invsimpson")                          # calculate Shannon index using vegan package
        alpha_diversity <- cbind(S.obs, data_shannon, data_hill, data_invsimpson, data_evenness,data_dominance) # combine all indices in one data table
        sample_data <- merge(data.frame(sample_data), alpha_diversity, by = 0, all = T) %>% column_to_rownames(var = "Row.names")
}
sample_data(phyloseq$phyloseq_count_prev) <- sample_data(alpha_diversity(phyloseq$phyloseq_count_prev))
sample_data(phyloseq$phyloseq_rel) <- sample_data(alpha_diversity(phyloseq$phyloseq_count))
sample_data(phyloseq$phyloseq_count) <- sample_data(alpha_diversity(phyloseq$phyloseq_count)) 
sample_data(phyloseq$phyloseq_path_rpk) <- sample_data(alpha_diversity(phyloseq$phyloseq_path_rpk))  
#sample_data(phyloseq$tree_phyloseq_count) <- sample_data(alpha_diversity(phyloseq$tree_phyloseq_count))
#phyloseq$tree_phyloseq_rel <- transform_sample_counts(phyloseq$tree_phyloseq_count, function (x) {x/sum(x)})
sample_data <- sample_data(phyloseq$phyloseq_count)

```

### Table 1 (revised)

**Revised Table 1 was constructed after employing decontam**

**Table 1 (revised)**. Sample number of each experimental group, total DNA quantified by PicoGreen, library prep QC result, low quality removed read counts, ratio of host mapped reads and final microbial reads.

```{r, warning=F, error=FALSE, results='asis', `QC2 Table - cnt.2`}

sample_data(phyloseq$phyloseq_count_prev)$sequencing_fail <- sample_data(phyloseq$phyloseq_count_prev)$S.obs == 0
sample_data(phyloseq$phyloseq_count_prev)$Final_reads <- ifelse(sample_data(phyloseq$phyloseq_count_prev)$S.obs == 0,
                                                                NA,
                                                                sample_data(phyloseq$phyloseq_count_prev)$Final_reads)
#sample_data(phyloseq$phyloseq_count_prev)$S.obs
#sample_data(phyloseq$phyloseq_count_prev)
table1 <- sample_data(phyloseq$phyloseq_count_prev) %>% data.frame() %>% 
        dplyr::filter(sample_type %in% c("Sputum", "Nasal", "BAL")) %>% 
        group_by (sample_type, treatment) %>%
        summarise(N = n(),
               `Total DNA <br>ng/µL` = paste(format(round(median(DNA_host_ng_uL + DNA_bac_ng_uL),2), nsmall = 2, big.mark = ","), "<br>(", format(round(quantile(DNA_host_ng_uL + DNA_bac_ng_uL, 0.25),2), nsmall = 2, big.mark = ","), ", ", format(round(quantile(DNA_host_ng_uL + DNA_bac_ng_uL, 0.75),2), nsmall = 2, big.mark = ","), ")", sep = ""),
              `Library/<br>sequencing<br>failure<br>N (%)` = paste(sum(lib_failed + sequencing_fail), " (", sum(lib_failed +sequencing_fail) / n() * 100, " %)", sep = ""),
              `QC'd reads<br>reads x 10<sup>6</sup>` = paste(format(round(median(Reads_after_trim/1000000),1), nsmall = 1, big.mark = ","), " (", format(round(quantile(Reads_after_trim/1000000, 0.25),1), nsmall = 1, big.mark = ","), ", ", format(round(quantile(Reads_after_trim/1000000, 0.75),1), nsmall = 1, big.mark = ","), ")", sep = ""),
              `Host reads %` = paste(format(round(median(sequencing_host_prop*100),1),
                                                nsmall = 1, big.mark = ","),
                                         " (",
                                         format(round(quantile(sequencing_host_prop * 100,
                                                               0.25),
                                                      1),
                                                nsmall = 1,
                                                big.mark = ","), 
                                         ", ", 
                                         format(round(quantile(sequencing_host_prop * 100, 0.75),1), 
                                                nsmall = 1, 
                                                big.mark = ","), 
                                         ")", 
                                         sep = ""),
             `Final reads<br>reads x 10<sup>6</sup>` = paste(format(round(median(Final_reads/1000000, na.rm = T),2),
                                                                    nsmall = 2, big.mark = ","),
                                                             " (", 
                                                             format(round(quantile(Final_reads/1000000,
                                                                                            0.25,
                                                                                            na.rm = T),
                                                                                   2),
                                                                             nsmall = 2,
                                                                             big.mark = ","),
                                                             ", ",
                                                             format(round(quantile(Final_reads/1000000, 
                                                                                   0.75, 
                                                                                   na.rm = T),
                                                                          2), 
                                                                    nsmall = 2, 
                                                                    big.mark = ","), 
                                                             ")",
                                                             sep = ""),
             `Species richness<br> N` = paste(median(S.obs, na.rm = T),
                                                             " (", 
                                                             quantile(S.obs, 0.25, na.rm = T),
                                                             ", ",
                                                             quantile(S.obs, 0.75, na.rm = T),
                                                             ")",
                                                             sep = "")
             ) %>% 
        data.frame(check.names = F) %>% 
        arrange(sample_type, treatment) %>%
        rename(`Sample type` = sample_type, Treatment = treatment) %>%
        mutate_all(linebreak) %>% kbl(format = "html", escape = F) %>% kable_styling(full_width = 0, html_font = "serif")

table1 


save_kable(table1, file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/table1.html", self_contained = T)


```

### {-}

## **+ α** *Does breakaay detects underestimated bugs in untreated sample?*

 - Waiting for Amy's analysis result

```{r, message=FALSE, warning=FALSE}
## Diversity estimation lab - STAMPS 2022
## Sarah Teichman, Amy Willis, and Pauline Trinh

# --------------- `breakaway` and `DivNet` ---------------------------

# We're going to learn about using `breakaway` and `DivNet` for diversity estimation
# and comparison!
# Let's begin by loading in all the packages we'll need for this tutorial.
#library(breakaway)
#library(DivNet)

# Let's make sure that `breakaway` and `DivNet` are loaded properly. If either of these
# cause an error, please put up your pink sticky note!

# Here we see that GlobalPatterns is a `phyloseq` object containing an otu table,
# taxonomy table, sample data, and a phylogenetic tree.

# Let's take a look at the sample data for GlobalPatterns to see some more
# information about the data we'll be using.



# It looks like there are 26 samples and 7 variables to describe those samples.
# We're particularly interested in looking at only the water samples so we are
# going to need to subset our samples using the variable SampleType.

#water <- phyloseq_unfiltered$phyloseq_count %>%
#  subset_samples(S.obs != 0)


# We need to decide at which taxonomic level we're interested in understanding
# diversity. Do we want to know about the diversity of our samples at the Kingdom,
# Phylum, Class, Order, Family, Genus, or Species level?

# Let's say we're interested in understanding the diversity in our water samples
# at the Order-level. We can use the `tax_glom()` function in `phyloseq` to
# summarize our data at the Order-level.

#water_order <- water %>% phyloseq::tax_glom("Order")



# A common goal in microbiome research is to look at the observed richness of our
# samples. `phyloseq` has some built-in tools for quantifying alpha diversity, but
# they're not great. They underestimate richness, underestimate uncertainty, and
# don't allow hypothesis testing.

# Let's take a look at a plot of our observed Order-level richness by SampleType
# using a built-in `plot` function from `breakaway`.

#observed_phyloseq <- sample_richness(water_order)
#plot(observed_phyloseq, water_order, color = "sample_type")
#plot(observed_phyloseq, water_order, color = "treatment")
# Based on the plot, Freshwater (creek) appears to have the highest observed
# richness. However, what might be an issue with just looking at observed richness
# across samples?

# Sequencing_depth!

# What might we expect to see if we plotted observed richness of each SampleType by
# their sequencing depth?

# We can use ggplot and phyloseq to plot our sample types by observed richness and
# sequencing depth.
#water_order %>% sample_data %>% .$S.obs

#data.frame("observed_richness" = (observed_phyloseq %>% summary)$estimate,
#           "depth" = phyloseq::sample_sums(water_order), # Easter egg! Phyloseq's function to get depth
#           "Sample type" = water_order %>% sample_data %>% get_variable("sample_type"),
#           "Treatment" = water_order %>% sample_data %>% get_variable("treatment")) %>%
#  ggplot(aes(x = depth, y = observed_richness, color = `Sample.type`, shape = `Treatment`)) +
#  geom_point()

# So what do we see? We see that some of our Freshwater (creek) samples which have
# the highest observed richness also have the highest sequencing depth and that our
# Sediment (estuary) samples which have the lowest observed richness also have the
# lowest sequencing depth. This isn't surprising! So what can we do about this?
# (hint: NOT rarefaction)

# ---------------------- answer: breakaway! ---------------------------

# Instead of randomly subsampling our data so that our samples have some uniform
# lower sequencing depth (does this seem like a good idea? seems like we might be
# losing some valuable information by doing this) why don't we use the information
# that we have to estimate the number of missing species with a species richness
# estimator. (Statistics is so cool)

# We can do that using `breakaway()`.

#ba <- breakaway(water_order)
#p1 <- plot(ba, water_order, color = "sample_type")
#p2 <- plot(observed_phyloseq, water_order, color = "sample_type") 
#grid.arrange(p2, p1, nrow = 2)

#ba_species <- breakaway(water)

#p1 <- plot(ba, water, color = "sample_type")
#p2 <- plot(sample_richness(water), water, color = "sample_type") 
#grid.arrange(p2, p1, nrow = 2)


#frequencytablelist <- build_frequency_count_tables(otu_table(water))
#frequencytablelist[1]
#frequencytablelist[12]
# Here we're comparing our results with the previous plot we generated using
# `phyloseq`'s observed richness estimator. What looks different? The most obvious
# is that there are error bars around our estimates of observed order-level richness!
# Makes sense to have them, right?

# `breakaway` goes through many different models to see which one is best for modeling
# our data. If you're interested in seeing what `breakaway` chooses to model the
# richness of a particular sample (in this case let's choose TRRsed3) you can look
# at that by selecting the sample of interest and `model` output!

```

## Calculations used for main text

## Figure 1 numbers {.tabset}

**These tables were generated to note average med(IQR) of qPCR results**

### Figure 1 numbers (Host DNA) - treated vs untreated


```{r}

sample_data %>% subset(., .$S.obs != 0) %>% group_by(sample_type, treated) %>% summarise(med = median(DNA_host_nondil + DNA_bac_nondil), lo = quantile(DNA_host_ng_uL + DNA_host_ng_uL, 0.25), high = quantile(DNA_host_nondil + DNA_bac_nondil, 0.75))

sample_data %>% subset(., .$S.obs != 0) %>%
        group_by(sample_type, treated) %>% summarise(med = median(DNA_host_ng_uL), lo = quantile(DNA_host_ng_uL, 0.25), high = quantile(DNA_host_ng_uL, 0.75))

```

### Figure 1 numbers (Host prooportion) - treated vs untreated


```{r}
sample_data %>% subset(., .$S.obs != 0) %>% group_by(sample_type, treated) %>% summarise(med = median(host_proportion), lo = quantile(host_proportion, 0.25), high = quantile(host_proportion, 0.75))
```


### Figure 1 numbers (bacterial DNA) - treated vs untreated

```{r}
sample_data %>% subset(., .$S.obs != 0) %>% group_by(sample_type, treated) %>% summarise(med = median(DNA_bac_ng_uL), lo = quantile(DNA_bac_ng_uL, 0.25), high = quantile(DNA_bac_ng_uL, 0.75))

```

### {-}

# **3.3. Effects of treatments on taxonomic composition**



# *vi. Did diversity matrices change? 



## Taxa change {.tabset}

### Alpha and beta diversity (Figure 3)


        1.	Figure of alpha and beta diversity changes (Figure 3)
                a.	Alpha - you currently have a box plot stratified by sample type, another approach is to use a forest plot using predictions from your model
                b.	Beta - can consider a forest plot for this one as well instead of a boxplot. For a forest plot you can annotate the x-axis to indicate which direction is better ie towards 0 you can annotate "less bias" and towards higher number you can annotate "more bias"


**Figure 3. ** (A) Species richness by sample type and treatment. (B) Horn-Morisita distances (mean and 95 % confidence interval) between untreated and treated within the sample subject by sample type and treatment.


```{r, warning=FALSE, message=FALSE}
f3a <-        ggplot(subset(sample_data(phyloseq$phyloseq_count) %>% data.frame, sample_data(phyloseq$phyloseq_count)$sample_type %in% c("Sputum", "Nasal", "BAL", "Mock")), aes(y = S.obs)) +
        geom_boxplot(aes(fill = treatment), lwd = 0.2) +
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "Host zero", "Molysis", "QIAaamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        ylab("Species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "serif") + 
        labs(tag = "A") +
        theme(plot.tag = element_text(size = 15),  axis.text.x = element_blank(), axis.ticks.x = element_blank()) +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(fill = guide_legend(nrow = 1))


#Making subset of non-zero samples without neg

phyloseq_rel_nz <- phyloseq$phyloseq_rel %>%
        subset_samples(S.obs != 0 & sample_type %in% c("Mock", "BAL", "Nasal", "Sputum"))

#distances of betadiversity - boxplots
horn_dist_long <- distance(phyloseq_rel_nz, method="horn") %>% as.matrix() %>% melt_dist() #making long data of distance matrices

#Adding sample type and treatment name. 
#this can be also done by merging metadata into the `horn_dist_long`
names <- data.frame(str_split_fixed(horn_dist_long$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(horn_dist_long$iso2, "_", 3))
horn_dist_long$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
horn_dist_long$method_1 <- ifelse(grepl("lyPMA", horn_dist_long$iso1),"lypma", 
                                         ifelse(grepl("ben", horn_dist_long$iso1),"benzonase", 
                                                ifelse(grepl("host", horn_dist_long$iso1),"host_zero", 
                                                       ifelse(grepl("qia", horn_dist_long$iso1),"qiaamp", 
                                                              ifelse(grepl("moly", horn_dist_long$iso1),"molysis", 
                                                                     "control")))))


#Adding data for iso 2 also should be done
horn_dist_long$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")
horn_dist_long$method_2 <-ifelse(grepl("lyPMA", horn_dist_long$iso2),"lypma", 
                                        ifelse(grepl("ben", horn_dist_long$iso2),"benzonase", 
                                               ifelse(grepl("host", horn_dist_long$iso2),"host_zero", 
                                                      ifelse(grepl("qia", horn_dist_long$iso2),"qiaamp", 
                                                             ifelse(grepl("moly", horn_dist_long$iso2),"molysis", 
                                                                    "control")))))


#subsetting distances of my interest
horn_dist_long$sample_id_1 <- ifelse(grepl("pos", horn_dist_long$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_1, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_1))
horn_dist_long$sample_id_2 <- ifelse(grepl("pos", horn_dist_long$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_2, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_2))


path_horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long, horn_dist_long$sample_id_1 == horn_dist_long$sample_id_2) # data within samples

path_horn_dist_long_within_sampleid_from_control <- subset(path_horn_dist_long_within_sampleid_from_control,
                                                           path_horn_dist_long_within_sampleid_from_control$method_1 != path_horn_dist_long_within_sampleid_from_control$method_2) # remove irrelevant association

path_horn_dist_long_within_sampleid_from_control <- subset(path_horn_dist_long_within_sampleid_from_control, (path_horn_dist_long_within_sampleid_from_control$method_1 == "control") + (path_horn_dist_long_within_sampleid_from_control$method_2 == "control") != 0)


path_horn_dist_long_within_sampleid_from_control$treatment <- path_horn_dist_long_within_sampleid_from_control$method_1

path_horn_dist_long_within_sampleid_from_control$treatment <- ifelse(path_horn_dist_long_within_sampleid_from_control$treatment == "control", path_horn_dist_long_within_sampleid_from_control$method_2, path_horn_dist_long_within_sampleid_from_control$treatment) 


#Setting key method
path_horn_dist_long_within_sampleid_from_control$sample_type <- ifelse(grepl("NS", path_horn_dist_long_within_sampleid_from_control$iso1), "Nasal",
                                                                  ifelse(grepl("CFB", path_horn_dist_long_within_sampleid_from_control$iso1), "Sputum",
                                                                         ifelse(grepl("BAL", path_horn_dist_long_within_sampleid_from_control$iso1), "BAL",
                                                                                ifelse(grepl("pos|POS", path_horn_dist_long_within_sampleid_from_control$iso1, ignore.case = T), "Mock",
                                                                                       ifelse(grepl("neg|N_EXT", path_horn_dist_long_within_sampleid_from_control$iso1), "Neg.",NA)))))

#Making a column for baseline (controls, from where?)
path_horn_dist_long_within_sampleid_from_control <- path_horn_dist_long_within_sampleid_from_control %>% 
        mutate(dist_from = case_when(method_1 == "control" ~ iso1,
                                     method_2 == "control" ~ iso2))

dummy <- data.frame(iso1 = path_horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           iso2 = path_horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           dist = 0,
           treatment = "Untreated",
           method_1 = "control",
           method_2 = "control"
           )
names <- data.frame(str_split_fixed(dummy$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(dummy$iso2, "_", 3))
dummy$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
#Adding data for iso 2 also should be done
dummy$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")


#subsetting distances of my interest
dummy$sample_id_1 <- ifelse(grepl("pos", dummy$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_1, ignore.case = T),"Neg.",
                                        dummy$sample_id_1))
dummy$sample_id_2 <- ifelse(grepl("pos", dummy$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_2, ignore.case = T),"Neg.",
                                        dummy$sample_id_2))
dummy$sample_type <- ifelse(grepl("NS", dummy$iso1), "Nasal",
                            ifelse(grepl("CFB", dummy$iso1), "Sputum",
                                   ifelse(grepl("BAL", dummy$iso1), "BAL",
                                          ifelse(grepl("pos|POS", dummy$iso1, ignore.case = T), "Mock",
                                                 ifelse(grepl("neg|N_EXT", dummy$iso1), "Neg.",NA)))))
dummy <- subset(dummy, !is.na(dummy$sample_type))
path_horn_dist_long_within_sampleid_from_control <- bind_rows(path_horn_dist_long_within_sampleid_from_control, dummy)



#Making figure of beta diversity distances
f3b2 <- path_horn_dist_long_within_sampleid_from_control %>% 
        mutate(across(sample_type, factor, levels=c("Mock", "BAL", "Nasal","Sputum"))) %>%
        subset(., .$sample_type != "Neg.") %>% 
        group_by(sample_type, treatment) %>%
        summarise(mean = mean(dist, na.rm = TRUE),
            sd = sd(dist, na.rm = TRUE),
            n = n()) %>%
  mutate(se = sd / sqrt(n),
         lower.ci = mean - qt(1 - (0.05 / 2), n - 1) * se,
         upper.ci = mean + qt(1 - (0.05 / 2), n - 1) * se,
         treatment = factor(treatment, levels = c("Untreated", "lypma", "benzonase", "host_zero", "molysis", "qiaamp"),
                            labels = c("Untreated", "lyPMA", "Benzonase", "Host zero", "Molysis", "QIAaamp"))) %>%#,
         #text = paste(sprintf("%.2f", round(mean, digits = 2)), " [", sprintf("%.2f", round(lower.ci, digits = 2)), ", ", sprintf("%.2f", round(upper.ci, digits = 2)), "]", sep = "")) %>%
        ggplot(aes(x = mean, y = treatment, col = treatment)) +
        geom_point(aes(x=mean), shape=15, size=3) +
        geom_linerange(aes(xmin=lower.ci, xmax=upper.ci)) +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated", "lyPMA", "Benzonase", "Host zero", "Molysis", "QIAaamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Distance from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "serif") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(), axis.ticks.y = element_blank(), legend.position = "none") +
        labs(tag = "B") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "serif") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "serif") +
        scale_x_continuous(breaks = c(0, 1), labels = c("Lower bias", "Higher bias"))

fig3 <- ggarrange(f3a, f3b2, ncol = 1, common.legend = T, align = "hv") +
        guides(fill = guide_legend(nrow = 1))
fig3

png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/Figure3.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 220, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

fig3
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()

```              

### {-}


# *vii.	& viii. Do these host depletion methods introduce bias in the sequenced community? *



+ Does bias differ by sample type?

        2.	Alpha diversity-
                a.	 Run below models to justify stratified analyses (just report the significant method*sample_type interaction term)
                        i.	species_richness ~ method + sample_type + method * sample_type + log10(reads) + (1|subjid)
                b.	Stratified Species richness - Does host depletion method introduce bias?
                        i.	LMER(species_richness ~ method + log10(reads) + (1|subjid))
                                1.	By sample type, for mock, bal, nasal, sputum
                c.	Inv Simp 
                        i.	Lmer(InvSimp~ sample type + treatment + sample type * treatment + (1|subject_id) (Table S12)

## Alpha diversity changes {.tabset}

### Calculation for log centered Final reads

**Distribution of centralized final reads**

```{r}
sample_data <- sample_data(phyloseq$phyloseq_count)
sample_data$log_centered_final_reads <- log(sample_data$Final_reads + 1) - median(log((subset(sample_data, sample_data$sample_type %in% c("BAL") & sample_data$treatment %in% c("Untreated")) %>% .$Final_reads) + 1))

sample_data$bal_log_centered_final_reads <- log(sample_data$Final_reads + 1) - median(log((subset(sample_data, sample_data$sample_type %in% c("BAL") & sample_data$treatment %in% c("Untreated"))%>% .$Final_reads) + 1))

sample_data$ns_log_centered_final_reads <- log(sample_data$Final_reads + 1) - median(log((subset(sample_data, sample_data$sample_type %in% c("Nasal") & sample_data$treatment %in% c("Untreated"))%>% .$Final_reads) + 1))

sample_data$spt_log_centered_final_reads <- log(sample_data$Final_reads + 1) - median(log((subset(sample_data, sample_data$sample_type %in% c("Sputum") & sample_data$treatment %in% c("Untreated"))%>% .$Final_reads) + 1))


subset(sample_data, sample_data$sample_type %in% c("BAL", "Nasal", "Sputum")) %>% .$log_centered_final_reads %>% hist (main = "Histogram of centered log10 final reads of BAL, Nasal, Sputum")

subset(sample_data, sample_data$sample_type %in% c("BAL")) %>% .$log_centered_final_reads %>% hist (main = "Histogram of centered log10 final reads of BAL")

subset(sample_data, sample_data$sample_type %in% c("Nasal")) %>% .$log_centered_final_reads %>% hist (main = "Histogram of centered log10 final reads of Nasal")

subset(sample_data, sample_data$sample_type %in% c("Sputum")) %>% .$log_centered_final_reads %>% hist (main = "Histogram of centered log10 final reads of Sputum")

```

### Species richness (all sample)

Effect size, standard error (SE) and p-value of a statistical test for species richness with an interaction term using linear mixed effect model (Species richness ~ sample_type * treatment + log10 (Final_reads) + (1|subject_id) ).

**Interaction term was highly significant (p = 2.2e-16)**

```{r}
lmer(S.obs ~ treatment * sample_type + log10(Final_reads) + (1|subject_id),
                    data = sample_data %>% 
                            data.frame %>% 
                            subset(., .$S.obs != 0)) %>% 
        anova()
        
```

### Species richness (stratified) without depth (Table S6)

**Table S6**. Effect size, standard error (SE) and p-value of a statistical test for species richness using linear mixed effect model stratified by sample type (Species richness ~ treatment + (1|subject_id) ).


```{r}

sr_lmer_bal <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("BAL") & S.obs != 0)) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = 1), 
                                " (", 
                                round(Estimate - 1.96 * SE, 1) %>% format(nsmall = 1),
                                ", ",
                                round(Estimate + 1.96 * SE, 1) %>% format(nsmall = 1),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " "))
        
sr_lmer_ns <- lmer(S.obs ~ treatment + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Nasal"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("ns_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = ), 
                                " (", 
                                round(Estimate - 1.96 * SE, 1) %>% format(nsmall = ),
                                ", ",
                                round(Estimate + 1.96 * SE, 1) %>% format(nsmall = ),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " "))


sr_lmer_spt <- lmer(S.obs ~ treatment + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Sputum"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("spt_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
                rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = 1), 
                                " (", 
                                round(Estimate - 1.96 * SE, 1) %>% format(nsmall = 1),
                                ", ",
                                round(Estimate + 1.96 * SE, 1) %>% format(nsmall = 1),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " "))
        

tables6  <- cbind(sr_lmer_bal, sr_lmer_ns, sr_lmer_spt) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "serif")

tables6

save_kable(tables6, file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tables6.html", self_contained = T)

```

### Species richness - stratified - with depth (Table S7)

**Table S7**. Sequencing depth adjusted effect size, standard error (SE) and p-value of a statistical test for species richness using linear mixed effect model stratified by sample type (Species richness ~ treatment + log10 (Final_reads) + (1|subject_id) ).

```{r}

sr_lmer_bal <- lmer(S.obs ~ treatment + bal_log_centered_final_reads + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("BAL") & S.obs != 0)) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = 1), 
                                " (", 
                                round(Estimate - 1.96 * SE, 1) %>% format(nsmall = 1),
                                ", ",
                                round(Estimate + 1.96 * SE, 1) %>% format(nsmall = 1),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " "))
        
sr_lmer_ns <- lmer(S.obs ~ treatment + ns_log_centered_final_reads + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Nasal"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("ns_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = ), 
                                " (", 
                                round(Estimate - 1.96 * SE, 1) %>% format(nsmall = ),
                                ", ",
                                round(Estimate + 1.96 * SE, 1) %>% format(nsmall = ),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " "))


sr_lmer_spt <- lmer(S.obs ~ treatment + spt_log_centered_final_reads + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Sputum"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("spt_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
                rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = 1), 
                                " (", 
                                round(Estimate - 1.96 * SE, 1) %>% format(nsmall = 1),
                                ", ",
                                round(Estimate + 1.96 * SE, 1) %>% format(nsmall = 1),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " "))
        

tableS7  <- cbind(sr_lmer_bal, sr_lmer_ns, sr_lmer_spt) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "serif")

tableS7

save_kable(tableS7, file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS7.html", self_contained = T)




```

### {-}


## Beta diversity distances - Morisita Horn index {.tabset}



### PERMANOVA - all samples (Table S8)

**Table S8. ** Degree of freedom, effect size (residual, R^2) and p-value of permutational ANOVA for Morisita-Horn distiances with an interaction term and strata term (BC-distance ~ sample_type * treatment + log10 (Final_reads), strata = subject_id).


```{r}

horn_perm_all <- vegan::adonis2(distance(phyloseq_rel_nz, method="horn") ~ sample_type * treatment + subject_id + log10(Final_reads),
                                  data = phyloseq_rel_nz %>% sample_data %>% data.frame(check.names = F),
                                  permutations = 10000)

horn_perm_ns <- vegan::adonis2(distance(subset_samples(phyloseq_rel_nz, sample_type == "Nasal"), method="horn") ~ lypma + benzonase + host_zero + molysis + qiaamp + log10(Final_reads),
                               data = subset_samples(phyloseq_rel_nz, sample_type == "Nasal") %>%
                                       sample_data %>% data.frame(check.names = F),
                               strata = subset_samples(phyloseq_rel_nz, sample_type == "Nasal") %>% 
                                       sample_data %>% data.frame(check.names = F) %>% .$subject_id, permutations = 10000)

horn_perm_bal  <- vegan::adonis2(distance(subset_samples(phyloseq_rel_nz, sample_type == "BAL"), method="horn") ~  lypma + benzonase + host_zero + molysis + qiaamp + log10(Final_reads),
                                 data = subset_samples(phyloseq_rel_nz, sample_type == "BAL") %>% sample_data %>% data.frame(check.names = F),
                                 strata = subset_samples(phyloseq_rel_nz, sample_type == "BAL") %>%
                                         sample_data %>% data.frame(check.names = F) %>% .$subject_id,
                                  permutations = 10000)

horn_perm_spt <- vegan::adonis2(distance(subset_samples(phyloseq_rel_nz, sample_type == "Sputum"), method="horn") ~ lypma + benzonase + host_zero + molysis + qiaamp + log10(Final_reads),
                                data = subset_samples(phyloseq_rel_nz, sample_type == "Sputum") %>% sample_data %>% data.frame(check.names = F),
                                strata = subset_samples(phyloseq_rel_nz, sample_type == "Sputum")
                                %>% sample_data %>% data.frame(check.names = F) %>% .$subject_id,
                                  permutations = 10000)



tableS8 <- horn_perm_all %>% data.frame(check.names = F) %>% rownames_to_column("row.names") %>% 
        mutate(row.names = case_when(row.names == "sample_type" ~ 'Sample type',
                                     row.names == "treatment" ~ 'Treatment',
                                     row.names == "subject_id" ~ 'Subject',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "sample_type:treatment" ~ 'Sample type * Treatment',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
        mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        mutate(`<i>p</i>-value` = format(`<i>p</i>-value`, nsmall = 3)) %>%
        select(c("Degree of freedom", "R<sup>2</sup>", "<i>p</i>-value", " ")) %>% 
        kbl(format = "html", escape = 0) %>%
        kable_styling(full_width = 0, html_font = "serif")

tableS8


save_kable(tableS8, file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS8.html", self_contained = T)


```

### PERMANOVA - Stratified (Table 3)

**Table 3**. Degree of freedom, effect size (residual, R^2) and p-value of permutational ANOVA for Morisita-Horn distiances  for species richness stratified by sample type (MH-distance ~ lyPMA + Benzoase + Host zero + Molysis + QIAamp + log10 (Final_reads), strata = subject_id).


```{r}

a <- horn_perm_bal %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'Host zero',
                                     row.names == "molysis" ~ 'Molysis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 

b <- horn_perm_ns %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'Host zero',
                                     row.names == "molysis" ~ 'Molysis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 

c <- horn_perm_spt %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'Host zero',
                                     row.names == "molysis" ~ 'Molysis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 


table2 <- cbind(a, b, c) %>% 
        kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "serif")

table2

save_kable(table2, file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/table2.html", self_contained = T)


```

### LM on M-H distance (Table S9)

**Table S9**. Effect size, standard error (SE) and p-value of a statistical test for Horn-Morisita distance from untreated to each treated within subject (BC-distance wihtin subject by treatment~ sample type + treatment + sampetype * treatment + log10 (Final_reads) + (1|subject_id) ).

```{r}

path_horn_dist_long_within_sampleid_from_control$treatment <- factor(path_horn_dist_long_within_sampleid_from_control$treatment,
                                                                     levels = c("Untreated",
                                                                                "lypma",
                                                                                "benzonase",
                                                                                "host_zero",
                                                                                "molysis",
                                                                                "qiaamp"))
path_horn_dist_long_within_sampleid_from_control$sample_type <- factor(path_horn_dist_long_within_sampleid_from_control$sample_type,
                                                                     levels = c("BAL",
                                                                                "Nasal",
                                                                                "Sputum"))

a <- lm(dist ~ treatment, data = path_horn_dist_long_within_sampleid_from_control %>% subset(., .$sample_type == "BAL")) %>% summary %>% .$ coefficients %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = gsub("host_zero", "Host zero", row.names)) %>% 
        mutate(row.names = gsub("benzonase", "Benzonase", row.names)) %>% 
        mutate(row.names = gsub("lypma", "lyPMA", row.names)) %>% 
        mutate(row.names = gsub("molysis", "Molysis", row.names)) %>% 
        mutate(row.names = gsub("qiaamp", "QIAamp", row.names)) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        mutate(`row.names` = gsub("treatment|sample_type", "", `row.names`)) %>% 
        column_to_rownames("row.names") %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                                            abs(`Pr(>|t|)`) < 0.01 ~ "**",
                                            abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 2) %>% format(nsmall = 2), 
                                " (", 
                                round(Estimate - 1.96 * SE, 2) %>% format(nsmall = 2),
                                ", ",
                                round(Estimate + 1.96 * SE, 2) %>% format(nsmall = 2),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) 

b <- lm(dist ~ treatment, data = path_horn_dist_long_within_sampleid_from_control %>% subset(., .$sample_type == "Nasal")) %>% summary %>% .$ coefficients %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = gsub("host_zero", "Host zero", row.names)) %>% 
        mutate(row.names = gsub("benzonase", "Benzonase", row.names)) %>% 
        mutate(row.names = gsub("lypma", "lyPMA", row.names)) %>% 
        mutate(row.names = gsub("molysis", "Molysis", row.names)) %>% 
        mutate(row.names = gsub("qiaamp", "QIAamp", row.names)) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        mutate(`row.names` = gsub("treatment|sample_type", "", `row.names`)) %>% 
        column_to_rownames("row.names") %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                                            abs(`Pr(>|t|)`) < 0.01 ~ "**",
                                            abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 2) %>% format(nsmall = 2), 
                                " (", 
                                round(Estimate - 1.96 * SE, 2) %>% format(nsmall = 2),
                                ", ",
                                round(Estimate + 1.96 * SE, 2) %>% format(nsmall = 2),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) 

c <- lm(dist ~ treatment, data = path_horn_dist_long_within_sampleid_from_control %>% subset(., .$sample_type == "Sputum")) %>% summary %>% .$ coefficients %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = gsub("host_zero", "Host zero", row.names)) %>% 
        mutate(row.names = gsub("benzonase", "Benzonase", row.names)) %>% 
        mutate(row.names = gsub("lypma", "lyPMA", row.names)) %>% 
        mutate(row.names = gsub("molysis", "Molysis", row.names)) %>% 
        mutate(row.names = gsub("qiaamp", "QIAamp", row.names)) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        mutate(`row.names` = gsub("treatment|sample_type", "", `row.names`)) %>% 
        column_to_rownames("row.names") %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                                            abs(`Pr(>|t|)`) < 0.01 ~ "**",
                                            abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 2) %>% format(nsmall = 2), 
                                " (", 
                                round(Estimate - 1.96 * SE, 2) %>% format(nsmall = 2),
                                ", ",
                                round(Estimate + 1.96 * SE, 2) %>% format(nsmall = 2),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " "))

tableS9 <- cbind(a, b, c) %>% 
        kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "serif")


save_kable(tableS9, file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS9.html", self_contained = T)

tableS9


```

### Permanova figure (Figure S6)

**Fig. S6. Stratified beta diversity PCA plot based on M-H index**

```{r}
figS6 <- ordinate(subset_samples(phyloseq_rel_nz, sample_type != "Neg." & sample_type != "Mock"), method = "PCoA", distance = "horn") %>%
        plot_ordination(phyloseq_rel_nz, ., col = "treatment") +
        #scale_color_viridis(discrete = 6, name = "Treatment", labels = c("Mock theoretical", "Control","lyPMA", "Benzonase", "Host zero", "Molysis", "QIAaamp")) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"),
                           name = "Treatment",
                           breaks = c("Untreated","lyPMA", "Benzonase", "Host zero", "Molysis", "QIAamp"),
                           labels = c("Untreated", "lyPMA", "Benzonase", "Host zero", "Molysis", "QIAaamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        #scale_shape(name = "Sample type", labels = c("Mock theoretical", "Mock")) +
        geom_point(size = 3) +
        theme_classic (base_size = 12, base_family = "serif") +
        facet_wrap(~sample_type, scales = "free") +
        theme(plot.tag = element_text(size = 15), legend.position = "top")# +
        #stat_ellipse(type = "norm") +
        #stat_ellipse(type = "t")

figS6

png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS6.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 90, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue
figS6 
dev.off()


```

### {-}



# *iv. Mediation analysis*






## mediation analysis {.tabset}

### Mediation analysis (treatment - binary)

outcome = S.obs
exposure = treatment (binary)
mediator = Final_reads
mediator-outcome confounders = sample_type 
exposure-mediator confounders = NA
outcome model = Mixed effects linear regression 
mediator model = Mixed effects linear regression

Mediation analysis will be conducted only for **samples**.
Sample type was considered as a mediator-outcome confounder, since it affected both S.obs and Final reads.
**Here, the model cannot consider there was sample-type X treatment effect**
Therefore none of the association was significant.

```{r}

# only mediator-outcome confounders
detach_package <- function(pkg, character.only = FALSE)
{
  if(!character.only)
  {
    pkg <- deparse(substitute(pkg))
  }
  search_item <- paste("package", pkg, sep = ":")
  while(search_item %in% search())
  {
    detach(search_item, unload = TRUE, character.only = TRUE)
  }
}

detach_package(lmerTest)

## all treatment groups

sample_data <- sample_data %>% data.frame()
sample_data_respiratory <- subset(sample_data, sample_data$sample_type == "Sputum" | sample_data$sample_type == "BAL" | sample_data$sample_type == "Nasal")


med.fit <- lmer(log10(Final_reads) ~ treatment + (1|subject_id),
                data = sample_data_respiratory) 

out.fit <- lmer(S.obs ~ treatment * log10(Final_reads) + sample_type + (1|subject_id), 
                data = sample_data_respiratory)

med.out <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "treatment",
                     mediator = "log10(Final_reads)",
                     sims = 1000)

out.sum <- summary(med.out)

final.out.all <- data.frame(indirect.est=out.sum$d.avg, 
                          indirect.P.val=out.sum$d.avg.p,
                          direct.est=out.sum$z.avg, 
                          direct.P.val=out.sum$z.avg.p, 
                          total.est=out.sum$tau.coef, 
                          total.P.val=out.sum$tau.p,
                          pm.est=out.sum$n.avg, 
                          pm.P.val=out.sum$n.avg.p) %>% 
    mutate_all(~round(., 3))

final.out.all

```


### Mediation analysis (treatment-stratified)

outcome = S.obs
exposure = treatment (stratified)
mediator = Final_reads
mediator-outcome confounders = sample_type
exposure-mediator confounders = NA
outcome model = Mixed effects linear regression 
mediator model = Mixed effects linear regression

**Mediation analysis was conducted stratified treatment.**


```{r}

## lypma


med.fit <- lmer(log10(Final_reads) ~ lypma + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$lypma == 1))

out.fit <- lmer(S.obs ~ lypma * log10(Final_reads) + sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$lypma == 1))

med.out <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "lypma",
                     mediator = "log10(Final_reads)",
                     sims = 1000)

out.sum <- summary(med.out)

final.out.lypma <- data.frame(indirect.est=out.sum$d.avg, 
                          indirect.P.val=out.sum$d.avg.p,
                          direct.est=out.sum$z.avg, 
                          direct.P.val=out.sum$z.avg.p, 
                          total.est=out.sum$tau.coef, 
                          total.P.val=out.sum$tau.p,
                          pm.est=out.sum$n.avg, 
                          pm.P.val=out.sum$n.avg.p) %>% 
    mutate_all(~round(., 3))

## benzonase


med.fit <- lmer(log10(Final_reads) ~ benzonase + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$benzonase == 1))

out.fit <- lmer(S.obs ~ benzonase * log10(Final_reads) +  sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$benzonase == 1))

med.out <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "benzonase",
                     mediator = "log10(Final_reads)",
                     sims = 1000)

out.sum <- summary(med.out)

final.out.benzonase <- data.frame(indirect.est=out.sum$d.avg, 
                          indirect.P.val=out.sum$d.avg.p,
                          direct.est=out.sum$z.avg, 
                          direct.P.val=out.sum$z.avg.p, 
                          total.est=out.sum$tau.coef, 
                          total.P.val=out.sum$tau.p,
                          pm.est=out.sum$n.avg, 
                          pm.P.val=out.sum$n.avg.p) %>% 
    mutate_all(~round(., 3))

## host zero


med.fit <- lmer(log10(Final_reads) ~ host_zero + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$host_zero == 1))

out.fit <- lmer(S.obs ~ host_zero * log10(Final_reads) +  sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$host_zero == 1))

med.out <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "host_zero",
                     mediator = "log10(Final_reads)",
                     sims = 1000)

out.sum <- summary(med.out)

final.out.host_zero <- data.frame(indirect.est=out.sum$d.avg, 
                          indirect.P.val=out.sum$d.avg.p,
                          direct.est=out.sum$z.avg, 
                          direct.P.val=out.sum$z.avg.p, 
                          total.est=out.sum$tau.coef, 
                          total.P.val=out.sum$tau.p,
                          pm.est=out.sum$n.avg, 
                          pm.P.val=out.sum$n.avg.p) %>% 
    mutate_all(~round(., 3))


## molysis


med.fit <- lmer(log10(Final_reads) ~ molysis + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$molysis == 1))

out.fit <- lmer(S.obs ~ molysis * log10(Final_reads) +  sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$molysis == 1))

med.out <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "molysis",
                     mediator = "log10(Final_reads)",
                     sims = 1000)

out.sum <- summary(med.out)

final.out.molysis <- data.frame(indirect.est=out.sum$d.avg, 
                          indirect.P.val=out.sum$d.avg.p,
                          direct.est=out.sum$z.avg, 
                          direct.P.val=out.sum$z.avg.p, 
                          total.est=out.sum$tau.coef, 
                          total.P.val=out.sum$tau.p,
                          pm.est=out.sum$n.avg, 
                          pm.P.val=out.sum$n.avg.p) %>% 
    mutate_all(~round(., 3))

## qiaamp


med.fit <- lmer(log10(Final_reads) ~ qiaamp + (1|subject_id),
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$qiaamp == 1))

out.fit <- lmer(S.obs ~ qiaamp * log10(Final_reads) +  sample_type + (1|subject_id), 
                data = subset(sample_data_respiratory, sample_data_respiratory$control == 1 | sample_data_respiratory$qiaamp == 1))

med.out <- mediate(model.m = med.fit,
                   model.y = out.fit,
                     treat = "qiaamp",
                     mediator = "log10(Final_reads)",
                     sims = 1000)

out.sum <- summary(med.out)

final.out.qiaamp <- data.frame(indirect.est=out.sum$d.avg, 
                          indirect.P.val=out.sum$d.avg.p,
                          direct.est=out.sum$z.avg, 
                          direct.P.val=out.sum$z.avg.p, 
                          total.est=out.sum$tau.coef, 
                          total.P.val=out.sum$tau.p,
                          pm.est=out.sum$n.avg, 
                          pm.P.val=out.sum$n.avg.p) %>% 
    mutate_all(~round(., 3))



rbind(final.out.lypma,
      final.out.benzonase,
      final.out.host_zero, 
      final.out.molysis,
      final.out.qiaamp) %>%
        mutate(treatment = c("lyPMA", "Benzonase", "Host zero", "Molysis", "QIAamp"), .before = indirect.est)
```




# *x.	How were changes of individual taxa?*

*x.	How were changes of individual taxa? Differential abundance analysis *
       
        1.	Volcano plot with Mock, BAL, Nasal and Sputum (Figure S5)
                a.	Maaslin (feature ~ log10(Final reads) + lyPMA + Benzonase + Host zero + Molysis + QIAamp, RE = subject_id)
                        i.	make sure that you are doing this analysis accounting for the compositional nature of this data
                b.	Or Maaslin ( feature ~ log10(Final reads) + sample type + treatment + sample type * treatment, RE = subject_id) 
        2.	Balloon plots Mock, BAL, Nasal and Sputum. (Figure)
                        a.	Add q-val, mean relative abundance, gram strain, phylogenetic information

## DA taxa {.tabset}

### Factors affecting DA taxa (main text)

```{r, warning=FALSE, message=FALSE}

filt_maaslin_all <- read.csv("data/filt_maaslin_all.csv")
filt_maaslin_interaction <- read.csv("data/filt_maaslin_interaction.csv")
filt_fit_data_bal <- read.csv("data/filt_fit_data_bal.csv")
filt_fit_data_spt <- read.csv("data/filt_fit_data_spt.csv")
filt_fit_data_ns <- read.csv("data/filt_fit_data_ns.csv")
filt_fit_data_pos <- read.csv("data/filt_fit_data_pos.csv")

cat("Factors affecting DA taxa (q<0.1)")

filt_maaslin_all %>% subset(., .$qval < 0.1 ) %>% .$metadata %>% table

```

### Volcano plot (Fugure S7)

**Figure S7**. Volcano plot of sequencing depth adjusted differential abundance of taxa by each treatment 


```{r warning=FALSE, message=FALSE, "MaAslin Function continued2"}

#Making significance table for figure
        # Define a function to make species names italicized
# Make a significance table for each figure (top 20 taxa)
species_italic <- function(data) {
  names <- gsub("_", " ", rownames(data))
  names <- gsub("[]]|[[]", "", names)
  names <- gsub(" sp", " sp.", names)
  names <- gsub(" sp.", "* sp.", names)
  names <- gsub(" group", "* group.", names)
  names <- ifelse(grepl("[*]", names), paste("*", names, sep = ""), paste("*", names, "*", sep = ""))
  rownames(data) <- names
  data
}

make_sig_table <- function(data) {
  sig_data <- spread(data[order(data$qval), c("feature", "metadata", "qval")], metadata, qval)
  sig_data$min <- apply(sig_data, 1, FUN = min)
  sig_data <- sig_data[order(sig_data$min),] %>% select("feature", "lypma", "benzonase", "host_zero", "molysis", "qiaamp") %>% .[1:20,]
  sig_data[["feature"]] <- ifelse(sig_data[["feature"]] == "X.Collinsella._massiliensis", "[Collinsella]_massiliensis", sig_data[["feature"]])
  sig_data_italic <- sig_data %>% rownames_to_column(var = "-") %>%
          column_to_rownames(var = "feature") %>% species_italic %>% select(-c("-")) %>%
          rename(lyPMA = lypma,  Benzonase = benzonase, `Host zero` = host_zero, Molysis = molysis, QIAamp = qiaamp)
  sig_data_sig <- ifelse(sig_data_italic < 0.1, "*", NA) %>% data.frame(check.names = F)
  return(list(data = sig_data, data_italic = sig_data_italic, data_sig = sig_data_sig))
}
filt_fit_data_pos <- make_sig_table(filt_fit_data_pos)
filt_fit_data_bal <- make_sig_table(filt_fit_data_bal)
filt_fit_data_ns <- make_sig_table(filt_fit_data_ns)
filt_fit_data_spt <- make_sig_table(filt_fit_data_spt)

filt_pos_sig <- subset_taxa(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Mock"),
                                       taxa_names(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Mock")) %in% filt_fit_data_pos$data$feature)

filt_fit_data_pos$rel <- cbind(filt_pos_sig %>% otu_table %>% t, filt_pos_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>% 
        .[row.names(filt_fit_data_pos$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")

filt_spt_sig <- subset_taxa(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Sputum"), taxa_names(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Sputum")) %in% filt_fit_data_spt$data$feature)

filt_fit_data_spt$rel <- cbind(filt_spt_sig %>% otu_table %>% t, filt_spt_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>% 
        .[row.names(filt_fit_data_spt$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")

filt_ns_sig <- subset_taxa(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Nasal"),
                                       taxa_names(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Nasal")) %in% filt_fit_data_ns$data$feature)

filt_fit_data_ns$rel <- cbind(filt_ns_sig %>% otu_table %>% t, filt_ns_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>% 
        .[row.names(filt_fit_data_ns$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")

filt_bal_sig <- subset_taxa(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "BAL"),
                                       taxa_names(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "BAL")) %in% filt_fit_data_bal$data$feature)

filt_fit_data_bal$rel <- cbind(filt_bal_sig %>% otu_table %>% t, filt_bal_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>%
        .[row.names(filt_fit_data_bal$data_italic),] %>%
        mutate_all(~na_if(., 0)) %>% rownames_to_column("feature") %>% subset(., !grepl("NA", .$feature))

```


```{r}

#Volcano plot

figS7 <- ggplot(filt_maaslin_all, aes(y = -log10(qval), x = coef, col = metadata)) +
        theme_classic(base_family = "serif") +
        #labs(tag = "A") +
        geom_point(size = 2) +
        xlab("MaAslin coefficient") +
        ylab("-log<sub>10</sub>(*q*-value)") +
        ylim(c(-1, 35)) +
        geom_hline(yintercept = 1, col = "gray") +
        geom_vline(xintercept = 0, col = "gray") +
        annotate(family = "serif",
                 geom='richtext',
                 x=0, y=80,
                 label = "<i>q</i>-value = 0.1, fold-change = 0") +
        theme(legend.position = "top", axis.title.y = ggtext::element_markdown(), legend.text = element_markdown()) +
        scale_color_manual(values = c("#4daf4a",  "#984ea3", "#f781bf", "#377eb8", "#ff7f00", "#ffff33", "#a65628"),
                           breaks = c("log10.Final_reads", "sample_type", "lypma", "benzonase", "host_zero",  "molysis", "qiaamp"), 
                           labels = c("log<sub>10</sub>(Final reads)", "Sample type", "lyPMA", "Benzonase", "Host zero",  "Molysis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        guides(col = guide_legend(title = "Covariates", title.position = "top", nrow = 1))

figS7
png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS7.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 90, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue
figS7
dev.off()

```


### Balloon plot (Fig 4)

**Fig. 4** Mean relative abundance of top 20 significant taxa identified by differential abundance analysis by sample type. (A) Mock community, (B) bronchoalveolor lavage, (C) nasal swabs, and (D) sputum.


```{r}



f5a <- merge(filt_fit_data_pos$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_pos$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(filt_fit_data_pos$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1"),
               feature = case_when(feature =="*Saccharomyces cerevisiae x Saccharomyces kudriavzevii*" ~ "*S. cerevisiae* X *S. kudriavzevii*",
                               .default = feature)) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "serif") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        ylab("Species") +
        labs(tag = "A") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 7))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "Molysis",
                                  "qiaamp" = "QIAamp")
                         )
#ffff33 qia

f5b <- merge(filt_fit_data_bal$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_bal$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(filt_fit_data_bal$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(is.na(qval) ~ "> 0.1",
                               qval < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%

#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        theme_classic(base_family = "serif") +
        #colors for qvalues
        xlab("Experimental group") +
        ylab("Species") +
        labs(tag = "B") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 7))  +
        
        scale_fill_manual(values = c("grey", "red"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "Molysis",
                                  "qiaamp" = "QIAamp")
                         )

f5c <- merge(filt_fit_data_ns$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_ns$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(filt_fit_data_ns$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "serif") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        ylab("Species") +
        labs(tag = "C") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 7))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "Molysis",
                                  "qiaamp" = "QIAamp")
                         )

f5d <- merge(filt_fit_data_spt$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_spt$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        merge(filt_fit_data_spt$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "serif") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        ylab("Species") +
        labs(tag = "D") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 7),
              axis.title.x = element_text(margin = margin(t = 20)))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "Molysis",
                                  "qiaamp" = "QIAamp")
                         )
fig4 <- ggarrange(f5a, f5b, f5c, f5d, align = "hv", common.legend = T)

fig4

png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/Figure4.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 170, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

fig4

# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()

```

### Balloon plot + α

**Ballon plot with strain information (Eukaryota, gram-stain)**


```{r}
#list of gram negative bacteria
list_gram_neg <- phyloseq_unfiltered$phyloseq_rel %>%
        tax_table %>%
        data.frame %>%
        subset(., .$gram_stain == "Gram-negative") %>%
        row.names()
#list of gram positives
list_gram_pos <- phyloseq_unfiltered$phyloseq_rel %>%
        tax_table %>%
        data.frame %>%
        subset(., .$gram_stain == "Gram-positive") %>%
        row.names()

#list of gram positives
list_gram_fungi <- phyloseq_unfiltered$phyloseq_rel %>%
        tax_table %>%
        data.frame %>%
        subset(., .$gram_stain == "Fungi") %>%
        row.names()




f5a <- merge(filt_fit_data_pos$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_pos$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(filt_fit_data_pos$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1"),
               feature = case_when(feature =="*Saccharomyces cerevisiae x Saccharomyces kudriavzevii*" ~ "*S. cerevisiae* X *S. kudriavzevii*",
                               .default = feature)) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "serif") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        ylab("Species") +
        labs(tag = "A") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 7))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "Molysis",
                                  "qiaamp" = "QIAamp")
                         )
#ffff33 qia

f5b <- merge(filt_fit_data_bal$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_bal$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(filt_fit_data_bal$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(is.na(qval) ~ "> 0.1",
                               qval < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%

#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        theme_classic(base_family = "serif") +
        #colors for qvalues
        xlab("Experimental group") +
        ylab("Species") +
        labs(tag = "B") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 7))  +
        
        scale_fill_manual(values = c("grey", "red"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "Molysis",
                                  "qiaamp" = "QIAamp")
                         )

f5c <- merge(filt_fit_data_ns$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_ns$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(filt_fit_data_ns$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "serif") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        ylab("Species") +
        labs(tag = "C") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 7))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "Molysis",
                                  "qiaamp" = "QIAamp")
                         )

f5d <- merge(filt_fit_data_spt$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_spt$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        merge(filt_fit_data_spt$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "serif") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        ylab("Species") +
        labs(tag = "D") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 7),
              axis.title.x = element_text(margin = margin(t = 20)))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "Molysis",
                                  "qiaamp" = "QIAamp")
                         )
fig4 <- ggarrange(f5a, f5b, f5c, f5d, align = "hv", common.legend = T)

fig4

png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/Figure4.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 170, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

fig4

# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()

```

### {-}

# **3.4. Effect of treatments on functional analysis results**




# *xi.	Did depletion methods change diversity, by sample type?*



        1.	How alpha-diversity changed along treatment?
                a.	Figure of alpha (Species richness) and beta diversity (Fig. S8)
                b.	Stratified LMER for alpha diversity changes - (Table S10)
                        i.	Lmer(S.obs~ sample type + treatment + sample type * treatment + (1|subject_id) ) (Table S6)
                        ii.	Lmer(S.obs~ sample type + treatment + (1|subject_id) ) (stratified)
                d.	Table of stratified beta diversity, Mock, BAL, Nasal, Sputum (Table S11)
                        i.	PERM(horn dist. ~ log10(Final reads) + lyPMA + Benzonase + Host zero + Molysis + QIAamp, strata = subject_id)
                                1.	Or, PERM(horn dist. ~ log10(Final reads) + sample type + treatment + sample_type * treatment, strata = subject_id)


## Function diveristy {.tabset}

### Figure of alpha and beta (Fig. S8)

**Figure S8**. Function richness and Horn-Morisita distances from untreated to treated by sample type

```{r warning=FALSE, "Function richness"}

sample_data <- sample_data(phyloseq$phyloseq_path_rpk) %>% data.frame(check.names = F) %>% subset(., !is.nan(.$simpson))
phyloseq_rel_nz <- subset_samples(phyloseq$phyloseq_path_rpk, S.obs != 0 & sample_type %in% c("BAL", "Nasal", "Sputum", "Mock"))

sample_data(phyloseq_rel_nz)$log10.Final_reads <- log10(sample_data(phyloseq_rel_nz)$Final_reads)
sample_data(phyloseq_rel_nz)$sampletype_treatment <- paste(sample_data(phyloseq_rel_nz)$sample_type, sample_data(phyloseq_rel_nz)$treatment, sep = ":")

```

```{r}

f4a <-        ggplot(subset(sample_data(phyloseq$phyloseq_path_rpk) %>% data.frame, sample_data(phyloseq$phyloseq_path_rpk)$sample_type %in% c("Sputum", "Nasal", "BAL", "Mock")), aes(y = S.obs)) +
        geom_boxplot(aes(fill = treatment), lwd = 0.2) +
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "Host zero", "Molysis", "QIAaamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        ylab("Species richness") +
        theme_classic (base_size = 12, base_family = "serif") + 
        labs(tag = "A") +
        theme(plot.tag = element_text(size = 15),  axis.text.x = element_blank(), axis.ticks.x = element_blank()) +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(fill = guide_legend(nrow = 1))



#distances of betadiversity - boxplots
horn_dist_long <- distance(phyloseq_rel_nz, method="horn") %>% as.matrix() %>% melt_dist() #making long data of distance matrices

#Adding sample type and treatment name. 
#this can be also done by merging metadata into the `horn_dist_long`
names <- data.frame(str_split_fixed(horn_dist_long$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(horn_dist_long$iso2, "_", 3))
horn_dist_long$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
horn_dist_long$method_1 <- ifelse(grepl("lyPMA", horn_dist_long$iso1),"lypma", 
                                         ifelse(grepl("ben", horn_dist_long$iso1),"benzonase", 
                                                ifelse(grepl("host", horn_dist_long$iso1),"host_zero", 
                                                       ifelse(grepl("qia", horn_dist_long$iso1),"qiaamp", 
                                                              ifelse(grepl("moly", horn_dist_long$iso1),"molysis", 
                                                                     "control")))))


#Adding data for iso 2 also should be done
horn_dist_long$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")
horn_dist_long$method_2 <-ifelse(grepl("lyPMA", horn_dist_long$iso2),"lypma", 
                                        ifelse(grepl("ben", horn_dist_long$iso2),"benzonase", 
                                               ifelse(grepl("host", horn_dist_long$iso2),"host_zero", 
                                                      ifelse(grepl("qia", horn_dist_long$iso2),"qiaamp", 
                                                             ifelse(grepl("moly", horn_dist_long$iso2),"molysis", 
                                                                    "control")))))


#subsetting distances of my interest
horn_dist_long$sample_id_1 <- ifelse(grepl("pos", horn_dist_long$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_1, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_1))
horn_dist_long$sample_id_2 <- ifelse(grepl("pos", horn_dist_long$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_2, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_2))


path_horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long, horn_dist_long$sample_id_1 == horn_dist_long$sample_id_2) # data within samples

path_horn_dist_long_within_sampleid_from_control <- subset(path_horn_dist_long_within_sampleid_from_control,
                                                           path_horn_dist_long_within_sampleid_from_control$method_1 != path_horn_dist_long_within_sampleid_from_control$method_2) # remove irrelevant association

path_horn_dist_long_within_sampleid_from_control <- subset(path_horn_dist_long_within_sampleid_from_control, (path_horn_dist_long_within_sampleid_from_control$method_1 == "control") + (path_horn_dist_long_within_sampleid_from_control$method_2 == "control") != 0)


path_horn_dist_long_within_sampleid_from_control$treatment <- path_horn_dist_long_within_sampleid_from_control$method_1

path_horn_dist_long_within_sampleid_from_control$treatment <- ifelse(path_horn_dist_long_within_sampleid_from_control$treatment == "control", path_horn_dist_long_within_sampleid_from_control$method_2, path_horn_dist_long_within_sampleid_from_control$treatment) 


#Setting key method
path_horn_dist_long_within_sampleid_from_control$sample_type <- ifelse(grepl("NS", path_horn_dist_long_within_sampleid_from_control$iso1), "Nasal",
                                                                  ifelse(grepl("CFB", path_horn_dist_long_within_sampleid_from_control$iso1), "Sputum",
                                                                         ifelse(grepl("BAL", path_horn_dist_long_within_sampleid_from_control$iso1), "BAL",
                                                                                ifelse(grepl("pos|POS", path_horn_dist_long_within_sampleid_from_control$iso1, ignore.case = T), "Mock",
                                                                                       ifelse(grepl("neg|N_EXT", path_horn_dist_long_within_sampleid_from_control$iso1), "Neg.",NA)))))

#Making a column for baseline (controls, from where?)
path_horn_dist_long_within_sampleid_from_control <- path_horn_dist_long_within_sampleid_from_control %>% 
        mutate(dist_from = case_when(method_1 == "control" ~ iso1,
                                     method_2 == "control" ~ iso2))

dummy <- data.frame(iso1 = path_horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           iso2 = path_horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           dist = 0,
           treatment = "Untreated",
           method_1 = "control",
           method_2 = "control"
           )
names <- data.frame(str_split_fixed(dummy$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(dummy$iso2, "_", 3))
dummy$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
#Adding data for iso 2 also should be done
dummy$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")


#subsetting distances of my interest
dummy$sample_id_1 <- ifelse(grepl("pos", dummy$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_1, ignore.case = T),"Neg.",
                                        dummy$sample_id_1))
dummy$sample_id_2 <- ifelse(grepl("pos", dummy$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_2, ignore.case = T),"Neg.",
                                        dummy$sample_id_2))
dummy$sample_type <- ifelse(grepl("NS", dummy$iso1), "Nasal",
                            ifelse(grepl("CFB", dummy$iso1), "Sputum",
                                   ifelse(grepl("BAL", dummy$iso1), "BAL",
                                          ifelse(grepl("pos|POS", dummy$iso1, ignore.case = T), "Mock",
                                                 ifelse(grepl("neg|N_EXT", dummy$iso1), "Neg.",NA)))))
dummy <- subset(dummy, !is.na(dummy$sample_type))
path_horn_dist_long_within_sampleid_from_control <- bind_rows(path_horn_dist_long_within_sampleid_from_control, dummy)


f4b2 <- path_horn_dist_long_within_sampleid_from_control %>% 
        mutate(across(sample_type, factor, levels=c("Mock", "BAL", "Nasal","Sputum"))) %>%
        subset(., .$sample_type != "Neg.") %>% 
        group_by(sample_type, treatment) %>%
        summarise(mean = mean(dist, na.rm = TRUE),
            sd = sd(dist, na.rm = TRUE),
            n = n()) %>%
  mutate(se = sd / sqrt(n),
         lower.ci = mean - qt(1 - (0.05 / 2), n - 1) * se,
         upper.ci = mean + qt(1 - (0.05 / 2), n - 1) * se,
         treatment = factor(treatment, levels = c("Untreated", "lypma", "benzonase", "host_zero", "molysis", "qiaamp"),
                            labels = c("Untreated", "lyPMA", "Benzonase", "Host zero", "Molysis", "QIAaamp"))) %>%#,
         #text = paste(sprintf("%.2f", round(mean, digits = 2)), " [", sprintf("%.2f", round(lower.ci, digits = 2)), ", ", sprintf("%.2f", round(upper.ci, digits = 2)), "]", sep = "")) %>%
        ggplot(aes(x = mean, y = treatment, col = treatment)) +
        geom_point(aes(x=mean), shape=15, size=3) +
        geom_linerange(aes(xmin=lower.ci, xmax=upper.ci)) +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated", "lyPMA", "Benzonase", "Host zero", "Molysis", "QIAaamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Distance from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "serif") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(), axis.ticks.y = element_blank(), legend.position = "none") +
        labs(tag = "B") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "serif") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "serif") +
        scale_x_continuous(breaks = c(0, 1), labels = c("Lower bias", "Higher bias"))


figS8 <- ggarrange(f4a, f4b2, ncol = 1, common.legend = T, align = "hv") +
        guides(fill = guide_legend(nrow = 1))


figS8

png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS8.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 170, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue
figS8

# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()


```


### Function richness - LMER all samples (Table S10)

**Effect of some treatment was neutralized by interaction term. Therefore, the association was sample_type specific.**

**Table S10**. Effect size, standard error (SE) and p-value of a statistical test for function richness with an interaction term using linear mixed effect model (Species richness ~ sample type * treatment + log10 (final reads) + (1|subject_id) ).

```{r}
library(lmerTest)
sample_data <- sample_data(phyloseq$phyloseq_path_rpk)
sample_data$log_centered_final_reads <- log(sample_data$Final_reads + 1) - median(log((subset(sample_data, sample_data$sample_type %in% c("BAL") & sample_data$treatment %in% c("Untreated")) %>% .$Final_reads) + 1))


tableS10 <- lmer(S.obs ~ sample_type * treatment + log_centered_final_reads + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("sample_type|treatment", "", x))  %>% mutate(x = gsub(":", " * ", x)) %>% 
        column_to_rownames(var = "x") %>%  
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = 1), 
                                " (", 
                                round(Estimate - 1.96 * SE, 1) %>% format(nsmall = 1),
                                ", ",
                                round(Estimate + 1.96 * SE, 1) %>% format(nsmall = 1),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>% 
        kbl(format = "html", escape = 0) %>% kable_styling(full_width = 0, html_font = "serif")


tableS10


save_kable(tableS10, file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS10.html", self_contained = T)


```

### Function richness - stratified

**Stratified analysis** will be conducted.


```{r}
sr_lmer_bal <- lmer(S.obs ~ treatment + log10 (Final_reads) + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("BAL"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("10", "<sub>10</sub>", x)) %>% mutate(x = gsub("_", " ", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate(across(is.numeric, round, digits=2))%>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error",
               "Effect size" = "Estimate") %>%
        select(c("Effect size", "SE", "<i>p</i>-value", " "))

sr_lmer_ns <- lmer(S.obs ~ treatment + log10 (Final_reads) + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("BAL"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("10", "<sub>10</sub>", x)) %>% mutate(x = gsub("_", " ", x)) %>%
        column_to_rownames(var = "x") %>% 
mutate(across(is.numeric, round, digits=2)) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error",
               "Effect size" = "Estimate") %>%
        select(c("Effect size", "SE", "<i>p</i>-value", " "))

sr_lmer_spt <- lmer(S.obs ~ treatment + log10 (Final_reads) + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("BAL"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("10", "<sub>10</sub>", x)) %>% mutate(x = gsub("_", " ", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate(across(is.numeric, round, digits=2)) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               SE = "Std. Error",
               "Effect size" = "Estimate") %>%
        select(c("Effect size", "SE", "<i>p</i>-value", " ")) 



cbind(sr_lmer_bal, sr_lmer_ns, sr_lmer_spt) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 4, "Nasal swab" = 4,"Sputum"= 4)) %>% 
        kable_styling(full_width = 0, html_font = "serif")

```

### Function beta - all samples (Table S11)

**Table S11**. Degree of freedom, effect size (residual, R^2) and p-value of permutational ANOVA for functional Horn-Morisita distances with an interaction term and strata term (BC-distance of functions ~ sample type * treatment + log10(final reads), strata = subject id).

```{r}

phyloseq_rel_nz <- transform_sample_counts(phyloseq$phyloseq_path_rpk, function(x) {x/sum(x)}) %>%
        subset_samples(S.obs != 0 & sample_type %in% c("Mock", "BAL", "Nasal", "Sputum"))

bray_perm_inter <- vegan::adonis2(distance(phyloseq_rel_nz, method="horn") ~ sample_type * treatment + subject_id + log10(Final_reads),
                                  data = phyloseq_rel_nz %>% sample_data %>% data.frame(check.names = F), 
                                  strata = phyloseq_rel_nz %>% sample_data %>% data.frame(check.names = F) %>% .$subject_id,
                                  permutations = 10000)

bray_perm_ns <- vegan::adonis2(distance(subset_samples(phyloseq_rel_nz, sample_type == "Nasal"), method="horn") ~ lypma + benzonase + host_zero + molysis + qiaamp + log10(Final_reads),
                               data = subset_samples(phyloseq_rel_nz, sample_type == "Nasal") %>%
                                       sample_data %>% data.frame(check.names = F),
                               strata = subset_samples(phyloseq_rel_nz, sample_type == "Nasal") %>% 
                                       sample_data %>% data.frame(check.names = F) %>% .$subject_id, permutations = 10000)

bray_perm_bal  <- vegan::adonis2(distance(subset_samples(phyloseq_rel_nz, sample_type == "BAL"), method="horn") ~  lypma + benzonase + host_zero + molysis + qiaamp + log10(Final_reads),
                                 data = subset_samples(phyloseq_rel_nz, sample_type == "BAL") %>% sample_data %>% data.frame(check.names = F),
                                 strata = subset_samples(phyloseq_rel_nz, sample_type == "BAL") %>%
                                         sample_data %>% data.frame(check.names = F) %>% .$subject_id,
                                  permutations = 10000)

bray_perm_spt <- vegan::adonis2(distance(subset_samples(phyloseq_rel_nz, sample_type == "Sputum"), method="horn") ~ lypma + benzonase + host_zero + molysis + qiaamp + log10(Final_reads),
                                data = subset_samples(phyloseq_rel_nz, sample_type == "Sputum") %>% sample_data %>% data.frame(check.names = F),
                                strata = subset_samples(phyloseq_rel_nz, sample_type == "Sputum")
                                %>% sample_data %>% data.frame(check.names = F) %>% .$subject_id,
                                  permutations = 10000)





tableS11 <- bray_perm_inter %>% data.frame(check.names = F) %>% rownames_to_column("row.names") %>% 
        mutate(row.names = case_when(row.names == "sample_type" ~ 'Sample type',
                                     row.names == "treatment" ~ 'Treatment',
                                     row.names == "subject_id" ~ 'Subject',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "sample_type:treatment" ~ 'Sample type * Treatment',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
        mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3),
               `Pr(>F)` = format(`Pr(>F)`, nsmall = 3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        select(c("Degree of freedom", "R<sup>2</sup>", "<i>p</i>-value", " ")) %>% 
        kbl(format = "html", escape = 0) %>%
        kable_styling(full_width = 0, html_font = "serif")

tableS11
save_kable(tableS11, file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS11.html", self_contained = T)

```

### Function beta - stratified

```{r}

a <- bray_perm_bal %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'Host zero',
                                     row.names == "molysis" ~ 'Molysis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 

b <- bray_perm_ns %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'Host zero',
                                     row.names == "molysis" ~ 'Molysis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 

c <- bray_perm_spt %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'Host zero',
                                     row.names == "molysis" ~ 'Molysis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 


cbind(a, b, c) %>% 
        kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "serif")


```


### {-}



# *xii.	What type of fuction were affected by the treatment?*





        1.	DA analysis result by sample type (Figure S9)

                a.	A: Mock
                b.	B: BAL
                c.	C: Nasal
                d.	D: Sputum
                
        2.	Volcano plot with Mock, BAL, Nasal and Sputum (Figure S8)
                a.	Maaslin ( feature ~ log10(Final reads) + sample type +  lyPMA + Benzonase + Host zero + Molysis + QIAamp, RE = subject_id)
                b.	Or Maaslin ( feature ~ log10(Final reads) + sample type + treatment + sample type * treatment, RE = subject_id)


## Function DA analysis {.tabset}

```{r warning=FALSE, message=FALSE,  "MaAslin Function"}

#DA analysis - MaAslin
sample_data(phyloseq_rel_nz)$log10.Final_reads <- log10(sample_data(phyloseq_rel_nz)$Final_reads)

#Running MaAslin for all sample without decontam
#for taxa differentially abundant by host depletion method, look to see which ones overlap with potential contaminant taxa

# Maaslin - # # y ~ log(final reads) + sample_type + treatment  -----------

#all samples
```


```{r}

f_maaslin_all <- read.csv("/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git/data/filt_f_maaslin_all.csv")
f_fit_data_bal <- read.csv("/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git/data/filt_f_fit_data_bal.csv")
f_fit_data_spt <- read.csv("/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git/data/filt_f_fit_data_spt.csv")
f_fit_data_ns <- read.csv("/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git/data/filt_f_fit_data_ns.csv")
f_fit_data_pos <- read.csv("/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git/data/filt_f_fit_data_pos.csv")


```


**Again, most of DA functions were sample type specific**

```{r warning=FALSE, message=FALSE, "MaAslin Function continued"}

#Making significance table for figure
        # Define a function to make species names italicized
# Make a significance table for each figure (top 20 taxa)
make_sig_table <- function(data) {
  sig_data <- spread(data[order(data$qval), c("feature", "metadata", "qval")], metadata, qval)
  sig_data$feature <- gsub("[.]", "-", sig_data$feature)
  sig_data$min <- apply(sig_data, 1, FUN = min)
  sig_data <- sig_data[order(sig_data$min),] %>% select("feature", "lypma", "benzonase", "host_zero", "molysis", "qiaamp") %>% .[1:20,]
  sig_data[["feature"]] <- ifelse(sig_data[["feature"]] == "X.Collinsella._massiliensis", "[Collinsella]_massiliensis", sig_data[["feature"]])
  sig_data_italic <- sig_data %>% rownames_to_column(var = "-") %>%
          column_to_rownames(var = "feature") %>% select(-c("-")) %>%
          rename(lyPMA = lypma,  Benzonase = benzonase, `Host zero` = host_zero, Molysis = molysis, QIAamp = qiaamp)
  sig_data_sig <- ifelse(sig_data_italic < 0.1, "*", NA) %>% data.frame(check.names = F)
  return(list(data = sig_data, data_italic = sig_data_italic, data_sig = sig_data_sig))
}

f_fit_data_pos <- make_sig_table(f_fit_data_pos)
f_fit_data_bal <- make_sig_table(f_fit_data_bal)
f_fit_data_ns <- make_sig_table(f_fit_data_ns)
f_fit_data_spt <- make_sig_table(f_fit_data_spt)

f_pos_sig <- subset_taxa(subset_samples(phyloseq_rel_nz, sample_type == "Mock"),
                                       taxa_names(subset_samples(phyloseq_rel_nz, sample_type == "Mock")) %in% f_fit_data_pos$data$feature)
f_fit_data_pos$rel <- cbind(f_pos_sig %>% otu_table %>% t, f_pos_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% data.frame(check.names = F) %>% 
        .[row.names(f_fit_data_pos$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")


f_spt_sig <- subset_taxa(subset_samples(phyloseq_rel_nz, sample_type == "Sputum"),
                                       taxa_names(subset_samples(phyloseq_rel_nz, sample_type == "Sputum")) %in% f_fit_data_spt$data$feature)


f_fit_data_spt$rel <- cbind(f_spt_sig %>% otu_table %>% t, f_spt_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% data.frame(check.names = F) %>% 
        .[row.names(f_fit_data_spt$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")

f_ns_sig <- subset_taxa(subset_samples(phyloseq_rel_nz, sample_type == "Nasal"),
                                       taxa_names(subset_samples(phyloseq_rel_nz, sample_type == "Nasal")) %in% f_fit_data_ns$data$feature)

f_fit_data_ns$rel <- cbind(f_ns_sig %>% otu_table %>% t, f_ns_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% data.frame(check.names = F) %>% 
        .[row.names(f_fit_data_ns$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")
f_fit_data_ns$rel$feature <- row.names(f_fit_data_ns$data_sig)

f_bal_sig <- subset_taxa(subset_samples(phyloseq_rel_nz, sample_type == "BAL"),
                                       taxa_names(subset_samples(phyloseq_rel_nz, sample_type == "BAL")) %in% f_fit_data_bal$data$feature)

f_fit_data_bal$rel <- cbind(f_bal_sig %>% otu_table %>% t, f_bal_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% data.frame(check.names = F) %>%
        .[row.names(f_fit_data_bal$data_italic),] %>%
        mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")



```

### MaAslin function - volcano plot (Fig. S9)

**Figure S9**. Volcano plot of sequencing depth adjusted differential abundance of function by each treatment 


```{r}

#Volcano plot

figS9 <- ggplot(f_maaslin_all, aes(y = -log10(qval), x = coef, col = metadata)) +
        theme_classic(base_family = "serif") +
        #labs(tag = "A") +
        geom_point(size = 2) +
        xlab("MaAslin coefficient") +
        ylab("-log<sub>10</sub>(*q*-value)") +
        geom_hline(yintercept = 1, col = "gray") +
        geom_vline(xintercept = 0, col = "gray") +
        annotate(family = "serif",
                 geom='richtext',
                 x=0, y=80,
                 label = "<i>q</i>-value = 0.1, fold-change = 0") +
        theme(legend.position = "top", axis.title.y = ggtext::element_markdown()) +
        scale_color_manual(values = c("#4daf4a",  "#984ea3", "#f781bf", "#377eb8", "#ff7f00", "#ffff33", "#a65628"),
                           breaks = c("log10.Final_reads", "sample_type", "lypma", "benzonase", "host_zero",  "molysis", "qiaamp"), 
                           labels = c("log10(Final reads)", "Sample type", "lyPMA", "Benzonase", "Host zero",  "Molysis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        guides(col = guide_legend(title = "Covariates", title.position = "top", nrow = 2))


png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS9.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 90, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figS9
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()
figS9

```

### Function baloon plot (Fig. S10)

**Figure S10**. Mean relative abundance of top 20 significant function identified by differential abundance analysis using MaAsLin. Analyses were stratified by sample type. (A) Mock community, (B) bronchoalveolor lavage, (C) nasal swabs, and (D) sputum. Statistical significances were noted at the level of q-value < 0.1

```{r}

f5a <- merge(f_fit_data_pos$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      f_fit_data_pos$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(f_fit_data_pos$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1"),
               value = value * 1000000) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "serif") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        ylab("Function") +
        labs(tag = "A") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
                axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 7),
              axis.title.x = element_text(margin = margin(t = 20)))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Copies per million",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "Molysis",
                                  "qiaamp" = "QIAamp")
                         )
#ffff33 qia

f5b <- merge(f_fit_data_bal$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      f_fit_data_bal$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(f_fit_data_bal$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1"),
               value = value * 1000000) %>%

#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        theme_classic(base_family = "serif") +
        #colors for qvalues
        xlab("Experimental group") +
        ylab("Function") +
        labs(tag = "B") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 7),
              axis.title.x = element_text(margin = margin(t = 20)))  +
        
        scale_fill_manual(values = c("grey", "red"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Copies per million",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "Molysis",
                                  "qiaamp" = "QIAamp")
                         )

f5c <- merge(f_fit_data_ns$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      f_fit_data_ns$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(f_fit_data_ns$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1"),
               value = value * 1000000) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "serif") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        ylab("Function") +
        labs(tag = "C") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 7),
              axis.title.x = element_text(margin = margin(t = 20)))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Copies per million",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "Molysis",
                                  "qiaamp" = "QIAamp")
                         )

f5d <- merge(f_fit_data_spt$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      f_fit_data_spt$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        merge(f_fit_data_spt$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1"),
               value = value * 1000000) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "serif") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        ylab("Function") +
        labs(tag = "D") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 7),
              axis.title.x = element_text(margin = margin(t = 20)))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Copies per millio",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "Molysis",
                                  "qiaamp" = "QIAamp")
                         )

figS10 <- ggarrange(f5a, f5b, f5c, f5d, align = "hv", common.legend = T)

figS10

png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS10.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 170, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

figS10
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()



```

### {-}


# Final results summary {.tabset}




### Rarefraction curve of species richenss (Fig S.11)

**Fig.S11. Rarefraction curve of species and function**

**As a sanity check, rarefaction curves were generated and seemed to be saturated**

```{r}

fig_rarefraction <- phyloseq$phyloseq_count %>% 
        sample_data %>% 
        data.frame %>%
        subset(., !is.na(.$treatment) & sample_type %in% c("BAL", "Nasal", "Sputum")) %>%
ggplot(., aes(x = Final_reads/1000000, y = S.obs, col = treatment)) +
        geom_point() +
        theme_classic(base_family = "serif") +
        xlab("Final reads x 10<sup>6</sup>") +
        ylab("Species richness") +
        labs(tag = "A") +
        theme(axis.title.x = element_markdown(), legend.position = "top") +
        guides(col = guide_legend(title = "Treatment", title.position = "top", nrow = 1)) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"),
                          name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "Host zero", "Molysis", "QIAaamp")) +
        facet_wrap(~sample_type, scales = "free", nrow = 1)

fig_rarefraction_function <- phyloseq$phyloseq_path_rpk %>% 
        sample_data %>% 
        data.frame %>%
        subset(., !is.na(.$treatment) & sample_type %in% c("BAL", "Nasal", "Sputum")) %>%
ggplot(., aes(x = Final_reads/1000000, y = S.obs, col = treatment)) +
        geom_point() +
        theme_classic(base_family = "serif") +
        xlab("Final reads x 10<sup>6</sup>") +
        ylab("Function richness") +
        labs(tag = "B") +
        theme(axis.title.x = element_markdown(), legend.position = "top") +
        guides(col = guide_legend(title = "Treatment", title.position = "top", nrow = 1)) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"),
                          name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "Host zero", "Molysis", "QIAaamp")) +
        facet_wrap(~sample_type, scales = "free", nrow = 1)

ggarrange(fig_rarefraction, fig_rarefraction_function, common.legend = T, ncol = 1)

png(file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS11.png",   # The directory you want to save the file in
    width = 180, # The width of the plot in inches
    height = 180, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

ggarrange(fig_rarefraction, fig_rarefraction_function, common.legend = T, ncol = 1)


dev.off()



```                   


### Mean change in species richness

```{r}
#Mean species richness change
sample_data %>% data.frame() %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")) %>% 
        select(c("sample_type", "subject_id", "treatment", "S.obs")) %>%
        spread(., treatment, S.obs) %>%
        group_by(sample_type) %>%
        summarise(lyPMA = mean(lyPMA - Untreated, na.rm = T),
                  Benzonase = mean(Benzonase - Untreated, na.rm = T),
                  `Host zero` = mean(`Host zero` - Untreated, na.rm = T),
                  Molysis = mean(Molysis - Untreated, na.rm = T),
                  QIAamp = mean(QIAamp - Untreated, na.rm = T)) %>%
        t() %>% 
        kbl(format = "html",caption = "Changes of species richenss by sample type and traetment (subject pairwise)") %>%
        kable_styling(full_width = 0, html_font = "serif")
        
```


### Sequencing summary (Table 4)

**Table 4**. Summary table

**3.5. Create a summary table like Table 5 in Birdy's DNA extraction methods paper to help drive home the takehome message. There are pros and cons to each method**

**Molysis for BAL, QIAamp for nasal sab, and Host zero for sputum. **

```{r}

summary_bal <- matrix(nrow=5,ncol=3) %>% data.frame() %>% 
        rename(Issues = X1, `Species richness` = X2, `Beta diversity change` = X3) %>%
        rownames_to_column("x") %>% mutate(x = c("lyPMA", "Benzonase", "Host zero", "Molysis", "QIAamp"),
                                           Issues = c("High host%",
                                                     "High host%",
                                                     "-",
                                                     "-",
                                                     "High host%"),
                                           `Species richness` = c("+0.8",
                                                                  "+6.4",
                                                                  "+9.2",
                                                                  "+18.8",
                                                                  "+10.0"),
                                           `Beta diversity change` = c("LM",
                                                                       "-",
                                                                       "LM",
                                                                       "-",
                                                                       "LM")) %>%
                                           column_to_rownames("x")
                                           

summary_ns <- matrix(nrow=5,ncol=3) %>% data.frame() %>% 
        rename(Issues = X1, `Species richness` = X2, `Beta diversity change` = X3) %>%
        rownames_to_column("x") %>% mutate(x = c("lyPMA", "Benzonase", "Host zero", "Molysis", "QIAamp"),
                                           Issues = c("High host %<br>gram-stain %",
                                                     "High host%",
                                                     "-",
                                                     "Library fail",
                                                     "Library fail"),
                                           `Species richness` = c("-4.2",
                                                                  "-0.8",
                                                                  "+8.8",
                                                                  "+5.4",
                                                                  "+7.4"),
                                           `Beta diversity change` = c("PERM<br>LM",
                                                                       "-",
                                                                       "-",
                                                                       "-",
                                                                       "-")) %>%
                                           column_to_rownames("x")
                                           

summary_spt <- matrix(nrow=5,ncol=3) %>% data.frame() %>% 
        rename(Issues = X1, `Species richness` = X2, `Beta diversity change` = X3) %>%
        rownames_to_column("x") %>% mutate(x = c("lyPMA", "Benzonase", "Host zero", "Molysis", "QIAamp"),
                                           Issues = c("Gram-stain %",
                                                     "Gram-stain %",
                                                     "Gram-stain %",
                                                     "Gram-stain %",
                                                     "Gram-stain %"),
                                           `Species richness` = c("+34.8",
                                                                  "+63.0",
                                                                  "+96.0",
                                                                  "+104.6",
                                                                  "+79.4"),
                                           `Beta diversity change` = c("-",
                                                                       "LM",
                                                                       "LM",
                                                                       "PERM<br>LM",
                                                                       "PERM<br>LM")) %>%
                                           column_to_rownames("x")
                                           

table3 <- cbind(summary_bal, summary_ns, summary_spt) %>% 
        kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
                kable_styling(full_width = 0, html_font = "serif")
table3
save_kable(table3, file = "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/table3.html", self_contained = T)


```



```



##  {.unnumbered}

Done.

# Bibliography

```{r warning=FALSE}
#===============================================================================
#BTC.LineZero.Footer.1.1.0
#===============================================================================
#R markdown citation generator.
#===============================================================================
#RLB.Dependencies:
#   magrittr, pacman, stringr
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#BTC.Dependencies:
#   LineZero.Header
#===============================================================================
#Generates citations for each explicitly loaded library.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
str_libraries <- c("r", str_libraries)
for (str_libraries in str_libraries) {
    str_libraries |>
        pacman::p_citation() |>
        print(bibtex = FALSE) |>
        capture.output() %>%
        .[-1:-3] %>% .[. != ""] |>
        stringr::str_squish() |>
        stringr::str_replace("_", "") |>
        cat()
    cat("\n")
}
#===============================================================================
```
