---
title: "COD_20240212_HOST_6_phage_analysis_figure_table"
author: "Minsik Kim"
date: "2024-02-12"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
        df_print: paged
editor_options: 
  chunk_output_type: inline
  markdown: 
    wrap: 72
---

# Loading packages

```{r warning=FALSE, message=FALSE, setup}
#===============================================================================
#BTC.LineZero.Header.1.1.0
#===============================================================================
#R Markdown environment setup and reporting utility.
#===============================================================================
#RLB.Dependencies:
#   knitr, magrittr, pacman, rio, rmarkdown, rmdformats, tibble, yaml
#===============================================================================
#Input for document parameters, libraries, file paths, and options.
#=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=
#knitr::opts_chunk$set(message=FALSE, warning = FALSE)


path_working <- 
        ifelse(sessionInfo()[1]$R.version$platform == "x86_64-pc-linux-gnu",
               "/home/bagel/minsik/",
               ifelse(sessionInfo()[1]$R.version$platform == "aarch64-apple-darwin20",
                      "/Volumes/macdrive/Dropbox/", 
                      "/Users/minsikkim/Dropbox (Personal)"))

path_library <- 
        ifelse(sessionInfo()[1]$R.version$platform == "x86_64-pc-linux-gnu",
               "/home/bagel/R/x86_64-pc-linux-gnu-library/4.1/",
               "/Library/Frameworks/R.framework/Resources/library/")

str_libraries <- c("readxl", "phyloseq", "tidyverse", "pacman", "yaml", "ggplot2", "vegan", "microbiome", "ggpubr", "viridis", "decontam", "gridExtra", "ggpubr", "lme4", "lmerTest", "writexl", "harrietr", "Maaslin2", "ggtext", "ggpmisc", "gamm4", "reshape2", "kableExtra", "knitr", "ggtree", "car", "mediation", "lemon", "qvalue")
        
YAML_header <-
'---
title: "Host-DNA depletion analysis of bacterio phage"
author: "Minsik Kim"
date: "2024.02.12"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
---'
seed <- "20230925"

#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Loads libraries, file paths, and other document options.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
FUN.LineZero.Boot <- function() {
    .libPaths(path_library)

    require(pacman)
    pacman::p_load(c("knitr", "rmarkdown", "rmdformats", "yaml"))

    knitr::opts_knit$set(root.dir = path_working)

    str_libraries |> unique() |> sort() -> str_libraries
    pacman::p_load(char = str_libraries)

    set.seed(seed)
}
FUN.LineZero.Boot()
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Outputs R environment report.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

FUN.LineZero.Report <- function() {
    cat("Line Zero Environment:\n\n")
    paste("R:", pacman::p_version(), "\n") |> cat()
    cat("Libraries:\n")
    for (str_libraries in str_libraries) {
        paste(
            "    ", str_libraries, ": ", pacman::p_version(package = str_libraries),
            "\n", sep = ""
        ) |> cat()
    }
    paste("\nOperating System:", pacman::p_detectOS(), "\n") |> cat()
    paste("    Library Path:", path_library, "\n") |> cat()
    paste("    Working Path:", path_working, "\n") |> cat()
    paste("Seed:", seed, "\n\n") |> cat()
    cat("YAML Header:\n")
    cat(YAML_header)
}
FUN.LineZero.Report()

```


# Contents

0. Exploratory data analysis

        •	Summary table of species richness
        •	Marker-MAGu microbial output
        •	Marker-MAGu phage output
        •	MetaPhlAnN3 output
        •	Mapping failure & viral mapped reads
        •	decontam

        
Question 1. How does the Marker-MAGu microbial community looks like?
        
        •	Microbial communities of samples at Domain level 
        •	Positive controls
        
Question 2. Was phage community changed the same as microbial community, by sample type and treatment?

        •	Alpha diversity (phage species richness)
        
Question 3. How was the changes of species richness, stratified by domain?

        •	Alpha diversity of microbial species richness, stratified by each domain
        
Conclusion


## Analysis prep {.tabset}

### Loading data

```{r warning=F, message=FALSE, "Data loading"}
# Loading files -----------------------------------------------------------
#loading tidy phyloseq object
phyloseq <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20240212_MGK_host_marker_magu_Jessica.rds")


m_phyloseq <- phyloseq$microbe_rpkm
v_phyloseq <- phyloseq$viral_rpkm
v_h_phyloseq <- phyloseq$viral_host_rpkm

m_phyloseq_rel <- phyloseq$microbe_rel
v_phyloseq_rel <- phyloseq$viral_rel
v_h_phyloseq_rel <- phyloseq$viral_host_rel

```


### Alpha diversity indices

```{r}           
alpha_diversity <- function(data) {
        otu_table <- otu_table(v_phyloseq) %>% .[, colSums(.) !=0]
        S.obs <- rowSums(t(otu_table) != 0)
        sample_data <- sample_data(data)
        data_evenness <- vegan::diversity(t(otu_table)) / log(vegan::specnumber(t(otu_table))) # calculate evenness index using vegan package
        data_shannon <- vegan::diversity(t(otu_table), index = "shannon") # calculate Shannon index using vegan package
        data_hill <- exp(data_shannon)                           # calculate Hills index
        data_dominance <- microbiome::dominance(otu_table, index = "all", rank = 1, aggregate = TRUE) # dominance (Berger-Parker index), etc.
        data_invsimpson <- vegan::diversity(t(otu_table), index = "invsimpson")                          # calculate Shannon index using vegan package
        alpha_diversity <- cbind(S.obs, data_shannon, data_hill, data_invsimpson, data_evenness,data_dominance) # combine all indices in one data table
        sample_data <- merge(data.frame(sample_data), alpha_diversity, by = 0, all = T) %>% column_to_rownames(var = "Row.names")
}

sample_data(m_phyloseq) <- sample_data(alpha_diversity(m_phyloseq)) 
sample_data(v_phyloseq) <- sample_data(alpha_diversity(v_phyloseq)) 
sample_data(v_h_phyloseq) <- sample_data(alpha_diversity(v_h_phyloseq)) 

sample_data(m_phyloseq_rel) <- sample_data(alpha_diversity(m_phyloseq_rel)) 
sample_data(v_phyloseq_rel) <- sample_data(alpha_diversity(v_phyloseq_rel)) 
sample_data(v_h_phyloseq_rel) <- sample_data(alpha_diversity(v_h_phyloseq_rel)) 

#sample data loading

m_sample_data <- sample_data(m_phyloseq) %>% data.frame()
sample_data(v_phyloseq)$Viral_reads <- sample_sums(v_phyloseq)

v_sample_data <- sample_data(v_phyloseq) %>% data.frame()
v_h_sample_data <- sample_data(v_h_phyloseq) %>% data.frame()

```

## {-}

# Summary of exploratory data analysis
## Summary table {.tabset}                

### Summary table of species richness

**Summary table of microbial community from Marker-MAGu**. 

```{r}

m_sample_data %>%
        group_by(sample_type, treatment) %>%
        summarise(Mean_SObs = mean(S.obs),
                  SD_SObs = sd(S.obs),
                  Max_SObs = max(S.obs),
                  Min_SObs = min(S.obs),
                  N = n())


```

### Summary table of viral species richness


```{r}

v_sample_data %>%
        group_by(sample_type, treatment) %>%
        summarise(Mean_SObs = mean(S.obs),
                  SD_SObs = sd(S.obs),
                  Max_SObs = max(S.obs),
                  Min_SObs = min(S.obs),
                  N = n())


```

### Summary table of viral host richness at genus level


```{r}

v_h_sample_data %>%
        group_by(sample_type, treatment) %>%
        summarise(Mean_SObs = mean(S.obs),
                  SD_SObs = sd(S.obs),
                  Max_SObs = max(S.obs),
                  Min_SObs = min(S.obs),
                  N = n())


```

**Summary table of phage community from Marker-MAGu**. 

```{r}
v_sample_data %>%
        group_by(sample_type, treatment) %>%
        summarise(Mean_SObs = mean(S.obs),
                  SD_SObs = sd(S.obs),
                  N = n())

```

1. Mock community had 8 microbial taxa in average. Did not show fungi reads.

2. The Marker-MAGu + metaphlan 4 outcome showed relatively lower viral speciese richness compared to microbial richness.

*3. The Marker-MAGu viral community seems less sensitive to contaminants. The increase in species richenss was lower than microbial community analysis result.*


### Marker-MAGu microbial output

*Marker-MAGu didn't detect Eukaryotes.*

```{r}
m_phyloseq %>% 
        tax_table %>%
        data.frame() %>%
        .$Kingdom %>% 
        table %>% 
        data.frame() %>% 
        rename(Domain = ".",
               Frequency = Freq)
```

### Marker-MAGu phage output

*Marker-MAGu didn't detect phage that infects Eukaryotes (There is no such phage).*

```{r}
v_phyloseq %>% 
        tax_table %>%
        data.frame() %>%
        .$H_Kingdom %>% 
        table(.,  useNA = "always") %>% 
        data.frame() %>% 
        rename(Domain = ".",
               Frequency = Freq)
```

### Metaphlan 3

```{r}

# Loading files -----------------------------------------------------------
#loading tidy phyloseq object
phyloseq_unfiltered <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20230521_MGK_host_tidy.rds")

rbind(

phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(sample_type == "BAL") %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>%
        tax_table %>%
        data.frame %>%
        .$Domain %>%
        table(.,  useNA = "always") %>% 
        data.frame() %>%
        mutate(`Sample type` = "BAL", .before = "."),

phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(sample_type == "Nasal") %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>%
        tax_table %>%
        data.frame %>%
        .$Domain %>%
        table(.,  useNA = "always") %>% 
        data.frame() %>%
        mutate(`Sample type` = "Nasal", .before = "."),


phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(sample_type == "Sputum") %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>%
        tax_table %>%
        data.frame %>%
        .$Domain %>%
        table(.,  useNA = "always") %>% 
        data.frame() %>%
        mutate(`Sample type` = "Sputum", .before = ".")
) %>% 
        rename(Domain = ".",
               Frequency = Freq)


```

*By sample type, the proportion of Eukaryota (mainly fungi) was different at Metaphlan3 output. Nasal swabs's community had 14 Eukaryotes out of 59 taxa.*


### {-}

# Exploratory QC

## Library failure {.tabset}


### Failure crosstab

*More samples showed 0 mapped reads, relative to metaphlan data. Samples with 0 taxa were not eliminated for further analyses*

```{r}
sample_data <- sample_data(v_phyloseq) %>% 
		  data.frame() %>%
		  mutate(
		      sample_sums = sample_sums(v_phyloseq),
		      lib_failed = case_match(lib_failed, TRUE ~ 1, FALSE ~0),
		      seq_failed = ifelse(sample_sums == 0, 1, 0),
		      lib_seq_failed = ifelse(lib_failed ==1 | seq_failed == 1, 1, 0),
		      prop_host = round((Host_mapped/Reads_after_trim), 1)
		      )
cat("Sequencing failed")		
xtabs(seq_failed ~ sample_type + treatment, data = sample_data)
cat("Sequencing and library failed")		
xtabs(lib_seq_failed ~ sample_type + treatment, data = sample_data)
```

### Figure of sequencing result (Viral reads)

Host depletion effects on total viral reads measured by shotgun metagenomic sequencing. 

```{r}

fs3a <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(Viral_reads))) +
        geom_jitter(aes(col = treatment, x = treatment),
                    lwd = 0.2, alpha = 0.3, stroke = 0)+
        stat_summary(aes(color = treatment, x = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Treatment")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              axis.text.x = element_blank()) +
        ylab("log<sub>10</sub>(Viral RPKM)") +
        labs(tag = "A") +
        facet_wrap(~sample_type, scale = "free_x") +
        guides(fill = guide_legend(nrow = 1))

#	- Host_mapped



fs3a


#png(file = "/Users/minsikkim/Dropbox (Personal)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS3.png",   # The directory you want to save the file in
#    width = 180, # The width of the plot in inches
#    height = 170, # The height of the plot in inches
#    units = "mm",
#    res = 600
#) #fixing multiple page issue

#figS3
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

#dev.off()



```

For BAL and Sputum, viral reads clearly increased after treatment. For nasal swabs, such pattern was not observed.

### {-}

## decontam {.tabset}

### Table of decontam - stratified by sample_type (prevalence method)

**Note: decontam was conducted only with prevalence method, as quantification on viruses was not conducted via qPCR.**

*At sample type stratified analysis, decontam identified 2 possible contaminants (that were identified in negative controls).*



```{r warning=FALSE, message=FALSE}
#Stratified by sample type
neg_number_decontam <- 
        v_phyloseq %>% 
        subset_samples(sample_type == "Neg.") %>%
        sample_names() %>%
        length

sample_number_decontam <- 
        v_phyloseq %>% 
        subset_samples(sample_type != "Mock") %>%
        subset_samples(sample_type != "Neg.") %>%
        #subset_samples(S.obs != 0) %>%
        sample_names() %>%
        length

prev_neg <- ((v_phyloseq %>% 
          subset_samples(sample_type == "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (negative controls)" = ".")

prev_all <- ((v_phyloseq %>%
                      subset_samples(sample_type != "Mock") %>%
                      subset_samples(sample_type != "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (all)" = ".")


sample_data(v_phyloseq)$is.neg <- grepl("Neg", sample_data(v_phyloseq)$sample_type)



phyloseq_decontam_bal <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(sample_type == "Neg." | sample_type == "BAL")
phyloseq_decontam_ns <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(sample_type == "Neg." | sample_type == "Nasal")
phyloseq_decontam_spt <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(sample_type == "Neg." | sample_type == "Sputum")



contaminant_combined_bal <- 
data.frame("BAL", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_bal, method="prev", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)


contaminant_combined_ns <- 
data.frame("Nasal", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_ns, method="prev", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)



contaminant_combined_spt <- 
data.frame("Sputum", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_spt, method="prev", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)

contaminants <- rbind(contaminant_combined_bal, contaminant_combined_ns, contaminant_combined_spt)

names(contaminants) <- c("Sample type", "Taxa")


merged_contaminants <- merge(contaminants, prev_all %>% 
                                     rownames_to_column("Taxa"), by = "Taxa") %>%
        merge(., prev_neg %>%
                      rownames_to_column("Taxa"), by = "Taxa") %>%
       dplyr::select(c("Sample type",
                       "Taxa",
                       "Prevalence (all)",
                       "Prevalence (negative controls)")) %>%
        subset(., .$"Prevalence (all)" != 0) %>%
        mutate(`Prevalence (all)` = paste0(`Prevalence (all)`,
                                           " (",
                                           round(`Prevalence (all)`/
                                                         sample_number_decontam * 100, 2),
                                           "%)"),
                `Prevalence (negative controls)` = paste0(`Prevalence (negative controls)`,
                                           " (",
                                           round(`Prevalence (negative controls)`/
                                                         neg_number_decontam * 100, 2),
                                           "%)")) %>%
        rename(`Prevalence (all) N = 94` = `Prevalence (all)`,
               `Prevalence (negative controls) N = 31` = `Prevalence (negative controls)`) %>%
        .[order(.$"Sample type", .$"Taxa"),] %>%
        remove_rownames()


merged_contaminants



```

### Decontam (stratified by treatment)

As a sanity check, decontam analysis was conducted after stratifying by treatment. Only 3 SGBs were nominated as potential candidates.

```{r, warning=FALSE}
#Stratified by sample type

prev_neg <- ((v_phyloseq %>% 
          subset_samples(sample_type == "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (negative controls)" = ".")

prev_all <- ((v_phyloseq %>%
                      subset_samples(sample_type != "Mock") %>%
                      subset_samples(sample_type != "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (all)" = ".")


sample_data(v_phyloseq)$is.neg <- grepl("Neg", sample_data(v_phyloseq)$sample_type)

phyloseq_decontam_lypma <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "lyPMA")

phyloseq_decontam_ben <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "Benzonase")

phyloseq_decontam_hostzero <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "HostZero")

phyloseq_decontam_molysis <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "MolYsis")

phyloseq_decontam_qiaamp <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "QIAamp")


#lypma showed zero contaminant
#isContaminant(phyloseq_decontam_lypma, method="prevalence", neg = "is.neg", threshold = 0.1) %>% subset(.,.$contaminant) 

contaminant_combined_ben <- 
data.frame("Benzonase", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_ben, method="prevalence", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)

#Host zero showed zero contaminant
#isContaminant(phyloseq_decontam_hostzero, method="prevalence", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant)

contaminant_combined_molysis <- 
data.frame("MolYsis", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_molysis, method="prevalence", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)

contaminant_combined_QIAamp <- 
data.frame("QIAamp", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_qiaamp, method="prevalence", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)

contaminants_strat_treat <- rbind(#contaminant_combined_lypma,
                      contaminant_combined_ben, 
                      #contaminant_combined_hostzero,
                      contaminant_combined_molysis,
                      contaminant_combined_QIAamp
                      )

names(contaminants_strat_treat) <- c("Treatment", "Taxa")

contaminants_strat_treat

```

### {-}


**Prevalence & abundance filtering was not conducted due to much lower species richness**

# Question 1. How does the Marker-MAGu microbial community looks like?


## Overview of sequencing results {.tabset}

### Fig. Viral community of samples

Relative abundances at SGB level by sample type after employing prevalence and abundance filtering and Top 10 phage in each sample type by mean abundance. (A) BAL, (B) nasal swabs, (C) sputum, and (D) mock community. Empty space indicates samples showed no microbial reads, i.e., sequencing failed samples.


```{r, warning=FALSE, message=FALSE}

my_plot_bar = function (physeq, x = "Sample", y = "Abundance", fill = NULL, title = NULL, 
                        facet_grid = NULL) {
    mdf = psmelt(physeq)
    p = ggplot(mdf, aes_string(x = x, y = y, fill = fill))
    p = p + geom_bar(stat = "identity")
    p = p + theme(axis.text.x = element_text(angle = -90, hjust = 0)) +
            scale_x_discrete(drop = F)
    if (!is.null(facet_grid)) {
        p <- p + facet_grid(facet_grid)
    }
    if (!is.null(title)) {
        p <- p + ggtitle(title)
    }
    return(p)
}

a <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "Mock")) %>%
        cbind(species20 = "Other") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq_rel,
                              sample_type == "Mock")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "Mock") 
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        my_plot_bar(., fill="species20") + 
        xlab("Sample") +
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.text.x = element_text(color = "white")) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        labs(tag = "A")
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +
        


b <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "Neg.")) %>%
        cbind(species20 = "Other") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq,
                              sample_type == "Neg.")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "Neg.") 
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>% 
        my_plot_bar(., fill="species20") + 
        ylab("") +
        xlab("Subject") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.text.x = element_text(color = "white")) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        #facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        facet_wrap ( ~ treatment,
                    scales= "free_x", nrow=1) +
        ggtitle("E    Negative control")

c <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "BAL")) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq,
                              sample_type == "BAL")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "BAL")
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E")) %>%
           as.character()
   phyloseq_temp
  } %>% 
        my_plot_bar(., x = "subject_id", fill="species20") + 
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        #facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        facet_wrap ( ~ treatment,
                    scales= "free_x", nrow=1) +
        ggtitle("B    BAL (viral host genus)")


d <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "Nasal")) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq,
                              sample_type == "Nasal")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "Nasal")
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E", "F", "G", "H", "I", "J")) %>%
           as.character()
   phyloseq_temp
  } %>% 
        my_plot_bar(., x = "subject_id", fill="species20") + 
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
#        scale_x_discrete(drop=) +
        facet_wrap(~ treatment,  nrow = 1, drop = T, scales = "free_x") +
        ggtitle("D    Nasal swabs (viral host genus)")

e <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "Sputum")) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq,
                              sample_type == "Sputum")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "Sputum")
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E"))
   phyloseq_temp
  } %>%
        my_plot_bar(., x = "subject_id", fill="species20")+
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap(~ treatment, scales= "free_x", nrow = 1) +
        ggtitle("F    Sputum (viral host genus)")


fig2 <- ggarrange(c, c %>% lemon::g_legend() %>% as_ggplot,
                  d, d %>% lemon::g_legend() %>% as_ggplot,
                  e, e %>% lemon::g_legend() %>% as_ggplot,
                  #a, a %>% lemon::g_legend() %>% as_ggplot,
                  #b, b %>% lemon::g_legend() %>% as_ggplot,
                  ncol=2,nrow=3, widths = c(3, 1),
                  legend = "none",
                  align = "hv")

annotate_figure(fig2,
                left = text_grob("Viral relative abundance",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Subject",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11)
)



```


### Fig. Viral host of samples

Relative abundances at host Genus level by sample type after employing prevalence and abundance filtering and Top 10 phage in each sample type by mean abundance. (A) BAL, (B) nasal swabs, (C) sputum, and (D) mock community. Empty space indicates samples showed no microbial reads, i.e., sequencing failed samples.


```{r, warning=FALSE, message=FALSE}

#{taxa_names(v_h_phyloseq_rel) <-  tax_table(v_h_phyloseq_rel) %>% data.frame %>% .$Genus
#v_h_phyloseq_rel
#} %>%
v_h_phyloseq_rel_renamed <- v_h_phyloseq_rel
taxa_names(v_h_phyloseq_rel_renamed) <-  tax_table(v_h_phyloseq_rel) %>% data.frame %>% .$Genus

a_vh <- tax_table(subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "Mock")) %>%
        cbind(species20 = "Other") %>%
        {top20species <- head(taxa_sums(subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "Mock")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Genus"])
        .[, 7] <- ifelse(grepl("Other",.[, 7]),
                         "Other",
                         paste0("*", .[, 7], "*")
                         )
        phyloseq_temp <- subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "Mock") 
        tax_table(phyloseq_temp) <- tax_table(.) 
        phyloseq_temp
  } %>% 
        my_plot_bar(., fill="species20") + 
        xlab("Sample") +
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.text.x = element_text(color = "white")) +
        guides(fill=guide_legend(title="Top 10 viral host at genus level")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        labs(tag = "A")
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +
        


b_vh <- tax_table(subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "Neg.")) %>%
        cbind(species20 = "Other") %>%
        {top20species <- head(taxa_sums(subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "Neg.")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Genus"])
        .[, 7] <- ifelse(grepl("Other",.[, 7]),
                         "Other",
                         paste0("*", .[, 7], "*")
                         )
        phyloseq_temp <- subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "Neg.") 
        tax_table(phyloseq_temp) <- tax_table(.) 
        phyloseq_temp
  } %>% 
        my_plot_bar(., fill="species20") + 
        ylab("") +
        xlab("Subject") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.text.x = element_text(color = "white")) +
        guides(fill=guide_legend(title="Top 10 viral host at genus level", ncol = 2)) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        #facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        facet_wrap ( ~ treatment,
                    scales= "free_x", nrow=1) +
        ggtitle("E    Negative control")

c_vh <- tax_table(subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "BAL")) %>%
        cbind(species20 = "Other") %>%
        {top20species <- head(taxa_sums(subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "BAL")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Genus"])
        .[, 7] <- ifelse(grepl("Other",.[, 7]),
                         "Other",
                         paste0("*", .[, 7], "*")
                         )
        phyloseq_temp <- subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "BAL") 
        tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E")) %>%
           as.character()
   phyloseq_temp
  } %>% 
        my_plot_bar(., x = "subject_id", fill="species20") + 
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 viral host at genus level", ncol = 2)) +
        scale_fill_manual(values = c('#ccebc5',# Bacteroides
                                     "#80b1d3",# Centipeda
                                     "#fdb462", # Enterococcus
                                     '#fccde5', # Fusobacterium
                                     "#bc80bd", # Leptotrichia
                                     "#b3de69", # Paujensenia
                                     '#fb8072', # Prevotella
                                     "#08306b",# Staphyloccocus
                                     "#cb181d", # Streptococcus
                                     "#d9d9d9", #Tannelrella
                                     "darkgrey" #Others
                                     )) +
        #facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        facet_wrap ( ~ treatment,
                    scales= "free_x", nrow=1) +
        ggtitle("B    BAL (viral host genus)")


d_vh <- tax_table(subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "Nasal")) %>%
        cbind(species20 = "Other") %>%
        {top20species <- head(taxa_sums(subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "Nasal")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Genus"])
        .[, 7] <- ifelse(grepl("Other",.[, 7]),
                         "Other",
                         paste0("*", .[, 7], "*")
                         )
        phyloseq_temp <- subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "Nasal") 
        tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E", "F", "G", "H", "I", "J")) %>%
           as.character()
   phyloseq_temp
  } %>% 
        my_plot_bar(., x = "subject_id", fill="species20") + 
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 viral host at genus level", ncol = 2)) +
                scale_fill_manual(values = c('#8dd3c7',# Anaerococcus
                                     "#80b1d3",# Bacillus
                                     "#B2DF8A", # Corynebacterium
                                     '#6baed6', # Cutibacterium
                                     "#bebada", # Dolosigranulum
                                     "#ffed6f", # Escherichia
                                     '#fb8072', # Klebsiella
                                     "#8c96c6",# Lawsonella
                                     "#006d2c", # Pseudomonas
                                     "#54278f", #Stagphylococcus
                                     "darkgrey" #Others
                                     )) +
#        scale_x_discrete(drop=) +
        facet_wrap(~ treatment,  nrow = 1, drop = T, scales = "free_x") +
        ggtitle("D    Nasal swabs (viral host genus)")

e_vh <- tax_table(subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "Sputum")) %>%
        cbind(species20 = "Other") %>%
        {top20species <- head(taxa_sums(subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "Sputum")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Genus"])
        .[, 7] <- ifelse(grepl("Other",.[, 7]),
                         "Other",
                         paste0("*", .[, 7], "*")
                         )
        phyloseq_temp <- subset_samples(v_h_phyloseq_rel_renamed,
                              sample_type == "Sputum") 
        tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E"))
   phyloseq_temp
  } %>%
        my_plot_bar(., x = "subject_id", fill="species20")+
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 viral host at genus level", ncol = 2)) +
                        scale_fill_manual(values = c('#B2DF8A',# Actinomyces
                                     "#fdb462", # Escherichia
                                     "#8dd3c7",# Gemella
                                     "#b3de69", # Granulicatella
                                     '#80b1d3', # Pauljensenia
                                     "#ffffb3", #Pseudomonas
                                     "#bebada", # Rothia
                                     "#fb8072", #Staphylococcus	
                                     "#08519c", #Streptococcus	
                                     "#fccde5", #UBA7748
                                     "darkgrey" #Others
                                     )) +
        facet_wrap(~ treatment, scales= "free_x", nrow = 1) +
        ggtitle("F    Sputum (viral host genus)")


fig2_vh <- ggarrange(c_vh, c_vh %>% lemon::g_legend() %>% as_ggplot,
                  d_vh, d_vh %>% lemon::g_legend() %>% as_ggplot,
                  e_vh, e_vh %>% lemon::g_legend() %>% as_ggplot,
                  #a, a %>% lemon::g_legend() %>% as_ggplot,
                  #b, b %>% lemon::g_legend() %>% as_ggplot,
                  ncol=2,nrow=3, widths = c(3, 1),
                  legend = "none",
                  align = "hv")

annotate_figure(fig2_vh,
                left = text_grob("Relative abundance of viral host at genus level",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Subject",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11)
) 
        


```

### Fig. Viral community of Zymo Mock

Lower species richness was observed, after treatment. 


```{r}

barplot_mock_viral <- a


barplot_mock_viral

```



```{r}

barplot_mock_viral_host <- a_vh


barplot_mock_viral_host

```

### {-}



# Question 2. Does phage community changed the same as microbial community, by sample type and treatment?

## Taxa change {.tabset}

### Fig. Phage species richness


```{r, warning=FALSE, message=FALSE}

f3a_v <- ggplot(subset(sample_data(v_phyloseq) %>% 
                             data.frame, sample_data(v_phyloseq)$sample_type %in% c(#"Mock",
                                     "Sputum", "Nasal", "BAL")), aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment),
                    position = position_jitter(0.2), size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        #stat_summary(fun = mean,
        #       geom = "pointrange",
        #       fun.max = function(x) mean(x) + sd(x),
        #       fun.min = function(x) mean(x) - sd(x))+
        ylab("Viral species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 

        ggtitle("B   Viral alpha diversity") +

        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 


dat_text <- data.frame(
  label = c(
          #"", "***", "***", "**", "***", #label for Mock
          "", "", "", "*", "", #label for BAL
          "*", "", ".", "", "", 
          "", "", "***", "***", "*"),
  sample_type = c(
          #"Mock", "Mock", "Mock", "Mock", "Mock", 
          "BAL", "BAL", "BAL", "BAL", "BAL", 
          "Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          "Sputum", "Sputum", "Sputum", "Sputum", "Sputum"),
  treatment     = c(
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
  S.obs = c(
          #50, 30, 35, 33, 31,
          2, 2, 2, 60, 2,
          10, 1, 35, 1, 1,
          1, 1, 180, 170, 140)
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f3a_v <- f3a_v + geom_text(
  data    = dat_text,
  mapping = aes(x = treatment, y = S.obs, label = label)
)

f3a_v


```

### Fig. Viral host genus richness


```{r, warning=FALSE, message=FALSE}

f3a_host <- ggplot(subset(sample_data(v_h_phyloseq_rel) %>% 
                             data.frame, sample_data(v_h_phyloseq_rel)$sample_type %in% c(#"Mock",
                                     "Sputum", "Nasal", "BAL")), aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment),
                    position = position_jitter(0.2), size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        #stat_summary(fun = mean,
        #       geom = "pointrange",
        #       fun.max = function(x) mean(x) + sd(x),
        #       fun.min = function(x) mean(x) - sd(x))+
        ylab("Viral host genus richness") +
        xlab("Treatment group") +
        labs(tag = "B") +
        ggtitle("Alpha diveristy of viral host genus") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 

dat_text <- data.frame(
  label = c(
          #"", "***", "***", "**", "***", #label for Mock
          "", "", "", "*", "", #label for BAL
          "*", "", ".", "", "", 
          "", "", "***", "***", "*"),
  sample_type = c(
          #"Mock", "Mock", "Mock", "Mock", "Mock", 
          "BAL", "BAL", "BAL", "BAL", "BAL", 
          "Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          "Sputum", "Sputum", "Sputum", "Sputum", "Sputum"),
  treatment     = c(
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
  S.obs = c(
          #50, 30, 35, 33, 31,
          2, 2, 2, 12, 2,
          5, 1, 7, 1, 1,
          1, 1, 25, 24, 20)
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f3a_host <- f3a_host + geom_text(
  data    = dat_text,
  mapping = aes(x = treatment, y = S.obs, label = label)
)


f3a_host


```
### Fig. Microbial species richness

*Microbial species richness did not increase at nasal swabs.*

```{r, warning=FALSE, message=FALSE}

f3a <- ggplot(subset(sample_data(m_phyloseq) %>% 
                             data.frame, sample_data(m_phyloseq)$sample_type %in% c(#"Mock",
                                     "Sputum", "Nasal", "BAL")), aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment),
                    position = position_jitter(0.2), size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        #stat_summary(fun = mean,
        #       geom = "pointrange",
        #       fun.max = function(x) mean(x) + sd(x),
        #       fun.min = function(x) mean(x) - sd(x))+
        ylab("Microbial species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        labs(tag = "A") +
        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 
f3a

```

### Viral species richness (stratified, suppl) 

**This table is too redundant. Removed from the manuscript.**. 

Effect size, standard error (SE) and p-value of a statistical test for species richness using linear mixed effect model stratified by sample type (Species richness ~ treatment + (1|subject_id) ). Stratified analyses were conducted for each sample type as an interaction term of sample type and treatment was significant at an ANOVA test (p-value < 0.001) using a model, lmer(species richness ~ sample type + treatment + sample type * treatment  + (1|subject_id)). Statistical significances were noted with `*: p-value < 0.05, **: p-value < 0.01, and ***: p-value < 0.001.`

LMER raw result - Mock

```{r}

sr_lmer_mock <- lm(S.obs ~ treatment,
                    data = v_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Mock") & S.obs != 0))

sr_lmer_mock %>% summary() 

```

LMER raw result - BAL

```{r}

sr_lmer_bal <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = v_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("BAL")))
sr_lmer_bal %>% summary()
        
```

LMER raw result - Nasal

```{r}
sr_lmer_ns <- lmer(S.obs ~ treatment + (1|subject_id),
                   data = v_sample_data %>% 
                           data.frame %>% 
                           subset(., .$sample_type %in% c("Nasal")))
sr_lmer_ns %>% summary()
```

LMER raw result - Sputum

```{r}
sr_lmer_spt <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = v_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Sputum")))
sr_lmer_spt %>% summary()
        
sr_lmer_spt %>% confint()

```

### Viral species richness (stratified) 

Tidy summarized table of LME tests stratified by sample type.

*The results shows some treatments significantly increased the species richess of phage, but the effect become smaller with nasal swab samples.*

```{r}

sr_lmer_mock_kbl <-  sr_lmer_mock %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_mock %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

sr_lmer_bal_kbl <-  sr_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        
sr_lmer_ns_kbl <-  sr_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               abs(`Pr(>|t|)`) < 0.1 ~ ".",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


sr_lmer_spt_kbl <-  sr_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

tables3 <- cbind(#sr_lmer_mock_kbl,
                 sr_lmer_bal_kbl,
                 sr_lmer_ns_kbl,
                 sr_lmer_spt_kbl) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1,
                           #"Mock" = 3,
                           "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")

tables3

#save_kable(tables3, file = "/Users/minsikkim/Dropbox (Personal)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS3.html", self_contained = T)

```


### Viral host genus richness

**This table is too redundant. Removed from the manuscript.**. 

Effect size, standard error (SE) and p-value of a statistical test for viral genus richness using linear mixed effect model stratified by sample type (Species richness ~ treatment + (1|subject_id) ). Stratified analyses were conducted for each sample type as an interaction term of sample type and treatment was significant at an ANOVA test (p-value < 0.001) using a model, lmer(species richness ~ sample type + treatment + sample type * treatment  + (1|subject_id)). Statistical significances were noted with `*: p-value < 0.05, **: p-value < 0.01, and ***: p-value < 0.001.`

LMER raw result - Mock

```{r}

sr_lmer_mock <- lm(S.obs ~ treatment,
                    data = v_h_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Mock") & S.obs != 0))

sr_lmer_mock %>% summary() 

```

LMER raw result - BAL

```{r}

sr_lmer_bal <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = v_h_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("BAL")))
sr_lmer_bal %>% summary()
        
```

LMER raw result - Nasal

```{r}

sr_lmer_ns <- lmer(S.obs ~ treatment + (1|subject_id),
                   data = v_h_sample_data %>% 
                           data.frame %>% 
                           subset(., .$sample_type %in% c("Nasal")))
sr_lmer_ns %>% summary()
```

LMER raw result - Sputum

```{r}
sr_lmer_spt <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = v_h_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Sputum")))
sr_lmer_spt %>% summary()
        
sr_lmer_spt %>% confint()

```

### Viral host genus richness (stratified) 

Tidy summarized table of LME tests stratified by sample type.

*The results shows some treatments significantly increased the host of viruses at genus level, but the effect become smaller with nasal swab samples.*

```{r}

sr_lmer_mock_kbl <-  sr_lmer_mock %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_mock %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

sr_lmer_bal_kbl <-  sr_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        
sr_lmer_ns_kbl <-  sr_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               abs(`Pr(>|t|)`) < 0.1 ~ ".",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


sr_lmer_spt_kbl <-  sr_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

tables3 <- cbind(#sr_lmer_mock_kbl,
                 sr_lmer_bal_kbl,
                 sr_lmer_ns_kbl,
                 sr_lmer_spt_kbl) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1,
                           #"Mock" = 3,
                           "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")

tables3

#save_kable(tables3, file = "/Users/minsikkim/Dropbox (Personal)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS3.html", self_contained = T)

```



### Species richness (stratified, suppl) 

**This table is too redundant. Removed from the manuscript.**. 

Effect size, standard error (SE) and p-value of a statistical test for species richness using linear mixed effect model stratified by sample type (Species richness ~ treatment + (1|subject_id) ). Stratified analyses were conducted for each sample type as an interaction term of sample type and treatment was significant at an ANOVA test (p-value < 0.001) using a model, lmer(species richness ~ sample type + treatment + sample type * treatment  + (1|subject_id)). Statistical significances were noted with `*: p-value < 0.05, **: p-value < 0.01, and ***: p-value < 0.001.`

LMER raw result - Mock

```{r}

sr_lmer_mock <- lm(S.obs ~ treatment,
                    data = m_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Mock") & S.obs != 0))

sr_lmer_mock %>% summary() 

```

LMER raw result - BAL

```{r}

sr_lmer_bal <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = m_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("BAL")))
sr_lmer_bal %>% summary()
        
```

LMER raw result - Nasal

```{r}

sr_lmer_ns <- lmer(S.obs ~ treatment + (1|subject_id),
                   data = m_sample_data %>% 
                           data.frame %>% 
                           subset(., .$sample_type %in% c("Nasal")))
sr_lmer_ns %>% summary()
```

LMER raw result - Sputum

```{r}
sr_lmer_spt <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = m_sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Sputum")))
sr_lmer_spt %>% summary()
        
sr_lmer_spt %>% confint()

```

### Bacterial species richness (stratified) 

Tidy summarized table of LME tests stratified by sample type.

*The results shows some treatments significantly increased the species richess of phage, but the effect become smaller with nasal swab samples.*

```{r}

sr_lmer_mock_kbl <-  sr_lmer_mock %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_mock %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

sr_lmer_bal_kbl <-  sr_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        
sr_lmer_ns_kbl <-  sr_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               abs(`Pr(>|t|)`) < 0.1 ~ ".",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


sr_lmer_spt_kbl <-  sr_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

tables3 <- cbind(#sr_lmer_mock_kbl,
                 sr_lmer_bal_kbl,
                 sr_lmer_ns_kbl,
                 sr_lmer_spt_kbl) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1,
                           #"Mock" = 3,
                           "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")

tables3

#save_kable(tables3, file = "/Users/minsikkim/Dropbox (Personal)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS3.html", self_contained = T)

```

### Beta diversity


```{r, warning=FALSE, message=FALSE}

#Making subset of non-zero samples without neg

phyloseq_rel_nz <- v_phyloseq_rel %>% 
        subset_samples(S.obs != 0 & sample_type %in% c("Mock", 
                                                       "BAL", "Nasal", "Sputum"))

#distances of betadiversity - boxplots
horn_dist_long <- distance(phyloseq_rel_nz, method="horn") %>% as.matrix() %>% melt_dist() #making long data of distance matrices

#Adding sample type and treatment name. 
#this can be also done by merging metadata into the `horn_dist_long`
names <- data.frame(str_split_fixed(horn_dist_long$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(horn_dist_long$iso2, "_", 3))
horn_dist_long$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
horn_dist_long$method_1 <- ifelse(grepl("lyPMA", horn_dist_long$iso1),"lypma", 
                                         ifelse(grepl("ben", horn_dist_long$iso1),"benzonase", 
                                                ifelse(grepl("host", horn_dist_long$iso1),"host_zero", 
                                                       ifelse(grepl("qia", horn_dist_long$iso1),"qiaamp", 
                                                              ifelse(grepl("moly", horn_dist_long$iso1),"molysis", 
                                                                     "control")))))


#Adding data for iso 2 also should be done
horn_dist_long$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")
horn_dist_long$method_2 <-ifelse(grepl("lyPMA", horn_dist_long$iso2),"lypma", 
                                        ifelse(grepl("ben", horn_dist_long$iso2),"benzonase", 
                                               ifelse(grepl("host", horn_dist_long$iso2),"host_zero", 
                                                      ifelse(grepl("qia", horn_dist_long$iso2),"qiaamp", 
                                                             ifelse(grepl("moly", horn_dist_long$iso2),"molysis", 
                                                                    "control")))))


#subsetting distances of my interest
horn_dist_long$sample_id_1 <- ifelse(grepl("pos", horn_dist_long$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_1, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_1))
horn_dist_long$sample_id_2 <- ifelse(grepl("pos", horn_dist_long$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_2, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_2))


horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long, horn_dist_long$sample_id_1 == horn_dist_long$sample_id_2) # data within samples

horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long_within_sampleid_from_control,
                                                           horn_dist_long_within_sampleid_from_control$method_1 != horn_dist_long_within_sampleid_from_control$method_2) # remove irrelevant association

horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long_within_sampleid_from_control, (horn_dist_long_within_sampleid_from_control$method_1 == "control") + (horn_dist_long_within_sampleid_from_control$method_2 == "control") != 0)


horn_dist_long_within_sampleid_from_control$treatment <- horn_dist_long_within_sampleid_from_control$method_1

horn_dist_long_within_sampleid_from_control$treatment <- ifelse(horn_dist_long_within_sampleid_from_control$treatment == "control", horn_dist_long_within_sampleid_from_control$method_2, horn_dist_long_within_sampleid_from_control$treatment) 


#Setting key method
horn_dist_long_within_sampleid_from_control$sample_type <- ifelse(grepl("NS", horn_dist_long_within_sampleid_from_control$iso1), "Nasal",
                                                                  ifelse(grepl("CFB", horn_dist_long_within_sampleid_from_control$iso1), "Sputum",
                                                                         ifelse(grepl("BAL", horn_dist_long_within_sampleid_from_control$iso1), "BAL",
                                                                                ifelse(grepl("pos|POS", horn_dist_long_within_sampleid_from_control$iso1, ignore.case = T), "Mock",
                                                                                       ifelse(grepl("neg|N_EXT", horn_dist_long_within_sampleid_from_control$iso1), "Neg.",NA)))))

#Making a column for baseline (controls, from where?)
horn_dist_long_within_sampleid_from_control <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(dist_from = case_when(method_1 == "control" ~ iso1,
                                     method_2 == "control" ~ iso2))

dummy <- data.frame(iso1 = horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           iso2 = horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           dist = 0,
           treatment = "Untreated",
           method_1 = "control",
           method_2 = "control"
           )
names <- data.frame(str_split_fixed(dummy$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(dummy$iso2, "_", 3))
dummy$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
#Adding data for iso 2 also should be done
dummy$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")


#subsetting distances of my interest
dummy$sample_id_1 <- ifelse(grepl("pos", dummy$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_1, ignore.case = T),"Neg.",
                                        dummy$sample_id_1))
dummy$sample_id_2 <- ifelse(grepl("pos", dummy$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_2, ignore.case = T),"Neg.",
                                        dummy$sample_id_2))
dummy$sample_type <- ifelse(grepl("NS", dummy$iso1), "Nasal",
                            ifelse(grepl("CFB", dummy$iso1), "Sputum",
                                   ifelse(grepl("BAL", dummy$iso1), "BAL",
                                          ifelse(grepl("pos|POS", dummy$iso1, ignore.case = T), "Mock",
                                                 ifelse(grepl("neg|N_EXT", dummy$iso1), "Neg.",NA)))))
dummy <- subset(dummy, !is.na(dummy$sample_type))
horn_dist_long_within_sampleid_from_control <- bind_rows(horn_dist_long_within_sampleid_from_control, dummy)

#Here, sample id is the same as subject id.
horn_dist_long_within_sampleid_from_control$subject_id <- horn_dist_long_within_sampleid_from_control$sample_id_1

horn_dist_long_within_sampleid_from_control$treatment <-
        factor(horn_dist_long_within_sampleid_from_control$treatment,
               levels = c("Untreated", "lypma", "benzonase", "host_zero", "molysis", "qiaamp"))

#Making figure of beta diversity distances
lmer(dist ~ treatment + (1|subject_id),
                data = horn_dist_long_within_sampleid_from_control %>%
                        subset(sample_type == "Sputum")) %>%  summary

#Making figure of beta diversity distances
lmer(dist ~ treatment + (1|subject_id),
                data = horn_dist_long_within_sampleid_from_control %>%
                        subset(sample_type == "Nasal")) %>%  summary


f3b2_box_v <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(across(sample_type, factor, levels=c(#"Mock", 
                                                    "BAL", "Nasal","Sputum"
                                                    ))) %>%
        subset(., .$sample_type != "Neg.") %>% 
        subset(., .$treatment != "Untreated") %>%
  mutate(treatment = factor(treatment, levels = c("lypma", "benzonase", "host_zero", "molysis", "qiaamp"),
                            labels = c("lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))) %>%
        ggplot(aes(x = dist, y = treatment, col = treatment)) +
        geom_boxplot(aes(x=dist)) +
        #geom_linerange(aes(xmin=`2.5%`, xmax=`97.5%`)) +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c(#"#e31a1c",
                                      "#fb9a99","#33a02c",
                                      "#b2df8a","#1f78b4","#a6cee3"),
                           name = "Treatment",
                           breaks = c(#"Untreated", 
                                      "lyPMA", "Benzonase",
                                      "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Morisita-Horn dissimilarity from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              legend.position = "none") +
        ggtitle("D   Viral beta diversity") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "sans") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "sans") +
        scale_x_continuous(breaks = c(-0.25, 0, 0.25, 0.5, 0.75),
                           labels = c(-0.25, "0 (low bias)", 0.25, 0.5, "0.75 (high bias)"))


dat_text <- data.frame(
  label = c(
          #"", "***", "***", "**", "***", #label for Mock
          "", "", "", "", "", #label for BAL
          "", "***", "", ".", "**", 
          "", ".", "**", "***", "**"),
  sample_type = c(
          #"Mock", "Mock", "Mock", "Mock", "Mock", 
          "BAL", "BAL", "BAL", "BAL", "BAL", 
          "Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          "Sputum", "Sputum", "Sputum", "Sputum", "Sputum"),
  treatment     = c(
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
  dist = c(
          #50, 30, 35, 33, 31,
          0.9, 0, 0, 0, 0,
          0.6, 0.4, 0, 0.73, 0.97,
          0.8, 0.65, 1, 1.01, 0.99)
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f3b2_box_v <- f3b2_box_v + geom_text(
  data    = dat_text,
  mapping = aes(y = treatment, x = dist, label = label),
  col = "black"
)


f3b2_box_v


```


### {-}

# DA analsyis

## DA viral species {.tabset}

### Factors affecting DA taxa (main text)

```{r, warning=FALSE, message=FALSE}
setwd("Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git")

#filt_maaslin_all <- read.csv("/Volumes/macdrive/Dropbox/Git/Host_depletion_macmini/data/mrkrmg_da_lmer_filt_maaslin_all.csv")
#filt_fit_data_bal <- read.csv("data/mrkrmg_da_lmer_filt_fit_data_bal.csv")
#filt_fit_data_spt <- read.csv("data/mrkrmg_da_lmer_filt_fit_data_spt.csv")
#filt_fit_data_ns <- read.csv("data/mrkrmg_da_lmer_filt_fit_data_ns.csv")




filt_maaslin_all_v <- read.csv("/Volumes/macdrive/Dropbox/Git/Host_depletion_macmini/data/mrkrmg_da_lmer_filt_maaslin_all.csv")
#filt_maaslin_interaction <- read.csv("data/da_lmer_filt_maaslin_interaction.csv")
filt_fit_data_bal_v <- read.csv("/Volumes/macdrive/Dropbox/Git/Host_depletion_macmini/data/mrkrmg_da_lmer_filt_fit_data_bal.csv")
filt_fit_data_spt_v <- read.csv("/Volumes/macdrive/Dropbox/Git/Host_depletion_macmini/data/mrkrmg_da_lmer_filt_fit_data_spt.csv")
filt_fit_data_ns_v <- read.csv("/Volumes/macdrive/Dropbox/Git/Host_depletion_macmini/data/mrkrmg_da_lmer_filt_fit_data_ns.csv")
#filt_fit_data_pos <- read.csv("data/da_lmer_filt_fit_data_pos.csv")

```

### MaAsLin output (raw data)

```{r}
filt_maaslin_all
```

### MaAsLin output (stratified)

BAL stratified MaAsLin result

```{r}
filt_fit_data_bal
```

Nasal stratified MaAsLin result

```{r}
filt_fit_data_ns
```

Sputum stratified MaAsLin result

```{r}
filt_fit_data_spt
```

### Number of taxa on some conditions

These calculations were made to write the details in the main text

```{r}

cat("Factors affecting DA taxa (q<0.1)")
filt_maaslin_all_v %>% subset(., .$qval < 0.1 ) %>% subset(., abs(.$Estimate) > 1) %>% .$metadata %>% table

cat("Number of positive & significant changes")
filt_maaslin_all_v %>% subset(., .$qval < 0.1 ) %>% subset(., .$Estimate > 0) %>% .$feature %>% unique

cat("Number of negative & significant changes")
filt_maaslin_all_v %>% subset(., .$qval < 0.1 ) %>% subset(., .$Estimate < 0) %>% .$feature %>% unique

cat("BAL highly changed taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_bal_v %>% subset(., .$qval < 0.1 ) %>% subset(., abs(.$Estimate) >1 ) %>% .$metadata %>% table

cat("Sputum highly changed taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_spt_v %>% subset(., .$qval < 0.1 ) %>% subset(., abs(.$Estimate) >1 ) %>% .$metadata %>% table

cat("NS highly changed taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_ns_v %>% subset(., .$qval < 0.1 ) %>% subset(., abs(.$Estimate) >1 ) %>% .$metadata %>% table



cat("NS taxa (q<0.1)")
filt_fit_data_bal_v %>% subset(., .$qval < 0.1 ) %>% .$feature %>% unique

cat("NS taxa (q<0.1)")
filt_fit_data_ns_v %>% subset(., .$qval < 0.1 ) %>% .$feature %>% unique

cat("Sputum taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_spt_v %>% subset(., .$qval < 0.1 ) %>% .$feature %>% unique


cat("Sputum taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_spt_v %>% subset(., .$qval < 0.1 ) %>% .$metadata %>% table




```

### Fig. S8. Volcano plot


**Fig. S8.** Volcano plot of differential abundance of microbes by each treatment with a model MaAsLin (relative abundance of each taxa ~ sample type + lyPMA + Benzonase + HostZERO + MolYsis + QIAamp, random effect = subject id).

```{r warning=FALSE, message=FALSE}

#Making significance table for figure
        # Define a function to make species names italicized
# Make a significance table for each figure (top 20 taxa)
species_italic <- function(data) {
  names <- gsub("_", " ", rownames(data))
  names <- gsub("[]]|[[]", "", names)
  names <- gsub(" sp", " sp.", names)
  names <- gsub(" sp.", "* sp.", names)
  names <- gsub(" group", "", names)
  names <- ifelse(grepl("[*]", names), paste("*", names, sep = ""), paste("*", names, "*", sep = ""))
  rownames(data) <- names
  data
}
species_revise <- function(data) {
  data$feature <- gsub("Saccharomyces_cerevisiae_x_Saccharomyces_kudriavzevii", "S._cerevisiae x S._kudriavzevii", data$feature)
  data$feature <- gsub("Pseudomonas_aeruginosa_group", "Pseudomonas_aeruginosa", data$feature)
  data$feature <- gsub("Pseudomonas_fluorescens_group", "Pseudomonas_fluorescens", data$feature)
  data
}



make_sig_table <- function(data) {
  sig_data <- spread(data[order(data$qval), c("feature", "metadata", "qval")], metadata, qval)
  sig_data <- species_revise(sig_data)
  sig_data$min <- apply(sig_data %>% dplyr::select(c("lypma", "benzonase", "molysis", "host_zero", "qiaamp")), 1, FUN = min)
  sig_data <- sig_data[order(sig_data$min),] %>% dplyr::select("feature", "lypma", "benzonase", "host_zero", "molysis", "qiaamp") %>% .[1:20,]
  sig_data[["feature"]] <- ifelse(sig_data[["feature"]] == "X.Collinsella._massiliensis", "[Collinsella]_massiliensis", sig_data[["feature"]])
  sig_data_italic <- sig_data %>% rownames_to_column(var = "-") %>%
          column_to_rownames(var = "feature") %>% species_italic %>% dplyr::select(-c("-")) %>%
          rename(lyPMA = lypma,  Benzonase = benzonase, `HostZERO` = host_zero, MolYsis = molysis, QIAamp = qiaamp)
  sig_data_sig <- ifelse(sig_data_italic < 0.1, "*", NA) %>% data.frame(check.names = F)
  return(list(data = sig_data, data_italic = sig_data_italic, data_sig = sig_data_sig))
}





make_sig_table <- function(data) {
  sig_data <- spread(data[order(data$qval), c("feature", "metadata", "qval")], metadata, qval)
  sig_data <- species_revise(sig_data)
  sig_data$min <- apply(sig_data %>% dplyr::select(c("lypma", "benzonase", "molysis", "host_zero", "qiaamp")), 1, FUN = min)
  sig_data <- sig_data[order(sig_data$min),] %>% dplyr::select("feature", "lypma", "benzonase", "host_zero", "molysis", "qiaamp") %>% .[1:20,]
  sig_data[["feature"]] <- ifelse(sig_data[["feature"]] == "X.Collinsella._massiliensis", "[Collinsella]_massiliensis", sig_data[["feature"]])
  sig_data_italic <- sig_data %>% rownames_to_column(var = "-") %>%
          column_to_rownames(var = "feature") %>% species_italic %>% dplyr::select(-c("-")) %>%
          rename(lyPMA = lypma,  Benzonase = benzonase, `HostZERO` = host_zero, MolYsis = molysis, QIAamp = qiaamp)
  sig_data_sig <- ifelse(sig_data_italic < 0.1, "*", NA) %>% data.frame(check.names = F)
  return(list(data = sig_data, data_italic = sig_data_italic, data_sig = sig_data_sig))
}




make_sig_table18 <- function(data) {
  sig_data <- spread(data[order(data$qval), c("feature", "metadata", "qval")], metadata, qval)
  sig_data <- species_revise(sig_data)
  sig_data$min <- apply(sig_data %>% dplyr::select(c("lypma", "benzonase", "molysis", "host_zero", "qiaamp")), 1, FUN = min)
  sig_data <- sig_data[order(sig_data$min),] %>% dplyr::select("feature", "lypma", "benzonase", "host_zero", "molysis", "qiaamp") %>% .[1:18,]
  sig_data[["feature"]] <- ifelse(sig_data[["feature"]] == "X.Collinsella._massiliensis", "[Collinsella]_massiliensis", sig_data[["feature"]])
  sig_data_italic <- sig_data %>% rownames_to_column(var = "-") %>%
          column_to_rownames(var = "feature") %>% species_italic %>% dplyr::select(-c("-")) %>%
          rename(lyPMA = lypma,  Benzonase = benzonase, `HostZERO` = host_zero, MolYsis = molysis, QIAamp = qiaamp)
  sig_data_sig <- ifelse(sig_data_italic < 0.1, "*", NA) %>% data.frame(check.names = F)
  return(list(data = sig_data, data_italic = sig_data_italic, data_sig = sig_data_sig))
}



make_sig_table8 <- function(data) {
  sig_data <- spread(data[order(data$qval), c("feature", "metadata", "qval")], metadata, qval)
  sig_data <- species_revise(sig_data)
  sig_data$min <- apply(sig_data %>% dplyr::select(c("lypma", "benzonase", "molysis", "host_zero", "qiaamp")), 1, FUN = min)
  sig_data <- sig_data[order(sig_data$min),] %>% dplyr::select("feature", "lypma", "benzonase", "host_zero", "molysis", "qiaamp") %>% .[1:8,]
  sig_data[["feature"]] <- ifelse(sig_data[["feature"]] == "X.Collinsella._massiliensis", "[Collinsella]_massiliensis", sig_data[["feature"]])
  sig_data_italic <- sig_data %>% rownames_to_column(var = "-") %>%
          column_to_rownames(var = "feature") %>% species_italic %>% dplyr::select(-c("-")) %>%
          rename(lyPMA = lypma,  Benzonase = benzonase, `HostZERO` = host_zero, MolYsis = molysis, QIAamp = qiaamp)
  sig_data_sig <- ifelse(sig_data_italic < 0.1, "*", NA) %>% data.frame(check.names = F)
  return(list(data = sig_data, data_italic = sig_data_italic, data_sig = sig_data_sig))
}



#filt_fit_data_pos <- make_sig_table(filt_fit_data_pos)
filt_fit_data_bal <- make_sig_table(filt_fit_data_bal_v)
filt_fit_data_ns <- make_sig_table(filt_fit_data_ns_v)
filt_fit_data_spt <- make_sig_table(filt_fit_data_spt_v)



```


```{r}

#Volcano plot

figS7 <- ggplot(filt_maaslin_all_v, aes(y = -log10(qval), x = Estimate, col = metadata)) +
        theme_classic(base_family = "sans") +
        #labs(tag = "A") +
        geom_point(size = 2, alpha = 0.3, stroke = 0) +
        xlab("Change estimate of CLR-normalized read counts") +
        ylab("-log<sub>10</sub>(*q*-value)") +
        #ylim(c(-1, 35)) +
        geom_hline(yintercept = 1, col = "gray") +
        geom_vline(xintercept = 0, col = "gray") +
        annotate(family = "sans",
                 geom='richtext',
                 x=0, y=2,
                 label = "<i>q</i>-value = 0.1, fold-change = 0") +
        theme(legend.position = "top", axis.title.y = ggtext::element_markdown(), legend.text = element_markdown()) +
        scale_color_manual(values = c("#a65628",
                                      "grey",
                                      "#fb9a99",
                                      "#33a02c",
                                      "#b2df8a",
                                      "#1f78b4",
                                      "#a6cee3"),
                           breaks = c("log10.Final_reads",
                                      "sample_type",
                                      "lypma",
                                      "benzonase",
                                      "host_zero",
                                      "molysis",
                                      "qiaamp"), 
                           labels = c("log<sub>10</sub>(Final reads)",
                                      "Sample type",
                                      "lyPMA",
                                      "Benzonase",
                                      "HostZERO",
                                      "MolYsis",
                                      "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        guides(col = guide_legend(title = "Factors", title.position = "top", nrow = 1))

figS7

```


## DA viral host at genus level {.tabset}

### Factors affecting DA taxa (main text)

```{r, warning=FALSE, message=FALSE}
setwd("Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git")

#filt_maaslin_all <- read.csv("data/mrkrmg_host_da_lmer_filt_maaslin_all.csv")
#filt_fit_data_bal <- read.csv("data/mrkrmg_host_da_lmer_filt_fit_data_bal.csv")
#filt_fit_data_spt <- read.csv("data/mrkrmg_host_da_lmer_filt_fit_data_spt.csv")
#filt_fit_data_ns <- read.csv("data/mrkrmg_host_da_lmer_filt_fit_data_ns.csv")




filt_maaslin_all_v_h <- read.csv("/Volumes/macdrive/Dropbox/Git/Host_depletion_macmini/data/mrkrmg_host_da_lmer_filt_maaslin_all.csv")
#filt_maaslin_interaction <- read.csv("data/da_lmer_filt_maaslin_interaction.csv")
filt_fit_data_bal_v_h <- read.csv("/Volumes/macdrive/Dropbox/Git/Host_depletion_macmini/data/mrkrmg_host_da_lmer_filt_fit_data_bal.csv")
filt_fit_data_spt_v_h <- read.csv("/Volumes/macdrive/Dropbox/Git/Host_depletion_macmini/data/mrkrmg_host_da_lmer_filt_fit_data_spt.csv")
filt_fit_data_ns_v_h <- read.csv("/Volumes/macdrive/Dropbox/Git/Host_depletion_macmini/data/mrkrmg_host_da_lmer_filt_fit_data_ns.csv")
#filt_fit_data_pos <- read.csv("data/da_lmer_filt_fit_data_pos.csv")

```

### MaAsLin output (raw data)

```{r}
filt_maaslin_all_v_h
```

### MaAsLin output (stratified)

BAL stratified MaAsLin result

```{r}
filt_fit_data_bal_v_h
```

Nasal stratified MaAsLin result

```{r}
filt_fit_data_ns_v_h
```

Sputum stratified MaAsLin result

```{r}
filt_fit_data_spt_v_h
```

### Number of taxa on some conditions

These calculations were made to write the details in the main text

```{r}

cat("Factors affecting DA taxa (q<0.1)")
filt_maaslin_all_v_h %>% subset(., .$qval < 0.1 ) %>% subset(., abs(.$Estimate) > 1) %>% .$metadata %>% table

cat("Number of positive & significant changes")
filt_maaslin_all_v_h %>% subset(., .$qval < 0.1 ) %>% subset(., .$Estimate > 0) %>% .$feature %>% unique

cat("Number of negative & significant changes")
filt_maaslin_all_v_h %>% subset(., .$qval < 0.1 ) %>% subset(., .$Estimate < 0) %>% .$feature %>% unique

cat("BAL highly changed taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_bal_v_h %>% subset(., .$qval < 0.1 ) %>% subset(., abs(.$Estimate) >1 ) %>% .$metadata %>% table

cat("Sputum highly changed taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_spt_v_h %>% subset(., .$qval < 0.1 ) %>% subset(., abs(.$Estimate) >1 ) %>% .$metadata %>% table

cat("NS highly changed taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_ns_v_h %>% subset(., .$qval < 0.1 ) %>% subset(., abs(.$Estimate) >1 ) %>% .$metadata %>% table



cat("NS taxa (q<0.1)")
filt_fit_data_bal_v_h %>% subset(., .$qval < 0.1 ) %>% .$feature %>% unique

cat("NS taxa (q<0.1)")
filt_fit_data_ns_v_h %>% subset(., .$qval < 0.1 ) %>% .$feature %>% unique

cat("Sputum taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_spt_v_h %>% subset(., .$qval < 0.1 ) %>% .$feature %>% unique


cat("Sputum taxa (q<0.1 & abs(.$coef) >1)")
filt_fit_data_spt_v_h %>% subset(., .$qval < 0.1 ) %>% .$metadata %>% table




```

### Fig. S8. Volcano plot


**Fig. S8.** Volcano plot of differential abundance of microbes by each treatment with a model MaAsLin (relative abundance of each taxa ~ sample type + lyPMA + Benzonase + HostZERO + MolYsis + QIAamp, random effect = subject id).

```{r warning=FALSE, message=FALSE}

#Making significance table for figure
        # Define a function to make species names italicized
# Make a significance table for each figure (top 20 taxa)
species_italic <- function(data) {
  names <- gsub("_", " ", rownames(data))
  names <- gsub("[]]|[[]", "", names)
  names <- gsub(" sp", " sp.", names)
  names <- gsub(" sp.", "* sp.", names)
  names <- gsub(" group", "", names)
  names <- ifelse(grepl("[*]", names), paste("*", names, sep = ""), paste("*", names, "*", sep = ""))
  rownames(data) <- names
  data
}
species_revise <- function(data) {
  data$feature <- gsub("Saccharomyces_cerevisiae_x_Saccharomyces_kudriavzevii", "S._cerevisiae x S._kudriavzevii", data$feature)
  data$feature <- gsub("Pseudomonas_aeruginosa_group", "Pseudomonas_aeruginosa", data$feature)
  data$feature <- gsub("Pseudomonas_fluorescens_group", "Pseudomonas_fluorescens", data$feature)
  data
}





make_sig_table <- function(data) {
  sig_data <- spread(data[order(data$qval), c("feature", "metadata", "qval")], metadata, qval)
  sig_data <- species_revise(sig_data)
  sig_data$min <- apply(sig_data %>% dplyr::select(c("lypma", "benzonase", "molysis", "host_zero", "qiaamp")), 1, FUN = min)
  sig_data <- sig_data[order(sig_data$min),] %>% dplyr::select("feature", "lypma", "benzonase", "host_zero", "molysis", "qiaamp") %>% .[1:20,]
  sig_data[["feature"]] <- ifelse(sig_data[["feature"]] == "X.Collinsella._massiliensis", "[Collinsella]_massiliensis", sig_data[["feature"]])
  sig_data_italic <- sig_data %>% rownames_to_column(var = "-") %>%
          column_to_rownames(var = "feature") %>% species_italic %>% dplyr::select(-c("-")) %>%
          rename(lyPMA = lypma,  Benzonase = benzonase, `HostZERO` = host_zero, MolYsis = molysis, QIAamp = qiaamp)
  sig_data_sig <- ifelse(sig_data_italic < 0.1, "*", NA) %>% data.frame(check.names = F)
  return(list(data = sig_data, data_italic = sig_data_italic, data_sig = sig_data_sig))
}




make_sig_table18 <- function(data) {
  sig_data <- spread(data[order(data$qval), c("feature", "metadata", "qval")], metadata, qval)
  sig_data <- species_revise(sig_data)
  sig_data$min <- apply(sig_data %>% dplyr::select(c("lypma", "benzonase", "molysis", "host_zero", "qiaamp")), 1, FUN = min)
  sig_data <- sig_data[order(sig_data$min),] %>% dplyr::select("feature", "lypma", "benzonase", "host_zero", "molysis", "qiaamp") %>% .[1:18,]
  sig_data[["feature"]] <- ifelse(sig_data[["feature"]] == "X.Collinsella._massiliensis", "[Collinsella]_massiliensis", sig_data[["feature"]])
  sig_data_italic <- sig_data %>% rownames_to_column(var = "-") %>%
          column_to_rownames(var = "feature") %>% species_italic %>% dplyr::select(-c("-")) %>%
          rename(lyPMA = lypma,  Benzonase = benzonase, `HostZERO` = host_zero, MolYsis = molysis, QIAamp = qiaamp)
  sig_data_sig <- ifelse(sig_data_italic < 0.1, "*", NA) %>% data.frame(check.names = F)
  return(list(data = sig_data, data_italic = sig_data_italic, data_sig = sig_data_sig))
}



make_sig_table8 <- function(data) {
  sig_data <- spread(data[order(data$qval), c("feature", "metadata", "qval")], metadata, qval)
  sig_data <- species_revise(sig_data)
  sig_data$min <- apply(sig_data %>% dplyr::select(c("lypma", "benzonase", "molysis", "host_zero", "qiaamp")), 1, FUN = min)
  sig_data <- sig_data[order(sig_data$min),] %>% dplyr::select("feature", "lypma", "benzonase", "host_zero", "molysis", "qiaamp") %>% .[1:8,]
  sig_data[["feature"]] <- ifelse(sig_data[["feature"]] == "X.Collinsella._massiliensis", "[Collinsella]_massiliensis", sig_data[["feature"]])
  sig_data_italic <- sig_data %>% rownames_to_column(var = "-") %>%
          column_to_rownames(var = "feature") %>% species_italic %>% dplyr::select(-c("-")) %>%
          rename(lyPMA = lypma,  Benzonase = benzonase, `HostZERO` = host_zero, MolYsis = molysis, QIAamp = qiaamp)
  sig_data_sig <- ifelse(sig_data_italic < 0.1, "*", NA) %>% data.frame(check.names = F)
  return(list(data = sig_data, data_italic = sig_data_italic, data_sig = sig_data_sig))
}



#filt_fit_data_pos <- make_sig_table(filt_fit_data_pos)
filt_fit_data_bal <- make_sig_table18(filt_fit_data_bal_v_h)
filt_fit_data_ns <- make_sig_table8(filt_fit_data_ns_v_h)
filt_fit_data_spt <- make_sig_table(filt_fit_data_spt_v_h)


filt_spt_sig <- subset_taxa(
        subset_samples(v_h_phyloseq_rel_renamed, sample_type == "Sputum"),
        (taxa_names(subset_samples(v_h_phyloseq_rel_renamed, sample_type == "Sputum")) %in%
                filt_fit_data_spt$data$feature))

filt_fit_data_spt$rel <- cbind(filt_spt_sig %>% otu_table %>% t, filt_spt_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>% 
        .[row.names(filt_fit_data_spt$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")

filt_ns_sig <- subset_taxa(subset_samples(v_h_phyloseq_rel_renamed, sample_type == "Nasal"),
                                       taxa_names(subset_samples(v_h_phyloseq_rel_renamed,
                                                                 sample_type == "Nasal")) %in%
                                   filt_fit_data_ns$data$feature)

filt_fit_data_ns$rel <- cbind(filt_ns_sig %>% otu_table %>% t, filt_ns_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>% 
        .[row.names(filt_fit_data_ns$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")

filt_bal_sig <- subset_taxa(subset_samples(v_h_phyloseq_rel_renamed, sample_type == "BAL"),
                                       taxa_names(subset_samples(v_h_phyloseq_rel_renamed, sample_type == "BAL")) %in% filt_fit_data_bal$data$feature)

filt_fit_data_bal$rel <- cbind(filt_bal_sig %>% otu_table %>% t, filt_bal_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>%
        .[row.names(filt_fit_data_bal$data_italic),] %>%
        mutate_all(~na_if(., 0)) %>% rownames_to_column("feature") %>% subset(., !grepl("NA", .$feature))


```


```{r}



species_italic <- function(data) {
  names <- gsub("_", "~", data)
  names <- ifelse(grepl("~", names), paste("italic('", gsub("~" , "')~", names),   sep = ""),
                  paste("italic('", names, "')", sep = ""))
  names <- ifelse(grepl("\\d", names),
                  names %>% gsub("[italic(']", "", .) %>% gsub("[')]", "", .),
                  names)
  names
}

filt_maaslin_all_v_h$group <-  ifelse(filt_maaslin_all_v_h$Estimate > 0 & filt_maaslin_all_v_h$qval < 0.1, "more",
                     ifelse(filt_maaslin_all_v_h$Estimate < 0 & filt_maaslin_all_v_h$qval < 0.1, "less",
                            "insignificant"))

#Volcano plot

figS7_host <- filt_maaslin_all_v_h %>% 
        mutate(feature = species_italic(feature),
                            feature = case_when(value == "Nasal"~ NA,
                                              value == "BAL"~ NA,
                                              value == "Sputum"~ NA,
                                              qval > 0.1 ~ NA,
                                              abs(Estimate) < 0.3 ~ NA,
                                              .default = feature),
                            label1 = case_when(group == "more" ~ feature,
                                                       .default = NA),
                            label2 = case_when(group == "less" ~ feature,
                                                       .default = NA)
                            ) %>%
        ggplot(., aes(y = -log10(qval), x = Estimate, col = metadata)) +
        theme_classic(base_family = "sans") +
        #labs(tag = "A") +
        geom_point(size = 2, alpha = 0.7, stroke = 0) +
        xlab("Change estimate of CLR-normalized RPKM") +
        ylab("-log<sub>10</sub>(*q*-value)") +
        ylim(c(-1, 6)) +
        xlim(c(-7, 8)) +
        geom_hline(yintercept = 1, col = "gray") +
        geom_vline(xintercept = 0, col = "gray") +
        labs(tag = "B")  +
        ggtitle("Viral host genus") +
        theme(legend.position = "top", axis.title.y = ggtext::element_markdown(), legend.text = element_markdown()) +
        scale_color_manual(values = c("#a65628",
                                      "grey",
                                      "#fb9a99",
                                      "#33a02c",
                                      "#b2df8a",
                                      "#1f78b4",
                                      "#a6cee3"),
                           breaks = c("log10.Final_reads",
                                      "sample_type",
                                      "lypma",
                                      "benzonase",
                                      "host_zero",
                                      "molysis",
                                      "qiaamp"), 
                           labels = c("log<sub>10</sub>(Final reads)",
                                      "Sample type",
                                      "lyPMA",
                                      "Benzonase",
                                      "HostZERO",
                                      "MolYsis",
                                      "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        guides(col = guide_legend(title = "Factors", title.position = "top", nrow = 1)) +
                ggrepel::geom_text_repel(aes(label = label1),
                                  point.padding = unit(0.5, "lines"),
                                  #nudge_y = 10, box.padding = unit(0.35, "lines"),
                                  na.rm = T,
                                  parse = T,
                                  max.overlaps = 34,
                                  grob = "richtext",
                                  show.legend = F,
                                  size=3, segment.size=0.25, nudge_x=5, direction="y") +
        ggrepel::geom_text_repel(aes(label = label2),
                                   box.padding = unit(0.35, "lines"),
                                  #point.padding = unit(0.5, "lines"),
                                  na.rm = T,
                                  parse = T,
                                  max.overlaps = 34,
                                  grob = "richtext",
                                  show.legend = F,
                                  #hjust=0,
                                  size=3,
                                  segment.size=0.25,
                                  nudge_x=-5,
                                  nudge_y = 2,
                                  direction="y")

figS7_host

```


### Fig. 4. Balloon plot

**Fig. 4.** Mean relative abundance of top 20 significant taxa by q-value identified by differential abundance analysis using MaAsLin. Analyses were stratified by sample type. (A) bronchoalveolor lavage, (B) nasal swabs, and (C) sputum. Statistical significances were noted at the level of q-value < 0.1



```{r}

#ffff33 qia

f5b_host <- merge(filt_fit_data_bal$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_bal$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(filt_fit_data_bal$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(is.na(qval) ~ "> 0.1",
                               qval < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%

#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        theme_classic(base_family = "sans") +
        #colors for qvalues
        #xlab("Experimental group") +
        #ylab("Species") +
        #labs(tag = "B")  +
        ggtitle("B  BAL (viral host)") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              #axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 8),
              axis.title.y = element_blank(),
              axis.title.x = element_blank(),
              plot.margin = unit(c(0,0.2,0,1), 'lines'))  +
        scale_fill_manual(values = c("grey", "red"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1)
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "MolYsis",
                                  "qiaamp" = "QIAamp")
                         )

f5c_host <- merge(filt_fit_data_ns$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_ns$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(filt_fit_data_ns$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "sans") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        ylab("Species") +
        #labs(tag = "D")  +
        ggtitle("Nasal (viral host)") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              #axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 8),
              axis.title.x = element_blank(),
              plot.margin = unit(c(0,0.2,0,1), 'lines'))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top",
                                   override.aes = list(size=5)),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "MolYsis",
                                  "qiaamp" = "QIAamp")
                         )

f5d_host <- merge(filt_fit_data_spt$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_spt$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        merge(filt_fit_data_spt$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "sans") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        #ylab("Species") +
        #labs(tag = "F")  +
        ggtitle("F  Sputum (viral host)") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              #axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 8),
              #axis.title.x = element_blank(),
              axis.title.x = element_text(margin = margin(t = 0)),
              axis.title.y = element_blank(),
              plot.margin = unit(c(0,0.2,0,1), 'lines'))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "MolYsis",
                                  "qiaamp" = "QIAamp")
                         )

fig4 <- ggarrange(f5c_host %>% lemon::g_legend() %>% as_ggplot,
                  f5b_host,
                  f5c_host,
                  f5d_host,
                  #b, b %>% lemon::g_legend() %>% as_ggplot,
                  ncol=1, heights = c(1.5, 4, 4, 4),
                  legend = "none",
                  align = "hv")

fig4

```


### Table S8. List of DA taxa

**Table S8.** List of differentially abundant taxa identified by MaAsLin (Beghini et al., 2021). Significant associations (q-value < 0.1) that is showing high change ( | log2(fold-change) | > 0.5) were listed.

```{r, warning=FALSE, message=FALSE}

filt_maaslin_all_v %>% subset(., .$qval < 0.1 & abs(.$Estimate) > 4) 

tableS8 <- filt_maaslin_all_v %>% subset(., .$qval < 0.1 & abs(.$Estimate) > 4) %>% dplyr::select(c("feature", "metadata", "Estimate", "qval")) %>%
        mutate(feature = paste("<i>", gsub("_", " ", feature), sep = "") %>%
                       gsub(" sp", "</i> sp.", .) %>% gsub(" group", "", .)) %>%
        mutate(feature = case_when(!grepl("</i>", feature) ~ paste(feature, "</i>", sep = ""),
                                   .default = feature))%>% 
        rename(Taxa = "feature",
                     `Fixed effect` = "metadata",
                     "Change estimate of CLR transformed count" = "Estimate",
                     `<i>q<i/>-value` = "qval") %>% 
        mutate(`Fixed effect` = case_when(
                         `Fixed effect` == "sample_type" ~ "Sample type",
                         `Fixed effect` == "lypma" ~ "lyPMA",
                         `Fixed effect` == "host_zero" ~ "HostZERO",
                         `Fixed effect` == "benzonase" ~ "Benzonase",
                         `Fixed effect` == "molysis" ~ "MolYsis",
                         `Fixed effect` == "qiaamp" ~ "QIAamp",
                         .default = `Fixed effect`),
               "Change estimate of CLR transformed count" = round(`Change estimate of CLR transformed count`, 3),
               `<i>q<i/>-value` = round(`<i>q<i/>-value`, 3)) %>%
        remove_rownames() %>% 
        kbl(format = "html", escape = FALSE) %>%
        kable_styling(full_width = 0, html_font = "sans")

tableS8


```

## Microbial - viral correlation

### Microbial file wrangling 

```{r}
setwd("Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git")
phyloseq_unfiltered <- read_rds("/Volumes/macdrive/Dropbox/Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20230521_MGK_host_tidy.rds")
phyloseq <- phyloseq_unfiltered
filt_maaslin_all <- read.csv("data/da_lmer_filt_maaslin_all.csv")
#filt_maaslin_interaction <- read.csv("data/da_lmer_filt_maaslin_interaction.csv")
filt_fit_data_bal <- read.csv("data/da_lmer_filt_fit_data_bal.csv")
filt_fit_data_spt <- read.csv("data/da_lmer_filt_fit_data_spt.csv")
filt_fit_data_ns <- read.csv("data/da_lmer_filt_fit_data_ns.csv")
#filt_fit_data_pos <- read.csv("data/da_lmer_filt_fit_data_pos.csv")
#Stratified by sample type



alpha_diversity <- function(data) {
        otu_table <- otu_table(data) %>% .[, colSums(.) !=0]
        S.obs <- rowSums(t(otu_table) != 0)
        sample_data <- sample_data(data)
        data_evenness <- vegan::diversity(t(otu_table)) / log(vegan::specnumber(t(otu_table))) # calculate evenness index using vegan package
        data_shannon <- vegan::diversity(t(otu_table), index = "shannon") # calculate Shannon index using vegan package
        data_hill <- exp(data_shannon)                           # calculate Hills index
        data_dominance <- microbiome::dominance(otu_table, index = "all", rank = 1, aggregate = TRUE) # dominance (Berger-Parker index), etc.
        data_invsimpson <- vegan::diversity(t(otu_table), index = "invsimpson")                          # calculate Shannon index using vegan package
        alpha_diversity <- cbind(S.obs, data_shannon, data_hill, data_invsimpson, data_evenness,data_dominance) # combine all indices in one data table
        sample_data <- merge(data.frame(sample_data), alpha_diversity, by = 0, all = T) %>% column_to_rownames(var = "Row.names")
}
phyloseq <- phyloseq_unfiltered

sample_data(phyloseq_unfiltered$phyloseq_rel) <- sample_data(alpha_diversity(phyloseq_unfiltered$phyloseq_rel))
sample_data(phyloseq_unfiltered$phyloseq_count) <- sample_data(alpha_diversity(phyloseq_unfiltered$phyloseq_count)) 
sample_data(phyloseq_unfiltered$phyloseq_path_rpk) <- sample_data(alpha_diversity(phyloseq_unfiltered$phyloseq_path_rpk))  



prev_neg <- ((phyloseq_unfiltered$phyloseq_count %>% 
          subset_samples(sample_type == "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (negative controls)" = ".")

prev_all <- ((phyloseq_unfiltered$phyloseq_count %>%
                      subset_samples(sample_type != "Mock") %>%
                      subset_samples(sample_type != "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (all)" = ".")




sample_data(phyloseq_unfiltered$phyloseq_rel)$is.neg <- grepl("Neg", sample_data(phyloseq_unfiltered$phyloseq_rel)$sample_type)



phyloseq_decontam_lypma <- phyloseq_unfiltered$phyloseq_rel %>% 
        subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "lyPMA")

phyloseq_decontam_ben <- phyloseq_unfiltered$phyloseq_rel %>% 
        subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "Benzonase")

phyloseq_decontam_hostzero <- phyloseq_unfiltered$phyloseq_rel %>% 
        subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "HostZero")

phyloseq_decontam_molysis <- phyloseq_unfiltered$phyloseq_rel %>% 
        subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "MolYsis")

phyloseq_decontam_qiaamp <- phyloseq_unfiltered$phyloseq_rel %>% 
        subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "QIAamp")


contaminant_combined_lypma <- 
data.frame("lyPMA", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_lypma, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) %>% row.names
)

contaminant_combined_ben <- 
data.frame("Benzonase", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_ben, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) %>% row.names
)


contaminant_combined_hostzero <- 
data.frame("HostZero", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_hostzero, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) %>% row.names
)


contaminant_combined_molysis <- 
data.frame("MolYsis", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_molysis, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) %>% row.names
)


contaminant_combined_QIAamp <- 
data.frame("QIAamp", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_qiaamp, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) %>% row.names
)

contaminants_strat_treat <- rbind(contaminant_combined_lypma,
                      contaminant_combined_ben, 
                      contaminant_combined_hostzero,
                      contaminant_combined_molysis,
                      contaminant_combined_QIAamp
                      )

names(contaminants_strat_treat) <- c("Treatment", "Taxa")

phyloseq_unfiltered$phyloseq_rel <- transform_sample_counts(phyloseq_unfiltered$phyloseq_rel,
                                                            function(x){x/sum(x)})
taxa_qc <- data.frame("species" =
                              otu_table(
                                      subset_samples(
                                              phyloseq_unfiltered$phyloseq_rel,S.obs != 0 &
                                                      sample_type %in% c("Mock", "BAL", "Nasal", "Sputum"))) %>%
                              t() %>% colnames(),
                      "prevalence" =
                              ifelse(subset_samples(phyloseq_unfiltered$phyloseq_rel, S.obs != 0 & 
                                                            sample_type %in% c("Mock", "BAL", "Nasal", "Sputum")) %>%
                                             otu_table() > 0, 1, 0)%>% 
                              t() %>%
                              colSums(), #Prevalence of taxa
                      "mean_rel_abd" = 
                              subset_samples(phyloseq_unfiltered$phyloseq_rel,
                                             S.obs != 0 & sample_type %in% c("Mock", "BAL", "Nasal", "Sputum")) %>%
                              otu_table() %>%
                              t() %>%
                              colMeans(na.rm = T) #mean relativ abundacne 
)


function_qc <- data.frame("function" =
                                  otu_table(
                                          subset_samples(
                                                  phyloseq_unfiltered$phyloseq_path_rpk,
                                                           S.obs != 0 & 
                                                          sample_type %in% 
                                                          c("Mock", "BAL", "Nasal", "Sputum")
                                                  )
                                          ) %>% 
                                  t() %>% 
                                  colnames(),
                          "prevalence" = 
                                  ifelse(subset_samples(phyloseq_unfiltered$phyloseq_path_rpk,
                                                        S.obs != 0 & 
                                                                sample_type %in% 
                                                                c("Mock", "BAL", "Nasal", "Sputum")
                                                        ) %>% 
                                                 otu_table() > 0, 
                                         1, 
                                         0
                                         ) %>% 
                                  t() %>% 
                                  colSums(), #Prevalence of taxa
                          "mean_rpk" = 
                                  subset_samples(phyloseq_unfiltered$phyloseq_path_rpk, 
                                                 S.obs != 0 & 
                                                         sample_type %in% 
                                                         c("Mock", "BAL", "Nasal", "Sputum")
                                                 ) %>% 
                                  otu_table() %>% 
                                  t() %>% 
                                  colMeans(na.rm = T), #mean relativ abundacne 
                          unidentified = 
                                  ifelse((subset_samples(phyloseq_unfiltered$phyloseq_path_rpk,
                                              S.obs != 0 & sample_type %in% c("Mock", "BAL", "Nasal", "Sputum")) %>% 
                                       otu_table() > 0) %>%
                                      row.names() %in% c("UNMAPPED", "UNINTEGRATED")
                              , 1, 0)
)

red_flag_taxa <- data.frame(species = taxa_qc$species,
                            prevalence = taxa_qc$prevalence,
                            mean_rel_abd = taxa_qc$mean_rel_abd,
                            red_flag_prev_abd = 
                                    ifelse(taxa_qc$prevalence < 
                                                   otu_table(
                                                           subset_samples(
                                                                   phyloseq_unfiltered$phyloseq_rel,
                                                                   S.obs != 0 & 
                                                                           sample_type %in% 
                                                                           c("Mock", "BAL", "Nasal", "Sputum"))) %>%
                                                               t %>% rownames() %>%
                                                   length * 0.05 &
                                                   #Removing taxa with zero prevalence - taxa from nasal swabs
                                                   taxa_qc$mean_rel_abd <
                                                   taxa_qc %>%
                                                   subset(., .$prevalence != 0) %>%
                                                   .$mean_rel_abd %>%
                                                   quantile(., 0.75), 1,0),
                           red_flag_prev =
                                    ifelse(taxa_qc$prevalence < 
                                                   otu_table(
                                                           subset_samples(
                                                                   phyloseq_unfiltered$phyloseq_rel,
                                                                   S.obs != 0 & 
                                                                           sample_type %in% 
                                                                           c("Mock", "BAL", "Nasal", "Sputum"))) %>%
                                                               t %>% rownames() %>%
                                                   length * 0.05,
                                           1,
                                           0)) %>%
        mutate(red_flag_decontam = species %in% (contaminants$Taxa %>% unique()))

subset(red_flag_taxa, red_flag_taxa$red_flag_prev == 1 & red_flag_taxa$red_flag_prev_abd == 0)

#Unampped function were removed

red_flag_function <- 
        data.frame(function. = function_qc$function.,
                   prevalence = function_qc$prevalence,
                   mean_rel_abd = function_qc$mean_rpk,
                   red_flag_prev_abd = 
                           ifelse(function_qc$prevalence < 
                                          otu_table(
                                                  subset_samples(
                                                          phyloseq_unfiltered$phyloseq_path_rpk,
                                                          S.obs != 0 & 
                                                                  sample_type %in% 
                                                                  c("Mock", "BAL", "Nasal", "Sputum"))) %>%
                                          t %>% 
                                          rownames() %>%
                                          length * 0.05 &
                                          #Removing taxa with zero prevalence - taxa from nasal swabs
                                          function_qc$mean_rpk <
                                          function_qc %>%
                                          subset(., .$prevalence != 0) %>%
                                          .$mean_rpk %>%
                                          quantile(., 0.75), 1,0),
                   red_flag_prev =
                           ifelse(function_qc$prevalence <
                                          otu_table(
                                                  subset_samples(
                                                          phyloseq_unfiltered$phyloseq_path_rpk,
                                                          S.obs != 0 & 
                                                                  sample_type %in% 
                                                                  c("Mock", "BAL", "Nasal", "Sputum"))) %>%
                                          t %>% 
                                          rownames() %>%
                                          length * 0.05,
                                          1,
                                          0)) %>%
        mutate(red_flag_prev_abd = case_when(function. %in% c("UNMAPPED", "UNINTEGRATED") ~ 1,
                                     .default = red_flag_prev_abd))

subset(red_flag_function, red_flag_function$red_flag_prev == 1 & red_flag_function$red_flag_prev_abd == 0)



#decontaminated phyloseq 
phyloseq_decontam <- phyloseq
phyloseq_decontam$phyloseq_count <- prune_taxa(subset(red_flag_taxa,
                                           red_flag_taxa$red_flag_prev_abd != 1 &
                                                   !red_flag_taxa$red_flag_decontam)$species,
                                    phyloseq$phyloseq_count)

phyloseq_decontam$phyloseq_rel <- prune_taxa(subset(red_flag_taxa,
                                           red_flag_taxa$red_flag_prev_abd != 1 &
                                                   !red_flag_taxa$red_flag_decontam)$species,
                                    phyloseq$phyloseq_rel) %>%
        transform_sample_counts(., function(x){x/sum(x)})
        
#phyloseq for analysis
phyloseq$phyloseq_count <- prune_taxa(subset(red_flag_taxa,
                                           red_flag_taxa$red_flag_prev_abd != 1)$species,
                                    phyloseq$phyloseq_count)
phyloseq$phyloseq_rel <- prune_taxa(subset(red_flag_taxa,
                                           red_flag_taxa$red_flag_prev_abd != 1)$species,
                                    phyloseq$phyloseq_rel) %>%
        transform_sample_counts(., function(x){x/sum(x)})


phyloseq$phyloseq_path_rpk <- prune_taxa(subset(red_flag_function, red_flag_function$red_flag_prev_abd != 1)$function., phyloseq$phyloseq_path_rpk)

#Calculation of alpha diversity indices for filtered samples

alpha_diversity <- function(data) {
        otu_table <- otu_table(data) %>% .[, colSums(.) !=0]
        S.obs <- rowSums(t(otu_table) != 0)
        sample_data <- sample_data(data)
        data_evenness <- vegan::diversity(t(otu_table)) / log(vegan::specnumber(t(otu_table))) # calculate evenness index using vegan package
        data_shannon <- vegan::diversity(t(otu_table), index = "shannon") # calculate Shannon index using vegan package
        data_hill <- exp(data_shannon)                           # calculate Hills index
        data_dominance <- microbiome::dominance(otu_table, index = "all", rank = 1, aggregate = TRUE) # dominance (Berger-Parker index), etc.
        data_invsimpson <- vegan::diversity(t(otu_table), index = "invsimpson")                          # calculate Shannon index using vegan package
        alpha_diversity <- cbind(S.obs, data_shannon, data_hill, data_invsimpson, data_evenness,data_dominance) # combine all indices in one data table
        sample_data <- merge(data.frame(sample_data), alpha_diversity, by = 0, all = T) %>% column_to_rownames(var = "Row.names")
}
#sample_data(phyloseq$phyloseq_count) <- sample_data(alpha_diversity(phyloseq$phyloseq_count))
sample_data(phyloseq$phyloseq_rel) <- sample_data(alpha_diversity(phyloseq$phyloseq_count))
sample_data(phyloseq$phyloseq_count) <-sample_data(alpha_diversity(phyloseq$phyloseq_count)) 
sample_data(phyloseq$phyloseq_path_rpk) <- sample_data(alpha_diversity(phyloseq$phyloseq_path_rpk))  

sample_data(phyloseq_decontam$phyloseq_rel) <- sample_data(alpha_diversity(phyloseq_decontam$phyloseq_count))  
sample_data(phyloseq_decontam$phyloseq_count) <- sample_data(alpha_diversity(phyloseq_decontam$phyloseq_count))

sample_data <- sample_data(phyloseq$phyloseq_count)

```

# Microbial filtering

All data

```{r}
#OTU table of microbial output
otu_bac <- phyloseq$phyloseq_rel %>%
        otu_table(., taxa_are_rows = F) %>%
        data.frame(., check.names = F) 

#Changing viral taxa name into their host genus
taxa_names(v_h_phyloseq_rel) <- v_h_phyloseq_rel %>%
        tax_table %>% 
        data.frame %>% 
        .$Genus

#OTU table of viral output
otu_v_h <- v_h_phyloseq_rel %>%
        otu_table(., taxa_are_rows = F) %>%
        data.frame(., check.names = F)

#Calculation of Spearman correlation matrix
spearman_corr <- Hmisc::rcorr(as.matrix(t(bind_rows(otu_bac, otu_v_h))), # 2 OTU tables were binned into 1. 
                              type="spearman") #bind_rows were used as 1 viral output is missing 1 sample
                                                # 
#Making a function that makes correlation matrix into tidy flat data (long format)
flattenCorrMatrix <- function(cormat, pmat) {
        cormat <- cormat %>% reshape2::melt() %>% rename(r = value)
        pmat <- pmat %>% reshape2::melt() %>% rename(p = value)
        cormat$p <- pmat$p
        cormat
}




#Making a flat table with adjusted p-values.
flat_corr <- flattenCorrMatrix(spearman_corr$r, spearman_corr$P) %>%
        mutate(qval = qvalue::qvalue(`p`)$qvalue,
               p_bh = p.adjust(p = `p`, method = "BH"),
               p_b = p.adjust(p = `p`, method = "bonferroni"),
               fdr = p.adjust(p = `p`, method = "fdr")) %>%
        subset(., .$Var1 %in% taxa_names(v_h_phyloseq_rel)) %>%
        subset(., !(.$Var2 %in% taxa_names(v_h_phyloseq_rel)))# %>%
        #subset(., .$p_b < 0.05) # bonferroni & 0.001 criteria was used as their correction was the strongest
      
heatmap_corr <- flat_corr %>% select(c("Var1", "Var2", "r")) %>%
        spread(., Var2, r) %>% 
        column_to_rownames("Var1") %>%
        as.matrix(.)


#pdf(file ="figures/FIG3C.pdf",
#    width = 13, height = 9)
ComplexHeatmap::Heatmap(heatmap_corr,
                        na_col = "grey",
        rect_gp = grid::gpar(col = "white"),
        row_names_gp = grid::gpar(fontsize = 10),
        column_names_gp = grid::gpar(fontsize = 10),
        column_names_rot = 45,
        column_dend_side = "bottom",
        column_names_side = "top",
        clustering_method_rows = "average",
        clustering_method_columns = "average",
        #col=colorRamp2(c(-.2, 0, .4), c("navy", "white", "red")),
        heatmap_legend_param = list(title = "Spearman\n(rho)"),
        cell_fun = function(j, i, x, y, width, height, fill) {
          grid::grid.text(sprintf("%.2f", heatmap_corr[i, j]), x, y, gp =
                      grid::gpar(fontsize = 7, col="white"))})
#dev.off()



```

p-Bonferroni < 0.5 removed data

```{r}


#Making a flat table with adjusted p-values.
flat_corr <- flattenCorrMatrix(spearman_corr$r, spearman_corr$P) %>%
        mutate(qval = qvalue::qvalue(`p`)$qvalue,
               p_bh = p.adjust(p = `p`, method = "BH"),
               p_b = p.adjust(p = `p`, method = "bonferroni"),
               fdr = p.adjust(p = `p`, method = "fdr")) %>%
        subset(., .$Var1 %in% taxa_names(v_h_phyloseq_rel)) %>%
        subset(., !(.$Var2 %in% taxa_names(v_h_phyloseq_rel))) %>%
        mutate(r = case_when(p_b > 0.05 ~ 0,
                             .default = r)) # bonferroni & 0.001 criteria was used as their correction was the strongest

heatmap_corr <- flat_corr %>% select(c("Var1", "Var2", "r")) %>%
        spread(., Var2, r) %>% 
        column_to_rownames("Var1") %>%
        #filter(all_vars(!is.na(.))) %>%
        as.matrix(.) 



#pdf(file ="figures/FIG3C.pdf",
#    width = 13, height = 9)
ComplexHeatmap::Heatmap(heatmap_corr,
                        na_col = "grey",
        rect_gp = grid::gpar(col = "grey"),
        row_names_gp = grid::gpar(fontsize = 10),
        column_names_gp = grid::gpar(fontsize = 10),
        column_names_rot = 45,
        column_dend_side = "bottom",
        column_names_side = "top",
        clustering_method_rows = "average",
        clustering_method_columns = "average",
        col=circlize::colorRamp2(c(-1, 0, 1), c("navy", "white", "red")),
        heatmap_legend_param = list(title = "Spearman\n(rho)"),
        cell_fun = function(j, i, x, y, width, height, fill) {
          grid::grid.text(sprintf("%.2f", heatmap_corr[i, j]), x, y, gp =
                      grid::gpar(fontsize = 7, col="white"))})
#dev.off()



```

### bacterial genus - viral host

```{r}

#genus level phyloseq object
phyloseq_m_genus <- tax_glom(phyloseq$phyloseq_count, taxrank = "Genus") %>% transform_sample_counts(., function(x){x/sum(x)})

taxa_names(phyloseq_m_genus) <- phyloseq_m_genus %>% tax_table() %>% data.frame() %>% .$Genus %>% gsub("g__", "", .) %>% gsub("_XIII_Incertae_Sedis_unclassified", "", .)

#OTU table of microbial output
otu_bac <- phyloseq_m_genus %>%
        otu_table(., taxa_are_rows = F) %>%
        data.frame(., check.names = F) 

#Changing viral taxa name into their host genus
taxa_names(v_h_phyloseq_rel) <- v_h_phyloseq_rel %>%
        tax_table %>% 
        data.frame %>% 
        .$Genus %>%
        paste0("v_", .)

#OTU table of viral output
otu_v_h <- v_h_phyloseq_rel %>%
        otu_table(., taxa_are_rows = F) %>%
        data.frame(., check.names = F)

#Calculation of Spearman correlation matrix
spearman_corr <- Hmisc::rcorr(as.matrix(t(bind_rows(otu_bac, otu_v_h))), # 2 OTU tables were binned into 1. 
                              type="spearman") #bind_rows were used as 1 viral output is missing 1 sample
                                                # 
#Making a function that makes correlation matrix into tidy flat data (long format)
flattenCorrMatrix <- function(cormat, pmat) {
        cormat <- cormat %>% reshape2::melt() %>% rename(r = value)
        pmat <- pmat %>% reshape2::melt() %>% rename(p = value)
        cormat$p <- pmat$p
        cormat
}


#Making a flat table with adjusted p-values.
flat_corr <- flattenCorrMatrix(spearman_corr$r, spearman_corr$P) %>%
        mutate(qval = qvalue::qvalue(`p`)$qvalue,
               p_bh = p.adjust(p = `p`, method = "BH"),
               p_b = p.adjust(p = `p`, method = "bonferroni"),
               fdr = p.adjust(p = `p`, method = "fdr")) %>%
        subset(., grepl("v_", .$Var1)) %>%
        subset(.,  !grepl("v_", .$Var2)) %>%
        mutate(r = case_when(p_b > 0.05 ~ 0,
                             .default = r)) # bonferroni & 0.001 criteria was used as their correction was the strongest
heatmap_corr <- flat_corr %>% select(c("Var1", "Var2", "r")) %>%
        spread(., Var2, r) %>% 
        column_to_rownames("Var1") %>%
        as.matrix(.)#
rownames(heatmap_corr) <- rownames(heatmap_corr) %>% gsub("v_", "", .)

#pdf(file ="figures/FIG3C.pdf",
#    width = 13, height = 9)



ComplexHeatmap::Heatmap(heatmap_corr,
                        column_order = colnames(heatmap_corr) %>% order,
                        row_order = rownames(heatmap_corr) %>% order,
                        na_col = "grey",
        rect_gp = grid::gpar(col = "grey"),
        row_names_gp = grid::gpar(fontsize = 10),
        column_names_gp = grid::gpar(fontsize = 10),
        column_names_rot = 45,
        column_dend_side = "bottom",
        column_names_side = "top",
        row_title = "Viral host",
        column_title =  "Microbial genus",
        clustering_method_rows = "average",
        clustering_method_columns = "average",
        col=circlize::colorRamp2(c(-1, 0, 1), c("navy", "white", "red")),
        heatmap_legend_param = list(title = "Spearman\n(rho)"),
        cell_fun = function(j, i, x, y, width, height, fill) {
          grid::grid.text(sprintf("%.2f", heatmap_corr[i, j]), x, y, gp =
                      grid::gpar(fontsize = 7, col="white"))})
#dev.off()



```

# List of updated figures

### Updated barplot

```{r}



#{taxa_names(v_h_phyloseq_rel) <-  tax_table(v_h_phyloseq_rel) %>% data.frame %>% .$Genus
#v_h_phyloseq_rel
#} %>%
v_h_phyloseq_rel_renamed <- v_h_phyloseq_rel
taxa_names(v_h_phyloseq_rel_renamed) <-  tax_table(v_h_phyloseq_rel) %>% data.frame %>% .$Genus


my_plot_bar = function (physeq, x = "Sample", y = "Abundance", fill = NULL, title = NULL, 
                        facet_grid = NULL) {
    mdf = psmelt(physeq)
    p = ggplot(mdf, aes_string(x = x, y = y, fill = fill))
    p = p + geom_bar(stat = "identity")
    p = p + theme(axis.text.x = element_text(angle = -90, hjust = 0)) +
            scale_x_discrete(drop = F)
    if (!is.null(facet_grid)) {
        p <- p + facet_grid(facet_grid)
    }
    if (!is.null(title)) {
        p <- p + ggtitle(title)
    }
    return(p)
}

a <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Mock")) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Mock") %>% 
           subset_samples(., S.obs != 0)) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Mock") %>% 
           subset_samples(., S.obs != 0)
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        my_plot_bar(., fill="species20") + 
        xlab("Sample") +
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.text.x = element_text(color = "white")) +
        guides(fill=guide_legend(title="Top 10 species")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        labs(tag = "A")
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +
        


b <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Neg.") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
        ) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Neg.") %>%
                                      subset_samples(.,
                                                     baylor_other_id != "20220606_Neg") %>%
                                      transform_sample_counts(.,
                                                              function(x){ifelse(is.na(x),
                                                                                 0, 
                                                                                 x)})
                              ) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Neg.") %>%
        subset_samples(.,
                       baylor_other_id != "20220606_Neg") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>% 
        my_plot_bar(., fill="species20") + 
        ylab("") +
        xlab("Subject") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.text.x = element_text(color = "white")) +
        guides(fill=guide_legend(title="Top 10 species")) +
        scale_fill_manual(values = c("#A6CEE3",
                                     "#1F78B4",
                                     "#B2DF8A",
                                     "#33A02C",
                                     "#FB9A99",
                                     "#E31A1C",
                                     "#FDBF6F",
                                     "#FF7F00",
                                     "#CAB2D6",
                                     "#6A3D9A",
                                     "#FFFF99")) +
        
        #facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        facet_wrap ( ~ treatment,
                    scales= "free_x", nrow=1) +
        ggtitle("E    Negative control")

c <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "BAL") %>%
                transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
        ) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "BAL") %>%
                                      transform_sample_counts(.,
                                                              function(x){ifelse(is.na(x),
                                                                                 0, 
                                                                                 x)})
                              ) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "BAL") %>%
           transform_sample_counts(., function(x){ifelse(is.na(x), 0, x)})
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E"))
   phyloseq_temp
  } %>% 
        my_plot_bar(., x = "subject_id", fill="species20") + 
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 species")) +

        scale_fill_manual(values = c('#8dd3c7',# Alloprevotella
                                     "#B2DF8A",# Candida
                                     "#33A02C",
                                     '#ffffb3', # Cupriavidus
                                     '#bebada', # Lactobacillus
                                     "#08519c", # Staphylococcus
                                     "#08306b",
                                     "#cb181d",# Streptococcus
                                     "#a50f15",
                                     "#67000d",
                                     "darkgrey" #Others
                                     )) +
        facet_wrap ( ~ treatment,
                    scales= "free_x", nrow=1) +
        ggtitle("A    BAL (microbial species)")


d <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Nasal") %>% 
           subset_samples(., S.obs != 0)) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Nasal") %>% 
           subset_samples(., S.obs != 0)) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Nasal") %>% 
           subset_samples(., S.obs != 0)
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E", "F", "G", "H", "I", "J")) %>%
           as.character()
   phyloseq_temp
  } %>% 
        my_plot_bar(., x = "subject_id", fill="species20") + 
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 species")) +
        
        scale_fill_manual(values = c('#8dd3c7',# Anaerococcus
                                     "#B2DF8A",# Corynebacterium
                                     "#33A02C",
                                     '#ffffb3', # Cupriavidus
                                     "#6baed6", # Cutibacterium
                                     "#1F78B4",
                                     '#bebada', # Dolosigranulum
                                     '#fb8072', # Malassezia
                                     "#54278f",# Straphyloccocus
                                     "#3f007d",
                                     "darkgrey" #Others
                                     )) +
        
#        scale_x_discrete(drop=) +
        facet_wrap(~ treatment,  nrow = 1, drop = T, scales = "free_x") +
        ggtitle("C    Nasal swabs (microbial species)")

e <- tax_table(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Sputum") %>% 
           subset_samples(., S.obs != 0)) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Sputum") %>% 
           subset_samples(., S.obs != 0)) %>%
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
   .[top20species, "species20"] <- as.character(.[top20species, "Species"])
   .[, 9] <- gsub("s__", " ", .[, 9])
   .[, 9] <- gsub("_", " ", .[, 9])
   .[, 9] <- gsub("[]]|[[]", "",  .[, 9])
   .[, 9] <- gsub(" sp", " sp.",  .[, 9])
   .[, 9] <- gsub(" sp.", "</i> sp.",  .[, 9])
   .[, 9] <- gsub(" group", "</i> group.",  .[, 9])
   .[, 9] <- ifelse(grepl("Other",.[, 9]),
                    "Other",
                    ifelse(grepl("</i>",  .[, 9]),
                           paste("<i>",  .[, 9], sep = ""),
                           paste("<i>",  .[, 9], "</i>", sep = "")) %>%
                            gsub("s__", "", .) %>%
                            gsub("_", " ", .)
   )
   phyloseq_temp <- subset_samples(phyloseq$phyloseq_rel,
                              sample_type == "Sputum") %>% 
           subset_samples(., S.obs != 0)
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E"))
   phyloseq_temp
  } %>%
        my_plot_bar(., x = "subject_id", fill="species20")+
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 species")) +
        
        scale_fill_manual(values = c('#B2DF8A',# Actinomyces
                                     "#33A02C",
                                     "#8dd3c7", #Gemella
                                     '#ffffb3', # Pseudomonas
                                     '#bebada', # Rothia
                                     '#fb8072', # Staphylococcus
                                     "#4292c6",# Streptococcus
                                     "#2171b5",
                                     "#08519c",
                                     "#08306b",
                                     "darkgrey" #Others
                                     )) +
        facet_wrap(~ treatment, scales= "free_x", nrow = 1) +
        ggtitle("E    Sputum (microbial species)")


fig2 <- ggarrange(c, c %>% lemon::g_legend() %>% as_ggplot,
                  d, d %>% lemon::g_legend() %>% as_ggplot,
                  e, e %>% lemon::g_legend() %>% as_ggplot,
                  #a, a %>% lemon::g_legend() %>% as_ggplot,
                  #b, b %>% lemon::g_legend() %>% as_ggplot,
                  ncol=2,nrow=3, widths = c(3, 1),
                  legend = "none",
                  align = "hv")

fig2_y <- annotate_figure(fig2,
                left = text_grob("Relative abundance",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Subject",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11)
)
ggarrange(fig2_y, fig2_vh) 

```

```{r}

png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/Figure2_updated.png",   # The directory you want to save the file in
    width = 170, # The width of the plot in inches
    height = 225, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue



fig2_updated <- ggarrange(c, c %>% lemon::g_legend() %>% as_ggplot,
                  c_vh, c_vh %>% lemon::g_legend() %>% as_ggplot,
                  d, d %>% lemon::g_legend() %>% as_ggplot,
                  d_vh, d_vh %>% lemon::g_legend() %>% as_ggplot,
                  e, e %>% lemon::g_legend() %>% as_ggplot,
                  e_vh, e_vh %>% lemon::g_legend() %>% as_ggplot,
                  #a, a %>% lemon::g_legend() %>% as_ggplot,
                  #b, b %>% lemon::g_legend() %>% as_ggplot,
                  ncol=2,nrow=6, widths = c(3, 1),
                  legend = "none",
                  align = "hv")


annotate_figure(fig2_updated,
                left = text_grob("Relative abundance",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Subject",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11)
)

# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()


```

### Updated alpha and beta-diversity plot 


```{r, warning=FALSE, message=FALSE}

f3a <- ggplot(subset(sample_data(phyloseq$phyloseq_count) %>% 
                             data.frame, sample_data(phyloseq$phyloseq_count)$sample_type %in% c(#"Mock",
                                     "Sputum", "Nasal", "BAL")), aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment),
                    position = position_jitter(0.2), size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        #stat_summary(fun = mean,
        #       geom = "pointrange",
        #       fun.max = function(x) mean(x) + sd(x),
        #       fun.min = function(x) mean(x) - sd(x))+
        ylab("Species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        ggtitle("A   Microbial alpha diversity") +
        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 

dat_text <- data.frame(
  label = c(
          #"", "***", "***", "**", "***", #label for Mock
          "", "", "", "*", "", #label for BAL
          "", "", "***", "*", "**", 
          "**", "***", "***", "***", "***"),
  sample_type = c(
          #"Mock", "Mock", "Mock", "Mock", "Mock", 
          "BAL", "BAL", "BAL", "BAL", "BAL", 
          "Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          "Sputum", "Sputum", "Sputum", "Sputum", "Sputum"),
  treatment     = c(
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
  S.obs = c(
          #50, 30, 35, 33, 31,
          30, 35, 50, 52, 50,
          30, 30, 30, 35, 30,
          100, 120, 147, 140, 125)
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f3a <- f3a + geom_text(
  data    = dat_text,
  mapping = aes(x = treatment, y = S.obs, label = label)
)



f5S_mock <- ggplot(subset(sample_data(phyloseq$phyloseq_count) %>% 
                             data.frame,
                          sample_data(phyloseq$phyloseq_count)$sample_type %in%
                                  c("Mock")),
                   aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment), position = position_jitter(0.2),
                    size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        ylab("Species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        labs(tag = "C") +
        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 

dat_text <- data.frame(
  label = c(
          "", "***", "***", "**", "***"#, #label for Mock
          #"", "", "", "*", "", #label for BAL
          #"", "", "***", "*", "**", 
          #"**", "***", "***", "***", "***"
          ),
  sample_type = c(
          "Mock", "Mock", "Mock", "Mock", "Mock"#, 
          #"BAL", "BAL", "BAL", "BAL", "BAL", 
          #"Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          #"Sputum", "Sputum", "Sputum", "Sputum", "Sputum"
          ),
  treatment = c(
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"#, 
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"
          ),
  S.obs = c(
          50, 30, 35, 33, 31
          #30, 35, 50, 52, 50,
          #30, 30, 30, 35, 30,
          #100, 120, 147, 140, 125
          )
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("Mock"#, 
                                                                #"BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f5S_mock <- f5S_mock + geom_text(
  data    = dat_text,
  mapping = aes(x = treatment, y = S.obs, label = label)
)



#Making subset of non-zero samples without neg

phyloseq_rel_nz <- phyloseq$phyloseq_rel %>%
        subset_samples(S.obs != 0 & sample_type %in% c("Mock", 
                                                       "BAL", "Nasal", "Sputum"))

#distances of betadiversity - boxplots
horn_dist_long <- distance(phyloseq_rel_nz, method="horn") %>% as.matrix() %>% melt_dist() #making long data of distance matrices

#Adding sample type and treatment name. 
#this can be also done by merging metadata into the `horn_dist_long`
names <- data.frame(str_split_fixed(horn_dist_long$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(horn_dist_long$iso2, "_", 3))
horn_dist_long$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
horn_dist_long$method_1 <- ifelse(grepl("lyPMA", horn_dist_long$iso1),"lypma", 
                                         ifelse(grepl("ben", horn_dist_long$iso1),"benzonase", 
                                                ifelse(grepl("host", horn_dist_long$iso1),"host_zero", 
                                                       ifelse(grepl("qia", horn_dist_long$iso1),"qiaamp", 
                                                              ifelse(grepl("moly", horn_dist_long$iso1),"molysis", 
                                                                     "control")))))


#Adding data for iso 2 also should be done
horn_dist_long$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")
horn_dist_long$method_2 <-ifelse(grepl("lyPMA", horn_dist_long$iso2),"lypma", 
                                        ifelse(grepl("ben", horn_dist_long$iso2),"benzonase", 
                                               ifelse(grepl("host", horn_dist_long$iso2),"host_zero", 
                                                      ifelse(grepl("qia", horn_dist_long$iso2),"qiaamp", 
                                                             ifelse(grepl("moly", horn_dist_long$iso2),"molysis", 
                                                                    "control")))))


#subsetting distances of my interest
horn_dist_long$sample_id_1 <- ifelse(grepl("pos", horn_dist_long$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_1, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_1))
horn_dist_long$sample_id_2 <- ifelse(grepl("pos", horn_dist_long$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_2, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_2))


horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long, horn_dist_long$sample_id_1 == horn_dist_long$sample_id_2) # data within samples

horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long_within_sampleid_from_control,
                                                           horn_dist_long_within_sampleid_from_control$method_1 != horn_dist_long_within_sampleid_from_control$method_2) # remove irrelevant association

horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long_within_sampleid_from_control, (horn_dist_long_within_sampleid_from_control$method_1 == "control") + (horn_dist_long_within_sampleid_from_control$method_2 == "control") != 0)


horn_dist_long_within_sampleid_from_control$treatment <- horn_dist_long_within_sampleid_from_control$method_1

horn_dist_long_within_sampleid_from_control$treatment <- ifelse(horn_dist_long_within_sampleid_from_control$treatment == "control", horn_dist_long_within_sampleid_from_control$method_2, horn_dist_long_within_sampleid_from_control$treatment) 


#Setting key method
horn_dist_long_within_sampleid_from_control$sample_type <- ifelse(grepl("NS", horn_dist_long_within_sampleid_from_control$iso1), "Nasal",
                                                                  ifelse(grepl("CFB", horn_dist_long_within_sampleid_from_control$iso1), "Sputum",
                                                                         ifelse(grepl("BAL", horn_dist_long_within_sampleid_from_control$iso1), "BAL",
                                                                                ifelse(grepl("pos|POS", horn_dist_long_within_sampleid_from_control$iso1, ignore.case = T), "Mock",
                                                                                       ifelse(grepl("neg|N_EXT", horn_dist_long_within_sampleid_from_control$iso1), "Neg.",NA)))))

#Making a column for baseline (controls, from where?)
horn_dist_long_within_sampleid_from_control <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(dist_from = case_when(method_1 == "control" ~ iso1,
                                     method_2 == "control" ~ iso2))

dummy <- data.frame(iso1 = horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           iso2 = horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           dist = 0,
           treatment = "Untreated",
           method_1 = "control",
           method_2 = "control"
           )
names <- data.frame(str_split_fixed(dummy$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(dummy$iso2, "_", 3))
dummy$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
#Adding data for iso 2 also should be done
dummy$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")


#subsetting distances of my interest
dummy$sample_id_1 <- ifelse(grepl("pos", dummy$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_1, ignore.case = T),"Neg.",
                                        dummy$sample_id_1))
dummy$sample_id_2 <- ifelse(grepl("pos", dummy$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_2, ignore.case = T),"Neg.",
                                        dummy$sample_id_2))
dummy$sample_type <- ifelse(grepl("NS", dummy$iso1), "Nasal",
                            ifelse(grepl("CFB", dummy$iso1), "Sputum",
                                   ifelse(grepl("BAL", dummy$iso1), "BAL",
                                          ifelse(grepl("pos|POS", dummy$iso1, ignore.case = T), "Mock",
                                                 ifelse(grepl("neg|N_EXT", dummy$iso1), "Neg.",NA)))))
dummy <- subset(dummy, !is.na(dummy$sample_type))
horn_dist_long_within_sampleid_from_control <- bind_rows(horn_dist_long_within_sampleid_from_control, dummy)

#Here, sample id is the same as subject id.
horn_dist_long_within_sampleid_from_control$subject_id <- horn_dist_long_within_sampleid_from_control$sample_id_1

horn_dist_long_within_sampleid_from_control$treatment <-
        factor(horn_dist_long_within_sampleid_from_control$treatment,
               levels = c("Untreated", "lypma", "benzonase", "host_zero", "molysis", "qiaamp"))

#Making figure of beta diversity distances

## This only includes samples (BAL, Nasal and Sputum)
f3b2 <- rbind(cbind("Sputum",
            lmer(dist ~ treatment + (1|subject_id),
                data = horn_dist_long_within_sampleid_from_control %>%
                        subset(sample_type == "Sputum")) %>% 
                    confint()
            ),
      cbind("BAL",
            lmer(dist ~ treatment + (1|subject_id),
                 data = horn_dist_long_within_sampleid_from_control %>%
                         subset(sample_type == "BAL")) %>% 
                    confint()
            ),
      cbind("Nasal",
            lmer(dist ~ treatment + (1|subject_id),
                 data = horn_dist_long_within_sampleid_from_control %>%
                         subset(sample_type == "Nasal")) %>% 
                    confint()
     )
) %>% {
        row_names <- rownames(.)
        data_frame <- data.frame(.)
        data_frame$treatment <- row_names
        data_frame %>% 
                remove_rownames() %>%
                rename(sample_type = "V1",
               "2.5%" = "X2.5..",
               "97.5%" = "X97.5..") %>% 
                mutate(`2.5%` = as.numeric(`2.5%`),
                       `97.5%` = as.numeric(`97.5%`),
                       mean = (`97.5%`+`2.5%`)/2,
                       treatment = case_when(treatment == "(Intercept)" ~ "Untreated",
                                             treatment == "treatmentlypma" ~ "lyPMA",
                                             treatment == "treatmentbenzonase" ~ "Benzonase",
                                             treatment == "treatmenthost_zero" ~ "HostZERO",
                                             treatment == "treatmentmolysis" ~ "MolYsis",
                                             treatment == "treatmentqiaamp" ~ "QIAamp"
                                             ),
                       treatment = factor(treatment,
                                          levels = c("Untreated", "lyPMA", "Benzonase",
                                      "HostZERO", "MolYsis", "QIAamp")),
                       sample_type = factor(sample_type,
                                          levels = c("BAL", "Nasal", "Sputum"))
                       ) %>%
                subset(treatment %in% c(#"Untreated", 
                                        "lyPMA", "Benzonase",
                                      "HostZERO", "MolYsis", "QIAamp"))
} %>%
        ggplot(aes(x = mean, y = treatment, col = treatment)) +
        geom_point(aes(x=mean), shape=15, size=3) +
        geom_linerange(aes(xmin=`2.5%`, xmax=`97.5%`)) +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c(#"#e31a1c",
                                      "#fb9a99","#33a02c",
                                      "#b2df8a","#1f78b4","#a6cee3"),
                           name = "Treatment",
                           breaks = c(#"Untreated", 
                                      "lyPMA", "Benzonase",
                                      "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Morisita-Horn dissimilarity from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              legend.position = "none") +
        labs(tag = "C") +
        ggtitle("Beta diveristy of microbial species") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "sans") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "sans") +
        scale_x_continuous(breaks = c(-0.25, 0, 0.25, 0.5, 0.75),
                           labels = c(-0.25, "0 (low bias)", 0.25, 0.5, "0.75 (high bias)"))


dat_text <- data.frame(
  label = c(
          #"", "***", "***", "**", "***", #label for Mock
          "*", "", "", "", "", #label for BAL
          "*", "*", "", "**", "", 
          "**", "***", "***", "***", "***"),
  sample_type = c(
          #"Mock", "Mock", "Mock", "Mock", "Mock", 
          "BAL", "BAL", "BAL", "BAL", "BAL", 
          "Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          "Sputum", "Sputum", "Sputum", "Sputum", "Sputum"),
  treatment     = c(
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
  mean = c(
          #50, 30, 35, 33, 31,
          0.6, 0, 0, 0, 0,
          0.32, 0.3, 0, 0.37, 0,
          0.58, 0.75, 0.85, 0.83, 0.85)
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f3b2 <- f3b2 + geom_text(
  data    = dat_text,
  mapping = aes(y = treatment, x = mean, label = label),
  col = "black"
)


f3b2_box <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(across(sample_type, factor, levels=c(#"Mock", 
                                                    "BAL", "Nasal","Sputum"
                                                    ))) %>%
        subset(., .$sample_type != "Neg.") %>% 
        subset(., .$treatment != "Untreated") %>%
  mutate(treatment = factor(treatment, levels = c("lypma", "benzonase", "host_zero", "molysis", "qiaamp"),
                            labels = c("lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))) %>%
        ggplot(aes(x = dist, y = treatment, col = treatment)) +
        geom_boxplot(aes(x=dist)) +
        #geom_linerange(aes(xmin=`2.5%`, xmax=`97.5%`)) +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c(#"#e31a1c",
                                      "#fb9a99","#33a02c",
                                      "#b2df8a","#1f78b4","#a6cee3"),
                           name = "Treatment",
                           breaks = c(#"Untreated", 
                                      "lyPMA", "Benzonase",
                                      "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Morisita-Horn dissimilarity from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              legend.position = "none") +
        #labs(tag = "C") +
        ggtitle("C   Microbial beta diversity") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "sans") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "sans") +
        scale_x_continuous(breaks = c(-0.25, 0, 0.25, 0.5, 0.75),
                           labels = c(-0.25, "0 (low bias)", 0.25, 0.5, "0.75 (high bias)"))


dat_text <- data.frame(
  label = c(
          #"", "***", "***", "**", "***", #label for Mock
          "*", "", "", "", "", #label for BAL
          "*", "*", "", "**", "", 
          "**", "***", "***", "***", "***"),
  sample_type = c(
          #"Mock", "Mock", "Mock", "Mock", "Mock", 
          "BAL", "BAL", "BAL", "BAL", "BAL", 
          "Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          "Sputum", "Sputum", "Sputum", "Sputum", "Sputum"),
  treatment     = c(
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
  dist = c(
          #50, 30, 35, 33, 31,
          0.9, 0, 0, 0, 0,
          0.48, 0.4, 0, 0.55, 0,
          0.8, 0.82,1, 1.02, 0.98)
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f3b2_box <- f3b2_box + geom_text(
  data    = dat_text,
  mapping = aes(y = treatment, x = dist, label = label),
  col = "black"
)



# Mock community's beta-diversity plot.
## This only includes positive control samples (Mock)

f5S_mock_mh <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(across(sample_type, factor, levels=c("Mock"#, 
                                                    #"BAL", "Nasal","Sputum"
                                                    ))) %>%
        subset(., .$sample_type != "Neg.") %>% 
        group_by(sample_type, treatment) %>%
        summarise(mean = mean(dist, na.rm = TRUE),
            sd = sd(dist, na.rm = TRUE),
            n = n()) %>%
        subset(., .$treatment != "Untreated") %>%
  mutate(se = sd / sqrt(n),
         lower.ci = mean - qt(1 - (0.05 / 2), n - 1) * se,
         upper.ci = mean + qt(1 - (0.05 / 2), n - 1) * se,
         treatment = factor(treatment, levels = c(#"Untreated",
                                                  "lypma", "benzonase", "host_zero", "molysis", "qiaamp"),
                            labels = c(#"Untreated",
                                       "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))) %>%
        #,
         #text = paste(sprintf("%.2f", round(mean, digits = 2)), " [", sprintf("%.2f", round(lower.ci, digits = 2)), ", ", sprintf("%.2f", round(upper.ci, digits = 2)), "]", sep = "")) %>%
        ggplot(aes(x = mean, y = treatment, col = treatment)) +
        geom_point(aes(x=mean), shape=15, size=3) +
        geom_linerange(aes(xmin=lower.ci, xmax=upper.ci)) +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c(#"#e31a1c", 
                                      "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", 
                           labels = c(#"Untreated",
                                      "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Morisita-Horn dissimilarity from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              legend.position = "none") +
        labs(tag = "D") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "sans") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "sans") +
        scale_x_continuous(breaks = c(-0.5, 0, 0.5, 1, 1.5), labels = c(-0.5, "0 (low bias)", 0.5, 1, "1.5 (high bias)"))



f5S_mock_mh_box <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(across(sample_type, factor, levels=c("Mock"#, 
                                                    #"BAL", "Nasal","Sputum"
                                                    ))) %>%
        subset(., .$sample_type != "Neg.") %>% 
        subset(., .$treatment != "Untreated") %>%
  mutate(treatment = factor(treatment, levels = c(#"Untreated",
                                                  "lypma", "benzonase", "host_zero", "molysis", "qiaamp"),
                            labels = c(#"Untreated",
                                       "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))) %>%
        #,
         #text = paste(sprintf("%.2f", round(mean, digits = 2)), " [", sprintf("%.2f", round(lower.ci, digits = 2)), ", ", sprintf("%.2f", round(upper.ci, digits = 2)), "]", sep = "")) %>%
        ggplot(aes(x = dist, y = treatment, col = treatment)) +
        geom_boxplot() +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c(#"#e31a1c", 
                                      "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", 
                           labels = c(#"Untreated",
                                      "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Morisita-Horn dissimilarity from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              legend.position = "none") +
        labs(tag = "D") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "sans") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "sans") +
        scale_x_continuous(breaks = c(-0.5, 0, 0.5, 1, 1.5), labels = c(-0.5, "0 (low bias)", 0.5, 1, "1.5 (high bias)"))


fig3 <- ggarrange(f3a, f3a_v, f3b2_box, f3b2_box_v, ncol = 2, nrow =  2, common.legend = T, align = "hv")

fig3

```

```{r}

png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/Figure3_updated.png",   # The directory you want to save the file in
    width = 170, # The width of the plot in inches
    height = 225, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue
fig3

# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()

```

### Updated volcano plot

```{r}

species_italic <- function(data) {
  names <- gsub("_", "~", data)
  names <- gsub("[]]|[[]", "", names)
  names <- gsub("X.Collinsella.", "Collinsella", names)
  names <- gsub("~sp", "')~'sp.'", names)
  names <- gsub(" group", "", names)
  names <- ifelse(grepl("')", names), paste("italic('", names, sep = ""),
                  paste("italic('", names, "')", sep = ""))
  names <- ifelse(grepl("sp[.]", names), names,
                  gsub("~", " ", names))
    names
}

species_italic(filt_maaslin_all$feature)
# Example usage
filt_maaslin_all$italic <- species_italic(filt_maaslin_all$feature)


filt_maaslin_all$group <-  ifelse(filt_maaslin_all$Estimate > 0 & filt_maaslin_all$qval < 0.1, "more",
                     ifelse(filt_maaslin_all$Estimate < 0 & filt_maaslin_all$qval < 0.1, "less",
                            "insignificant"))

figS7 <- filt_maaslin_all %>% mutate(feature = species_italic(feature),
                            feature = case_when(value == "Nasal"~ NA,
                                              value == "BAL"~ NA,
                                              value == "Sputum"~ NA,
                                              qval > 0.01 ~ NA,
                                              abs(Estimate) < 2 ~ NA,
                                              .default = feature),
                            label1 = case_when(group == "more" ~ feature,
                                                       .default = NA),
                            label2 = case_when(group == "less" ~ feature,
                                                       .default = NA)
                            ) %>%
        ggplot(., aes(y = -log10(qval), x = Estimate, col = metadata)) +
        theme_classic(base_family = "sans") +
        #labs(tag = "A") +
        geom_point(size = 2, alpha = 0.7, stroke = 0) +
        xlab("Change estimate of CLR-normalized read counts") +
        ylab("-log<sub>10</sub>(*q*-value)") +
        ylim(c(-1, 25)) +
        xlim(c(-20, 22)) +
        geom_hline(yintercept = 1, col = "gray") +
        geom_vline(xintercept = 0, col = "gray") +
        labs(tag = "A")  +
        ggtitle("Microbial species") +
        #annotate(family = "sans",
        #         geom='richtext',
        #         x=0, y=7,
        #         label = "<i>q</i>-value = 0.1, fold-change = 0") +
        theme(legend.position = "top", axis.title.y = ggtext::element_markdown(), legend.text = element_markdown()) +
        scale_color_manual(values = c("#a65628",
                                      "grey",
                                      "#fb9a99",
                                      "#33a02c",
                                      "#b2df8a",
                                      "#1f78b4",
                                      "#a6cee3"),
                           breaks = c("log10.Final_reads",
                                      "sample_type",
                                      "lypma",
                                      "benzonase",
                                      "host_zero",
                                      "molysis",
                                      "qiaamp"), 
                           labels = c("log<sub>10</sub>(Final reads)",
                                      "Sample type",
                                      "lyPMA",
                                      "Benzonase",
                                      "HostZERO",
                                      "MolYsis",
                                      "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        guides(col = guide_legend(title = "Factors", title.position = "top", nrow = 1)) +
        ggrepel::geom_text_repel(aes(label = label1),
                                  point.padding = unit(0.5, "lines"),
                                  #nudge_y = 10, box.padding = unit(0.35, "lines"),
                                  na.rm = T,
                                  parse = T,
                                  max.overlaps = 34,
                                  grob = "richtext",
                                  show.legend = F,
                                  size=3, segment.size=0.25, nudge_x=15, direction="y") +
        ggrepel::geom_text_repel(aes(label = label2),
                                   box.padding = unit(0.35, "lines"),
                                  #point.padding = unit(0.5, "lines"),
                                  na.rm = T,
                                  parse = T,
                                  max.overlaps = 34,
                                  grob = "richtext",
                                  show.legend = F,
                                  #hjust=0,
                                  size=3,
                                  segment.size=0.25,
                                  nudge_x=-10,
                                  nudge_y = 10,
                                  direction="y")
        

ggarrange(figS7, figS7_host,
          common.legend = T, ncol = 1,align = "hv")
figS7_host
```

```{r}


png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/Figure4_updated.png",   # The directory you want to save the file in
    width = 170, # The width of the plot in inches
    height = 225, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue


ggarrange(figS7, figS7_host,
          common.legend = T, ncol = 1,align = "hv")

# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()
```


### Updated figure 4


```{r, warning=FALSE, message=FALSE}

species_italic <- function(data) {
  names <- gsub("_", " ", rownames(data))
  names <- gsub("[]]|[[]", "", names)
  names <- gsub(" sp", " sp.", names)
  names <- gsub(" sp.", "* sp.", names)
  names <- gsub(" group", "", names)
  names <- ifelse(grepl("[*]", names), paste("*", names, sep = ""), paste("*", names, "*", sep = ""))
  rownames(data) <- names
  data
}
species_revise <- function(data) {
  data$feature <- gsub("Saccharomyces_cerevisiae_x_Saccharomyces_kudriavzevii", "S._cerevisiae x S._kudriavzevii", data$feature)
  data$feature <- gsub("Pseudomonas_aeruginosa_group", "Pseudomonas_aeruginosa", data$feature)
  data$feature <- gsub("Pseudomonas_fluorescens_group", "Pseudomonas_fluorescens", data$feature)
  data
}



make_sig_table <- function(data) {
  sig_data <- spread(data[order(data$qval), c("feature", "metadata", "qval")], metadata, qval)
  sig_data <- species_revise(sig_data)
  sig_data$min <- apply(sig_data %>% dplyr::select(c("lypma", "benzonase", "molysis", "host_zero", "qiaamp")), 1, FUN = min)
  sig_data <- sig_data[order(sig_data$min),] %>% dplyr::select("feature", "lypma", "benzonase", "host_zero", "molysis", "qiaamp") %>% .[1:20,]
  sig_data[["feature"]] <- ifelse(sig_data[["feature"]] == "X.Collinsella._massiliensis", "[Collinsella]_massiliensis", sig_data[["feature"]])
  sig_data_italic <- sig_data %>% rownames_to_column(var = "-") %>%
          column_to_rownames(var = "feature") %>% species_italic %>% dplyr::select(-c("-")) %>%
          rename(lyPMA = lypma,  Benzonase = benzonase, `HostZERO` = host_zero, MolYsis = molysis, QIAamp = qiaamp)
  sig_data_sig <- ifelse(sig_data_italic < 0.1, "*", NA) %>% data.frame(check.names = F)
  return(list(data = sig_data, data_italic = sig_data_italic, data_sig = sig_data_sig))
}



#filt_fit_data_pos <- make_sig_table(filt_fit_data_pos)
filt_fit_data_bal <- make_sig_table(filt_fit_data_bal)
filt_fit_data_ns <- make_sig_table(filt_fit_data_ns)
filt_fit_data_spt <- make_sig_table(filt_fit_data_spt)

#filt_pos_sig <- subset_taxa(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Mock"),
#                                       taxa_names(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Mock")) %in% filt_fit_data_pos$data$feature)

#filt_fit_data_pos$rel <- cbind(filt_pos_sig %>% otu_table %>% t, filt_pos_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>% 
#        .[row.names(filt_fit_data_pos$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")

filt_spt_sig <- subset_taxa(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Sputum"), taxa_names(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Sputum")) %in% filt_fit_data_spt$data$feature)

filt_fit_data_spt$rel <- cbind(filt_spt_sig %>% otu_table %>% t, filt_spt_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>% 
        .[row.names(filt_fit_data_spt$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")

filt_ns_sig <- subset_taxa(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Nasal"),
                                       taxa_names(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "Nasal")) %in% filt_fit_data_ns$data$feature)

filt_fit_data_ns$rel <- cbind(filt_ns_sig %>% otu_table %>% t, filt_ns_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>% 
        .[row.names(filt_fit_data_ns$data_italic),] %>%  mutate_all(~na_if(., 0)) %>% rownames_to_column("feature")

filt_bal_sig <- subset_taxa(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "BAL"),
                                       taxa_names(subset_samples(phyloseq_unfiltered$phyloseq_rel, sample_type == "BAL")) %in% filt_fit_data_bal$data$feature)

filt_fit_data_bal$rel <- cbind(filt_bal_sig %>% otu_table %>% t, filt_bal_sig %>% sample_data) %>% group_by(treatment) %>% summarise_if(is.numeric, mean, na.rm = TRUE) %>% .[, 1:21] %>% column_to_rownames(., "treatment") %>% t () %>% species_italic() %>%  data.frame(check.names = F) %>%
        .[row.names(filt_fit_data_bal$data_italic),] %>%
        mutate_all(~na_if(., 0)) %>% rownames_to_column("feature") %>% subset(., !grepl("NA", .$feature))

#ffff33 qia

f5b <- merge(filt_fit_data_bal$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_bal$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(filt_fit_data_bal$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(is.na(qval) ~ "> 0.1",
                               qval < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%

#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        theme_classic(base_family = "sans") +
        #colors for qvalues
        #xlab("Experimental group") +
        #ylab("Species") +
        #labs(tag = "A")  +
        ggtitle("A  BAL (microbe)") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              #axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 8),
              axis.title.y = element_blank(),
              axis.title.x = element_blank(),
              plot.margin = unit(c(0,0.2,0,1), 'lines'))  +
        scale_fill_manual(values = c("grey", "red"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1)
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "MolYsis",
                                  "qiaamp" = "QIAamp")
                         )

f5c <- merge(filt_fit_data_ns$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_ns$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        
        merge(filt_fit_data_ns$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "sans") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        #ylab("Species") +
        #labs(tag = "C")  +
        ggtitle("C  Nasal swab (microbe)") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              #axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 8),
              axis.title.x = element_blank(),
              plot.margin = unit(c(0,0.2,0,1), 'lines'))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top",
                                   override.aes = list(size=5)),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "MolYsis",
                                  "qiaamp" = "QIAamp")
                         )

f5d <- merge(filt_fit_data_spt$rel %>%
              gather(treatment,
                     value,
                     Untreated:QIAamp,
                     factor_key=TRUE),
      filt_fit_data_spt$data_italic %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     qval,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
      by.x = c('feature', 'treatment'),
      by.y = c('feature', 'treatment'),
      all = T) %>%
        merge(filt_fit_data_spt$data_sig %>%
              rownames_to_column("feature") %>%
              gather(treatment,
                     sig,
                     lyPMA:QIAamp,
                     factor_key=TRUE),
              by.x = c('feature', 'treatment'),
              by.y = c('feature', 'treatment'),
              all = T) %>%
        mutate(sig = case_when(sig < 0.1 ~ "< 0.1",
                               .default = "> 0.1")) %>%
#Baloon plot
        ggballoonplot(size = "value", y = "feature", x= "treatment", fill = "sig") +
        
        theme_classic(base_family = "sans") +
        #colors for qvalues
        gradient_fill(c("#006d2c", "#edf8fb")) +
        xlab("Experimental group") +
        #ylab("Species") +
        #labs(tag = "E")  +
        ggtitle("E  Sputum (microbe)") +
        theme(panel.grid.major = element_line(colour = "grey"),
              legend.position = "top",
              #axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
              #Element markdown for taxa name italicizing
              axis.text.y = ggtext::element_markdown(size = 8),
              #axis.title.x = element_blank(),
              axis.title.x = element_text(margin = margin(t = 0)),
              axis.title.y = element_blank(),
              plot.margin = unit(c(0,0.2,0,1), 'lines'))  +
        #Adding significance asterisks
        scale_fill_manual(values = c("red", "grey"), aes(y = feature,
                      x = treatment,
                      label = sig)) +
        guides(fill = guide_legend(title = c(expression(paste(italic("q"),
                                                       "-value",
                                                       sep = ""))),
                                   title.position = "top"),
               size = guide_legend(title = "Relative abundance",
                                   title.position = "top",
                                   order = 1,
                                   nrow = 1),
               ) + 
        scale_x_discrete(labels=c("control" = "Untreated",
                                  "lypma" = "lyPMA",
                                  "benzonase" = "Benzonase",
                                  "host_zero" = "Host-zero",
                                  "molysis" = "MolYsis",
                                  "qiaamp" = "QIAamp")
                         )



fig4 <- ggarrange(f5c %>% lemon::g_legend() %>% as_ggplot,
                  ggarrange(
                          ggarrange(f5b, f5c, f5d, legend = "none", ncol = 1, align = "hv"),
                          ggarrange(f5b_host, f5c_host, f5d_host, legend = "none", ncol = 1, align = "hv"),
                          ncol = 2, widths = c(2.5,2)),
                  ncol=1, heights = c(1.5, 12),
                  align = "v")




png(file = "Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS8_updated.png",   # The directory you want to save the file in
    width = 170, # The width of the plot in inches
    height = 225, # The height of the plot in inches
    units = "mm",
    res = 600
) #fixing multiple page issue

fig4
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()

fig4


```




### {-}

# Question 3. How was the changes of species richness, stratified by domain?

## Species richenss (metaphlan) by domain {.tabset}

### Stratrifying phyloseq object

Increased species richness in Eukaryota was not observed

```{r}
metaphlan_euk <- phyloseq_unfiltered$phyloseq_count %>% 
        subset_taxa(., Domain == "k__Eukaryota")
sample_data(metaphlan_euk)$S.obs <- ((metaphlan_euk %>% otu_table) != 0 )%>% colSums()
sample_data(metaphlan_euk) %>% group_by(sample_type, treatment) %>% summarise(mean_Sobs = mean(S.obs))

```

*Subsetting phyloseq object having no Eukaryotes*

```{r}
metaphlan_bac_arc <- phyloseq_unfiltered$phyloseq_count %>% 
        subset_taxa(., Domain != "k__Eukaryota")

sample_data(metaphlan_bac_arc) <- alpha_diversity(metaphlan_bac_arc)

```


### Analysis on higher level taxonomy

*Were there differences in phylum richness after treatment?*

No.

```{r}
metaphlan_bac_arc_p <- metaphlan_bac_arc %>% tax_glom(., taxrank = "Phylum")
taxa_names(metaphlan_bac_arc_p) <- metaphlan_bac_arc_p %>% tax_table() %>% data.frame %>% .$Phylum
sample_data(metaphlan_bac_arc_p)$Phylum.obs <- ((metaphlan_bac_arc_p %>% otu_table) != 0)%>% colSums()
sample_data(metaphlan_bac_arc_p) %>% group_by(sample_type, treatment) %>% summarise(mean_phylum_richness = mean(Phylum.obs))
```

*Were there differences in class richness after treatment?*

It seems like a new class was observed at QIAamp treated nasal swab, by metaphan3.

```{r}
metaphlan_bac_arc_class <- metaphlan_bac_arc %>% tax_glom(., taxrank = "Class")
taxa_names(metaphlan_bac_arc_class) <- metaphlan_bac_arc_class %>% tax_table() %>% data.frame %>% .$Class
sample_data(metaphlan_bac_arc_class)$Class.obs <- ((metaphlan_bac_arc_class %>% otu_table) != 0)%>% colSums()
sample_data(metaphlan_bac_arc_class) %>% group_by(sample_type, treatment) %>% 
        summarise(mean_class_richness = mean(Class.obs),
                  sd_class_richness = sd(Class.obs))
```

*Were there differences in class richness after treatment?*

It seems like a new class was observed at QIAamp treated nasal swab, by metaphan3.

### Listing newly found taxa at nasal swab after QIAamp treatment

```{r}
NS_qiaamp_samples <- sample_data %>% 
        subset(., .$sample_type == "Nasal") %>% 
        subset(., .$treatment == "QIAamp") %>%
        .$subject_id


metaphlan3_ns6 <- phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(subject_id == "NS_6") %>% 
        prune_taxa(taxa_sums(.) != 0, .)
control_ns_species <- metaphlan3_ns6 %>%  
        subset_samples(treatment == "Untreated")  %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>% taxa_names()
new_taxa_ns6 <- prune_taxa(!(taxa_names(metaphlan3_ns6) %in% control_ns_species), metaphlan3_ns6)  %>% 
        prune_taxa(taxa_sums(.) != 0, .) 

metaphlan3_ns8 <- phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(subject_id == "NS_8") %>% 
        prune_taxa(taxa_sums(.) != 0, .)
control_ns_species <- metaphlan3_ns8 %>%  
        subset_samples(treatment == "Untreated")  %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>% taxa_names()
new_taxa_ns8 <- prune_taxa(!(taxa_names(metaphlan3_ns8) %in% control_ns_species), metaphlan3_ns8)  %>% 
        prune_taxa(taxa_sums(.) != 0, .) 

metaphlan3_ns17 <- phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(subject_id == "NS_17") %>% 
        prune_taxa(taxa_sums(.) != 0, .)
control_ns_species <- metaphlan3_ns17 %>%  
        subset_samples(treatment == "Untreated")  %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>% taxa_names()
new_taxa_ns17 <- prune_taxa(!(taxa_names(metaphlan3_ns17) %in% control_ns_species), metaphlan3_ns17)  %>% 
        prune_taxa(taxa_sums(.) != 0, .) 

metaphlan3_ns18 <- phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(subject_id == "NS_18") %>% 
        prune_taxa(taxa_sums(.) != 0, .)
control_ns_species <- metaphlan3_ns18 %>%  
        subset_samples(treatment == "Untreated")  %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>% taxa_names()
new_taxa_ns18 <- prune_taxa(!(taxa_names(metaphlan3_ns18) %in% control_ns_species), metaphlan3_ns18)  %>% 
        prune_taxa(taxa_sums(.) != 0, .) 

metaphlan3_ns37 <- phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(subject_id == "NS_37") %>% 
        prune_taxa(taxa_sums(.) != 0, .)
control_ns_species <- metaphlan3_ns37 %>%  
        subset_samples(treatment == "Untreated")  %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>% taxa_names()
new_taxa_ns37 <- prune_taxa(!(taxa_names(metaphlan3_ns37) %in% control_ns_species), metaphlan3_ns37)  %>% 
        prune_taxa(taxa_sums(.) != 0, .) 

newly_found_taxa <- merge_phyloseq(new_taxa_ns6, new_taxa_ns8, new_taxa_ns17, new_taxa_ns18, new_taxa_ns37) %>%
        tax_table %>%
        data.frame


marker_magu_db <- read.csv("@minsik/project_ucas/Marker-magu_20240112/Marker-magu_DB_v1.1_metadata/mpa_vOct22_CHOCOPhlAnSGB_202212_SGB_metadata.tsv", sep = "\t") %>% .$lineage %>% str_split_fixed(., "[|]", 8) %>% 
        data.frame() %>% .[,7]


newly_found_taxa$Species[!(newly_found_taxa$Species %in% marker_magu_db)]

```


Corynebacterium_pseudogenitalium:In Metaphan4 dataset, not in Marker-MAGu DB (from mpa_v_Oct22_CHOCOPhlAnSGB)
Propionibacterium_humerusii:In Metaphan4 dataset, not in Marker-MAGu DB (from mpa_v_Oct22_CHOCOPhlAnSGB)
Propionibacterium_namnetense: Not in both Metaphan4 dataset and Marker-MAGu DB (from mpa_v_Oct22_CHOCOPhlAnSGB)
Olsenella_scatoligenes: In Metaphan4 dataset, not in Marker-MAGu DB (from mpa_v_Oct22_CHOCOPhlAnSGB)
Collinsella_massiliensis: In both Metaphan4 dataset and Marker-MAGu DB (from mpa_v_Oct22_CHOCOPhlAnSGB)

### {-}

# Summary

1. Viral species richness increases for

`BAL + MolYsis` and `Sputum + HostZERO or MolYsis or QIAamp.` 

For nasal swabs, `Nasal swabs + QIAamp` was the most effect combination but the p-value become less significant (Effect size 3.9, p-value: 0.073)

2. With Marker-MAGu, effect of treatments on increase of microbial species richness decreased.

The hypotheiss `Marker-MAGu does not have Eukaryotes in the databse therefore the change was underestimated`, and `Marker-MAGu does not have suffient amount of data base` was rejected.

It could be due to the options of Marker-MAGu: 

Marker-MAGu run with `--detection relaxed`: 33% of marker genes required for detection. 

With Marker-MAGu `--detection default`: 75% of marker genes required for detection.

# To do list


1. It would be helpful if we can run the another option (Marker-MAGu run with `--detection relaxed`) but it would take several more weeks

2. I suggest adding phage data (stratified LMR run) as the last main table to the manuscript. The figures (community data and boxplots of viral species richness) can be added as a supplementary material.


## EXTRA

### PERMANOVA stratified by samples

For it is unable to run statistical test on beta diversity as distances of phage community in BAL samples can not be calculted (too many sample with 0 reads)


```{r}
set.seed(seed)

horn_perm_ns <- vegan::adonis2(distance(subset_samples(v_phyloseq_rel, sample_type == "Nasal"), method="horn") ~ lypma + benzonase + host_zero + molysis + qiaamp,
                               data = subset_samples(v_phyloseq_rel, sample_type == "Nasal") %>%
                                       sample_data %>% data.frame(check.names = F),
                               strata = subset_samples(v_phyloseq_rel, sample_type == "Nasal") %>% 
                                       sample_data %>% data.frame(check.names = F) %>% .$subject_id, permutations = 10000)
#set.seed(seed)
#horn_perm_bal  <- vegan::adonis2(distance(subset_samples(v_phyloseq_rel, sample_type == "BAL"), method="horn") ~  lypma + benzonase + host_zero + molysis + qiaamp, data = subset_samples(v_phyloseq_rel, sample_type == "BAL") %>% sample_data %>% data.frame(check.names = F), strata = subset_samples(v_phyloseq_rel, sample_type == "BAL") %>% sample_data %>% data.frame(check.names = F) %>% .$subject_id, permutations = 10000)
set.seed(seed)
horn_perm_spt <- vegan::adonis2(distance(subset_samples(v_phyloseq_rel, sample_type == "Sputum"), method="horn") ~ lypma + benzonase + host_zero + molysis + qiaamp,
                                data = subset_samples(v_phyloseq_rel, sample_type == "Sputum") %>% sample_data %>% data.frame(check.names = F),
                                strata = subset_samples(v_phyloseq_rel, sample_type == "Sputum")
                                %>% sample_data %>% data.frame(check.names = F) %>% .$subject_id,
                                  permutations = 10000)

```



### {-}

Done.

# Bibliography

```{r warning=FALSE}
#===============================================================================
#BTC.LineZero.Footer.1.1.0
#===============================================================================
#R markdown citation generator.
#===============================================================================
#RLB.Dependencies:
#   magrittr, pacman, stringr
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#BTC.Dependencies:
#   LineZero.Header
#===============================================================================
#Generates citations for each explicitly loaded library.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
str_libraries <- c("r", str_libraries)
for (str_libraries in str_libraries) {
    str_libraries |>
        pacman::p_citation() |>
        print(bibtex = FALSE) |>
        capture.output() %>%
        .[-1:-3] %>% .[. != ""] |>
        stringr::str_squish() |>
        stringr::str_replace("_", "") |>
        cat()
    cat("\n")
}
#===============================================================================
```
