---
title: "COD_20240120_HOST_5_phage_QC"
author: "Minsik Kim"
date: "2024-01-20"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
        df_print: paged
editor_options: 
  chunk_output_type: inline
  markdown: 
    wrap: 72
---

# Loading packages

```{r warning=FALSE, message=FALSE, setup}
#===============================================================================
#BTC.LineZero.Header.1.1.0
#===============================================================================
#R Markdown environment setup and reporting utility.
#===============================================================================
#RLB.Dependencies:
#   knitr, magrittr, pacman, rio, rmarkdown, rmdformats, tibble, yaml
#===============================================================================
#Input for document parameters, libraries, file paths, and options.
#=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=
#knitr::opts_chunk$set(message=FALSE, warning = FALSE)


path_working <- 
        ifelse(sessionInfo()[1]$R.version$platform == "x86_64-pc-linux-gnu",
               "/home/bagel/minsik/",
               ifelse(sessionInfo()[1]$R.version$platform == "aarch64-apple-darwin20",
                      "/Volumes/macdrive/Dropbox/", 
                      "/Dropbox (Personal)"))

path_library <- 
        ifelse(sessionInfo()[1]$R.version$platform == "x86_64-pc-linux-gnu",
               "/home/bagel/R/x86_64-pc-linux-gnu-library/4.1/",
               "/Library/Frameworks/R.framework/Resources/library/")

str_libraries <- c("readxl", "phyloseq", "tidyverse", "pacman", "yaml", "ggplot2", "vegan", "microbiome", "ggpubr", "viridis", "decontam", "gridExtra", "ggpubr", "lme4", "lmerTest", "writexl", "harrietr", "Maaslin2", "ggtext", "ggpmisc", "gamm4", "reshape2", "kableExtra", "knitr", "ggtree", "car", "mediation", "lemon", "qvalue")
        
YAML_header <-
'---
title: "Host-DNA depletion analysis"
author: "Minsik Kim"
date: "2024.01.20"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
---'
seed <- "20230925"

#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Loads libraries, file paths, and other document options.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
FUN.LineZero.Boot <- function() {
    .libPaths(path_library)

    require(pacman)
    pacman::p_load(c("knitr", "rmarkdown", "rmdformats", "yaml"))

    knitr::opts_knit$set(root.dir = path_working)

    str_libraries |> unique() |> sort() -> str_libraries
    pacman::p_load(char = str_libraries)

    set.seed(seed)
}
FUN.LineZero.Boot()
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Outputs R environment report.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

FUN.LineZero.Report <- function() {
    cat("Line Zero Environment:\n\n")
    paste("R:", pacman::p_version(), "\n") |> cat()
    cat("Libraries:\n")
    for (str_libraries in str_libraries) {
        paste(
            "    ", str_libraries, ": ", pacman::p_version(package = str_libraries),
            "\n", sep = ""
        ) |> cat()
    }
    paste("\nOperating System:", pacman::p_detectOS(), "\n") |> cat()
    paste("    Library Path:", path_library, "\n") |> cat()
    paste("    Working Path:", path_working, "\n") |> cat()
    paste("Seed:", seed, "\n\n") |> cat()
    cat("YAML Header:\n")
    cat(YAML_header)
}
FUN.LineZero.Report()

```



## Analysis prep {.tabset}

### Loading data

```{r warning=F, message=FALSE, "Data loading"}
phage_example <- read.csv("@minsik/project_ucas/Marker-magu_20240112/P1039_markger_magu_output/p1039.combined_profile.tsv", sep = "\t")

```

# exploratory data analysis 

```{r warning=F, message=FALSE, "Data wrangling"}
phage_example$SGB
phage_example$P_phase4_control
phage_example$marker.magu_genes
phage_example$S_high
phage_example$lineage %>% str_split_fixed(pattern = "[|]", 8) %>% data.frame %>% .[,8] %>% duplicated
phage_example$lineage %>% str_split_fixed(pattern = "[|]", 8) %>% data.frame %>% .[350,] 
phage_example$lineage %>% str_split_fixed(pattern = "[|]", 8) %>% data.frame %>% .[100,] 
phage_example <- read.csv("@minsik/project_ucas/Marker-magu_20240112/P1039_markger_magu_output/p1039.combined_profile.tsv", sep = "\t")
phage_example$SGB %>% str_split_fixed(pattern = "[|]", 2) %>% data.frame %>% .[100,] 
grep("59748", phage_example$SGB)

```

# Subsetting Viral tax_table & making viral phyloseq object

```{r warning=F, message=FALSE, "Data wrangling"}
phage_example_virus <- subset(phage_example, phage_example$lineage %>% grepl("virus", .))
tax_table_virus <- phage_example_virus$lineage %>% str_split_fixed(pattern = "[|]", 7) %>% {
        data_frame <- data.frame("Kingdom" = .[,1],
                   "Phylum" = .[,2],
                   "Class" = .[,3],
                   "Order" = .[,4],
                   "Family" = .[,5],
                   "Genus" = .[,6],
                   "Species" = .[,7])
        data_frame
}
rownames(tax_table_virus) <- tax_table_virus$Species


row.names(phage_example_virus) <- phage_example_virus$SGB

data_virus <- phage_example_virus%>% dplyr::select(names(phage_example_virus)[365:373])
data_virus

phage_example_virus <- phage_example_virus %>%
        dplyr::select(-c(names(phage_example_virus)[365:373])) %>% 
        dplyr::select(-c("SGB"))
```

# Generating sample_data

the inference about sample type was given by my first impression from the sample_ids (as this is preliminary analysis)

```{r}

sample_data <- data.frame(sample_names = phage_example_virus %>% names) %>%
        mutate(sample_type = ifelse(grepl("mock", sample_names), "Mock DNA",
                                    ifelse(grepl("PosCtrl", sample_names), "Mock community",
                                           ifelse(grepl("dust", sample_names), "Dust samples",
                                                  ifelse(grepl("phase", sample_names), "Ambiguous Ctrls",
                                           "Other samples"))))) %>%
        column_to_rownames("sample_names")
                                    
phage_example_virus
vphyloseq <- merge_phyloseq(sample_data = sample_data(sample_data),
               otu_table = otu_table(phage_example_virus, taxa_are_rows = T),
               tax_table = tax_table(tax_table_virus %>% as.matrix()))


```


# Calculating species richness

```{r}

data <- otu_table(vphyloseq) %>% data.frame
S.obs <- (as.data.frame(otu_table(vphyloseq)) != 0) %>% colSums() %>% data.frame()

#mergind daata - species richenss with sample_data
sample_data(vphyloseq) <- {
        data <- sample_data(vphyloseq)        
        data$species_richness <- S.obs$.
        sample_data(data)
        data
}



sample_data(vphyloseq)

```

# plotting species richenss (viral)

```{r}

sample_names(vphyloseq)
sample_data(vphyloseq) %>%
        as.data.frame() %>%
        ggplot(., aes(x = sample_type, y = species_richness, fill = sample_type)) +
        geom_boxplot() +
        theme_classic(base_size = 10) +
        scale_fill_brewer(type = "qual", palette = 6) +
        ylab("Viral species richness") +
        #theme(axis.text.x = element_text(angle=20, ))
        xlab("Sample type") 


```

```{r}


summary_dataframe <- sample_data(vphyloseq) %>% data.frame %>% group_by(sample_type) %>% summarise(`Species richness` = paste(round(mean(species_richness), 2), "Â±", round(sd(species_richness), 2))) %>% data.frame()


```
# Marker-Magu data processing

## Laoding Marker-MAGu data

```{r}
marker_magu_reads1 <- read.csv("Project_SICAS2_microbiome/4_Data/1_Raw/Sequencing_Baylor_Metagenomics/20240209_PQ00331_marker_magu_output/PQ00331.AlignedReads.tsv", sep = "\t", check.names = T)
marker_magu_reads2 <- read.csv("Project_SICAS2_microbiome/4_Data/1_Raw/Sequencing_Baylor_Metagenomics/20240209_PQ00430_marker_magu_output/PQ00430.without.spikeins.AlignedReads.tsv", sep = "\t", check.names = T)

marker_magu_rpkm1 <- read.csv("Project_SICAS2_microbiome/4_Data/1_Raw/Sequencing_Baylor_Metagenomics/20240209_PQ00331_marker_magu_output/PQ00331.RPKM.tsv", sep = "\t", check.names = T)
marker_magu_rpkm2 <- read.csv("Project_SICAS2_microbiome/4_Data/1_Raw/Sequencing_Baylor_Metagenomics/20240209_PQ00430_marker_magu_output/PQ00430.without.spikeins.RPKM.tsv", sep = "\t", check.names = T)

marker_magu_tax1 <- read.csv("Project_SICAS2_microbiome/4_Data/1_Raw/Sequencing_Baylor_Metagenomics/20240209_PQ00331_marker_magu_output/PQ00331.combined_profile.tsv", check.names = F)
marker_magu_tax2 <- read.csv("Project_SICAS2_microbiome/4_Data/1_Raw/Sequencing_Baylor_Metagenomics/20240209_PQ00430_marker_magu_output/PQ00430.without.spikeins.combined_profile.tsv", check.names = F)

```


## Loading phyloseq data

Loading phylose data to extract sample metadata

```{r}
phyloseq_unfiltered <- readRDS("Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20230521_MGK_host_tidy.rds")

sample_data <- phyloseq_unfiltered$phyloseq_count %>% sample_data


sample_data$baylor_id
```

## QC column names

Some data are duplicated (with .1 as their variable name)


```{r}

marker_magu_rpkm1 %>% names %>% .[grep("[.]1",.)]
marker_magu_rpkm2 %>% names %>% .[grep("[.]1",.)]

```


```{r}
marker_magu_reads1 <-  marker_magu_reads1 %>% 
        dplyr::select(-c((marker_magu_reads1 %>% names %>% .[grep("[.]1", .)])))

marker_magu_reads2 <-  marker_magu_reads2 %>% 
        dplyr::select(-c((marker_magu_reads2 %>% names %>% .[grep("[.]1", .)])))


marker_magu_rpkm1 <-  marker_magu_rpkm1 %>% 
        dplyr::select(-c((marker_magu_rpkm1 %>% names %>% .[grep("[.]1", .)])))

marker_magu_rpkm2 <-  marker_magu_rpkm2 %>% 
        dplyr::select(-c((marker_magu_rpkm2 %>% names %>% .[grep("[.]1", .)])))


names(marker_magu_reads1) <- names(marker_magu_reads1)  %>% gsub("X", "", .)

names(marker_magu_reads2) <- names(marker_magu_reads2)  %>% gsub("X", "", .)


names(marker_magu_rpkm1) <- names(marker_magu_rpkm1)  %>% gsub("X", "", .)

names(marker_magu_rpkm2) <- names(marker_magu_rpkm2)  %>% gsub("X", "", .)


```


# QC on the names of SGBs

```{r}

sbg_table_1 <- marker_magu_rpkm1$SGB %>% 
        str_split_fixed(., "[|]", 2) %>% 
        data.frame(check.names = F,
                   species = .[,1],
                   SBG = .[,2])

sbg_table_2 <- marker_magu_rpkm2$SGB %>% 
        str_split_fixed(., "[|]", 2) %>% 
        data.frame(check.names = F,
                   species = .[,1],
                   SBG = .[,2])


sbg_table_1 %>% subset(., grepl("s__Bifidobacterium", .$species))
sbg_table_2 %>% subset(., grepl("s__Bifidobacterium", .$species))


```


# Subsetting viral tax table



```{r}
magu_microbe1 <- subset(marker_magu_rpkm1, !(marker_magu_rpkm1$lineage %>% grepl("Virus", .)))
magu_microbe2 <- subset(marker_magu_rpkm2, !(marker_magu_rpkm2$lineage %>% grepl("Virus", .)))

magu_virus1 <- subset(marker_magu_rpkm1, marker_magu_rpkm1$lineage %>% grepl("Virus", .)) # This is RPKM data
magu_virus2 <- subset(marker_magu_rpkm2, marker_magu_rpkm2$lineage %>% grepl("Virus", .))

magu_virus1_reads <- subset(marker_magu_reads1, marker_magu_rpkm1$lineage %>% grepl("Virus", .)) # This is the aligned reads
magu_virus2_reads <- subset(marker_magu_reads2, marker_magu_rpkm2$lineage %>% grepl("Virus", .))

```

+ Taxnomic table can be made but it is not informative (all viruses were unclassified)

```{r}

tax_table_microbe1 <- magu_microbe1$lineage %>% str_split_fixed(pattern = "[|]", 8) %>% {
        data_frame <- data.frame("Kingdom" = .[,1],
                   "Phylum" = .[,2],
                   "Class" = .[,3],
                   "Order" = .[,4],
                   "Family" = .[,5],
                   "Genus" = .[,6],
                   "Species" = .[,7],
                   "SBG" = .[,8])
        data_frame
}

tax_table_microbe2 <- magu_microbe2$lineage %>% str_split_fixed(pattern = "[|]", 8) %>% {
        data_frame <- data.frame("Kingdom" = .[,1],
                   "Phylum" = .[,2],
                   "Class" = .[,3],
                   "Order" = .[,4],
                   "Family" = .[,5],
                   "Genus" = .[,6],
                   "Species" = .[,7],
                   "SBG" = .[,8])
        data_frame
}



tax_table_virus1 <- magu_virus1$lineage %>% str_split_fixed(pattern = "[|]", 7) %>% {
        data_frame <- data.frame("Kingdom" = .[,1],
                   "Phylum" = .[,2],
                   "Class" = .[,3],
                   "Order" = .[,4],
                   "Family" = .[,5],
                   "Genus" = .[,6],
                   "Species" = .[,7])
        host_tax <- magu_virus1$iphop_host_genus %>% 
                str_split_fixed(pattern = "[;]", 6) %>% data.frame()
        host_tax <- data.frame("H_Kingdom" = host_tax[,1],
                   "H_Phylum" = host_tax[,2],
                   "H_Class" = host_tax[,3],
                   "H_Order" = host_tax[,4],
                   "H_Family" = host_tax[,5],
                   "H_Genus" = host_tax[,6])
        data_frame <- cbind(data_frame, 
                            host_tax,
                            dplyr::select(.data = magu_virus1, c("contig_id", "marker.magu_genes", "contig_length", "checkv_completeness",
                                                          "iphop_confidence", "Virulence_score")))
}

tax_table_virus2 <- magu_virus2$lineage %>% str_split_fixed(pattern = "[|]", 7) %>% {
        data_frame <- data.frame("Kingdom" = .[,1],
                   "Phylum" = .[,2],
                   "Class" = .[,3],
                   "Order" = .[,4],
                   "Family" = .[,5],
                   "Genus" = .[,6],
                   "Species" = .[,7])
        host_tax <- magu_virus2$iphop_host_genus %>% 
                str_split_fixed(pattern = "[;]", 6) %>% data.frame()
        host_tax <- data.frame("H_Kingdom" = host_tax[,1],
                   "H_Phylum" = host_tax[,2],
                   "H_Class" = host_tax[,3],
                   "H_Order" = host_tax[,4],
                   "H_Family" = host_tax[,5],
                   "H_Genus" = host_tax[,6])
        data_frame <- cbind(data_frame, 
                            host_tax,
                            dplyr::select(.data = magu_virus2, c("contig_id", "marker.magu_genes", "contig_length", "checkv_completeness",
                                                          "iphop_confidence", "Virulence_score")))
}

```

Separating lineage, contig-ids, etc.

```{r}

tax_table_microbe1$SBG <- tax_table_microbe1$SBG %>% gsub("t__", "", .)
tax_table_microbe2$SBG <- tax_table_microbe2$SBG %>% gsub("t__", "", .)

tax_table_virus1$Species <- tax_table_virus1$Species %>% gsub("s__", "", .) %>% gsub("_", "", .)
tax_table_virus2$Species <- tax_table_virus2$Species %>% gsub("s__", "", .) %>% gsub("_", "", .)


rownames(tax_table_microbe1) <- tax_table_microbe1$SBG
rownames(tax_table_microbe2) <- tax_table_microbe2$SBG

rownames(tax_table_virus1) <- tax_table_virus1$Species
rownames(tax_table_virus2) <- tax_table_virus2$Species

tax_table_microbe1 <- tax_table(tax_table_microbe1 %>% as.matrix())
tax_table_microbe2 <- tax_table(tax_table_microbe2 %>% as.matrix())

tax_table_virus1 <- tax_table(tax_table_virus1 %>% as.matrix())
tax_table_virus2 <- tax_table(tax_table_virus2 %>% as.matrix())


row.names(magu_microbe1) <- magu_microbe1$SGB %>% gsub(".*[|]", "", .) %>% gsub("t__", "", .)
row.names(magu_microbe2) <- magu_microbe2$SGB %>% gsub(".*[|]", "", .) %>% gsub("t__", "", .)

row.names(magu_virus1) <- magu_virus1$SGB %>% gsub("s__", "", .) %>% gsub("_", "", .)
row.names(magu_virus2) <- magu_virus2$SGB %>% gsub("s__", "", .) %>% gsub("_", "", .)

row.names(magu_virus1_reads) <- magu_virus1_reads$SGB %>% gsub("s__", "", .) %>% gsub("_", "", .)
row.names(magu_virus2_reads) <- magu_virus2_reads$SGB %>% gsub("s__", "", .) %>% gsub("_", "", .)

magu_microbe1 <- magu_microbe1 %>%
        dplyr::select(-c("SGB", "lineage", "contig_id", "marker.magu_genes", "contig_length", "checkv_completeness", "iphop_host_genus", "iphop_confidence", "Virulence_score"))

magu_microbe2 <- magu_microbe2 %>%
        dplyr::select(-c("SGB", "lineage", "contig_id", "marker.magu_genes", "contig_length", "checkv_completeness", "iphop_host_genus", "iphop_confidence", "Virulence_score"))

magu_virus1 <- magu_virus1 %>%
        dplyr::select(-c("SGB", "lineage", "contig_id", "marker.magu_genes", "contig_length", "checkv_completeness", "iphop_host_genus", "iphop_confidence", "Virulence_score"))

magu_virus2 <- magu_virus2 %>%
        dplyr::select(-c("SGB", "lineage", "contig_id", "marker.magu_genes", "contig_length", "checkv_completeness", "iphop_host_genus", "iphop_confidence", "Virulence_score"))


magu_virus1_reads <- magu_virus1_reads %>%
        dplyr::select(-c("SGB", "lineage", "contig_id", "marker.magu_genes", "contig_length", "checkv_completeness", "iphop_host_genus", "iphop_confidence", "Virulence_score"))

magu_virus2_reads <- magu_virus2_reads %>%
        dplyr::select(-c("SGB", "lineage", "contig_id", "marker.magu_genes", "contig_length", "checkv_completeness", "iphop_host_genus", "iphop_confidence", "Virulence_score"))



```

# Making 2 phyloseq objects

```{r}

m_otu1 <- otu_table(magu_microbe1, taxa_are_rows = T)

m_otu2 <- otu_table(magu_microbe2, taxa_are_rows = T)



v_otu1 <- otu_table(magu_virus1, taxa_are_rows = T)

v_otu2 <- otu_table(magu_virus2, taxa_are_rows = T)


v_otu1_reads <- otu_table(magu_virus1_reads, taxa_are_rows = T)

v_otu2_reads <- otu_table(magu_virus2_reads, taxa_are_rows = T)


```

# Double-checking if SBG matches

```{r}
rbind(tax_table_virus1 %>% data.frame %>% subset(., .$Species == "vSGB67035"),
      tax_table_virus2 %>% data.frame %>% subset(., .$Species == "vSGB67035"))

tax_table_virus2

rbind(tax_table_virus1 %>% data.frame %>% subset(., .$Species == "vSGB31342"),
      tax_table_virus2 %>% data.frame %>% subset(., .$Species == "vSGB31342"))


```

Generating & merging phyloseq object

```{r}


m_phy1 <- merge_phyloseq(m_otu1, sample_data(phyloseq_unfiltered$phyloseq_count), tax_table_microbe1)
v_phy1 <- merge_phyloseq(v_otu1, sample_data(phyloseq_unfiltered$phyloseq_count), tax_table_virus1)
v_phy1_reads <- merge_phyloseq(v_otu1_reads, sample_data(phyloseq_unfiltered$phyloseq_count), tax_table_virus1)

dummy_phyloseq <- phyloseq_unfiltered$phyloseq_count
sample_names(dummy_phyloseq) <- sample_data(dummy_phyloseq)$baylor_id

m_phy2 <- merge_phyloseq(m_otu2, sample_data(dummy_phyloseq), tax_table_microbe2)
v_phy2 <- merge_phyloseq(v_otu2, sample_data(dummy_phyloseq), tax_table_virus2)
v_phy2_reads <- merge_phyloseq(v_otu2_reads, sample_data(dummy_phyloseq), tax_table_virus2)


m_phyloseq <- merge_phyloseq(m_phy1, m_phy2)
v_phyloseq <- merge_phyloseq(v_phy1, v_phy2)
v_phyloseq_reads <- merge_phyloseq(v_phy1_reads, v_phy2_reads)

m_phyloseq <- prune_taxa(taxa_sums(m_phyloseq) != 0, m_phyloseq)
v_phyloseq <- prune_taxa(taxa_sums(v_phyloseq) != 0, v_phyloseq)
v_phyloseq_reads <- prune_taxa(taxa_sums(v_phyloseq_reads) != 0, v_phyloseq_reads)

sample_names(m_phyloseq) <- sample_data(m_phyloseq)$baylor_other_id
sample_names(v_phyloseq) <- sample_data(v_phyloseq)$baylor_other_id
sample_names(v_phyloseq_reads) <- sample_data(v_phyloseq_reads)$baylor_other_id

sample_sums(m_phyloseq)
sample_sums(v_phyloseq)
sample_sums(v_phyloseq_reads)

phyloseq_unfiltered$phyloseq_count %>% 
        sample_names %>%
        .[!(phyloseq_unfiltered$phyloseq_count %>% sample_names) %in% (v_phyloseq %>% sample_names)]


phyloseq <- list()

phyloseq$microbe_rpkm <- m_phyloseq
phyloseq$viral_rpkm <- v_phyloseq
phyloseq$viral_reads <- v_phyloseq_reads
```

1 sample is missing (from the original .tsv file)

# Save RDS

```{r}
saveRDS(phyloseq, "Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20240212_MGK_host_marker_magu.rds")
```



# Bibliography

```{r warning=FALSE}
#===============================================================================
#BTC.LineZero.Footer.1.1.0
#===============================================================================
#R markdown citation generator.
#===============================================================================
#RLB.Dependencies:
#   magrittr, pacman, stringr
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#BTC.Dependencies:
#   LineZero.Header
#===============================================================================
#Generates citations for each explicitly loaded library.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
str_libraries <- c("r", str_libraries)
for (str_libraries in str_libraries) {
    str_libraries |>
        pacman::p_citation() |>
        print(bibtex = FALSE) |>
        capture.output() %>%
        .[-1:-3] %>% .[. != ""] |>
        stringr::str_squish() |>
        stringr::str_replace("_", "") |>
        cat()
    cat("\n")
}
#===============================================================================
```
