---
title: "COD_20240212_HOST_6_phage_analysis_figure_table"
author: "Minsik Kim"
date: "2024-02-12"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
        df_print: paged
editor_options: 
  chunk_output_type: inline
  markdown: 
    wrap: 72
---

# Loading packages

```{r warning=FALSE, message=FALSE, setup}
#===============================================================================
#BTC.LineZero.Header.1.1.0
#===============================================================================
#R Markdown environment setup and reporting utility.
#===============================================================================
#RLB.Dependencies:
#   knitr, magrittr, pacman, rio, rmarkdown, rmdformats, tibble, yaml
#===============================================================================
#Input for document parameters, libraries, file paths, and options.
#=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=
#knitr::opts_chunk$set(message=FALSE, warning = FALSE)


path_working <- 
        ifelse(sessionInfo()[1]$R.version$platform == "x86_64-pc-linux-gnu",
               "/home/bagel/minsik/",
               ifelse(sessionInfo()[1]$R.version$platform == "aarch64-apple-darwin20",
                      "/Volumes/macdrive/Dropbox/", 
                      "/Dropbox (Personal)"))

path_library <- 
        ifelse(sessionInfo()[1]$R.version$platform == "x86_64-pc-linux-gnu",
               "/home/bagel/R/x86_64-pc-linux-gnu-library/4.1/",
               "/Library/Frameworks/R.framework/Resources/library/")

str_libraries <- c("readxl", "phyloseq", "tidyverse", "pacman", "yaml", "ggplot2", "vegan", "microbiome", "ggpubr", "viridis", "decontam", "gridExtra", "ggpubr", "lme4", "lmerTest", "writexl", "harrietr", "Maaslin2", "ggtext", "ggpmisc", "gamm4", "reshape2", "kableExtra", "knitr", "ggtree", "car", "mediation", "lemon", "qvalue")
        
YAML_header <-
'---
title: "Host-DNA depletion analysis of bacterio phage"
author: "Minsik Kim"
date: "2024.02.12"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
---'
seed <- "20230925"

#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Loads libraries, file paths, and other document options.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
FUN.LineZero.Boot <- function() {
    .libPaths(path_library)

    require(pacman)
    pacman::p_load(c("knitr", "rmarkdown", "rmdformats", "yaml"))

    knitr::opts_knit$set(root.dir = path_working)

    str_libraries |> unique() |> sort() -> str_libraries
    pacman::p_load(char = str_libraries)

    set.seed(seed)
}
FUN.LineZero.Boot()
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Outputs R environment report.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

FUN.LineZero.Report <- function() {
    cat("Line Zero Environment:\n\n")
    paste("R:", pacman::p_version(), "\n") |> cat()
    cat("Libraries:\n")
    for (str_libraries in str_libraries) {
        paste(
            "    ", str_libraries, ": ", pacman::p_version(package = str_libraries),
            "\n", sep = ""
        ) |> cat()
    }
    paste("\nOperating System:", pacman::p_detectOS(), "\n") |> cat()
    paste("    Library Path:", path_library, "\n") |> cat()
    paste("    Working Path:", path_working, "\n") |> cat()
    paste("Seed:", seed, "\n\n") |> cat()
    cat("YAML Header:\n")
    cat(YAML_header)
}
FUN.LineZero.Report()

```




# 1. Loading data

1.1. phyloseq obejct

1.2. qPCR data (controls)

## LIST OF PRIMARY QUESTIONS AND CORRESPONDING ANALYSES {.tabset}

STUDY AIMS

Aim 1. What is the efficiency of host depletion for each method?
        
        •	% host DNA measured by mNGS
        o	Sequencing failure rates
        •	% host DNA measured by qPCR 

Aim 2. Did host depletion change microbial community composition?

        •	Alpha diversity (microbial species richness, microbial predicted functional richness)
        •	Beta diversity (Morisita-Horn distance compared to control (not host-depleted) sample)
        •	Differential abundance

Aim 3: Is there effect modification by sample type?

Aim 4. Does host depletion increase the risk of contamination?



# Aim 1: What is the efficiency of host depletion for each method?

## 1a. Did treatment change % host DNA?

## Figure

        A: Raw reads (do you really need to log10 transform y-axis?)
        B: Host mapped reads
        C: Final reads (QC’d, non-human)
        D: % Host DNA 

## Statistical model

        ### linear mixed effects model to account for repeated measures (multiple aliquots per individual)
        ### Outcome = % host DNA (mNGS reads mapping to human genome/QC’d reads)
        ### Predictors
                Host depletion method (categorical, comparison group = control not host-depleted)
                Sample type
        ### testing for interaction term to justify stratified analysis
        ###  % host DNA ~ method + sample_type + method*sample_type + (1|subjid), report interaction p-value.
        ### Stratified analysis
        ###  %host ~ method + (1|subjid), report beta [95% CI], p-value for each sample type
        
## 1b. Did host depletion work successfully for sequencing?
        
## Statistical Model 
        
        ### Logistics mixed effects model
                ### Outcome = library prep and sequencing failure rate
                ### (==1 if failed library prep, ==0 if not)
                ### Report in text sequencing failure (n) stratified by sample type and treatment method.
        ## testing for interaction term to justify stratified analysis
                ### fail ~ method + sample_type + method*sample_type + (1|subjid), report interaction p-value.
        ### stratified analysis: 
                ### fail ~ method + (1|subjid), report OR [95% CI], p-value for each sample type

## Final reads

        ### Investigate distribution of final reads and apply proper transformation.

### Figure 

        A: Raw reads
        B: Host mapped reads
        C: Final reads
        D: Host DNA ratio
        
### model (to justify stratified analysis)
        
        ### final reads ~ method + sample_type + method*sample_type + (1|subjid), just report interaction p-value.

### model (stratified analysis)

        # final reads ~ method + (1|subjid), report effect size [95% CI], p-value for each sample type.
        
# Aim 2. Did host depletion change microbial community composition?

## Figure 

        # Alpha diversity.  Species richness by treatment, facetted by sample type

        # Beta diversity. Morisita-Horn distance of each treated compared to untreated sample  
        # forest plot - x-axis as distance (size of bias) and y axis as treatment. Facet data by sample type

## Alpha diversity

        # Linear mixed effects model
                ## Outcome 
                ### species richness
                ## inverse simpson
        # stratified analyses by sample type
                ## report the significant method*sample_type interaction term to justify stratified analysis)
                ## species_richness ~ method + sample_type + method * sample_type + (1|subjid)
                ## InvSimp~ sample type + treatment + sample type * treatment + (1|subject_id) 

## Beta diversity 
        
        ### Outcome = Morisita Horn 
        # PERMANOVA
                ## Overall: MH ~ sample type + treatment + sample type * treatment, strata = subject_id ) 
                ## Stratified: MH ~ method + strata = subject_id
        # Calculate Morisita-Horn distance between control and host depleted sample and use that as an outcome in linear model
                ## Change in MH ~ method + sample_type + method*sample_type + log10(reads))
## Differential abundance of species 

        # Outcome = relative abundance of species

## Figure

        # Volcano plot with Mock, BAL, Nasal and Sputum 
                ## Note: Mock community placed in Zymo DNA/RNA Protect prior to freezing. Zymo DNA/RNA protect is a mild detergent thus increases susceptibility of microbial cells to lysis and is not recommended to put these samples through host depletion 

## Statistical model
        
        # MaAsLin2
                ## Stratified by sample type
                        ### Species ~ + lyPMA + Benzonase + HostZERO + MolYsis + QIAamp + (1|subjid). Comparison group is untreated sample
                ## Figure: Balloon plots for BAL, Nasal and Sputum (Figure). Add q-val, mean relative abundance
        
## Proportion gram negative 
        
        # Species collapsed into single category: gram negative bacteria, gram positive bacteria, fungi
        # Statistical model
                ## % gram negative ~ sample type + treatment + sample type * treatment + (1|subject_id) ) 
                ## Stratified analysis
                        ### % gram negative ~ treatment + (1|subject_id) ) 

## Does change differ by sample type for results of predicted function as well? Repeat all analyses above analyses but use predicted microbial function (KEGG, CPM) instead

# Aim 3: Is there effect modification by sample type?
        
## All above analyses with interaction term and stratified analyses

## If effect modification present, which treatment is the best for each treatment? 

## Make a summary result by treatment, stratified by sample type

        ### Issues from sequencing (library failure, no change in host depletion, etc.)
        ### Changes in alpha diversity (Mean species richness change by each subject)
        ### Changes in beta diversity (statistical test results with significant factor)
        
# Secondary analyses
## 1.	Is qPCR an alternative to mNGS for estimating % Host DNA? 
### Figure
        
        ### A: Correlation Plot. x-axis with host DNA proportion measured with shotgun metagenomic sequencing vs y-axis as that by qPCR.
### Statistics
        ### Correlation coefficient
        ### Bland-Altman statistics

## 2. Mediation analysis. Host depletion treatment increases species and predicted microbial functional richness due higher effective sequencing depth 

        # Mediation R package
                ## outcome = species richness
                ## exposure = each host depletion treatment compared to untreated control
                ## mediator = final reads
                ## mediator-outcome confounders = sample type
                ## exposure-mediator confounders = NA
                ## outcome model = Mixed effects linear regression 
                ## mediator model = Mixed effects linear regression

# Aim 4. Does host depletion increase the risk of contamination?

1. Were there any contaminants in the sequencing result? If species richness were increased after treatment, is it due to increased coverage with higher final reads?

        # Run decontam (Davis 2018. PMID: 30558668)
                ## List potential contaminants with their prevalences in samples and negative controls 
                ## Sensitivity analysis where potential contaminant species identified removed
                        ### species richness  ~ treatment + (1|subjid)). 
        # Decontaminate data by estimating microbial population using mock community data.
                ## Run Tinyvamp  (arXiv: 2204.12733)
                ## Make adjusted relative abundance table by calculating taxon-specific detection efficiencies, relative to a reference taxa (Enterococcus).
                ## Known community for efficiency estimation: negative (no taxa) + positive (mock community
        

# Data inputs

*Meta data*

-   qPCR - bacteria

-   qPCR - human

-   qPCR host %

-   Raw reads

-   final reads

-   sequencing host %

-   library prep failure status

-   Raw reads

-   subject_id

-   treatment

-   sample_type

-   subject_id

*Sequencing result*

-   samples

-   controls

###  {-}

# Aanalysis preparation


## Analysis prep {.tabset}

### Loading data

```{r warning=F, message=FALSE, "Data loading"}
# Loading files -----------------------------------------------------------
#loading tidy phyloseq object
phyloseq <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20240212_MGK_host_marker_magu.rds")

m_phyloseq <- phyloseq$microbe_rpkm
v_phyloseq <- phyloseq$viral_rpkm


```


### Alpha diversity indices

```{r}           
alpha_diversity <- function(data) {
        otu_table <- otu_table(data) %>% .[colSums(.) !=0]
        S.obs <- rowSums(t(otu_table) != 0)
        sample_data <- sample_data(data)
        data_evenness <- vegan::diversity(t(otu_table)) / log(vegan::specnumber(t(otu_table))) # calculate evenness index using vegan package
        data_shannon <- vegan::diversity(t(otu_table), index = "shannon") # calculate Shannon index using vegan package
        data_hill <- exp(data_shannon)                           # calculate Hills index
        data_dominance <- microbiome::dominance(otu_table, index = "all", rank = 1, aggregate = TRUE) # dominance (Berger-Parker index), etc.
        data_invsimpson <- vegan::diversity(t(otu_table), index = "invsimpson")                          # calculate Shannon index using vegan package
        alpha_diversity <- cbind(S.obs, data_shannon, data_hill, data_invsimpson, data_evenness,data_dominance) # combine all indices in one data table
        sample_data <- merge(data.frame(sample_data), alpha_diversity, by = 0, all = T) %>% column_to_rownames(var = "Row.names")
}

sample_data(m_phyloseq) <- sample_data(alpha_diversity(m_phyloseq)) 

sample_data(v_phyloseq) <- sample_data(alpha_diversity(v_phyloseq)) 



#sample data loading
m_sample_data <- sample_data(m_phyloseq) %>% data.frame()

sample_data(v_phyloseq)$Viral_reads <- sample_sums(v_phyloseq)
v_sample_data <- sample_data(v_phyloseq) %>% data.frame()




```

## {-}

# **3.1. Screening of treatment effect**

# Summary

## Summary table {.tabset}                

### Summary table of species richness

**Table Sx**. 

```{r}

m_sample_data %>%
        group_by(sample_type, treatment) %>%
        summarise(Mean_SObs = mean(S.obs),
                  SD_SObs = sd(S.obs),
                  Max_SObs = max(S.obs),
                  Min_SObs = min(S.obs),
                  N = n())

v_sample_data %>%
        group_by(sample_type, treatment) %>%
        summarise(Mean_SObs = mean(S.obs),
                  SD_SObs = sd(S.obs),
                  N = n())
        
```
1. Mock community had 8 microbial taxa in average. Did not show fungi reads.

2. The marker-magu + metaphlan 4 outcome showed relatively lower viral speciese richness compared to microbial richness.




```{r}

m_phyloseq %>%
        subset_samples(sample_type == "Mock") %>%
        otu_table %>%   view

m_phyloseq %>%
        subset_samples(sample_type == "Nasal") %>% 
        otu_table %>% 
        head

m_phyloseq %>% 
        tax_table %>%
        head
        
```

# Metaphlan 3

```{r}

# Loading files -----------------------------------------------------------
#loading tidy phyloseq object
phyloseq_unfiltered <- read_rds("Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20230521_MGK_host_tidy.rds")

rbind(

phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(sample_type == "BAL") %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>%
        tax_table %>%
        data.frame %>%
        .$Domain %>%
        table %>% 
        data.frame() %>%
        mutate(`Sample type` = "BAL", .before = "."),

phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(sample_type == "Nasal") %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>%
        tax_table %>%
        data.frame %>%
        .$Domain %>%
        table %>% 
        data.frame() %>%
        mutate(`Sample type` = "Nasal", .before = "."),


phyloseq_unfiltered$phyloseq_count %>%
        subset_samples(sample_type == "Sputum") %>% 
        prune_taxa(taxa_sums(.) != 0, .) %>%
        tax_table %>%
        data.frame %>%
        .$Domain %>%
        table %>% 
        data.frame() %>%
        mutate(`Sample type` = "Sputum", .before = ".")
) %>% 
        rename(Domain = ".",
               Frequency = Freq)


```

By sample type, the proportion of Eukaryota (mainly fungi) was different from Metaphlan3 output.


### {-}

# *ii.	How were changes in sequencing results?*        

## Changes in sequencing output {.tabset}

**Table 1 is generated later (after QCing to remove sequencing failed samples, etc.)**

## Library failure {.tabset}


### Failure crosstab

```{r}
sample_data <- sample_data(v_phyloseq) %>% 
		  data.frame() %>%
		  mutate(
		      sample_sums = sample_sums(v_phyloseq),
		      lib_failed = case_match(lib_failed, TRUE ~ 1, FALSE ~0),
		      seq_failed = ifelse(sample_sums == 0, 1, 0),
		      lib_seq_failed = ifelse(lib_failed ==1 | seq_failed == 1, 1, 0),
		      prop_host = round((Host_mapped/Reads_after_trim), 1)
		      )
cat("Sequencing failed")		
xtabs(seq_failed ~ sample_type + treatment, data = sample_data)
cat("Sequencing and library failed")		
xtabs(lib_seq_failed ~ sample_type + treatment, data = sample_data)
```


### {-}

## Figure of sequencing result {.tabset}

### Fig. SX. Figure of sequencing result (Viral reads)

*ii.	How were changes in sequencing results?*

 **Fig. SX.**  Host depletion effects on total viral reads measured by shotgun metagenomic sequencing. 

```{r}

fs3a <- ggplot(sample_data %>% subset(., .$sample_type %in% c("BAL", "Nasal", "Sputum")), aes(x = sample_type, y = log10(Viral_reads))) +
        geom_jitter(aes(col = treatment, x = treatment),
                    lwd = 0.2, alpha = 0.3, stroke = 0)+
        stat_summary(aes(color = treatment, x = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        theme_classic (base_size = 12, base_family = "sans")+ 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        scale_x_discrete(name ="Treatment")+
        theme(axis.title.y = element_markdown(),
              plot.tag = element_text(size = 15),
              axis.text.x = element_blank()) +
        ylab("log<sub>10</sub>(Viral RPKM)") +
        labs(tag = "A") +
        facet_wrap(~sample_type, scale = "free_x") +
        guides(fill = guide_legend(nrow = 1))

#	- Host_mapped



fs3a


#png(file = "/Users/minsikkim/Dropbox (Personal)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/FigureS3.png",   # The directory you want to save the file in
#    width = 180, # The width of the plot in inches
#    height = 170, # The height of the plot in inches
#    units = "mm",
#    res = 600
#) #fixing multiple page issue

#figS3
# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

#dev.off()



```

For BAL and Sputum, viral reads clearly increased after treatment. For nasal swabs, such pattern was not observed.

### {-}

## decontam {.tabset}

### Table SX. decontam - stratified by sample_type (prevalence method)

Note: decontam was conducted only with prevalence method, as quantification on viruses was not conducted via qPCR.



```{r warning=FALSE, message=FALSE}
#Stratified by sample type
neg_number_decontam <- 
        v_phyloseq %>% 
        subset_samples(sample_type == "Neg.") %>%
        sample_names() %>%
        length

sample_number_decontam <- 
        v_phyloseq %>% 
        subset_samples(sample_type != "Mock") %>%
        subset_samples(sample_type != "Neg.") %>%
        #subset_samples(S.obs != 0) %>%
        sample_names() %>%
        length

prev_neg <- ((v_phyloseq %>% 
          subset_samples(sample_type == "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (negative controls)" = ".")

prev_all <- ((v_phyloseq %>%
                      subset_samples(sample_type != "Mock") %>%
                      subset_samples(sample_type != "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (all)" = ".")


sample_data(v_phyloseq)$is.neg <- grepl("Neg", sample_data(v_phyloseq)$sample_type)



phyloseq_decontam_bal <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(sample_type == "Neg." | sample_type == "BAL")
phyloseq_decontam_ns <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(sample_type == "Neg." | sample_type == "Nasal")
phyloseq_decontam_spt <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(sample_type == "Neg." | sample_type == "Sputum")



contaminant_combined_bal <- 
data.frame("BAL", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_bal, method="prev", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)


contaminant_combined_ns <- 
data.frame("Nasal", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_ns, method="prev", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)



contaminant_combined_spt <- 
data.frame("Sputum", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_spt, method="prev", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)

contaminants <- rbind(contaminant_combined_bal, contaminant_combined_ns, contaminant_combined_spt)

names(contaminants) <- c("Sample type", "Taxa")


merged_contaminants <- merge(contaminants, prev_all %>% 
                                     rownames_to_column("Taxa"), by = "Taxa") %>%
        merge(., prev_neg %>%
                      rownames_to_column("Taxa"), by = "Taxa") %>%
       dplyr::select(c("Sample type",
                       "Taxa",
                       "Prevalence (all)",
                       "Prevalence (negative controls)")) %>%
        subset(., .$"Prevalence (all)" != 0) %>%
        mutate(`Prevalence (all)` = paste0(`Prevalence (all)`,
                                           " (",
                                           round(`Prevalence (all)`/
                                                         sample_number_decontam * 100, 2),
                                           "%)"),
                `Prevalence (negative controls)` = paste0(`Prevalence (negative controls)`,
                                           " (",
                                           round(`Prevalence (negative controls)`/
                                                         neg_number_decontam * 100, 2),
                                           "%)")) %>%
        rename(`Prevalence (all) N = 94` = `Prevalence (all)`,
               `Prevalence (negative controls) N = 31` = `Prevalence (negative controls)`) %>%
        .[order(.$"Sample type", .$"Taxa"),] %>%
        remove_rownames()


merged_contaminants



```

At sample type stratified analysis, decontam identified 2 possible contaminants (that were identified in negative controls).

### Decontam result (combined method)

Combined methods showd slightly different result but as total concentration of viruses are not directly correlated with bacteiral qPCR result this data was aborted.

Stratified-BAL


```{r}

isContaminant(phyloseq_decontam_bal, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) 

```

Stratified-Nasal

```{r}

isContaminant(phyloseq_decontam_ns, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) 

```

Stratified-Sputum

```{r}

isContaminant(phyloseq_decontam_spt, method="combined", neg = "is.neg", threshold = 0.1, conc = "DNA_bac_ng_uL") %>% subset(.,.$contaminant) 

```
### Decontam -stratified by treatment

As a sanity check, decontam analysis was conducted after stratifying by treatment. Only 3 SGBs were nominated as potential candidates.

```{r}
#Stratified by sample type

prev_neg <- ((v_phyloseq %>% 
          subset_samples(sample_type == "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (negative controls)" = ".")

prev_all <- ((v_phyloseq %>%
                      subset_samples(sample_type != "Mock") %>%
                      subset_samples(sample_type != "Neg.") %>%
          otu_table %>% 
          data.frame(.)) != 0) %>%
        rowSums %>% data.frame %>% rename("Prevalence (all)" = ".")


sample_data(v_phyloseq)$is.neg <- grepl("Neg", sample_data(v_phyloseq)$sample_type)

phyloseq_decontam_lypma <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "lyPMA")

phyloseq_decontam_ben <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "Benzonase")

phyloseq_decontam_hostzero <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "HostZero")

phyloseq_decontam_molysis <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "MolYsis")

phyloseq_decontam_qiaamp <- v_phyloseq %>% 
        #subset_samples(S.obs != 0) %>%
        subset_samples(treatment == "Untreated" | treatment == "QIAamp")


#lypma showed zero contaminant
isContaminant(phyloseq_decontam_lypma, method="prevalence", neg = "is.neg", threshold = 0.1) %>% subset(.,.$contaminant) 

contaminant_combined_ben <- 
data.frame("Benzonase", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_ben, method="prevalence", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)

#Host zero showed zero contaminant
isContaminant(phyloseq_decontam_hostzero, method="prevalence", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant)

contaminant_combined_molysis <- 
data.frame("MolYsis", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_molysis, method="prevalence", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)

contaminant_combined_QIAamp <- 
data.frame("QIAamp", fix.empty.names = F, 
        isContaminant(phyloseq_decontam_qiaamp, method="prevalence", neg = "is.neg", threshold = 0.5) %>% subset(.,.$contaminant) %>% row.names
)

contaminants_strat_treat <- rbind(#contaminant_combined_lypma,
                      contaminant_combined_ben, 
                      #contaminant_combined_hostzero,
                      contaminant_combined_molysis,
                      contaminant_combined_QIAamp
                      )

names(contaminants_strat_treat) <- c("Treatment", "Taxa")

contaminants_strat_treat

```

5 taxa identified as contaminants.

### Decontam summary

Summary table of potential contaminants with all sample types and stratified sample types in all methods (prevalence, frequence, and combined)

```{r}

tableS10 <- merged_contaminants %>% 
        mutate(Taxa = species_italic2(Taxa)
               ) %>%
        kbl(format = "html", escape = F) %>%
        kable_styling(full_width = 0, html_font = "sans") 
tableS10


```

### {-}


## Filtering taxa {.tabset}

### Prevalence filtering 

**Prevalence & abundance filtering was conducted**

v.	M&M - Prevalence filtering

        1.	Prevalence filtration at 5%, except its abundance is over 0.75 quantile.
                a.	Information in the main text

```{r warning=FALSE, "Red flag"}

v_phyloseq_rel <- transform_sample_counts(v_phyloseq,
                                                            function(x){x/sum(x)})


taxa_qc <- data.frame("species" =
                              otu_table(
                                      subset_samples(
                                              v_phyloseq,S.obs != 0 &
                                                      sample_type %in% c("Mock", "BAL", "Nasal", "Sputum"))) %>%
                              t() %>% colnames(),
                      "prevalence" =
                              ifelse(subset_samples(v_phyloseq, S.obs != 0 & 
                                                            sample_type %in% c("Mock", "BAL", "Nasal", "Sputum")) %>%
                                             otu_table() > 0, 1, 0)%>% 
                              t() %>%
                              colSums(), #Prevalence of taxa
                      "mean_rel_abd" = 
                              subset_samples(v_phyloseq,
                                             S.obs != 0 & sample_type %in% c("Mock", "BAL", "Nasal", "Sputum")) %>%
                              otu_table() %>%
                              t() %>%
                              colMeans(na.rm = T) #mean relativ abundacne 
)

red_flag_taxa <- data.frame(species = taxa_qc$species,
                            prevalence = taxa_qc$prevalence,
                            mean_rel_abd = taxa_qc$mean_rel_abd,
                            red_flag_prev_abd = 
                                    ifelse(taxa_qc$prevalence < 
                                                   otu_table(
                                                           subset_samples(
                                                                   v_phyloseq,
                                                                   S.obs != 0 & 
                                                                           sample_type %in% 
                                                                           c("Mock", "BAL", "Nasal", "Sputum"))) %>%
                                                               t %>% rownames() %>%
                                                   length * 0.05 &
                                                   #Removing taxa with zero prevalence - taxa from nasal swabs
                                                   taxa_qc$mean_rel_abd <
                                                   taxa_qc %>%
                                                   subset(., .$prevalence != 0) %>%
                                                   .$mean_rel_abd %>%
                                                   quantile(., 0.75), 1,0),
                           red_flag_prev =
                                    ifelse(taxa_qc$prevalence < 
                                                   otu_table(
                                                           subset_samples(
                                                                   v_phyloseq,
                                                                   S.obs != 0 & 
                                                                           sample_type %in% 
                                                                           c("Mock", "BAL", "Nasal", "Sputum"))) %>%
                                                               t %>% rownames() %>%
                                                   length * 0.05,
                                           1,
                                           0)) %>%
        mutate(red_flag_decontam = species %in% (contaminants$Taxa %>% unique()))


```

### Filtration result

Taxa were filtered wehn `red_flag_prev_abd == 1`.

```{r}
red_flag_taxa
```

### {-}

# **3.3. Effects of treatments on taxonomic composition**


# *Did taxonomic composition change? 

## Fig. X Overview of sequencing results


**Fig. SX. ** Relative abundances at SGB level by sample type after employing prevalence and abundance filtering and Top 10 phage in each sample type by mean abundance. (A) BAL, (B) nasal swabs, (C) sputum, and (D) mock community. Empty space indicates samples showed no microbial reads, i.e., sequencing failed samples.


```{r, warning=FALSE, message=FALSE}

my_plot_bar = function (physeq, x = "Sample", y = "Abundance", fill = NULL, title = NULL, 
                        facet_grid = NULL) {
    mdf = psmelt(physeq)
    p = ggplot(mdf, aes_string(x = x, y = y, fill = fill))
    p = p + geom_bar(stat = "identity")
    p = p + theme(axis.text.x = element_text(angle = -90, hjust = 0)) +
            scale_x_discrete(drop = F)
    if (!is.null(facet_grid)) {
        p <- p + facet_grid(facet_grid)
    }
    if (!is.null(title)) {
        p <- p + ggtitle(title)
    }
    return(p)
}

a <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "Mock")) %>%
        cbind(species20 = "Other") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq_rel,
                              sample_type == "Mock")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "Mock") 
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        my_plot_bar(., fill="species20") + 
        xlab("Sample") +
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.text.x = element_text(color = "white")) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        labs(tag = "A")
        #facet_wrap (~ factor(sample_type, levels = c("Mock", "BAL", "Nasal", "Sputum")),
        #            scales= "free_x", nrow=2) +
        


b <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "Neg.")) %>%
        cbind(species20 = "Other") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq,
                              sample_type == "Neg.")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "Neg.") 
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>% 
        my_plot_bar(., fill="species20") + 
        ylab("") +
        xlab("Subject") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.text.x = element_text(color = "white")) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        #facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        facet_wrap ( ~ treatment,
                    scales= "free_x", nrow=1) +
        ggtitle("E    Negative control")

c <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "BAL")) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq,
                              sample_type == "BAL")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "BAL")
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E")) %>%
           as.character()
   phyloseq_temp
  } %>% 
        my_plot_bar(., x = "subject_id", fill="species20") + 
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        #facet_wrap (~ treatment, scales = "free_x", nrow = 1) +
        facet_wrap ( ~ treatment,
                    scales= "free_x", nrow=1) +
        ggtitle("A    BAL")


d <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "Nasal")) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq,
                              sample_type == "Nasal")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "Nasal")
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E", "F", "G", "H", "I", "J")) %>%
           as.character()
   phyloseq_temp
  } %>% 
        my_plot_bar(., x = "subject_id", fill="species20") + 
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
#        scale_x_discrete(drop=) +
        facet_wrap(~ treatment,  nrow = 1, drop = T, scales = "free_x") +
        ggtitle("B    Nasal swabs")

e <- tax_table(subset_samples(v_phyloseq_rel,
                              sample_type == "Sputum")) %>%
        cbind(species20 = "[Other]") %>%
        {top20species <- head(taxa_sums(subset_samples(v_phyloseq,
                              sample_type == "Sputum")) %>% 
                                data.frame %>%
                                arrange(-.) %>%
                                row.names(), 10)
        .[top20species, "species20"] <- as.character(.[top20species, "Species"])
        phyloseq_temp <- subset_samples(v_phyloseq_rel,
                              sample_type == "Sputum")
   tax_table(phyloseq_temp) <- tax_table(.) 
   sample_data(phyloseq_temp)$subject_id <- sample_data(phyloseq_temp)$subject_id %>% 
           factor(labels = c("A", "B", "C", "D", "E"))
   phyloseq_temp
  } %>%
        my_plot_bar(., x = "subject_id", fill="species20")+
        ylab("") +
        theme_classic(base_size = 11, base_family = "sans") +
        theme(legend.text = element_markdown(size = 6),
              legend.key.size = unit(3, "mm"),
              legend.title = element_text(size = 6),
              axis.title.x = element_blank()) +
        guides(fill=guide_legend(title="Top 10 phage")) +
        scale_fill_manual(values = c(RColorBrewer::brewer.pal(n = 11, name = "Paired"))) +
        facet_wrap(~ treatment, scales= "free_x", nrow = 1) +
        ggtitle("C    Sputum")


fig2 <- ggarrange(c, c %>% lemon::g_legend() %>% as_ggplot,
                  d, d %>% lemon::g_legend() %>% as_ggplot,
                  e, e %>% lemon::g_legend() %>% as_ggplot,
                  #a, a %>% lemon::g_legend() %>% as_ggplot,
                  #b, b %>% lemon::g_legend() %>% as_ggplot,
                  ncol=2,nrow=3, widths = c(3, 1),
                  legend = "none",
                  align = "hv")

annotate_figure(fig2,
                left = text_grob("Relative abundance",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Subject",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11)
)


          

#png(file = "/Users/minsikkim/Dropbox (Personal)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/Figure2.png",   # The directory you want to save the file in
#    width = 180, # The width of the plot in inches
#    height = 160, # The height of the plot in inches
#    units = "mm",
#    res = 600
#) #fixing multiple page issue

annotate_figure(fig2,
                left = text_grob("Relative abundance",
                                 rot = 90,
                                 family = "sans", 
                                 size = 11),
                bottom = text_grob("Subject",
                                 rot = 0,
                                 family = "sans", 
                                 size = 11)
)

# alpha diversity plots
#ggarrange(f4ad, ggarrange(f4e, f4f, ncol = 2),
#          ncol = 1) # alpha diversity plots

dev.off()

```


### Fig. S5. Changes of Mock microbial community

This figure will be updated, after calculating species richenss of all samples

```{r}

barplot_mock_microbe <- a


barplot_mock_microbe

```


### {-}



# *vi. Did diversity matrices change? 



## Taxa change {.tabset}

### Fig. 3 Alpha and beta diversity


**Fig. 3.** Alpha and beta diversity by sample type and treatment method after removing potential contaminants and rare taxa. (A) Species richness with statistical test results (linear mixed effect model stratified by sample type), (B) Morisita-Horn dissimilarity within subject between treatment, representing squares for median value and bars for 95% confidence intervals.  

- PSL comments (20230630): 
		- Figure 3A: it is hard to see some of the boxplot colors due to the narrow interquartile range. Can you use stat_summary and geom = "pointrange" like what Maghini DG et al did for their Figure 3b or create dotplot + pointrange geom like their Figure 3c? I found some example code goodgling though you might want to show median + iqr rather than mean + sd
			stat_summary(fun = mean,
               geom = "pointrange",
               fun.max = function(x) mean(x) + sd(x),
               fun.min = function(x) mean(x) - sd(x))
    - Figure 3A: we also need to talk because the mock community is only supposed to have a species richness = 10. I know that some of this may be due to taxonomic misclassification but that immediately will raise issues with the reviewer so we need to anticipate how to address this in advance
    - Figure 3A: add relevant p-values using horizontal bars and stars for significance like Maghini DG et al did for Figure 3B
    - Figure 3A: why not just do what Maghini DG et al did for Figure 3e? But have one panel per sample type


```{r, warning=FALSE, message=FALSE}

f3a <- ggplot(subset(sample_data(v_phyloseq) %>% 
                             data.frame, sample_data(v_phyloseq)$sample_type %in% c(#"Mock",
                                     "Sputum", "Nasal", "BAL")), aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment),
                    position = position_jitter(0.2), size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        #stat_summary(fun = mean,
        #       geom = "pointrange",
        #       fun.max = function(x) mean(x) + sd(x),
        #       fun.min = function(x) mean(x) - sd(x))+
        ylab("Species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        scale_fill_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        labs(tag = "A") +
        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 

dat_text <- data.frame(
  label = c(
          #"", "***", "***", "**", "***", #label for Mock
          "", "", "", "*", "", #label for BAL
          "", "", "***", "*", "**", 
          "**", "***", "***", "***", "***"),
  sample_type = c(
          #"Mock", "Mock", "Mock", "Mock", "Mock", 
          "BAL", "BAL", "BAL", "BAL", "BAL", 
          "Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          "Sputum", "Sputum", "Sputum", "Sputum", "Sputum"),
  treatment     = c(
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
  S.obs = c(
          #50, 30, 35, 33, 31,
          30, 35, 50, 52, 50,
          30, 30, 30, 35, 30,
          100, 120, 147, 140, 125)
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f3a <- f3a + geom_text(
  data    = dat_text,
  mapping = aes(x = treatment, y = S.obs, label = label)
)



f5S_mock <- ggplot(subset(sample_data(phyloseq$phyloseq_count) %>% 
                             data.frame,
                          sample_data(phyloseq$phyloseq_count)$sample_type %in%
                                  c("Mock")),
                   aes(x = treatment, y = S.obs)) +
        geom_jitter(aes(color = treatment), position = position_jitter(0.2),
                    size = 1.2, alpha = 0.3, stroke = 0) +
        stat_summary(aes(color = treatment),
                             fun.data="mean_sdl",  fun.args = list(mult=1), 
                             geom = "pointrange",  size = 0.4) +
        ylab("Species richness") +
        xlab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", labels = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using 
        labs(tag = "C") +
        theme(plot.tag = element_text(size = 15),
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank(),
              legend.position = "top") +
        facet_wrap(~sample_type, nrow = 1) + 
        guides(col = guide_legend(nrow = 1)) 

dat_text <- data.frame(
  label = c(
          "", "***", "***", "**", "***"#, #label for Mock
          #"", "", "", "*", "", #label for BAL
          #"", "", "***", "*", "**", 
          #"**", "***", "***", "***", "***"
          ),
  sample_type = c(
          "Mock", "Mock", "Mock", "Mock", "Mock"#, 
          #"BAL", "BAL", "BAL", "BAL", "BAL", 
          #"Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          #"Sputum", "Sputum", "Sputum", "Sputum", "Sputum"
          ),
  treatment = c(
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"#, 
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"
          ),
  S.obs = c(
          50, 30, 35, 33, 31
          #30, 35, 50, 52, 50,
          #30, 30, 30, 35, 30,
          #100, 120, 147, 140, 125
          )
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("Mock"#, 
                                                                #"BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f5S_mock <- f5S_mock + geom_text(
  data    = dat_text,
  mapping = aes(x = treatment, y = S.obs, label = label)
)



#Making subset of non-zero samples without neg

phyloseq_rel_nz <- phyloseq$phyloseq_rel %>%
        subset_samples(S.obs != 0 & sample_type %in% c("Mock", 
                                                       "BAL", "Nasal", "Sputum"))

#distances of betadiversity - boxplots
horn_dist_long <- distance(phyloseq_rel_nz, method="horn") %>% as.matrix() %>% melt_dist() #making long data of distance matrices

#Adding sample type and treatment name. 
#this can be also done by merging metadata into the `horn_dist_long`
names <- data.frame(str_split_fixed(horn_dist_long$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(horn_dist_long$iso2, "_", 3))
horn_dist_long$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
horn_dist_long$method_1 <- ifelse(grepl("lyPMA", horn_dist_long$iso1),"lypma", 
                                         ifelse(grepl("ben", horn_dist_long$iso1),"benzonase", 
                                                ifelse(grepl("host", horn_dist_long$iso1),"host_zero", 
                                                       ifelse(grepl("qia", horn_dist_long$iso1),"qiaamp", 
                                                              ifelse(grepl("moly", horn_dist_long$iso1),"molysis", 
                                                                     "control")))))


#Adding data for iso 2 also should be done
horn_dist_long$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")
horn_dist_long$method_2 <-ifelse(grepl("lyPMA", horn_dist_long$iso2),"lypma", 
                                        ifelse(grepl("ben", horn_dist_long$iso2),"benzonase", 
                                               ifelse(grepl("host", horn_dist_long$iso2),"host_zero", 
                                                      ifelse(grepl("qia", horn_dist_long$iso2),"qiaamp", 
                                                             ifelse(grepl("moly", horn_dist_long$iso2),"molysis", 
                                                                    "control")))))


#subsetting distances of my interest
horn_dist_long$sample_id_1 <- ifelse(grepl("pos", horn_dist_long$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_1, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_1))
horn_dist_long$sample_id_2 <- ifelse(grepl("pos", horn_dist_long$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", horn_dist_long$sample_id_2, ignore.case = T),"Neg.",
                                        horn_dist_long$sample_id_2))


horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long, horn_dist_long$sample_id_1 == horn_dist_long$sample_id_2) # data within samples

horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long_within_sampleid_from_control,
                                                           horn_dist_long_within_sampleid_from_control$method_1 != horn_dist_long_within_sampleid_from_control$method_2) # remove irrelevant association

horn_dist_long_within_sampleid_from_control <- subset(horn_dist_long_within_sampleid_from_control, (horn_dist_long_within_sampleid_from_control$method_1 == "control") + (horn_dist_long_within_sampleid_from_control$method_2 == "control") != 0)


horn_dist_long_within_sampleid_from_control$treatment <- horn_dist_long_within_sampleid_from_control$method_1

horn_dist_long_within_sampleid_from_control$treatment <- ifelse(horn_dist_long_within_sampleid_from_control$treatment == "control", horn_dist_long_within_sampleid_from_control$method_2, horn_dist_long_within_sampleid_from_control$treatment) 


#Setting key method
horn_dist_long_within_sampleid_from_control$sample_type <- ifelse(grepl("NS", horn_dist_long_within_sampleid_from_control$iso1), "Nasal",
                                                                  ifelse(grepl("CFB", horn_dist_long_within_sampleid_from_control$iso1), "Sputum",
                                                                         ifelse(grepl("BAL", horn_dist_long_within_sampleid_from_control$iso1), "BAL",
                                                                                ifelse(grepl("pos|POS", horn_dist_long_within_sampleid_from_control$iso1, ignore.case = T), "Mock",
                                                                                       ifelse(grepl("neg|N_EXT", horn_dist_long_within_sampleid_from_control$iso1), "Neg.",NA)))))

#Making a column for baseline (controls, from where?)
horn_dist_long_within_sampleid_from_control <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(dist_from = case_when(method_1 == "control" ~ iso1,
                                     method_2 == "control" ~ iso2))

dummy <- data.frame(iso1 = horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           iso2 = horn_dist_long_within_sampleid_from_control$dist_from %>% unique,
           dist = 0,
           treatment = "Untreated",
           method_1 = "control",
           method_2 = "control"
           )
names <- data.frame(str_split_fixed(dummy$iso1, "_", 3))
names2 <- data.frame(str_split_fixed(dummy$iso2, "_", 3))
dummy$sample_id_1 <- paste(names$X1, names$X2, sep = "_")
#Adding data for iso 2 also should be done
dummy$sample_id_2 <- paste(names2$X1, names2$X2, sep = "_")


#subsetting distances of my interest
dummy$sample_id_1 <- ifelse(grepl("pos", dummy$sample_id_1, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_1, ignore.case = T),"Neg.",
                                        dummy$sample_id_1))
dummy$sample_id_2 <- ifelse(grepl("pos", dummy$sample_id_2, ignore.case = T),"Mock", 
                                 ifelse(grepl("neg|n_", dummy$sample_id_2, ignore.case = T),"Neg.",
                                        dummy$sample_id_2))
dummy$sample_type <- ifelse(grepl("NS", dummy$iso1), "Nasal",
                            ifelse(grepl("CFB", dummy$iso1), "Sputum",
                                   ifelse(grepl("BAL", dummy$iso1), "BAL",
                                          ifelse(grepl("pos|POS", dummy$iso1, ignore.case = T), "Mock",
                                                 ifelse(grepl("neg|N_EXT", dummy$iso1), "Neg.",NA)))))
dummy <- subset(dummy, !is.na(dummy$sample_type))
horn_dist_long_within_sampleid_from_control <- bind_rows(horn_dist_long_within_sampleid_from_control, dummy)

#Here, sample id is the same as subject id.
horn_dist_long_within_sampleid_from_control$subject_id <- horn_dist_long_within_sampleid_from_control$sample_id_1

horn_dist_long_within_sampleid_from_control$treatment <-
        factor(horn_dist_long_within_sampleid_from_control$treatment,
               levels = c("Untreated", "lypma", "benzonase", "host_zero", "molysis", "qiaamp"))

#Making figure of beta diversity distances

## This only includes samples (BAL, Nasal and Sputum)
f3b2 <- rbind(cbind("Sputum",
            lmer(dist ~ treatment + (1|subject_id),
                data = horn_dist_long_within_sampleid_from_control %>%
                        subset(sample_type == "Sputum")) %>% 
                    confint()
            ),
      cbind("BAL",
            lmer(dist ~ treatment + (1|subject_id),
                 data = horn_dist_long_within_sampleid_from_control %>%
                         subset(sample_type == "BAL")) %>% 
                    confint()
            ),
      cbind("Nasal",
            lmer(dist ~ treatment + (1|subject_id),
                 data = horn_dist_long_within_sampleid_from_control %>%
                         subset(sample_type == "Nasal")) %>% 
                    confint()
     )
) %>% {
        row_names <- rownames(.)
        data_frame <- data.frame(.)
        data_frame$treatment <- row_names
        data_frame %>% 
                remove_rownames() %>%
                rename(sample_type = "V1",
               "2.5%" = "X2.5..",
               "97.5%" = "X97.5..") %>% 
                mutate(`2.5%` = as.numeric(`2.5%`),
                       `97.5%` = as.numeric(`97.5%`),
                       mean = (`97.5%`+`2.5%`)/2,
                       treatment = case_when(treatment == "(Intercept)" ~ "Untreated",
                                             treatment == "treatmentlypma" ~ "lyPMA",
                                             treatment == "treatmentbenzonase" ~ "Benzonase",
                                             treatment == "treatmenthost_zero" ~ "HostZERO",
                                             treatment == "treatmentmolysis" ~ "MolYsis",
                                             treatment == "treatmentqiaamp" ~ "QIAamp"
                                             ),
                       treatment = factor(treatment,
                                          levels = c("Untreated", "lyPMA", "Benzonase",
                                      "HostZERO", "MolYsis", "QIAamp")),
                       sample_type = factor(sample_type,
                                          levels = c("BAL", "Nasal", "Sputum"))
                       ) %>%
                subset(treatment %in% c(#"Untreated", 
                                        "lyPMA", "Benzonase",
                                      "HostZERO", "MolYsis", "QIAamp"))
} %>%
        ggplot(aes(x = mean, y = treatment, col = treatment)) +
        geom_point(aes(x=mean), shape=15, size=3) +
        geom_linerange(aes(xmin=`2.5%`, xmax=`97.5%`)) +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c(#"#e31a1c",
                                      "#fb9a99","#33a02c",
                                      "#b2df8a","#1f78b4","#a6cee3"),
                           name = "Treatment",
                           breaks = c(#"Untreated", 
                                      "lyPMA", "Benzonase",
                                      "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Morisita-Horn dissimilarity from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              legend.position = "none") +
        labs(tag = "B") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "sans") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "sans") +
        scale_x_continuous(breaks = c(-0.25, 0, 0.25, 0.5, 0.75),
                           labels = c(-0.25, "0 (low bias)", 0.25, 0.5, "0.75 (high bias)"))


dat_text <- data.frame(
  label = c(
          #"", "***", "***", "**", "***", #label for Mock
          "*", "", "", "", "", #label for BAL
          "*", "*", "", "**", "", 
          "**", "***", "***", "***", "***"),
  sample_type = c(
          #"Mock", "Mock", "Mock", "Mock", "Mock", 
          "BAL", "BAL", "BAL", "BAL", "BAL", 
          "Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          "Sputum", "Sputum", "Sputum", "Sputum", "Sputum"),
  treatment     = c(
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
  mean = c(
          #50, 30, 35, 33, 31,
          0.6, 0, 0, 0, 0,
          0.32, 0.3, 0, 0.37, 0,
          0.58, 0.75, 0.85, 0.83, 0.85)
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f3b2 <- f3b2 + geom_text(
  data    = dat_text,
  mapping = aes(y = treatment, x = mean, label = label),
  col = "black"
)


f3b2_box <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(across(sample_type, factor, levels=c(#"Mock", 
                                                    "BAL", "Nasal","Sputum"
                                                    ))) %>%
        subset(., .$sample_type != "Neg.") %>% 
        subset(., .$treatment != "Untreated") %>%
  mutate(treatment = factor(treatment, levels = c("lypma", "benzonase", "host_zero", "molysis", "qiaamp"),
                            labels = c("lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))) %>%
        ggplot(aes(x = dist, y = treatment, col = treatment)) +
        geom_boxplot(aes(x=dist)) +
        #geom_linerange(aes(xmin=`2.5%`, xmax=`97.5%`)) +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c(#"#e31a1c",
                                      "#fb9a99","#33a02c",
                                      "#b2df8a","#1f78b4","#a6cee3"),
                           name = "Treatment",
                           breaks = c(#"Untreated", 
                                      "lyPMA", "Benzonase",
                                      "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Morisita-Horn dissimilarity from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              legend.position = "none") +
        labs(tag = "B") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "sans") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "sans") +
        scale_x_continuous(breaks = c(-0.25, 0, 0.25, 0.5, 0.75),
                           labels = c(-0.25, "0 (low bias)", 0.25, 0.5, "0.75 (high bias)"))


dat_text <- data.frame(
  label = c(
          #"", "***", "***", "**", "***", #label for Mock
          "*", "", "", "", "", #label for BAL
          "*", "*", "", "**", "", 
          "**", "***", "***", "***", "***"),
  sample_type = c(
          #"Mock", "Mock", "Mock", "Mock", "Mock", 
          "BAL", "BAL", "BAL", "BAL", "BAL", 
          "Nasal", "Nasal", "Nasal", "Nasal", "Nasal", 
          "Sputum", "Sputum", "Sputum", "Sputum", "Sputum"),
  treatment     = c(
          #"lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp", 
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp",
          "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
  dist = c(
          #50, 30, 35, 33, 31,
          0.9, 0, 0, 0, 0,
          0.48, 0.4, 0, 0.55, 0,
          0.8, 0.82,1, 1.02, 0.98)
)



dat_text$sample_type <- factor(dat_text$sample_type, levels = c("BAL", "Nasal", "Sputum"
                                                                ))
dat_text$treatment <- factor(dat_text$treatment, levels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))


f3b2_box <- f3b2_box + geom_text(
  data    = dat_text,
  mapping = aes(y = treatment, x = dist, label = label),
  col = "black"
)



# Mock community's beta-diversity plot.
## This only includes positive control samples (Mock)

f5S_mock_mh <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(across(sample_type, factor, levels=c("Mock"#, 
                                                    #"BAL", "Nasal","Sputum"
                                                    ))) %>%
        subset(., .$sample_type != "Neg.") %>% 
        group_by(sample_type, treatment) %>%
        summarise(mean = mean(dist, na.rm = TRUE),
            sd = sd(dist, na.rm = TRUE),
            n = n()) %>%
        subset(., .$treatment != "Untreated") %>%
  mutate(se = sd / sqrt(n),
         lower.ci = mean - qt(1 - (0.05 / 2), n - 1) * se,
         upper.ci = mean + qt(1 - (0.05 / 2), n - 1) * se,
         treatment = factor(treatment, levels = c(#"Untreated",
                                                  "lypma", "benzonase", "host_zero", "molysis", "qiaamp"),
                            labels = c(#"Untreated",
                                       "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))) %>%
        #,
         #text = paste(sprintf("%.2f", round(mean, digits = 2)), " [", sprintf("%.2f", round(lower.ci, digits = 2)), ", ", sprintf("%.2f", round(upper.ci, digits = 2)), "]", sep = "")) %>%
        ggplot(aes(x = mean, y = treatment, col = treatment)) +
        geom_point(aes(x=mean), shape=15, size=3) +
        geom_linerange(aes(xmin=lower.ci, xmax=upper.ci)) +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c(#"#e31a1c", 
                                      "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", 
                           labels = c(#"Untreated",
                                      "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Morisita-Horn dissimilarity from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              legend.position = "none") +
        labs(tag = "D") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "sans") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "sans") +
        scale_x_continuous(breaks = c(-0.5, 0, 0.5, 1, 1.5), labels = c(-0.5, "0 (low bias)", 0.5, 1, "1.5 (high bias)"))



f5S_mock_mh_box <- horn_dist_long_within_sampleid_from_control %>% 
        mutate(across(sample_type, factor, levels=c("Mock"#, 
                                                    #"BAL", "Nasal","Sputum"
                                                    ))) %>%
        subset(., .$sample_type != "Neg.") %>% 
        subset(., .$treatment != "Untreated") %>%
  mutate(treatment = factor(treatment, levels = c(#"Untreated",
                                                  "lypma", "benzonase", "host_zero", "molysis", "qiaamp"),
                            labels = c(#"Untreated",
                                       "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"))) %>%
        #,
         #text = paste(sprintf("%.2f", round(mean, digits = 2)), " [", sprintf("%.2f", round(lower.ci, digits = 2)), ", ", sprintf("%.2f", round(upper.ci, digits = 2)), "]", sep = "")) %>%
        ggplot(aes(x = dist, y = treatment, col = treatment)) +
        geom_boxplot() +
        facet_wrap(~sample_type, nrow = 4) +
        scale_y_discrete(limits=rev) +
        scale_color_manual(values = c(#"#e31a1c", 
                                      "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"), name = "Treatment", 
                           labels = c(#"Untreated",
                                      "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        xlab("Morisita-Horn dissimilarity from untreated") +
        ylab("Treatment group") +
        theme_classic (base_size = 12, base_family = "sans") + 
        theme(plot.tag = element_text(size = 15),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              legend.position = "none") +
        labs(tag = "D") +
        geom_vline(xintercept = 0, col = "black", linetype="dotted") +
        #coord_cartesian(xlim=c(-0.5, 1)) +
        #geom_text(aes(x = 0, label = treatment), hjust = 0, nudge_x = -.55, size = 3, color = "black", family = "sans") +
        #geom_text(aes(x = 0, label = text), hjust = 0, nudge_x = -0.4, size = 3, color = "black", family = "sans") +
        scale_x_continuous(breaks = c(-0.5, 0, 0.5, 1, 1.5), labels = c(-0.5, "0 (low bias)", 0.5, 1, "1.5 (high bias)"))


fig3 <- ggarrange(f3a, f3b2_box, ncol = 1, common.legend = T, align = "hv")

fig3

#png(file = "/Users/minsikkim/Dropbox (Personal)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/Figure3.png",   # #The directory you want to save the file in
#    width = 180, # The width of the plot in inches
#    height = 220, # The height of the plot in inches
#    units = "mm",
#    res = 600
#) #fixing multiple page issue

fig3

#dev.off()
```


### {-}


# *vii.	& viii. Do these host depletion methods introduce bias in the sequenced community? *

+ Does bias differ by sample type?


## Alpha diversity changes {.tabset}

### Species richness (all sample)

**This table is too redundant. Removed from the manuscript.**. 

Effect size, standard error (SE) and p-value of a statistical test for species richness with an interaction term using linear mixed effect model (Species richness ~ sample_type * treatment + log10 (Final_reads) + (1|subject_id) ).

**Interaction term was highly significant (p = 2.2e-16)**

Raw result

```{r}
lmer_sr <- lmer(S.obs ~ treatment * sample_type + log10(Viral_reads) + (1|subject_id),
                    data = sample_data %>% 
                            data.frame %>% 
                            subset(., .$S.obs != 0))

lmer_sr %>% summary
        
```

ANOVA result on LMER result for species richness

```{r}
lmer_sr %>% anova()
        
```

### Table S3. Species richness (stratified) without depth

**Table S3.** Effect size, standard error (SE) and p-value of a statistical test for species richness using linear mixed effect model stratified by sample type (Species richness ~ treatment + (1|subject_id) ). Stratified analyses were conducted for each sample type as an interaction term of sample type and treatment was significant at an ANOVA test (p-value < 0.001) using a model, lmer(species richness ~ sample type + treatment + sample type * treatment  + (1|subject_id)). Statistical significances were noted with `*: p-value < 0.05, **: p-value < 0.01, and ***: p-value < 0.001.`

LMER raw result - Mock

```{r}

sr_lmer_mock <- lm(S.obs ~ treatment,
                    data = sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Mock") & S.obs != 0))

sr_lmer_mock %>% summary() 

```

LMER raw result - BAL

```{r}

sr_lmer_bal <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("BAL")))
sr_lmer_bal %>% summary()
        
```

LMER raw result - Nasal

```{r}

sr_lmer_ns <- lmer(S.obs ~ treatment + (1|subject_id),
                   data = sample_data %>% 
                           data.frame %>% 
                           subset(., .$sample_type %in% c("Nasal")))
sr_lmer_ns %>% summary()
```

LMER raw result - Sputum

```{r}
sr_lmer_spt <- lmer(S.obs ~ treatment + (1|subject_id),
                    data = sample_data %>% 
                            data.frame %>% 
                            subset(., .$sample_type %in% c("Sputum")))
sr_lmer_spt %>% summary()
        
sr_lmer_spt %>% confint()

```

Tidy summarized table

```{r}

sr_lmer_mock_kbl <-  sr_lmer_mock %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_mock %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

sr_lmer_bal_kbl <-  sr_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        
sr_lmer_ns_kbl <-  sr_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               abs(`Pr(>|t|)`) < 0.1 ~ ".",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


sr_lmer_spt_kbl <-  sr_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              sr_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("(Intercept)",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]
        

tables3 <- cbind(#sr_lmer_mock_kbl,
                 sr_lmer_bal_kbl,
                 sr_lmer_ns_kbl,
                 sr_lmer_spt_kbl) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1,
                           #"Mock" = 3,
                           "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")

tables3

#save_kable(tables3, file = "/Users/minsikkim/Dropbox (Personal)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS3.html", self_contained = T)

```

### Species richness - stratified - with depth

**This table is too redundant. Removed from the manuscript.**. 

Sequencing depth adjusted effect size, standard error (SE) and p-value of a statistical test for species richness using linear mixed effect model stratified by sample type (Species richness ~ treatment + log10 (Final_reads) + (1|subject_id) ).

```{r}

sr_lmer_bal_w_depth <- lmer(S.obs ~ treatment + bal_log_centered_final_reads + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("BAL") & S.obs != 0)) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = 1), 
                                " (", 
                                round(Estimate - 1.96 * abs(t_value), 1) %>% format(nsmall = 1),
                                ", ",
                                round(Estimate + 1.96 * abs(t_value), 1) %>% format(nsmall = 1),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " "))
        
sr_lmer_ns_w_depth <- lmer(S.obs ~ treatment + ns_log_centered_final_reads + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Nasal"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("ns_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = ), 
                                " (", 
                                round(Estimate - 1.96 * abs(t_value), 1) %>% format(nsmall = ),
                                ", ",
                                round(Estimate + 1.96 * abs(t_value), 1) %>% format(nsmall = ),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " "))


sr_lmer_spt_w_depth <- lmer(S.obs ~ treatment + spt_log_centered_final_reads + (1|subject_id), data = sample_data %>% data.frame %>% subset(., .$sample_type %in% c("Sputum"))) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("spt_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
                rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        mutate("Effect size (95% CI)" = paste(round(Estimate, 1) %>% format(nsmall = 1), 
                                " (", 
                                round(Estimate - 1.96 * abs(t_value), 1) %>% format(nsmall = 1),
                                ", ",
                                round(Estimate + 1.96 * abs(t_value), 1) %>% format(nsmall = 1),
                                ")", 
                                sep = ""),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " "))
        

cbind(sr_lmer_bal_w_depth, sr_lmer_ns_w_depth, sr_lmer_spt_w_depth) %>%
    kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")



#save_kable(tableS7, file = "/Users/minsikkim/Dropbox (Personal)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS7.html", self_contained = T)




```

### {-}


## Beta diversity distances - Morisita Horn dissimilarity {.tabset}

### PERMANOVA - all samples 

**This table is too redundant. Removed from the manuscript.**. 

Degree of freedom, effect size (residual, R2) and p-value of permutational ANOVA for Morisita-Horn dissimilarity with an interaction term and strata term (MH-index of species composition ~ sample type `*` treatment + subject + log10(final reads), strata = subject id). Statistical significances were noted with `***:` p-value < 0.001.



```{r}
set.seed(seed)
horn_perm_all <- vegan::adonis2(distance(phyloseq_rel_nz, method="horn") ~ sample_type * treatment + subject_id,
                                  data = phyloseq_rel_nz %>% sample_data %>% data.frame(check.names = F),
                                  permutations = 10000)
set.seed(seed)
horn_perm_ns <- vegan::adonis2(distance(subset_samples(phyloseq_rel_nz, sample_type == "Nasal"), method="horn") ~ lypma + benzonase + host_zero + molysis + qiaamp,
                               data = subset_samples(phyloseq_rel_nz, sample_type == "Nasal") %>%
                                       sample_data %>% data.frame(check.names = F),
                               strata = subset_samples(phyloseq_rel_nz, sample_type == "Nasal") %>% 
                                       sample_data %>% data.frame(check.names = F) %>% .$subject_id, permutations = 10000)
set.seed(seed)
horn_perm_bal  <- vegan::adonis2(distance(subset_samples(phyloseq_rel_nz, sample_type == "BAL"), method="horn") ~  lypma + benzonase + host_zero + molysis + qiaamp,
                                 data = subset_samples(phyloseq_rel_nz, sample_type == "BAL") %>% sample_data %>% data.frame(check.names = F),
                                 strata = subset_samples(phyloseq_rel_nz, sample_type == "BAL") %>%
                                         sample_data %>% data.frame(check.names = F) %>% .$subject_id,
                                  permutations = 10000)
set.seed(seed)
horn_perm_spt <- vegan::adonis2(distance(subset_samples(phyloseq_rel_nz, sample_type == "Sputum"), method="horn") ~ lypma + benzonase + host_zero + molysis + qiaamp,
                                data = subset_samples(phyloseq_rel_nz, sample_type == "Sputum") %>% sample_data %>% data.frame(check.names = F),
                                strata = subset_samples(phyloseq_rel_nz, sample_type == "Sputum")
                                %>% sample_data %>% data.frame(check.names = F) %>% .$subject_id,
                                  permutations = 10000)

```

PERMANOVA raw result

```{r}
horn_perm_all 
```

Tidy permanova result

```{r}

tableS5_A <- horn_perm_all %>% data.frame(check.names = F) %>% rownames_to_column("row.names") %>% 
        mutate(row.names = case_when(row.names == "sample_type" ~ 'Sample type',
                                     row.names == "treatment" ~ 'Treatment',
                                     row.names == "subject_id" ~ 'Subject',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "sample_type:treatment" ~ 'Sample type * Treatment',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
        mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        mutate(`<i>p</i>-value` = format(`<i>p</i>-value`, nsmall = 3)) %>%
        dplyr::select(c("Degree of freedom", "R<sup>2</sup>", "<i>p</i>-value", " ")) %>% 
        kbl(format = "html", escape = 0) %>%
        kable_styling(full_width = 0, html_font = "sans")

tableS5_A



```

### Table S6. PERMANOVA - Stratified

Raw result - BAL

```{r}
horn_perm_bal
```


Raw result - Nasal

```{r}
horn_perm_ns
```


Raw result - BAL

```{r}
horn_perm_spt
```



**This table was additionally added to the manuscript as to determine bias after treatment with sputum**. 

**Table S6.** Degree of freedom, effect size (residual, R^2) and p-value of permutational ANOVA for Morisita-Horn distiances for species richness (MH-distance ~ lyPMA + Benzoase + HostZERO + MolYsis + QIAamp, strata = subject_id). Sample types were stratified as interaction term of (MH-distance ~ sample type + treatment + sample type * treatment, strata = subject_id) was significant (p < 0.001).

Tidy permanova result (stratified)

```{r}

a <- horn_perm_bal %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'HostZERO',
                                     row.names == "molysis" ~ 'MolYsis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        dplyr::select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 

b <- horn_perm_ns %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'HostZERO',
                                     row.names == "molysis" ~ 'MolYsis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        dplyr::select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 

c <- horn_perm_spt %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'HostZERO',
                                     row.names == "molysis" ~ 'MolYsis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
                mutate(` ` = case_when(abs(`Pr(>F)`) < 0.001 ~ "***",
                                            abs(`Pr(>F)`) < 0.01 ~ "**",
                                            abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        mutate(across(is.numeric, round, digits=3)) %>% 
        rename("<i>p</i>-value" = "Pr(>F)",
               "R<sup>2</sup>" = "R2",
               "Degree of freedom" = "Df") %>% 
        dplyr::select(c("R<sup>2</sup>", "<i>p</i>-value", " ")) 


tableS6 <- cbind(a, b, c) %>% 
        kbl(format = "html", escape = 0) %>%
        add_header_above(c(" " = 1, "BAL" = 3, "Nasal swab" = 3,"Sputum"= 3)) %>% 
        kable_styling(full_width = 0, html_font = "sans")


#save_kable(tableS6, file = "/Users/minsikkim/Dropbox (Personal)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Figures/tableS6.html", self_contained = T)


tableS6


```

### Table S7. LM on M-H distance 

**Table S7.** Effect size, standard error (SE) and p-value of a statistical test for Morisita-Horn dissimilarity from untreated to each treated within subject, stratified by sample type. Stratified analyses were conducted for each sample type as an interaction term of sample type and treatment was significant (p-value < 0.001) using a model, ANOVA(species richness ~ sample type + treatment + sample type * treatment). The tested was conducted with a linear mixed effect model LM(distance within subject by treatment ~ treatment + (1|subject)). The baseline of sample type is untreated (0 distance, untreated from untreated), and statistical significance were noted with `*: p-value < 0.05, **: p-value < 0.01 and ***: `p-value < 0.001.

```{r}

horn_dist_long_within_sampleid_from_control$treatment <- factor(horn_dist_long_within_sampleid_from_control$treatment,
                                                                     levels = c("Untreated",
                                                                                "lypma",
                                                                                "benzonase",
                                                                                "host_zero",
                                                                                "molysis",
                                                                                "qiaamp"))
horn_dist_long_within_sampleid_from_control$sample_type <- factor(horn_dist_long_within_sampleid_from_control$sample_type,
                                                                     levels = c("BAL",
                                                                                "Nasal",
                                                                                "Sputum"))
```

Raw lmer result on M-H distances

```{r}

lmer(dist ~ treatment * sample_type + (1|subject_id),
     data = horn_dist_long_within_sampleid_from_control %>%
             subset(sample_type != "Mock")) %>%
        summary() 


```

Justifying stratified analysis - ANOVA(LM(dist ~ sample type * treatment))

```{r}
lmer(dist ~ treatment * sample_type + (1|subject_id),
     data = horn_dist_long_within_sampleid_from_control %>%
             subset(sample_type != "Mock")) %>%
        anova() 

```

Stratified analysis - BAL

```{r}
mh_lmer_bal <- lmer(dist ~ treatment + (1|subject_id),
     data = horn_dist_long_within_sampleid_from_control %>%
             subset(sample_type == "BAL")) 

mh_lmer_bal %>% 
        summary
```


Stratified analysis - Nasal

```{r}
mh_lmer_ns <- lmer(dist ~ treatment + (1|subject_id),
     data = horn_dist_long_within_sampleid_from_control %>%
             subset(sample_type == "Nasal")) 

mh_lmer_ns %>% 
        summary

```

Stratified analysis - Sputum

```{r}
mh_lmer_spt <- lmer(dist ~ treatment + (1|subject_id),
     data = horn_dist_long_within_sampleid_from_control %>%
             subset(sample_type == "Sputum")) 

mh_lmer_spt %>% 
        summary

```

Tidy, summarized-stratified analysis

```{r}
mh_lmer_bal_kbl <-  mh_lmer_bal %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              mh_lmer_bal %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
        column_to_rownames("Row.names") %>%
        rownames_to_column("treatment") %>%
        mutate(treatment = case_when(treatment == "(Intercept)" ~ "Untreated",
                                             treatment == "treatmentlypma" ~ "lyPMA",
                                             treatment == "treatmentbenzonase" ~ "Benzonase",
                                             treatment == "treatmenthost_zero" ~ "HostZERO",
                                             treatment == "treatmentmolysis" ~ "MolYsis",
                                             treatment == "treatmentqiaamp" ~ "QIAamp"
                                             )) %>%
        column_to_rownames("treatment") %>%
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("Untreated",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


mh_lmer_ns_kbl <-  mh_lmer_ns %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              mh_lmer_ns %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
                column_to_rownames("Row.names") %>%
        rownames_to_column("treatment") %>%
        mutate(treatment = case_when(treatment == "(Intercept)" ~ "Untreated",
                                             treatment == "treatmentlypma" ~ "lyPMA",
                                             treatment == "treatmentbenzonase" ~ "Benzonase",
                                             treatment == "treatmenthost_zero" ~ "HostZERO",
                                             treatment == "treatmentmolysis" ~ "MolYsis",
                                             treatment == "treatmentqiaamp" ~ "QIAamp"
                                             )) %>%
        column_to_rownames("treatment")  %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("Untreated",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


mh_lmer_spt_kbl <-  mh_lmer_spt %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.001 ~ "***",
                               abs(`Pr(>|t|)`) < 0.01 ~ "**",
                               abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        rename("<i>p</i>-value" = "Pr(>|t|)",
               t_value = "t value") %>%
        merge(., 
              mh_lmer_spt %>% confint() %>%
                        round(digits = 1) %>% 
                        format(nsmall = 1) %>%
                        as.data.frame() %>%
                        mutate(conf = paste("(",
                                            gsub(" ","", .[,1]),
                                            ", ",
                                            gsub(" ","", .[,2]),
                                            ")",
                                            sep = "")),
              by = 0
              ) %>%
                column_to_rownames("Row.names") %>%
        rownames_to_column("treatment") %>%
        mutate(treatment = case_when(treatment == "(Intercept)" ~ "Untreated",
                                             treatment == "treatmentlypma" ~ "lyPMA",
                                             treatment == "treatmentbenzonase" ~ "Benzonase",
                                             treatment == "treatmenthost_zero" ~ "HostZERO",
                                             treatment == "treatmentmolysis" ~ "MolYsis",
                                             treatment == "treatmentqiaamp" ~ "QIAamp"
                                             )) %>%
        column_to_rownames("treatment") %>% 
        rownames_to_column(var = "x") %>% mutate(x = gsub("treatment|sample_type", "", x)) %>% mutate(x = gsub(":", " * ", x)) %>%
        mutate(x = gsub("bal_log_centered_final_reads", "log<sub>10</sub>(Final reads)", x)) %>%
        column_to_rownames(var = "x") %>% 
        mutate("Effect size (95% CI)" = 
                       paste(round(Estimate, digits = 1) %>%
                                     format(nsmall = 1),
                             conf, 
                             sep = " "),
               "<i>p</i>-value" = round(`<i>p</i>-value`, 3)) %>%
        dplyr::select(c("Effect size (95% CI)", "<i>p</i>-value", " ")) %>%
        .[c("Untreated",
                  "lyPMA",
                  "Benzonase",
                  "HostZERO",
                  "MolYsis",
                  "QIAamp"),]


tableS7


```

### Fig. S7 PCA figure

**Fig. S7.** Principal coordinate analysis plot based on Morisita-Horn dissimilarity of taxonomic sequencing results stratified by sample type.

```{r}
figS7 <- ordinate(subset_samples(phyloseq_rel_nz, sample_type != "Neg." & sample_type != "Mock"), method = "PCoA", distance = "horn") %>%
        plot_ordination(phyloseq_rel_nz, ., col = "treatment") +
        #scale_color_viridis(discrete = 6, name = "Treatment", labels = c("Mock theoretical", "Control","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) +
        scale_color_manual(values = c("#e31a1c", "#fb9a99", "#33a02c", "#b2df8a", "#1f78b4", "#a6cee3"),
                           name = "Treatment",
                           breaks = c("Untreated","lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp"),
                           labels = c("Untreated", "lyPMA", "Benzonase", "HostZERO", "MolYsis", "QIAamp")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        #scale_shape(name = "Sample type", labels = c("Mock theoretical", "Mock")) +
        geom_point(size = 3) +
        theme_classic (base_size = 12, base_family = "sans") +
        facet_wrap(~sample_type, scales = "free") +
        theme(plot.tag = element_text(size = 15), legend.position = "top")# +
        #stat_ellipse(type = "norm") +
        #stat_ellipse(type = "t")

figS7


```

### {-}

Done.

# Bibliography

```{r warning=FALSE}
#===============================================================================
#BTC.LineZero.Footer.1.1.0
#===============================================================================
#R markdown citation generator.
#===============================================================================
#RLB.Dependencies:
#   magrittr, pacman, stringr
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#BTC.Dependencies:
#   LineZero.Header
#===============================================================================
#Generates citations for each explicitly loaded library.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
str_libraries <- c("r", str_libraries)
for (str_libraries in str_libraries) {
    str_libraries |>
        pacman::p_citation() |>
        print(bibtex = FALSE) |>
        capture.output() %>%
        .[-1:-3] %>% .[. != ""] |>
        stringr::str_squish() |>
        stringr::str_replace("_", "") |>
        cat()
    cat("\n")
}
#===============================================================================
```
