---
title: "COD_20230301_HOST_1_QC_analysis"
author: "Minsik Kim"
date: "2023-03-14"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
        df_print: paged
editor_options: 
  chunk_output_type: inline
---


## Loading packages

```{r setup}
#===============================================================================
#BTC.LineZero.Header.1.1.0
#===============================================================================
#R Markdown environment setup and reporting utility.
#===============================================================================
#RLB.Dependencies:
#   knitr, magrittr, pacman, rio, rmarkdown, rmdformats, tibble, yaml
#===============================================================================
#Input for document parameters, libraries, file paths, and options.
#=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=
path_working <- "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git/"
path_library <- "/Library/Frameworks/R.framework/Resources/library"
str_libraries <- c(
    "readxl", "phyloseq", "tidyverse", "pacman", "yaml"
)

path_working <- "/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/5_Scripts/MGK/Host_depletion_git"
path_library <- "/Library/Frameworks/R.framework/Resources/library"
str_libraries <- c("readxl", "phyloseq", "tidyverse", "pacman", "yaml", "ggplot2", "vegan", "microbiome", "ggpubr", "viridis", "decontam", "gridExtra", "ggpubr", "lme4", "lmerTest", "writexl", "harrietr", "Maaslin2", "ggtext", "ggpmisc", "gridExtra", "gamm4", "reshape2", "kableExtra", "knitr")
        
YAML_header <-
'---
title: "Host-DNA depletion 1: data wrangling"
author: "Minsik Kim"
date: "2032.03.10"
output:
    rmdformats::downcute:
        downcute_theme: "chaos"
        code_folding: hide
        fig_width: 6
        fig_height: 6
---'
seed <- "20230314"

#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Loads libraries, file paths, and other document options.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
FUN.LineZero.Boot <- function() {
    .libPaths(path_library)

    require(pacman)
    pacman::p_load(c("knitr", "rmarkdown", "rmdformats", "yaml"))

    knitr::opts_knit$set(root.dir = path_working)

    str_libraries |> unique() |> sort() -> str_libraries
    pacman::p_load(char = str_libraries)

    set.seed(seed)
}
FUN.LineZero.Boot()
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
#Outputs R environment report.
#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
FUN.LineZero.Report <- function() {
    cat("Line Zero Environment:\n\n")
    paste("R:", pacman::p_version(), "\n") |> cat()
    cat("Libraries:\n")
    for (str_libraries in str_libraries) {
        paste(
            "    ", str_libraries, ": ", pacman::p_version(package = str_libraries),
            "\n", sep = ""
        ) |> cat()
    }
    paste("\nOperating System:", pacman::p_detectOS(), "\n") |> cat()
    paste("    Library Path:", path_library, "\n") |> cat()
    paste("    Working Path:", path_working, "\n") |> cat()
    paste("Seed:", seed, "\n\n") |> cat()
    cat("YAML Header:\n")
    cat(YAML_header)
}
FUN.LineZero.Report()


```


#Script description#

*1. Loading data
        1.1. phyloseq obejct
        1.2. qPCR data (controls)

2. QC
        Q1. How many samples failed sequencing
        Q2. How were changes in read stats and host DNA proportion?
        Q3. How were the extraction controls
        Q4. Prevalence / abundance filtering - red flag

3. Exploratory analysis *

***data input required:***

*Meta data*

* qPCR - bacteria

* qPCR - human

* qPCR host %

* Raw reads

* final reads

* sequencing host %

* library prep failure status

* Raw reads

* subject_id

* treatment

* sample_type

* subject_id

*Sequencing result*

* samples

* controls

***Analysis***

0. calculation of alpha-diversity indices

1. qPCR, by smaple type and treatment
        A. Total DNA
        B. Host DNA
        C. Bacterial DNA
        D. Host %

2. Sequencing results (treatment, raw reads, host reads, % host, final reads)
	--> both stratified and nonstratified
	 library prep % vs sample_type + treatment + sample_type * treatment + subject_id (binary)
	log10(Final reads) vs sample_type + treatment + sample_type * treatment + subject_id (linear or binary, see regular and cumulative distribution)
	Host DNA % vs sample_type + treatment + sample_type * treatment + subject_id (linear or binary, see regular and cumulative distribution)
	--> both stratified and nonstratified
	
3. LM of taxa alpha diversity (richness)
	--> both stratified and nonstratified
	
4. Permanova (Taxa dist ~ log10(final reads) + sample_type + treatment + sample_type * treatment + subject_id)
	--> both stratified and nonstratified
	
5. DA analysis for taxa, by sample type and treatment 
	--> both stratified and nonstratified
	--> look at level groups as well - phylum, etc.
	
6. Decontam - stratified by treatment
	--> both stratified and nonstratified
	
7. LM of taxa alpha diversity (Shannon)
	--> both stratified and nonstratified
	
8. LM of taxa alpha diversity (InvSimpson)
	--> both stratified and nonstratified
	
9. LM of taxa alpha diversity (BPI) 
	--> both stratified and nonstratified
	
10. LM of function alpha diversity (richness)
	--> both stratified and nonstratified
	
11. LM of function alpha diversity (Shannon)
	--> both stratified and nonstratified

12. LM of function alpha diversity (InvSimpson)
	--> both stratified and nonstratified
	
13. LM of function alpha diversity (BPI) 
	--> both stratified and nonstratified
	
14.  permanova of function alpha diversity
	--> both stratified and nonstratified



#Loading data#


```{r cars}
# Loading files -----------------------------------------------------------
#loading tidy phyloseq object
phyloseq <- read_rds("/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/4_Data/2_Tidy/Phyloseq/PHY_20221129_MGK_host_tidy_tax.rds")
                
        ####Need to be removed after adding sequenicing data
                data_qPCR <- read_csv("/Users/minsikkim/Dropbox (Partners HealthCare)/Project_SICAS2_microbiome/7_Manuscripts/2022_MGK_Host_Depletion/Tables/DAT_20230122_MGK_host_control_qPCR.csv")
                #qPCR data of all the samples sent out sequencing
                data_qPCR <- subset(data_qPCR, data_qPCR$baylor_other_id %in% c(sample_names(phyloseq$phyloseq_count)) | data_qPCR$baylor_other_id %in% c(read_excel("/Users/minsikkim/Dropbox (Partners HealthCare)/Project_Baylor/3_Documentation/Communications/2023-01-24_baylor_shipping_sicas2_nasal_host_depleted/CMMR_MetadataCapture_20230124_LaiP-PQ00430_SICAS2_NS.xlsx", skip = 27) %>% data.frame %>% .$`Optional..............secondary.ID`))
                
                data_qPCR <- subset(data_qPCR, data_qPCR$sample_type %in% c("BAL", "nasal_swab", "Sputum", "neg_depletion", "pos_depletion"))
                data_qPCR$sample_type <- factor(data_qPCR$sample_type, levels = c("BAL", "nasal_swab", "Sputum", "pos_depletion", "neg_depletion"),
                                                labels = c("BAL", "Nasal swab", "Sputum", "P depletion", "N depletion"))
                data_qPCR$treatment <- factor(data_qPCR$treatment, levels = c("control", "lyPMA", "benzonase", "host_zero", "molysis", "qiaamp"),
                                                labels = c("Control", "lyPMA", "Benzonase", "Host zero", "Molysis", "QIAamp"))

#sample data loading
sample_data <- sample_data(phyloseq$phyloseq_count)
        
```


2. QC
        Q1. How many samples failed sequencing
        Q2. How were changes in read stats and host DNA proportion?
        Q3. How were the extraction controls
        Q4. Decontam - were there any contaminants?

# Q1. How many samples failed in sequencing?#


```{r warning=F, QC1}
# Initail QC --------------------------------------------------------------
        #Quesetions - QC
        
#Q0. How many samples failed in sequencing

## figures -----raw data

sample_data %>% 
        subset(., !is.na(.$subject_id)) %>%
        data.frame() %>%
        mutate(host_seq_percent = sequencing_host_prop * 100, 
               .after = sequencing_host_prop,) %>% 
        gather(feature, value, Raw_reads:host_seq_percent) %>%
        group_by(feature, sample_type) %>% 
        subset(., .$feature %in% c("Raw_reads", "Host_mapped", "Final_reads", "host_seq_percent")) %>%
        mutate(feature = factor(feature, levels = c("Raw_reads", "Host_mapped", "Final_reads", "host_seq_percent"), labels = c("Raw reads", "Host mapped", "Final reads", "Host %"))) %>%
        ggplot(aes(x = value, fill = treatment)) +
                geom_histogram() +
                theme(legend.position = "top") +
                guides(fill=guide_legend(title="Treatment", nrow = 1)) +
                facet_grid(sample_type~feature, scales = "free") +
                ggtitle("log10 transfromed histrogram")
             
 
## figures -----log10

sample_data %>% 
        subset(., !is.na(.$subject_id)) %>%
        data.frame() %>%
        mutate(host_seq_percent = sqrt(sequencing_host_prop), 
               .after = sequencing_host_prop,) %>% 
        gather(feature, value, Raw_reads:host_seq_percent) %>%
        group_by(feature, sample_type) %>% 
        subset(., .$feature %in% c("Raw_reads", "Host_mapped", "Final_reads", "host_seq_percent")) %>%
        mutate(feature = factor(feature, levels = c("Raw_reads", "Host_mapped", "Final_reads", "host_seq_percent"), labels = c("Raw reads", "Host mapped", "Final reads", "Host %"))) %>%
        ggplot(aes(x = log10(value), fill = treatment)) +
                geom_histogram() +
                theme(legend.position = "top") +
                guides(fill=guide_legend(title="Treatment", nrow = 1)) +
                facet_grid(sample_type~feature, scales = "free") +
                ggtitle("log10 for reads, sqrt for host % transfromed histrogram")
             
             

# no somaple showed 0 reads

ggarrange(ggplot(sample_data %>% subset(., !is.na(.$subject_id)) %>% data.frame(), aes(x = Final_reads, fill = treatment)) +
                geom_histogram(bins = 97) +
                facet_wrap(~sample_type) +
                theme_classic(base_family = "serif") +
                ggtitle("Histogram of final reads by sample type and treatment") +
                scale_fill_manual(values = c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ff7f00", "#ffff33"), name = "Treatment"),
             ggplot(sample_data %>% subset(., !is.na(.$subject_id)) %>% data.frame(), aes(x = log10(Final_reads), fill = treatment)) +
                geom_histogram(bins = 97) +
                facet_wrap(~sample_type) +
                theme_classic(base_family = "serif") +
                ggtitle("Histogram of log10(final reads) by sample type and treatment") +
                scale_fill_manual(values = c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ff7f00", "#ffff33"), name = "Treatment"),
          common.legend = T, ncol = 1)


hist((log10((phyloseq$phyloseq_count %>% otu_table %>% colSums()) + 1)),100, main = "Histogram of total reads (sum of OTU table)") # 2 samples showed 0 total reads (sum of otu_table)

#how were the samples failed in library prep?
sample_data %>% data.frame %>% mutate(total_read = phyloseq$phyloseq_count %>% otu_table %>% colSums()) %>% rownames_to_column(var = "baylor_other_id") %>%
        ggplot(aes(x = reorder(baylor_other_id, -total_read),
                               y = log10(total_read + 1),
                               col = lib_failed)) +
                geom_point() +
                theme_classic(base_family = "serif") +
                theme(axis.title.y = element_markdown(), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 4)) +
                ylab("log<sub>10</sub>(Sum of OTU table reads)") +
                xlab("Sample ID") +
        guides(col=guide_legend(title="Library failed")) +
        ggtitle("Sum of OTU reads by library failure status")


sample_data %>% data.frame %>% filter(phyloseq$phyloseq_count %>% otu_table %>% colSums() == 0) # two BAL sampels showed 0 total reads
sample_data(phyloseq$phyloseq_count) %>% data.frame() %>% subset(., .$lib_failed)

```
*Results:*

*2.1.1 Modeling final read should be conducted with both linear and logistic regression. Host % need no transformation, and should be tested for both linear and binomial methods*

*2.1.2 13 samples failed in library prep*

*2.1.3. Two BAL sampels showed 0 total reads*

*2.1.4. Sequencing fail ≠ library prep fail*

Comments from Baylor:

Q: What was the lab's criteria for deciding which samples failed library prep.? There were 13 samples that you pointed as failed but their sequencing result actually looks pretty good (ie similar to samples that didn't fail library prep)

A: To determine whether a library attempt “passed or failed” the lab looks at the picogreen concentrations and a library fragment size distribution trace. The trace files are an output from either the Fragment Analyzer or TapeStation  (a copy of the trace files for PQ00331 is attached). If a sample has a background level pico concentration and no discernable fragment concentration on the trace  (i.e. a flat trace line) it is considered failed library. If the sample is below the level of detection of our standard library QC methods, it is considered failure. It’s still possible that there is some small amounts of library in those samples that were successfully sequenced, but often those samples do not generate a meaningful amount of sequence data.


```{r, warning=F, error=FALSE, results='asis', `QC2 Table`}

#sequencing result by sample type and control (1/0)
options(dplyr.summarise.inform = FALSE)

sample_data %>% data.frame() %>% 
        group_by(sample_type, treated) %>% 
        summarise(N = n(),
                  `Raw reads<br>(median [IQR])<br>[reads x 10<sup>7</sup>]` = paste(format(round(median(Raw_reads/10000000),2),nsmall = 2, big.mark = ","), " [", format(round(quantile(Raw_reads/10000000, 0.25),2), nsmall = 2, big.mark = ","), ", ", format(round(quantile(Raw_reads/10000000, 0.75),2), nsmall = 2, big.mark = ","), "]", sep = ""),
                  `Host reads<br>(median [IQR])<br>[reads x 10<sup>7</sup>]` = paste(format(round(median(Host_mapped/10000000),2), nsmall = 2, big.mark = ","), " [", format(round(quantile(Host_mapped/10000000, 0.25),2), nsmall = 2, big.mark = ","), ", ", format(round(quantile(Host_mapped/10000000, 0.75),2), nsmall = 2, big.mark = ","), "]", sep = ""),
                  `Host reads proportion<br>(median [IQR])<br>[%]` = paste(format(round(median(sequencing_host_prop * 100),2), nsmall = 2, big.mark = ","), " [", format(round(quantile(sequencing_host_prop * 100, 0.25),2), nsmall = 2, big.mark = ","), ", ", format(round(quantile(sequencing_host_prop * 100, 0.75),2), nsmall = 2, big.mark = ","), "]", sep = ""),
                  `Final reads<br>(median [IQR])<br>[reads x 10<sup>7</sup>]` = paste(format(round(median(Final_reads/10000000),2), nsmall = 2, big.mark = ","), " [", format(round(quantile(Final_reads/10000000, 0.25),2), nsmall = 2, big.mark = ","), ", ", format(round(quantile(Final_reads/10000000, 0.75),2), nsmall = 2, big.mark = ","), "]", sep = ""),
        ) %>%
        rename(`Sample type` = sample_type, Treated = treated) %>%
        data.frame(check.names = F) %>% mutate_all(linebreak) %>% kbl(format = "html", escape = F) %>% kable_styling(full_width = 0, html_font = "serif")

sample_data %>% data.frame() %>% 
        #dplyr::filter(sample_type %in% c("Sputum", "nasal_swab", "BAL")) %>% 
        group_by (sample_type, treatment) %>%
        summarise(N = n(),
              `Raw reads<br>(median [IQR])<br>[reads x 10<sup>7</sup>]` = paste(format(round(median(Raw_reads/10000000),2), nsmall = 2, big.mark = ","), " [", format(round(quantile(Raw_reads/10000000, 0.25),2), nsmall = 2, big.mark = ","), ", ", format(round(quantile(Raw_reads/10000000, 0.75),2), nsmall = 2, big.mark = ","), "]", sep = ""),
              `Host reads<br>(median [IQR])<br>[reads x 10<sup>7</sup>]` = paste(format(round(median(Host_mapped/10000000),2), nsmall = 2, big.mark = ","), " [", format(round(quantile(Host_mapped/10000000, 0.25),2), nsmall = 2, big.mark = ","), ", ", format(round(quantile(Host_mapped/10000000, 0.75),2), nsmall = 2, big.mark = ","), "]", sep = ""),
              `Host reads proportion<br>(median [IQR])<br>[%]` = paste(format(round(median(sequencing_host_prop * 100),2), nsmall = 2, big.mark = ","), " [", format(round(quantile(sequencing_host_prop * 100, 0.25),2), nsmall = 2, big.mark = ","), ", ", format(round(quantile(sequencing_host_prop * 100, 0.75),2), nsmall = 2, big.mark = ","), "]", sep = ""),
              `Final reads<br>(median [IQR])<br>[reads x 10<sup>7</sup>]` = paste(format(round(median(Final_reads/10000000),2), nsmall = 2, big.mark = ","), " [", format(round(quantile(Final_reads/10000000, 0.25),2), nsmall = 2, big.mark = ","), ", ", format(round(quantile(Final_reads/10000000, 0.75),2), nsmall = 2, big.mark = ","), "]", sep = ""),
        ) %>% data.frame(check.names = F) %>% 
        arrange(sample_type, treatment) %>%
        rename(`Sample type` = sample_type, Treatment = treatment) %>%
        mutate_all(linebreak) %>% kbl(format = "html", escape = F) %>% kable_styling(full_width = 0, html_font = "serif")

```

```{r, warning=F, `QC2 Figure`}

# Summary figures - facet and z-score -------------------------------------

sample_data %>% 
        subset(., !is.na(.$subject_id)) %>%
        data.frame() %>%
        gather(feature, value, Raw_reads:sequencing_host_prop) %>%
        group_by(feature, sample_type) %>% 
        subset(., .$feature %in% c("Raw_reads", "Host_mapped", "Final_reads", "sequencing_host_prop")) %>%
        mutate(z_score = scale(value),
               feature = factor(feature, levels = c("Raw_reads", "Host_mapped", "Final_reads", "sequencing_host_prop"), labels = c("Raw reads", "Host mapped", "Final reads", "Host %"))) %>%
        ggplot(aes(x = treatment, y = z_score, fill = treatment)) +
                geom_boxplot(lwd = 0.2) +
                guides(fill=guide_legend(title="Treatment", nrow = 1)) +
                facet_grid(sample_type~feature) +
                xlab("Treatment") +
                ylab("Z score") +
                theme_classic(base_family = "serif", base_size = 14) +
                guides( x =  guide_axis(angle = 90)) + 
                theme(legend.position = "top") +
                scale_fill_manual(values = c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ff7f00", "#ffff33"), name = "Treatment") #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6

```

*Results:*

*2.2.1. There were no differences between raw reads, however, final reads increased after some treatment, and host DNA proportion decreased *

*2.2.2. The changes in final reads were varied by treatment and sample type *

comment: Should I normalize z-score to mean value of control?


# Q3. How was the extraction controls, both pos and neg?#


```{r, warning=FALSE, QC3}

#Loading theoretical mock community

zymo_mock <- read_excel("/Users/minsikkim/Dropbox (Partners HealthCare)/@minsik/project_sicas2/data_raw/DAR_20210929_zymo_mock_data.xlsx") %>%
        data.frame(row.names = T) %>% rename(mock_theoretical = Mock) %>% mutate(mock_theoretical = mock_theoretical/100) %>%
        merge_phyloseq(otu_table(., taxa_are_rows = T), tax_table(phyloseq$phyloseq_count))

phyloseq_control_rel <- rbind(c("mock_theoretical", "mock")) %>% data.frame() %>%
        column_to_rownames(var = "X1") %>% rename(sample_type = X2) %>% #making sample_data of mock community
        merge_phyloseq(sample_data(.), zymo_mock) %>%        
        merge_phyloseq(., subset_samples(phyloseq$phyloseq_rel, sample_type == "Pos. ext." | sample_type == "Neg. ext." )) #adding data of controls


#Species richness of each control groups
cat("Species richness\n")
rowSums(t(otu_table(phyloseq_control_rel)) != 0)

sample_data(phyloseq_control_rel)$sample_type <- factor(sample_data(phyloseq_control_rel)$sample_type,
                                                        levels = c("mock", "2", "1"),
                                                        labels = c("Mock, 10 spp.", "Pos., 41 spp.", "Neg., 3 spp."))

#Manipulating phyloseq - only top 10 

tax_table(phyloseq_control_rel) %>%
  cbind(species10 = "[Others]") %>%
  {top10species <- head(taxa_sums(phyloseq_control_rel), 10) %>% names()
   .[top10species, "species10"] <- as.character(.[top10species, "Species"])
   .[, 8] <- .[, 8] %>% gsub("s__", "", .) %>% gsub("_", " ", .) %>% paste("<i>", ., "</i>", sep = "")
   phyloseq_temp <- phyloseq_control_rel
   tax_table(phyloseq_temp) <- tax_table(.) 
   phyloseq_temp
  } %>%
        plot_bar(., fill="species10") + 
        ylab("Relative abundancne") +
        theme_classic(base_size = 11, base_family = "serif") +
        ggtitle("Species richness of control data") +
        theme(legend.text = element_markdown()) +
        guides(fill=guide_legend(title="Top 10 species")) +
        facet_wrap (~ sample_type, scales= "free_x", nrow=1)

#there could be opportunistic pathogens...
                                

```

*Results*

*2.3.1. Negative control showed minimal number of possible contaminants*

*2.3.2. Positive control contained various contaminants*



*Results:*

*2.2.1. There were no differences between raw reads, however, final reads increased after some treatment, and host DNA proportion decreased *

*2.2.2. The changes in final reads were varied by treatment and sample type *


# Q4. Prevalence filtering - red flag #

```{r, warning=F, QC4}
#Calculation of sample prevalence, standard deviation, median abundance across all samples for all bugs and making into a table.
#
#•	In initial analysis we will not perform prevalence or abundance filtering (though we may consider this for secondary differential abundance analyses to manage p (features) > n (sample size) problem and issues with multiple hypothesis correction)
taxa_qc <- data.frame("species" =  otu_table(phyloseq$phyloseq_rel) %>% t() %>% colnames(),
        "prevalence" = ifelse(phyloseq$phyloseq_count %>% otu_table() > 0, 1, 0) %>% t() %>% colSums(), #Prevalence of taxa
        "mean_rel_abd" = phyloseq$phyloseq_rel %>% otu_table() %>% t() %>% colMeans(na.rm = T) #mean relativ abundacne 
)


hist(log10(taxa_qc$prevalence), xlab = "log10(Taxa prevalence)", main = "Histogram of prevalence of taxa")
hist(log10(taxa_qc$mean_rel_abd), xlab = "log10(Mean relative abundance)", main = "Histogram of mean relative abundance")

```


•	In initial analysis we will not perform prevalence or abundance filtering (though we may consider this for secondary differentialabundance analyses to manage p (features) > n (sample size) problem and issues with multiple hypothesis correction)
•	Calculation of sample prevalence, standard deviation, median abundance across all samples for all bugs and making into a table.
        
Although we don't consider the prevalence of abundance at this time, we can consider their red-flags after running the DA analysis

* 2. red flag

```{r warning=FALSE, "Red flag"}
red_flag_taxa <- data.frame(species = taxa_qc$species,
                          red_flag_prev_abd = ifelse(taxa_qc$prevalence < otu_table(phyloseq$phyloseq_rel) %>% t %>% rownames() %>% length * 0.05 & taxa_qc$mean_rel_abd < quantile(taxa_qc$mean_rel_abd, 0.75), 1, 0)
)
```

```

# Decontam package --------------------------------------------------------

        # common contaminants across all the treatment methods





#Decontam - were there any contaminants?#

input of DNA concentration: 16S qPCR data

 https://github.com/benjjneb/decontam/issues/33

Ben Callahan: But in the more limited testing on qPCR data the method still seems to work, and other publications report strong patterns of inverse frequency of contaminants using qPCR data - which is the pattern the frequency method relies on.


Strategy:

2.4.1. run decontam for all samples (common contaminants, by extraction)

2.4.2. stratify decontam analysis per each treatment method (contaminants by depletion methods)


# Decontam package --------------------------------------------------------

        # common contaminants across all the treatment methods



# Decontam package --------------------------------------------------------

        # common contaminants across all the treatment methods


#Manipulating phyloseq 

decontam_prev_lypma <- subset_samples(phyloseq, treatment == "lypma") %>%
        isContaminant(method="prevalence", sample_type="Neg. etx.", threshold = 0.5)        
decontam_prev_benzonase <- subset_samples(phyloseq, treatment == "beonzoanse") %>%
        isContaminant(method="prevalence", sample_type="Neg. etx.", threshold = 0.5)        
decontam_prev_host_zero <- subset_samples(phyloseq, treatment == "host_zero") %>%
        isContaminant(method="prevalence", sample_type="Neg. etx.", threshold = 0.5)        
decontam_prev_molysis <- subset_samples(phyloseq, treatment == "molysis") %>%
        isContaminant(method="prevalence", sample_type="Neg. etx.", threshold = 0.5)        
decontam_prev_qiaamp <- subset_samples(phyloseq, treatment == "qiaamp") %>%
        isContaminant(method="prevalence", sample_type="Neg. etx.", threshold = 0.5)        



#stratified decontam
phyloseq_nomock <- merge_phyloseq(phyloseq_sample, phyloseq_control_nomock)
contamdf.prev <- isContaminant(phy_decontam, method="prevalence", sample_type="Neg. etx.", threshold = 0.5)        



#species level decontama
#Control phyloseq object
#decontam - frequency method
contamdf.freq <- isContaminant(phy_decontam, method="frequency", conc="DNA_bac_nondil")


contamdf.freq$contaminant %>% sum()
which(contamdf.freq$contaminant)#list of contam
                                        
                                        plot_frequency(phy_decontam, taxa_names(phy_decontam_freq)[c(which(contamdf.freq$contaminant))], conc="DNA_bac_nondil") + 
                                                xlab("DNA concentration (qPCR; ) [ng / uL]")        
                                        
                                        #decontam - prevalence method
                                        sample_data(phy_decontam)$is.neg <- sample_data(phy_decontam)$is.neg == 1
                                        
                                        
                                        
                                        table(contamdf.prev$contaminant)
                                        ps.pa <- transform_sample_counts(phy_decontam, function(abund) 1*(abund>0))
                                        ps.pa.neg <- prune_samples(sample_data(ps.pa)$is.neg == 1, ps.pa)
                                        ps.pa.pos <- prune_samples(sample_data(ps.pa)$is.neg == 0, ps.pa)
                                        # Make data.frame of prevalence in positive and negative samples
                                        df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                                                            contaminant=contamdf.prev$contaminant)
                                        ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() +
                                                xlab("Prevalence (Negative Controls)") + ylab("Prevalence (True Samples)")
                                        
                                        #decontam - combined method
                                        phy_decontam_comb <- subset_samples(phy_decontam, total_reads != 0)
                                        phy_decontam_comb <- subset_samples(phy_decontam_comb, !is.na(DNA_bac_nondil))
                                        contamdf.comb <- isContaminant(phy_decontam_comb, method="combined", neg="is.neg", conc = "DNA_bac_nondil")
                                        table(contamdf.comb$contaminant)
                                        #using combined method, 6 potential candidates were made.
                                        
                                        otu_table <- otu_table(phyloseq)
                                        otu_table_decontam <- subset(otu_table, !(row.names(otu_table) %in% row.names(subset(contamdf.comb, contamdf.comb$contaminant))))
                                        
                                        phyloseq_decontam <- merge_phyloseq(sample_data(phyloseq), tax_table(phyloseq), otu_table(otu_table_decontam, taxa_are_rows = T))
                                        phyloseq_decontam
                                        #Getting info of both methods
                                        contamdf.freq_true <- subset(contamdf.freq, contamdf.freq$contaminant)
                                        contamdf.prev_true <- subset(contamdf.prev, contamdf.prev$contaminant)
                                        contamdf.comb_true <- subset(contamdf.comb, contamdf.comb$contaminant)
                                        
                                        otu_table_decontam2 <- subset(otu_table, !(row.names(otu_table) %in% row.names(contamdf.prev_true)) & !(row.names(otu_table) %in% row.names(contamdf.freq_true)))
                                        phyloseq_decontam2 <- merge_phyloseq(sample_data(phyloseq), tax_table(phyloseq), otu_table(otu_table_decontam, taxa_are_rows = T))

# Decontam - sensitivity analysis -----------------------------------------

                                        
                                        #decontam-sensitivity analysis
                                        
                                        
                                                        phy_decontam <- subset_samples(phy_decontam, !grepl("control", Row.names))
                                                        contamdf.freqs <- isContaminant(phy_decontam, method="frequency", conc="DNA_bac_nondil")
                                                        contamdf.freqs$contaminant %>% sum()
                                                        which(contamdf.freqs$contaminant)#list of contam
                                                        plot_frequency(phy_decontam, taxa_names(phy_decontam_freq)[c(which(contamdf.freqs$contaminant))], conc="DNA_bac_nondil") + 
                                                                xlab("DNA concentration (pigogreen fluorescent intensity) [ng / uL]")        
                                                        
                                                        #decontam - prevalence method
                                                        sample_data(phy_decontam)$is.neg <- sample_data(phy_decontam)$is.neg == 1
                                                        
                                                        contamdf.prev_s <- isContaminant(phy_decontam, method="prevalence", neg="is.neg", threshold = 0.5)
                                                        
                                                        table(contamdf.prev_s$contaminant)
                                                        ps.pa <- transform_sample_counts(phy_decontam, function(abund) 1*(abund>0))
                                                        ps.pa.neg <- prune_samples(sample_data(ps.pa)$is.neg == 1, ps.pa)
                                                        ps.pa.pos <- prune_samples(sample_data(ps.pa)$is.neg == 0, ps.pa)
                                                        # Make data.frame of prevalence in positive and negative samples
                                                        df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                                                                            contaminant=contamdf.prev$contaminant)
                                                        ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() +
                                                                xlab("Prevalence (Negative Controls)") + ylab("Prevalence (True Samples)")
                                                        
                                                        #decontam - combined method
                                                        phy_decontam_comb_s <- subset_samples(phy_decontam, total_reads != 0)
                                                        phy_decontam_comb_s <- subset_samples(phy_decontam_comb_s, !is.na(DNA_bac_nondil))
                                                        contamdf.comb <- isContaminant(phy_decontam_comb, method="combined", neg="is.neg", conc = "DNA_bac_nondil")
                                                        table(contamdf.comb$contaminant)
                                                        #using combined method, 6 potential candidates were made.
                                                        
                                                        otu_table_decontam <- subset(otu_table, !(row.names(otu_table) %in% row.names(subset(contamdf.comb, contamdf.comb$contaminant))))
                                                        phyloseq_decontam <- merge_phyloseq(sample_data(phyloseq), tax_table(phyloseq), otu_table(otu_table_decontam, taxa_are_rows = T))
                                                        
                                                        #Getting info of both methods
                                                        
                                                        contamdf.freq_true_s <- subset(contamdf.freqs, contamdf.freqs$contaminant)
                                                        contamdf.prev_true_s <- subset(contamdf.prev_s, contamdf.prev_s$contaminant)
                                                        contamdf.comb_true_s <- subset(contamdf.comb, contamdf.comb$contaminant)
                                                        

# Decontam - for each sample type -----------------------------------------

                                                        
                                #Running decontam for differnet sample types
                                                        
                                                        phy_decontam %>% sample_data
                                              
                                                        #species level decontama
                                                        #decontam - frequency method
                                                        
                                                        sample_data(phyloseq_nomock)$total_dna_ngul <- sample_data(phyloseq_nomock)$DNA_bac_nondil + sample_data(phyloseq_nomock)$DNA_host_nondil
                                                        phy_decontam <- subset_samples(phy_decontam, total_reads != 0)
                                                        phy_decontam_ns <- subset_samples(phy_decontam, sample_type == "nasal_swab")
                                                        phy_decontam_ns_prev <- subset_samples(phy_decontam, sample_type != "BAL" & sample_type != "Sputum" )
                                                        phy_decontam_bal <- subset_samples(phy_decontam, sample_type == "BAL")
                                                        phy_decontam_bal_prev <- subset_samples(phy_decontam, sample_type != "nasal_swab" & sample_type != "Sputum")
                                                        phy_decontam_spt <- subset_samples(phy_decontam, sample_type == "Sputum")
                                                        phy_decontam_spt_prev <- subset_samples(phy_decontam, sample_type != "BAL" & sample_type != "nasal_swab")
                                                        
                                                        contamdf.freq$contaminant %>% sum()
                                                        which(contamdf.freq$contaminant)#list of contam
                                                        
                                                        #negative control
                                                        phyloseq_control %>% otu_table()
                                                        phyloseq %>% otu_table() %>% subset(.,grepl("Sutter", row.names(.)))
                                                        phyloseq %>% otu_table() %>% subset(.,grepl("Sutter", row.names(.)))
                                        #frequency method
                                                #All sample type
                                                        plot_frequency(phy_decontam, taxa_names(phy_decontam_freq)[c(which(contamdf.freq$contaminant))], conc="DNA_bac_nondil") + 
                                                                xlab("Bacteiral DNA concentration (qPCR) [ng / uL]") 

                                                        
                                                #Nasal swab                        
                                                        #Bacterial DNA
                                                        contamdf.freq <- isContaminant(phy_decontam_ns, method="frequency", conc="DNA_bac_nondil")
                                                        plot_frequency(phy_decontam_ns, taxa_names(phy_decontam_freq)[c(which(contamdf.freq$contaminant))], conc="DNA_bac_nondil") + 
                                                                xlab("Bacteiral DNA concentration (qPCR) [ng / uL]")        
                                                        #Total DNA
                                                        contamdf.freq <- isContaminant(phy_decontam_ns, method="frequency", conc="total_dna_ngul")
                                                        plot_frequency(phy_decontam_ns, taxa_names(phy_decontam_freq)[c(which(contamdf.freq$contaminant))], conc="DNA_bac_nondil") + 
                                                                xlab("Total DNA concentration (qPCR) [ng / uL]")        
                                                        taxa_names(phy_decontam_freq)[c(which(contamdf.freq$contaminant))]
                                                        
                                                        
                                                #BAL
                                                        #Bacterial DNA
                                                        contamdf.freq <- isContaminant(phy_decontam_bal, method="frequency", conc="DNA_bac_nondil")
                                                        plot_frequency(phy_decontam_bal, taxa_names(phy_decontam_freq)[c(which(contamdf.freq$contaminant))], conc="DNA_bac_nondil") + 
                                                                xlab("Bacteiral DNA concentration (qPCR) [ng / uL]")        
                                                        #Total DNA
                                                        contamdf.freq <- isContaminant(phy_decontam_bal, method="frequency", conc="total_dna_ngul")
                                                        plot_frequency(phy_decontam_bal, taxa_names(phy_decontam_freq)[c(which(contamdf.freq$contaminant))], conc="DNA_bac_nondil") + 
                                                                xlab("Total DNA concentration (qPCR) [ng / uL]")        
                                                        taxa_names(phy_decontam_freq)[c(which(contamdf.freq$contaminant))]
                                                        
                                                #Sputum
                                                        #Bacterial DNA
                                                        contamdf.freq <- isContaminant(phy_decontam_spt, method="frequency", conc="DNA_bac_nondil")
                                                        plot_frequency(phy_decontam_spt, taxa_names(phy_decontam_freq)[c(which(contamdf.freq$contaminant))], conc="DNA_bac_nondil") + 
                                                                xlab("Bacteiral DNA concentration (qPCR) [ng / uL]")        
                                                        #Total DNA
                                                        
                                                        contamdf.freq <- isContaminant(phy_decontam_spt, method="frequency", conc="total_dna_ngul")
                                                        plot_frequency(phy_decontam_spt, taxa_names(phy_decontam_freq)[c(which(contamdf.freq$contaminant))], conc="DNA_bac_nondil") + 
                                                                xlab("Total DNA concentration (qPCR) [ng / uL]")        
                                                        taxa_names(phy_decontam_freq)[c(which(contamdf.freq$contaminant))]
                                                         
                                        #decontam - prevalence method
                                                        decontam_prev <- function (data) {
                                                                sample_data(data)$is.neg <- sample_data(data)$is.neg == 1
                                                                contamdf.prev <- isContaminant(data, method="prevalence", neg="is.neg", threshold = 0.5)
                                                                table(contamdf.prev$contaminant)
                                                                ps.pa <- transform_sample_counts(data, function(abund) 1*(abund>0))
                                                                ps.pa.neg <- prune_samples(sample_data(ps.pa)$is.neg == 1, ps.pa)
                                                                ps.pa.pos <- prune_samples(sample_data(ps.pa)$is.neg == 0, ps.pa)
                                                                # Make data.frame of prevalence in positive and negative samples
                                                                df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                                                                                    contaminant=contamdf.prev$contaminant)
                                                                ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() +
                                                                        xlab("Prevalence (Negative Controls)") + ylab("Prevalence (True Samples)")
                                                         #       subset(df.pa, df.pa$contaminant)
                                                        }
                                                        decontam_prev(phy_decontam_bal_prev)
                                                        decontam_prev(phy_decontam_ns_prev)
                                                        decontam_prev(phy_decontam_spt_prev)
                                                #decontam - combined method
                                                        phy_decontam_comb <- subset_samples(phy_decontam, total_reads != 0)
                                                        phy_decontam_comb <- subset_samples(phy_decontam_comb, !is.na(DNA_bac_nondil))
                                                        contamdf.comb <- isContaminant(phy_decontam_comb, method="combined", neg="is.neg", conc = "DNA_bac_nondil")
                                                        table(contamdf.comb$contaminant)
                                                        
```                                                        


***Analysis***



0. calculation of alpha-diversity indices

```{r warning=FALSE, `Analysis 0`}

alpha_diversity <- function(data) {
        otu_table <- otu_table(data) %>% .[colSums(.) !=0]
        S.obs <- rowSums(t(otu_table) != 0)
        sample_data <- sample_data(data)
        data_evenness <- vegan::diversity(t(otu_table)) / log(vegan::specnumber(t(otu_table))) # calculate evenness index using vegan package
        data_shannon <- vegan::diversity(t(otu_table), index = "shannon") # calculate Shannon index using vegan package
        data_hill <- exp(data_shannon)                           # calculate Hills index
        
        data_dominance <- microbiome::dominance(otu_table, index = "all", rank = 1, aggregate = TRUE) # dominance (Berger-Parker index), etc.
        data_invsimpson <- vegan::diversity(t(otu_table), index = "invsimpson")                          # calculate Shannon index using vegan package
        alpha_diversity <- cbind(S.obs, data_shannon, data_hill, data_invsimpson, data_evenness,data_dominance) # combine all indices in one data table
        sample_data <- merge(data.frame(sample_data), alpha_diversity, by = 0, all = T) %>% column_to_rownames(var = "Row.names")
}

sample_data(phyloseq$phyloseq_rel) <- sample_data(alpha_diversity(phyloseq$phyloseq_count))
sample_data(phyloseq$phyloseq_count) <- sample_data(alpha_diversity(phyloseq$phyloseq_count)) 
sample_data(phyloseq$phyloseq_path_rpkm) <- sample_data(alpha_diversity(phyloseq$phyloseq_path_rpkm))


```


1. qPCR, by smaple type and treatment
        A. Total DNA
        B. Host DNA
        C. Bacterial DNA
        D. Host %

```{r}
   
#2A: Change in total DNA (qPCR)


f2a <- ggplot(data_qPCR, aes(x = sample_type, y = log10(DNA_host_nondil + DNA_bac_nondil))) +
        geom_boxplot(aes(fill = treatment), lwd = 0.2) +
        scale_fill_manual(values = c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ff7f00", "#ffff33")) + #color using https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=6
        ylab("log<sub>10</sub>(qPCR total DNA)<br>(ng/μL)") +
        xlab("Sample type") +
        theme_classic (base_size = 12, base_family = "serif") + 
        labs(tag = "A") +
        scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
        theme(plot.tag = element_text(size = 15), axis.title.y = element_markdown()) +              # Plot title size
        guides(fill = guide_legend(nrow = 1, title = "Treatment"))


#2B: Change in human DNA (qPCR)
f2b <- ggplot(data_qPCR, aes(x = sample_type, y = log10(DNA_host_nondil))) +
        geom_boxplot(aes(fill = treatment), lwd = 0.2) +
        scale_fill_manual(values = c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ff7f00", "#ffff33")) +
        ylab("log<sub>10</sub>(qPCR host DNA)<br>(ng/μL)") +
        xlab("Sample type") +
        theme_classic (base_size = 12, base_family = "serif")+ 
        labs(tag = "B") +
        scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
        theme(plot.tag = element_text(size = 15), axis.title.y = element_markdown()) +              # Plot title size
        guides(fill = guide_legend(nrow = 1, title = "Treatment"))
#2C: Change in 16S DNA (qPCR)
f2c <- ggplot(data_qPCR, aes(x = sample_type, y = log10(DNA_bac_nondil))) +
        geom_boxplot(aes(fill = treatment), lwd = 0.2) +
        scale_fill_manual(values = c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ff7f00", "#ffff33")) +
        ylab("log<sub>10</sub>(qPCR bacterial DNA)<br>(ng/μL)") +
        xlab("Sample type") +
        theme_classic (base_size = 12, base_family = "serif")+ 
        labs(tag = "C") +
        scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
        theme(plot.tag = element_text(size = 15), axis.title.y = element_markdown()) +              # Plot title size
        guides(fill = guide_legend(nrow = 1, title = "Treatment"))

#2D. Change in % host (qPCR)
f2d <- ggplot(data_qPCR, aes(x = sample_type, y = log10(host_proportion * 100))) +
        geom_boxplot(aes(fill = treatment), lwd = 0.2) +
        scale_fill_manual(values = c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ff7f00", "#ffff33")) +
        ylab("log<sub>10</sub>(host DNA) (%)") +
        xlab("Sample type") +
        theme_classic (base_size = 12, base_family = "serif") + 
        labs(tag = "D") +
        scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
        theme(plot.tag = element_text(size = 15), axis.title.y = element_markdown()) +              # Plot title size
        guides(fill = guide_legend(nrow = 1, title = "Treatment"))

#output for markdown
ggarrange(f2a, f2b, f2c, f2d, common.legend = T , align = "hv")

```

2. Sequencing results (treatment, raw reads, host reads, % host, final reads)
	--> both stratified and nonstratified
	 library prep % vs sample_type + treatment + sample_type * treatment + subject_id (binary)
	log10(Final reads) vs sample_type + treatment + sample_type * treatment + subject_id (linear or binary, see regular and cumulative distribution)
	Host DNA % vs sample_type + treatment + sample_type * treatment + subject_id (linear or binary, see regular and cumulative distribution)
	--> both stratified and nonstratified

```{r warning=F, '2 Figure'}

cat("Binomial - library failed? \n\n")

glmer(lib_failed ~ sample_type + treatment + sample_type * treatment + (1|subject_id), data = sample_data %>% data.frame) %>%
        summary %>% .$coefficients %>% data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`t value`) > 2 ~ "*", .default = " ")) %>% 
        kbl(format = "html") %>% kable_styling(full_width = 0, html_font = "serif")
cat("Binomial - Final reads? \n\n")
glmer(log10(Final_reads) ~ sample_type + treatment + sample_type * treatment + (1|subject_id), data = sample_data %>% data.frame) %>%
        summary %>% .$coefficients %>% data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`t value`) > 2 ~ "*", .default = " ")) %>% 
        kbl(format = "html") %>% kable_styling(full_width = 0, html_font = "serif")
cat("Linear - Final reads? \n\n")
lmer(log10(Final_reads) ~ sample_type + treatment + sample_type * treatment + (1|subject_id), data = sample_data %>% data.frame) %>%
        summary %>% .$coefficients %>% data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.05 ~ "*", .default = " ")) %>% 
        kbl(format = "html") %>% kable_styling(full_width = 0, html_font = "serif")
cat("Binomial - Host DNA proportion? \n\n")
glmer(sequencing_host_prop ~ sample_type + treatment + sample_type * treatment + (1|subject_id), data = sample_data %>% data.frame) %>%
        summary %>% .$coefficients %>% data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`t value`) > 2 ~ "*", .default = " ")) %>% 
        kbl(format = "html") %>% kable_styling(full_width = 0, html_font = "serif")
cat("Linear - Host DNA proportion? \n\n")
lmer(sequencing_host_prop ~ sample_type + treatment + sample_type * treatment + (1|subject_id), data = sample_data %>% data.frame) %>%
        summary %>% .$coefficients %>% data.frame(check.names = F) %>%
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.05 ~ "*", .default = " ")) %>% 
        kbl(format = "html") %>% kable_styling(full_width = 0, html_font = "serif")

```
	
3. LM of taxa alpha diversity
	--> both stratified and nonstratified

``` {r}

sample_data <- sample_data(phyloseq$phyloseq_count) %>% data.frame(check.names = F) %>% subset(., !is.nan(.$simpson))

cat("Species richness of all samples\n\n")
cat("S.obs ~ sample_type * treatment + log10 (Final_reads) + (1|original_sample)\n\n")
lmer(S.obs ~ sample_type * treatment + log10 (Final_reads) + (1|subject_id), data = sample_data) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "serif")

#Species richness  - stratified 
cat("Species richness at NS\n\n")
cat("S.obs ~ sample_type + log10 (Final_reads) + (1|original_sample)\n\n")
lmer(S.obs ~ treatment + log10 (Final_reads) + (1|subject_id), data = subset(sample_data, sample_data$sample_type == "Nasal swab")) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "serif")

cat("Species richness at BAL\n\n")
cat("S.obs ~ sample_type + log10 (Final_reads) + (1|original_sample)\n\n")
lmer(S.obs ~ treatment + log10 (Final_reads) + (1|original_sample), data = subset(sample_data, sample_data$sample_type == "BAL")) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "serif")

cat("Species richness at sputum\n\n")
cat("S.obs ~ sample_type + log10 (Final_reads) + (1|original_sample)\n\n")
lmer(S.obs ~ treatment + log10 (Final_reads) + (1|original_sample), data = subset(sample_data, sample_data$sample_type == "Sputum")) %>% 
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "serif")

        
#Shannon
cat("Shannon index\n\n")
cat("Shannon ~ sample_type + log10 (Final_reads) + (1|original_sample)\n\n")
lmer(data_shannon ~ sample_type * treatment + log10 (Final_reads) + (1|original_sample), data = sample_data) %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "serif")


#Simpson
cat("Inverse Simpson index\n\n")
cat("InvSimp ~ sample_type + log10 (Final_reads) + (1|original_sample)\n\n")
lmer(data_invsimpson ~ sample_type * treatment + log10 (Final_reads) + (1|original_sample), data = sample_data) %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "serif")


#BPI
cat("Beger Parker index\n\n")
cat("BPI ~ sample_type + log10 (Final_reads) + (1|original_sample)\n\n")
lmer(dbp ~ sample_type * treatment + log10 (Final_reads) + (1|original_sample), data = sample_data) %>%
        summary() %>%
        .$coefficients %>%
        data.frame(check.names = F) %>% 
        mutate(` ` = case_when(abs(`Pr(>|t|)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "serif")


        
        

```
	
4. Permanova (Taxa dist ~ log10(final reads) + sample_type + treatment + sample_type * treatment + subject_id)
	--> both stratified and nonstratified

``` {r}

phyloseq_rel_nz <- subset_samples(phyloseq$phyloseq_rel, S.obs != 0 & sample_type %in% c("BAL", "Nasal swab", "Sputum"))

bray_perm_uni <- vegan::adonis2(distance(phyloseq_rel_nz, method="bray") ~ sample_type + log10(Final_reads) + treatment + subject_id,
                            data = phyloseq_rel_nz %>% sample_data %>% data.frame(check.names = F), permutations = 10000)

bray_perm <- vegan::adonis2(distance(phyloseq_rel_nz, method="bray") ~ sample_type + log10(Final_reads) + lypma + benzonase + host_zero + molysis + qiaamp + subject_id,
                            data = phyloseq_rel_nz %>% sample_data %>% data.frame(check.names = F), permutations = 10000)

bray_perm_strata <- vegan::adonis2(distance(phyloseq_rel_nz, method="bray") ~ sample_type + log10(Final_reads) + lypma + benzonase + host_zero + molysis + qiaamp,
                            data = phyloseq_rel_nz %>% sample_data %>% data.frame(check.names = F),
                            strata = phyloseq_rel_nz %>% sample_data %>% data.frame(check.names = F) %>% .$subject_id, permutations = 10000)

bray_perm_inter <- vegan::adonis2(distance(phyloseq_rel_nz, method="bray") ~ sample_type * treatment + log10(Final_reads) + subject_id,
                                  data = phyloseq_rel_nz %>% sample_data %>% data.frame(check.names = F), permutations = 10000) 

bray_perm_ns <- vegan::adonis2(distance(subset_samples(phyloseq_rel_nz, sample_type == "Nasal swab"), method="bray") ~ lypma + benzonase + host_zero + molysis + qiaamp + log10(Final_reads) + subject_id,
                               data = subset_samples(phyloseq_rel_nz, sample_type == "Nasal swab") %>% sample_data %>% data.frame(check.names = F), permutations = 10000) 

bray_perm_bal  <- vegan::adonis2(distance(subset_samples(phyloseq_rel_nz, sample_type == "BAL"), method="bray") ~  lypma + benzonase + host_zero + molysis + qiaamp + log10(Final_reads) + subject_id,
                                 data = subset_samples(phyloseq_rel_nz, sample_type == "BAL") %>% sample_data %>% data.frame(check.names = F), permutations = 10000) 

bray_perm_spt <- vegan::adonis2(distance(subset_samples(phyloseq_rel_nz, sample_type == "Sputum"), method="bray") ~ lypma + benzonase + host_zero + molysis + qiaamp + log10(Final_reads) + subject_id,
                                data = subset_samples(phyloseq_rel_nz, sample_type == "Sputum") %>% sample_data %>% data.frame(check.names = F), permutations = 10000) 

cat("\nUnivariate analysis\n")

bray_perm_uni %>% data.frame(check.names = F) %>% rownames_to_column("row.names") %>% 
        mutate(row.names = case_when(row.names == "sample_type" ~ 'Sample type',
                                     row.names == "treatment" ~ 'Treatment',
                                     row.names == "subject_id" ~ 'Subject',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
        round(3) %>% mutate(` ` = case_when(abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "serif")

cat("\nDetailed treatment\n")
bray_perm %>% data.frame(check.names = F) %>% rownames_to_column("row.names") %>% 
        mutate(row.names = case_when(row.names == "sample_type" ~ 'Sample type',
                                     row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'Host zero',
                                     row.names == "molysis" ~ 'Molysis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
        round(3) %>% mutate(` ` = case_when(abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "serif")

cat("\n Strata -detailed treatment\n")
bray_perm_strata %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "sample_type" ~ 'Sample type',
                                     row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'Host zero',
                                     row.names == "molysis" ~ 'Molysis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
        round(3) %>% mutate(` ` = case_when(abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "serif")

cat("\n Interaction term \n")

bray_perm_inter %>% data.frame(check.names = F) %>% rownames_to_column("row.names") %>% 
        mutate(row.names = case_when(row.names == "sample_type" ~ 'Sample type',
                                     row.names == "treatment" ~ 'Treatment',
                                     row.names == "subject_id" ~ 'Subject',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "sample_type:treatment" ~ 'Sample type X treatment',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
        round(3) %>% mutate(` ` = case_when(abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "serif")


cat("\n Stratified - nasal swab \n")

bray_perm_ns %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'Host zero',
                                     row.names == "molysis" ~ 'Molysis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
        round(3) %>% mutate(` ` = case_when(abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "serif")

cat("\n Stratified - BAL \n")

bray_perm_bal %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'Host zero',
                                     row.names == "molysis" ~ 'Molysis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
        round(3) %>% mutate(` ` = case_when(abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "serif")

cat("\n Stratified - sputum \n")

bray_perm_spt %>% data.frame(check.names = F) %>% rownames_to_column('row.names') %>% 
        mutate(row.names = case_when(row.names == "lypma" ~ 'lyPMA',
                                     row.names == "benzonase" ~ 'Benzonase',
                                     row.names == "host_zero" ~ 'Host zero',
                                     row.names == "molysis" ~ 'Molysis',
                                     row.names == "qiaamp" ~ 'QIAamp',
                                     row.names == "subject_id" ~ 'Subject id',
                                     row.names == "log10(Final_reads)" ~ 'log10(Final reads)',
                                     row.names == "Residual" ~ 'Residual',
                                     row.names == "Total" ~ 'Total')) %>% column_to_rownames('row.names') %>% 
        round(3) %>% mutate(` ` = case_when(abs(`Pr(>F)`) < 0.05 ~ "*",
                               .default = " ")) %>% 
        kbl(format = "html") %>%
        kable_styling(full_width = 0, html_font = "serif")

```

	
5. DA analysis for taxa, by sample type and treatment 
	--> both stratified and nonstratified
	--> look at level groups as well - phylum, etc.
	
	
```{r warning=F}
#DA analysis - MaAslin

sample_data(phyloseq_rel_nz)$log10.Final_reads <- log10(sample_data(phyloseq_rel_nz)$Final_reads)
sample_data(phyloseq_rel_nz)$sampletype_treatment <- paste(sample_data(phyloseq_rel_nz)$sample_type, sample_data(phyloseq_rel_nz)$treatment, sep = ":")

#Running MaAslin for all sample without decontam
#for taxa differentially abundant by host depletion method, look to see which ones overlap with potential contaminant taxa

# Maaslin - # # y ~ log(final reads) + sample_type + treatment  -----------
phyloseq_rel_nz %>% sample_data %>% data.frame(check.names = F)
#all samples
maaslin_all = Maaslin2(input_data = otu_table(phyloseq_rel_nz) %>% t %>% data.frame(), 
                input_metadata = phyloseq_rel_nz %>% sample_data %>% data.frame(check.names = F), 
                output = "/Users/minsikkim/Dropbox (Partners HealthCare)/@minsik/project_host_dna_depletion/Data/maaslin_output", 
                fixed_effects = c("sample_type", "log10.Final_reads","lypma", "benzonase", "host_zero", "molysis", "qiaamp"), 
                transform = "LOG", #default
                normalization = "TSS", # default
                random_effects = c("subject_id"), 
                reference = c("sample_type,BAL"), 
                plot_heatmap = F,
                plot_scatter = F)

#Checking number of bugs for the main text of the manuscript 
cat("Number of differentially abundant bugs by each metadata")
maaslin_all$results %>% subset(., .$qval < 0.1) %>% .$metadata %>% table()

maaslin_interaction = Maaslin2(input_data = otu_table(phyloseq_rel_nz) %>% t %>% data.frame(), 
                input_metadata = phyloseq_rel_nz %>% sample_data %>% data.frame(check.names = F), 
                output = "/Users/minsikkim/Dropbox (Partners HealthCare)/@minsik/project_host_dna_depletion/Data/maaslin_output", 
                fixed_effects = c("sample_type", "log10.Final_reads", "treatment", "sampletype_treatment"), 
                transform = "LOG", #default
                normalization = "TSS", # default
                random_effects = c("subject_id"), 
                reference = c("sample_type,BAL", "treatment,Control", "sampletype_treatment,BAL:Control"), 
                plot_heatmap = F,
                plot_scatter = F)

#Checking number of bugs for the main text of the manuscript 
cat("Number of differentially abundant bugs by each metadata, MaAslin with interaction term")
maaslin_interaction$results %>% subset(., .$qval < 0.1) %>% .$metadata %>% table()

#NS
# # y ~ log(final reads) + sample_type + treatment 

fit_data_ns = Maaslin2(input_data = otu_table(subset_samples(phyloseq_rel_nz, sample_type == "Nasal swab")) %>% t %>% data.frame(),
                       input_metadata = sample_data(subset_samples(phyloseq_rel_nz, sample_type == "Nasal swab")) %>% data.frame(), 
                       output = "/Users/minsikkim/Dropbox (Partners HealthCare)/@minsik/project_host_dna_depletion/Data/maaslin_output", 
                       fixed_effects = c("log10.Final_reads","lypma", "benzonase", "host_zero", "molysis", "qiaamp"), 
                       transform = "LOG", #default
                       normalization = "TSS", # default
                       random_effects = c("subject_id"), 
                       plot_heatmap = F,
                       plot_scatter = F)

#Checking number of bugs for the main text of the manuscript 
cat("Number of differentially abundant bugs by each metadata in NS")
fit_data_ns$results %>% subset(., .$qval < 0.1) %>% .$metadata %>% table()

#bal
# # y ~ log(final reads) + sample_type + treatment 

fit_data_bal = Maaslin2(input_data = otu_table(subset_samples(phyloseq_rel_nz, sample_type == "BAL")) %>% t %>% data.frame(),
                       input_metadata = sample_data(subset_samples(phyloseq_rel_nz, sample_type == "BAL")) %>% data.frame(), 
                       output = "/Users/minsikkim/Dropbox (Partners HealthCare)/@minsik/project_host_dna_depletion/Data/maaslin_output", 
                       fixed_effects = c("log10.Final_reads","lypma", "benzonase", "host_zero", "molysis", "qiaamp"), 
                       transform = "LOG", #default
                       normalization = "TSS", # default
                       random_effects = c("subject_id"), 
                       plot_heatmap = F,
                       plot_scatter = F)

cat("Number of differentially abundant bugs by each metadata in BAL")
fit_data_bal$results %>% subset(., .$qval < 0.1) %>% .$metadata %>% table()


#sputum
# # y ~ log(final reads) + sample_type + treatment 

fit_data_spt = Maaslin2(input_data = otu_table(subset_samples(phyloseq_rel_nz, sample_type == "Sputum")) %>% t %>% data.frame(),
                       input_metadata = sample_data(subset_samples(phyloseq_rel_nz, sample_type == "Sputum")) %>% data.frame(), 
                       output = "/Users/minsikkim/Dropbox (Partners HealthCare)/@minsik/project_host_dna_depletion/Data/maaslin_output", 
                       fixed_effects = c("log10.Final_reads","lypma", "benzonase", "host_zero", "molysis", "qiaamp"), 
                       transform = "LOG", #default
                       normalization = "TSS", # default
                       random_effects = c("subject_id"),
                       plot_heatmap = F,
                       plot_scatter = F)

cat("Number of differentially abundant bugs by each metadata in sputum")
fit_data_spt$results %>% subset(., .$qval < 0.1) %>% .$metadata %>% table()

#Making significance table for figure
        # Define a function to make species names italicized
species_italic <- function(data) {
  names <- gsub("_", " ", rownames(data))
  names <- gsub("[]]|[[]", "", names)
  names <- gsub(" sp", " sp.", names)
  names <- gsub(" sp.", "* sp.", names)
  names <- gsub(" group", "* group.", names)
  names <- ifelse(grepl("[*]", names), paste("*", names, sep = ""), paste("*", names, "*", sep = ""))
  rownames(data) <- names
  data
}

#volcano plot of MaAslin with all data

ggplot(maaslin_all$results, aes(y = -log10(qval), x = coef, col = metadata)) +
        theme_classic(base_family = "serif") +
        #labs(tag = "A") +
        ggtitle("MaAslin with treatment types")+
        geom_point(size = 2) +
        xlab("MaAslin coefficient") +
        ylab("-log<sub>10</sub>(*q*-value)") +
        geom_hline(yintercept = 1, col = "gray") +
        geom_vline(xintercept = 0, col = "gray") +
        geom_richtext(aes( 4, 8, label = "*q*-value = 0.1, fold-change = 0", vjust = -1, fontface = 1), col = "grey", size = 3, family = "serif") +
        theme(legend.position = "top", axis.title.y = ggtext::element_markdown()) +
        scale_color_manual(values = c("#4daf4a",  "#984ea3", "#f781bf", "#377eb8", "#ff7f00", "#ffff33", "#a65628")) +
        guides(col = guide_legend(title = "Fixed effects", title.position = "top", nrow = 3))

#volcano plot of MaAslin with all data and an interaction term

ggplot(maaslin_interaction$results, aes(y = -log10(qval), x = coef, col = metadata)) +
        theme_classic(base_family = "serif") +
        #labs(tag = "A") +
        ggtitle("MaAslin with interaction term")+
        geom_point(size = 2) +
        xlab("MaAslin coefficient") +
        ylab("-log<sub>10</sub>(*q*-value)") +
        geom_hline(yintercept = 1, col = "gray") +
        geom_vline(xintercept = 0, col = "gray") +
        geom_richtext(aes( 4, 8, label = "*q*-value = 0.1, fold-change = 0", vjust = -1, fontface = 1), col = "grey", size = 3, family = "serif") +
        theme(legend.position = "top", axis.title.y = ggtext::element_markdown()) +
        scale_color_manual(values = c("#e41a1c",  "#377eb8", "#4daf4a", "#984ea3")) +
        guides(col = guide_legend(title = "Fixed effects", title.position = "top", nrow = 3))
        

# Make a significance table for each figure (top 20 taxa)
make_sig_table <- function(data) {
  sig_data <- spread(data$results[order(data$results$qval), c("feature", "metadata", "qval")], metadata, qval)
  sig_data$min <- apply(sig_data, 1, FUN = min)
  sig_data <- sig_data[order(sig_data$min),] %>% .[1:20,]
  sig_data[["feature"]] <- ifelse(sig_data[["feature"]] == "X.Collinsella._massiliensis", "[Collinsella]_massiliensis", sig_data[["feature"]])
  sig_data_italic <- sig_data %>% rownames_to_column(var = "-") %>% column_to_rownames(var = "feature") %>% species_italic %>% select(-c("-", "min", "log10.Final_reads"))
  sig_data_sig <- ifelse(sig_data_italic < 0.1, "*", NA)
  return(list(data = sig_data, data_italic = sig_data_italic, data_sig = sig_data_sig))
}

fit_data_bal <- make_sig_table(fit_data_bal)
fit_data_ns <- make_sig_table(fit_data_ns)
fit_data_spt <- make_sig_table(fit_data_spt)

cat("\n\nMasslin run results with q < 0.1 by variable\n\n")
maaslin_all$results %>% subset(., .$qval < 0.1) %>% .$name %>% table


cat("\n\nMasslin run results with q < 0.1 by variable, with interaction term\n")
maaslin_interaction$results %>% subset(., .$qval < 0.1) %>% .$name %>% table
cat("\n\nMasslin run results with q < 0.1 by variable, with interaction term\n")
maaslin_interaction$results %>% subset(., .$qval < 0.1 & .$coef > 0 & .$metadata == "treatment") %>% select(c("feature", "name", "coef", "qval"))
cat("\n\nThese taxa were highly prevenlent by each treatmment.\n")
cat("\n\nBut they are not contaminants, if 1) they are present in most of the treatments, or, they are included in other interaction terms.\n")
cat("\n\nChecking Case 1 \n")
maaslin_interaction$results %>% subset(., .$qval < 0.1 & .$coef > 0 & .$metadata == "treatment") %>% select(c("feature", "name", "coef", "qval")) %>% .$feature %>% table
cat("\nMost of taxa were treatment specific, however they were found on most of treatments. They could be lab contaminants or newly found taxa. Further analysis is required with control groups.\n")
cat("\n\nChecking hypothesis 2\n")
subset(maaslin_interaction$results,
       maaslin_interaction$results$feature %in% (maaslin_interaction$results %>% 
                                                         subset(., .$qval < 0.1 & .$metadata == "treatment") %>%
                                                         .$feature %>% table %>% names)) %>%
        subset(., .$metadata == "sampletype_treatment" & .$qval < 0.1) %>% select(c("feature", "name", "coef", "qval"))
cat("\nAmong taxa that were treatment specific, some were `sampletype * treatment` specific at the same time. This may be due to the bacterial community is slightly weighted to the treated samples, as they have larger sample numbers. In detail, they were sample specific, therefore it is less likely they are contaminants appeared during the host-DNA depletion step.\n")


cat("\n\nStratified Masslin - significant of each but by treatment in BAL\n")
fit_data_bal$data_sig
cat("\n\nStratified Masslin - significant of each but by treatment in nasal swab\n")
fit_data_ns$data_sig
cat("\n\nStratified Masslin - significant of each but by treatment in Sputum\n")
fit_data_spt$data_sig


```
	
Results
After adding control data, MaAslin needs to be reanalyzed. Adding controls (mock communities) for each treatment group will show more statistically valid results in 
y ~ log(final reads) + sample_type + treatment + sample_type * treatment, (re = subject_id))
*with interaction term*. 
Here, 
interaction term --> the sample type and treatment specific results --> native to that sample type, not contaminants
treatment --> Taxa generally changed in each treatment. To assess them as contaminants, need to doulbe check the coeffciecnt of that taxa associated with with                 sampletype_pos.depletion:treatment methods. If
                1) taxa were positive in `treatment` and having no other significant changes by interaction term --> that taxa is contaminant
                2) taxa were positive in `treatment` and having significant negative interaction term such as `pos.depletion:host_zero` --> that taxa is not a contaminant
                3) taxa were negative in `treatment` --> the total abundance was decreased, it could be due to higher diversity of sequencing. Negligible with small amount of negative changes.
                




6. Decontam - stratified by treatment
	--> both stratified and nonstratified

```
   

```
	
10. LM of function alpha diversity (richness)
	--> both stratified and nonstratified
```
   

```
11. LM of function alpha diversity (Shannon)
	--> both stratified and nonstratified
	
```
   

```
	
12. LM of function alpha diversity (InvSimpson)
	--> both stratified and nonstratified
```
   

```
	
13. LM of function alpha diversity (BPI) 
	--> both stratified and nonstratified

```
   

```
	
14.  permanova of function alpha diversity
	--> both stratified and nonstratified
```
   

```
15. DA analysis for function, by sample type and treatment 
	--> both stratified and nonstratified
	--> multiple levels?
```
   

```
	
